<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Tools for the Efficient Analysis of High-Resolution Genomics Data • BRGenomics</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Tools for the Efficient Analysis of High-Resolution Genomics Data">
<meta property="og:description" content="This package provides useful and efficient utilites for the 
    analysis of high-resolution genomic data using standard Bioconductor 
    methods and classes.">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="brgenomics" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#brgenomics" class="anchor"></a>BRGenomics</h1></div>
<!-- badges: start -->
<!-- badges: end -->
<p>Efficient tools for the analysis of high-resolution genomics data in R.</p>
<p>Explore the documentation: <a href="https://mdeber.github.io/" class="uri">https://mdeber.github.io/</a></p>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>Install development version from <a href="https://github.com/mdeber/BRGenomics">GitHub</a>:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="co"># install.packages("devtools")</span></a>
<a class="sourceLine" id="cb1-2" data-line-number="2">devtools<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/devtools/man/remote-reexports.html">install_github</a></span>(<span class="st">"mdeber/BRGenomics"</span>)</a></code></pre></div>
</div>
<div id="features" class="section level2">
<h2 class="hasAnchor">
<a href="#features" class="anchor"></a>Features</h2>
<p>A full introductory vignette is in the works, but some quick highlights include:</p>
<ul>
<li>Data import</li>
<li>Read counting by genes and positions in genes (e.g. for heatmaps)</li>
<li>Bootstrapping mean signals at positions within genes (e.g. for metaplotting)</li>
<li>Straightforward integration with DESeq2, including:
<ul>
<li>Added support for discontinuous gene regions, so you can do things like filter out dREG peaks within a gene</li>
<li>Automatic use of reduced dispersion matrices, which means that DESeq2 won’t calculate dispersion from all samples within the dataset, but only those being compared – <em>important for perturbations with genome-wide consequences, whether or not there’s a global shift</em>
</li>
<li>Easy, simultaneous generation of numerous sets of DESeq results</li>
</ul>
</li>
<li>Random subsampling of reads from datasets</li>
<li>Convenient, easy-to-use modification of gene regions, e.g. to define promoters or genebodies</li>
<li>Consistent support for binning measurements throughout</li>
</ul>
<p>For more, see the <a href="https://mdeber.github.io/reference/index.html">documentation</a>, which includes demonstrative example code. The package currently includes example PRO-seq data<sup>[1]</sup>.</p>
</div>
<div id="coming-soon" class="section level2">
<h2 class="hasAnchor">
<a href="#coming-soon" class="anchor"></a>Coming Soon</h2>
<ul>
<li>Support for importing bam files directly</li>
<li>Calculation of spike-in normalization factors using several different methods</li>
<li>Functions to assess sufficiency of sequencing depth using meaningful metrics, e.g. pausing indices</li>
<li>Support for length-scaled, bootstrapped signal counting, i.e. signal is binned by percentage of gene length</li>
<li>Function to summarize/aggregate data in bins generated using <code>binNdimensions</code>
</li>
<li>
<em>Potentially:</em> Support of “Views” objects to avoid loading data into memory</li>
</ul>
</div>
<div id="known-issues-and-limitations" class="section level2">
<h2 class="hasAnchor">
<a href="#known-issues-and-limitations" class="anchor"></a>Known issues and limitations</h2>
<ul>
<li>Currently no support for multiple fields in <code>getStrandedCoverage</code>
</li>
<li>Certain gene names can cause <code>getDESeqDataSet</code> to return an error
<ul>
<li>Systematic naming schemes work (e.g. ensembl IDs) while some lists of conventional gene names, i.e. “symbols”, will cause failure.</li>
</ul>
</li>
<li>In some contexts, using multicore causes <code>getDESeqResults</code> to return errors. If this occurs in your environment, set <code>ncores = 1</code>.</li>
</ul>
<hr>
<ol>
<li>Hojoong Kwak, Nicholas J. Fuda, Leighton J. Core, John T. Lis (2013). Precise Maps of RNA Polymerase Reveal How Promoters Direct Initiation and Pausing. <em>Science</em> <strong>339</strong>(6122): 950–953. <a href="https://doi.org/10.1126/science.1229386" class="uri">https://doi.org/10.1126/science.1229386</a>
</li>
</ol>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Mike DeBerardine <br><small class="roles"> Author, maintainer </small>  </li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
