<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Changelog â€¢ BRGenomics</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Changelog"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-news">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.11.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Overview.html">Overview</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Guides
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/GettingStarted.html">Getting Started</a>
    </li>
    <li>
      <a href="../articles/ImportingModifyingAnnotations.html">Importing &amp; Modifying Annotations</a>
    </li>
    <li>
      <a href="../articles/ImportingProcessingData.html">Importing &amp; Processing Data</a>
    </li>
    <li>
      <a href="../articles/SignalCounting.html">Signal Counting</a>
    </li>
    <li>
      <a href="../articles/AnalyzingMultipleDatasets.html">Analyzing Multiple Datasets</a>
    </li>
    <li>
      <a href="../articles/ProfilePlotsAndBootstrapping.html">Profile Plots &amp; Bootstrapping</a>
    </li>
    <li>
      <a href="../articles/SpikeInNormalization.html">Spike-in Normalization</a>
    </li>
    <li>
      <a href="../articles/DESeq2WithGlobalPerturbations.html">DESeq2 with Global Perturbations</a>
    </li>
    <li>
      <a href="../articles/SequenceExtraction.html">Sequence Extraction</a>
    </li>
  </ul></li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/mdeber/BRGenomics" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
      <h1 data-toc-skip>Changelog <small></small></h1>
      <small>Source: <a href="https://github.com/mdeber/BRGenomics/blob/HEAD/NEWS.md" class="external-link"><code>NEWS.md</code></a></small>
    </div>

    <div class="section level2">
<h2 class="page-header" data-toc-text="1.10.1" id="brgenomics-1101-release--1111-devel">BRGenomics 1.10.1 (release) &amp; 1.11.1 (devel)<a class="anchor" aria-label="anchor" href="#brgenomics-1101-release--1111-devel"></a></h2>
<ul><li>Updated the <code>import_bam_ATAC()</code> function with slightly revised options and operations</li>
<li>Fixed broken links in documentation</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.3" id="brgenomics-103-release--113-devel">BRGenomics 1.0.3 (release) &amp; 1.1.3 (devel)<a class="anchor" aria-label="anchor" href="#brgenomics-103-release--113-devel"></a></h2>
<ul><li>Increased safeguards against automatic vector coercion of matrices and dataframes</li>
<li>Increased tolerance of the precision workaround in <code><a href="../reference/getStrandedCoverage.html">getStrandedCoverage()</a></code> (workaround releveant to GenomicRanges <a href="https://github.com/Bioconductor/GenomicRanges/issues/39" class="external-link">issue </a><a href="https://github.com/mdeber/BRGenomics/issues/39" class="external-link">#39</a>; will be obviated in next major Bioconductor release)</li>
<li>Various minor efficiency &amp; formatting changes</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.1" id="brgenomics-101-release--111-devel">BRGenomics 1.0.1 (release) &amp; 1.1.1 (devel)<a class="anchor" aria-label="anchor" href="#brgenomics-101-release--111-devel"></a></h2>
<ul><li>Merge changes from 0.99.33 into release branch</li>
<li>Bug workaround for <code><a href="../reference/getStrandedCoverage.html">getStrandedCoverage()</a></code> and dependent methods for getting coverage of normalized data (apparent bug in <code><a href="https://rdrr.io/pkg/IRanges/man/coverage-methods.html" class="external-link">IRanges::coverage()</a></code> for weighting by normalized values)</li>
<li>Bug fix in <code><a href="../reference/getCountsByPositions.html">getCountsByPositions()</a></code> for getting counts over an unstranded region with <code>expand_ranges=TRUE</code>
</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.33" id="brgenomics-09933">BRGenomics 0.99.33<a class="anchor" aria-label="anchor" href="#brgenomics-09933"></a></h2>
<ul><li>Add support for log fold-change shrinkage in <code><a href="../reference/getDESeqResults.html">getDESeqResults()</a></code>
</li>
<li>Minor bugfix generating a warning under some conditions when non-integer binsizes are used</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.30" id="brgenomics-09930">BRGenomics 0.99.30<a class="anchor" aria-label="anchor" href="#brgenomics-09930"></a></h2>
<ul><li>Change all <code>ncores</code> options to default to <code>getOption("mc.cores", 2L)</code>
</li>
<li>Rename n-dimensional binning functions <code><a href="../reference/binNdimensions.html">aggregateByNdimBins()</a></code> and <code><a href="../reference/binNdimensions.html">densityInNdimBins()</a></code>
</li>
<li>Redone documentation website, splitting up the vignette</li>
<li>Updated package vignettes/guide pages in several places</li>
<li>Other minor internal changes</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.25" id="brgenomics-09925">BRGenomics 0.99.25<a class="anchor" aria-label="anchor" href="#brgenomics-09925"></a></h2>
<ul><li>Bug fixes with <code>expand_ranges</code> arguments affecting <code><a href="../reference/getCountsByRegions.html">getCountsByRegions()</a></code>, <code><a href="../reference/subsampleGRanges.html">subsampleGRanges()</a></code>, and <code><a href="../reference/getSpikeInNFs.html">getSpikeInNFs()</a></code>
</li>
<li>Expanded testing, particularly for <code>expand_ranges</code> arguments and <code><a href="../reference/import_bam.html">import_bam()</a></code>
</li>
<li>Added options in <code><a href="../reference/mergeReplicates.html">mergeReplicates()</a></code>
</li>
<li>Various small doc updates and minor internal changes</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.23" id="brgenomics-09923">BRGenomics 0.99.23<a class="anchor" aria-label="anchor" href="#brgenomics-09923"></a></h2>
<ul><li>Added support for non-basepair-resolution GRanges throughout, via the <code>expand_ranges</code> argument<ul><li>Substantial performance benefits for less-sparse datasets (e.g.Â whole read coverage)</li>
<li>Supported everywhere, including counting functions, subsampling, merging, normalization, etc.</li>
</ul></li>
<li>Rewrite of <code><a href="../reference/mergeGRangesData.html">mergeGRangesData()</a></code>:<ul><li>Substantial performance improvements for most datasets</li>
<li>No longer requires basepair-resolution GRanges objects</li>
<li>Added options and flexibility for merging reads as well as coverage data</li>
</ul></li>
<li>Add a <code><a href="../reference/mergeReplicates.html">mergeReplicates()</a></code> function</li>
<li>Rewrite of <code><a href="../reference/makeGRangesBRG.html">makeGRangesBRG()</a></code> that significantly improves performance for sparser datasets (the datasets for which using the function makes the most sense)</li>
<li>
<code><a href="../reference/subsampleGRanges.html">subsampleGRanges()</a></code> no longer returns a basepair-resolution GRanges by default</li>
<li>When <code>field=NULL</code>, <code><a href="../reference/applyNFsGRanges.html">applyNFsGRanges()</a></code> no longer returns a basepair-resolution GRanges by default</li>
<li>Add <code>use_bin_numbers</code> option to n-dimension binning functions; setting to false allows returning of bin values (the bin center) instead of the ordinal bin numbers (indexes)</li>
<li>Quietly adding support for <code>GRangesList</code> objects (not fully tested)</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.15" id="brgenomics-09915">BRGenomics 0.99.15<a class="anchor" aria-label="anchor" href="#brgenomics-09915"></a></h2>
<ul><li>Add pre-filtering to counting functions for performance</li>
<li>Some additional clarification of readcounts vs.Â coverage signal in counting and import functions</li>
<li>Change tidyChromosomes test</li>
<li>Remove indirect links in doc pages (use only exact names of man pages)</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.10" id="brgenomics-09910">BRGenomics 0.99.10<a class="anchor" aria-label="anchor" href="#brgenomics-09910"></a></h2>
<ul><li>Code modifications to pass Bioconductor test builds on Windows:<ul><li>Make all examples and tests single core</li>
<li>Internally (not exported) redefine mcMap (current implementation in package parallel needs to be modified)</li>
<li>In tests and examples, test if on Windows before attempting any bigWig file import</li>
</ul></li></ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.0" id="brgenomics-0990">BRGenomics 0.99.0<a class="anchor" aria-label="anchor" href="#brgenomics-0990"></a></h2>
<ul><li>Changes for Bioconductor submission:<ul><li>Move to package versioning 0.99.x</li>
<li>Update R requirement to version 4.0</li>
<li>Add new branch <code>R3</code> to allow users to install under R version &gt;=3.5</li>
</ul></li>
<li>Various minor formatting changes to codebase</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.8.1" id="brgenomics-081">BRGenomics 0.8.1<a class="anchor" aria-label="anchor" href="#brgenomics-081"></a></h2>
<ul><li>Fixed bug in <code><a href="../reference/import_bam.html">import_bam()</a></code> that produced warnings when <code>shift</code> argument used to shift both 5â€™ and 3â€™ ends of reads (i.e.Â when <code>length(shift) == 2</code>)</li>
<li>Updated included external datasets to be much smaller<ul><li>Included bam file now &lt;200 reads</li>
<li>Included bigWig and bedGraph files derived from the bam file</li>
</ul></li>
<li>Minor update to vignette to reflect the change in the bam file</li>
<li>Updated the included <code><a href="https://rdrr.io/r/utils/data.html" class="external-link">data()</a></code> objects (<code>PROseq</code> and <code>PROseq_paired</code>)<ul><li>Shifted PRO-seq 3â€™ bases to remove the run-on base, and updated associated package tests</li>
<li>Added xz compression to the files</li>
</ul></li>
<li>Streamlined some method dispatch in <code><a href="../reference/genebodies.html">genebodies()</a></code> and <code><a href="../reference/getDESeqDataSet.html">getDESeqDataSet()</a></code> functions</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.8.0" id="brgenomics-080">BRGenomics 0.8.0<a class="anchor" aria-label="anchor" href="#brgenomics-080"></a></h2>
<ul><li>Add support for lists in data import functions</li>
<li>Add the convenience function <code><a href="../reference/applyNFsGRanges.html">applyNFsGRanges()</a></code>
</li>
<li>Significant internal changes to <code><a href="../reference/import_bam.html">import_bam()</a></code><ul><li>New test for paired-end reads (deprecated use of <code><a href="https://rdrr.io/pkg/Rsamtools/man/testPairedEndBam.html" class="external-link">Rsamtools::testPairedEndBam()</a></code>)</li>
<li>Avoids any internal use of <code>bpiterate()</code>
</li>
<li>Dropped dependency on <code>GenomicFiles</code> package</li>
</ul></li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.7.10" id="brgenomics-0710">BRGenomics 0.7.10<a class="anchor" aria-label="anchor" href="#brgenomics-0710"></a></h2>
<ul><li>Add <code><a href="../reference/intersectByGene.html">intersectByGene()</a></code> and <code><a href="../reference/intersectByGene.html">reduceByGene()</a></code> functions</li>
<li>Minor vignette updates</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.7.8" id="brgenomics-078">BRGenomics 0.7.8<a class="anchor" aria-label="anchor" href="#brgenomics-078"></a></h2>
<ul><li>Substantially updated vignette</li>
<li>Fully load <code>rtracklayer</code> (so completely exported to users)</li>
<li>Add <code><a href="../reference/makeGRangesBRG.html">isBRG()</a></code> function</li>
<li>Fixed bug in <code><a href="../reference/getSpikeInNFs.html">spikeInNormGRanges()</a></code> that failed to remove spike-in reads (aside from maintaining those reads, normalization was otherwise correct)</li>
<li>Fixed minor bug in which <code><a href="../reference/bootstrap-signal-by-position.html">metaSubsample()</a></code> automatically added rownames with list input</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.7.7" id="brgenomics-077">BRGenomics 0.7.7<a class="anchor" aria-label="anchor" href="#brgenomics-077"></a></h2>
<ul><li>Bug fix in <code>aggregateByNdimensionalBins()</code> affecting simultaneous aggregation of multiple data</li>
<li>Minor updates to documentation, including an error in <code><a href="../reference/getDESeqResults.html">getDESeqResults()</a></code>
</li>
<li>Slightly expanded vignette</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.7.5" id="brgenomics-075">BRGenomics 0.7.5<a class="anchor" aria-label="anchor" href="#brgenomics-075"></a></h2>
<ul><li>Update bootstrapping functions<ul><li>Add blacklisting support for <code><a href="../reference/bootstrap-signal-by-position.html">metaSubsample()</a></code>
</li>
<li>Related to blacklisting, NA values now ignored in bootstrapping</li>
</ul></li>
<li>Add additional <code>melt</code> options for signal counting functions</li>
<li>Further expanded support for list inputs (lists of GRanges datasets), including in <code><a href="../reference/getStrandedCoverage.html">getStrandedCoverage()</a></code>
</li>
<li>Add explicit support for blacklisting in <code><a href="../reference/getDESeqDataSet.html">getDESeqDataSet()</a></code>
</li>
<li>Rewrite n-dimensional binning functions, and add function for aggregating data within n-dimensional bins<ul><li>Changed arguments in <code><a href="../reference/binNdimensions.html">binNdimensions()</a></code> to only accept dataframe inputs</li>
<li>Add <code>densityInNdimensionalBins()</code> function to count points in each bin</li>
<li>Add <code>aggregateByNdimensionalBins()</code> function to aggregate data within bins using arbitrary functions</li>
</ul></li>
<li>Added arguments for setting sample names in spike-in/normalization functions</li>
<li>Various improvements and streamlining for method dispatch and flexibility</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.7.0" id="brgenomics-070">BRGenomics 0.7.0<a class="anchor" aria-label="anchor" href="#brgenomics-070"></a></h2>
<ul><li>Added functions for counting and filtering spike-in reads</li>
<li>Added functions for generating spike-in normalization factors</li>
<li>Added support for lists of GRanges datasets throughout, including all signal counting functions</li>
<li>Updating signal counting functions with a <code>blacklist</code> argument, for ignoring reads from user-supplied regions</li>
<li>Added wrappers for <code><a href="../reference/import_bam.html">import_bam()</a></code> for several common use cases</li>
<li>Update <code><a href="../reference/getCountsByPositions.html">getCountsByPositions()</a></code>:<ul><li>Added a <code>melt</code> option for returning melted dataframes</li>
<li>Now returns an error by default if multi-width regions are given (must be explicit)</li>
</ul></li>
<li>Changed argument order in <code><a href="../reference/getMaxPositionsBySignal.html">getMaxPositionsBySignal()</a></code>
</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.5.6" id="brgenomics-056">BRGenomics 0.5.6<a class="anchor" aria-label="anchor" href="#brgenomics-056"></a></h2>
<ul><li>Update <code><a href="../reference/import_bam.html">import_bam()</a></code> function<ul><li>Added support for paired-end bam files</li>
<li>Added the <code>shift</code> argument</li>
</ul></li>
<li>Made <code><a href="../reference/bootstrap-signal-by-position.html">metaSubsample()</a></code> functions robust to unevaluated inputs</li>
<li>Small performance improvement to <code><a href="../reference/genebodies.html">genebodies()</a></code> function</li>
<li>Multicore usage is again the default for <code><a href="../reference/getDESeqResults.html">getDESeqResults()</a></code>
</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.5.3" id="brgenomics-053">BRGenomics 0.5.3<a class="anchor" aria-label="anchor" href="#brgenomics-053"></a></h2>
<ul><li>Substantial performance improvement for <code><a href="../reference/mergeGRangesData.html">mergeGRangesData()</a></code>
</li>
<li>Make single-core the default for <code><a href="../reference/getDESeqResults.html">getDESeqResults()</a></code>
</li>
<li>Fixed errant warning message in <code><a href="../reference/binNdimensions.html">binNdimensions()</a></code> for integer inputs</li>
<li>Update namespace to fully import GenomicRanges and S4Vectors</li>
<li>Changed R dependency, evidently required by the updated data objects</li>
<li>Various documentation updates</li>
<li>Added package documentation page</li>
<li>Minor changes to examples</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.5.0" id="brgenomics-050">BRGenomics 0.5.0<a class="anchor" aria-label="anchor" href="#brgenomics-050"></a></h2>
<ul><li>Increased support for multiplexed GRanges objects across all functions</li>
<li>Increased performance for bootstrapping over multiple fields/multiplexed GRanges objects</li>
<li>Removed requirement to set â€˜fieldâ€™ argument for gettings counts over multiplexed GRanges</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.4.7" id="brgenomics-047">BRGenomics 0.4.7<a class="anchor" aria-label="anchor" href="#brgenomics-047"></a></h2>
<ul><li>Rewrote <code><a href="../reference/mergeGRangesData.html">mergeGRangesData()</a></code> to support the creation of multiplexed GRanges objects</li>
<li>Made <code><a href="../reference/getCountsByRegions.html">getCountsByRegions()</a></code> and <code><a href="../reference/getCountsByPositions.html">getCountsByPositions()</a></code> to return integers if input signal is integer</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.4.5" id="brgenomics-045">BRGenomics 0.4.5<a class="anchor" aria-label="anchor" href="#brgenomics-045"></a></h2>
<ul><li>Added and modified numerous arguments</li>
<li>Increased support for normalization factors across signal counting functions</li>
<li>Modified behavior of bootstrapping over multiple fields by removing explicit access to .Random.seed</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.4.1" id="brgenomics-041">BRGenomics 0.4.1<a class="anchor" aria-label="anchor" href="#brgenomics-041"></a></h2>
<ul><li>Various documentation updates</li></ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.4.0" id="brgenomics-040">BRGenomics 0.4.0<a class="anchor" aria-label="anchor" href="#brgenomics-040"></a></h2>
<ul><li>Added a <code>NEWS.md</code> file to track changes</li>
<li>Added support for bam files</li>
</ul></div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>

</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

