<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Changelog • BRGenomics</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Changelog" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-news">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.33</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Overview.html">Overview</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Guides
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/GettingStarted.html">Getting Started</a>
    </li>
    <li>
      <a href="../articles/ImportingModifyingAnnotations.html">Importing &amp; Modifying Annotations</a>
    </li>
    <li>
      <a href="../articles/ImportingProcessingData.html">Importing &amp; Processing Data</a>
    </li>
    <li>
      <a href="../articles/SignalCounting.html">Signal Counting</a>
    </li>
    <li>
      <a href="../articles/AnalyzingMultipleDatasets.html">Analyzing Multiple Datasets</a>
    </li>
    <li>
      <a href="../articles/ProfilePlotsAndBootstrapping.html">Profile Plots &amp; Bootstrapping</a>
    </li>
    <li>
      <a href="../articles/SpikeInNormalization.html">Spike-in Normalization</a>
    </li>
    <li>
      <a href="../articles/DESeq2WithGlobalPerturbations.html">DESeq2 with Global Perturbations</a>
    </li>
    <li>
      <a href="../articles/SequenceExtraction.html">Sequence Extraction</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mdeber/BRGenomics">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
      <h1>Changelog <small></small></h1>
      
    </div>

    <div id="brgenomics-0-99-33" class="section level2">
<h2 class="hasAnchor">
<a href="#brgenomics-0-99-33" class="anchor"></a>BRGenomics 0.99.33</h2>
<ul>
<li>Add support for log fold-change shrinkage in <code><a href="../reference/getDESeqResults.html">getDESeqResults()</a></code>
</li>
<li>Minor bugfix generating a warning under some conditions when non-integer binsizes are used</li>
</ul>
</div>
    <div id="brgenomics-0-99-30" class="section level2">
<h2 class="hasAnchor">
<a href="#brgenomics-0-99-30" class="anchor"></a>BRGenomics 0.99.30</h2>
<ul>
<li>Change all <code>ncores</code> options to default to <code><a href="https://rdrr.io/r/base/options.html">getOption("mc.cores", 2L)</a></code>
</li>
<li>Rename n-dimensional binning functions <code><a href="../reference/binNdimensions.html">aggregateByNdimBins()</a></code> and <code><a href="../reference/binNdimensions.html">densityInNdimBins()</a></code>
</li>
<li>Redone documentation website, splitting up the vignette</li>
<li>Updated package vignettes/guide pages in several places</li>
<li>Other minor internal changes</li>
</ul>
</div>
    <div id="brgenomics-0-99-25" class="section level2">
<h2 class="hasAnchor">
<a href="#brgenomics-0-99-25" class="anchor"></a>BRGenomics 0.99.25</h2>
<ul>
<li>Bug fixes with <code>expand_ranges</code> arguments affecting <code><a href="../reference/getCountsByRegions.html">getCountsByRegions()</a></code>, <code><a href="../reference/subsampleGRanges.html">subsampleGRanges()</a></code>, and <code><a href="../reference/getSpikeInNFs.html">getSpikeInNFs()</a></code>
</li>
<li>Expanded testing, particularly for <code>expand_ranges</code> arguments and <code><a href="../reference/import_bam.html">import_bam()</a></code>
</li>
<li>Added options in <code><a href="../reference/mergeReplicates.html">mergeReplicates()</a></code>
</li>
<li>Various small doc updates and minor internal changes</li>
</ul>
</div>
    <div id="brgenomics-0-99-23" class="section level2">
<h2 class="hasAnchor">
<a href="#brgenomics-0-99-23" class="anchor"></a>BRGenomics 0.99.23</h2>
<ul>
<li>Added support for non-basepair-resolution GRanges throughout, via the <code>expand_ranges</code> argument
<ul>
<li>Substantial performance benefits for less-sparse datasets (e.g. whole read coverage)</li>
<li>Supported everywhere, including counting functions, subsampling, merging, normalization, etc.</li>
</ul>
</li>
<li>Rewrite of <code><a href="../reference/mergeGRangesData.html">mergeGRangesData()</a></code>:
<ul>
<li>Substantial performance improvements for most datasets</li>
<li>No longer requires basepair-resolution GRanges objects</li>
<li>Added options and flexibility for merging reads as well as coverage data</li>
</ul>
</li>
<li>Add a <code><a href="../reference/mergeReplicates.html">mergeReplicates()</a></code> function</li>
<li>Rewrite of <code><a href="../reference/makeGRangesBRG.html">makeGRangesBRG()</a></code> that significantly improves performance for sparser datasets (the datasets for which using the function makes the most sense)</li>
<li>
<code><a href="../reference/subsampleGRanges.html">subsampleGRanges()</a></code> no longer returns a basepair-resolution GRanges by default</li>
<li>When <code>field=NULL</code>, <code><a href="../reference/applyNFsGRanges.html">applyNFsGRanges()</a></code> no longer returns a basepair-resolution GRanges by default</li>
<li>Add <code>use_bin_numbers</code> option to n-dimension binning functions; setting to false allows returning of bin values (the bin center) instead of the ordinal bin numbers (indexes)</li>
<li>Quietly adding support for <code>GRangesList</code> objects (not fully tested)</li>
</ul>
</div>
    <div id="brgenomics-0-99-15" class="section level2">
<h2 class="hasAnchor">
<a href="#brgenomics-0-99-15" class="anchor"></a>BRGenomics 0.99.15</h2>
<ul>
<li>Add pre-filtering to counting functions for performance</li>
<li>Some additional clarification of readcounts vs. coverage signal in counting and import functions</li>
<li>Change tidyChromosomes test</li>
<li>Remove indirect links in doc pages (use only exact names of man pages)</li>
</ul>
</div>
    <div id="brgenomics-0-99-10" class="section level2">
<h2 class="hasAnchor">
<a href="#brgenomics-0-99-10" class="anchor"></a>BRGenomics 0.99.10</h2>
<ul>
<li>Code modifications to pass Bioconductor test builds on Windows:
<ul>
<li>Make all examples and tests single core</li>
<li>Internally (not exported) redefine mcMap (current implementation in package parallel needs to be modified)</li>
<li>In tests and examples, test if on Windows before attempting any bigWig file import</li>
</ul>
</li>
</ul>
</div>
    <div id="brgenomics-0-99-0" class="section level2">
<h2 class="hasAnchor">
<a href="#brgenomics-0-99-0" class="anchor"></a>BRGenomics 0.99.0</h2>
<ul>
<li>Changes for Bioconductor submission:
<ul>
<li>Move to package versioning 0.99.x</li>
<li>Update R requirement to version 4.0</li>
<li>Add new branch <code>R3</code> to allow users to install under R version &gt;=3.5</li>
</ul>
</li>
<li>Various minor formatting changes to codebase</li>
</ul>
</div>
    <div id="brgenomics-0-8-1" class="section level2">
<h2 class="hasAnchor">
<a href="#brgenomics-0-8-1" class="anchor"></a>BRGenomics 0.8.1</h2>
<ul>
<li>Fixed bug in <code><a href="../reference/import_bam.html">import_bam()</a></code> that produced warnings when <code>shift</code> argument used to shift both 5’ and 3’ ends of reads (i.e. when <code>length(shift) == 2</code>)</li>
<li>Updated included external datasets to be much smaller
<ul>
<li>Included bam file now &lt;200 reads</li>
<li>Included bigWig and bedGraph files derived from the bam file</li>
</ul>
</li>
<li>Minor update to vignette to reflect the change in the bam file</li>
<li>Updated the included <code><a href="https://rdrr.io/r/utils/data.html">data()</a></code> objects (<code>PROseq</code> and <code>PROseq_paired</code>)
<ul>
<li>Shifted PRO-seq 3’ bases to remove the run-on base, and updated associated package tests</li>
<li>Added xz compression to the files</li>
</ul>
</li>
<li>Streamlined some method dispatch in <code><a href="../reference/genebodies.html">genebodies()</a></code> and <code><a href="../reference/getDESeqDataSet.html">getDESeqDataSet()</a></code> functions</li>
</ul>
</div>
    <div id="brgenomics-0-8-0" class="section level2">
<h2 class="hasAnchor">
<a href="#brgenomics-0-8-0" class="anchor"></a>BRGenomics 0.8.0</h2>
<ul>
<li>Add support for lists in data import functions</li>
<li>Add the convenience function <code><a href="../reference/applyNFsGRanges.html">applyNFsGRanges()</a></code>
</li>
<li>Significant internal changes to <code><a href="../reference/import_bam.html">import_bam()</a></code>
<ul>
<li>New test for paired-end reads (deprecated use of <code><a href="https://rdrr.io/pkg/Rsamtools/man/testPairedEndBam.html">Rsamtools::testPairedEndBam()</a></code>)</li>
<li>Avoids any internal use of <code>bpiterate()</code>
</li>
<li>Dropped dependency on <code>GenomicFiles</code> package</li>
</ul>
</li>
</ul>
</div>
    <div id="brgenomics-0-7-10" class="section level2">
<h2 class="hasAnchor">
<a href="#brgenomics-0-7-10" class="anchor"></a>BRGenomics 0.7.10</h2>
<ul>
<li>Add <code><a href="../reference/intersectByGene.html">intersectByGene()</a></code> and <code><a href="../reference/intersectByGene.html">reduceByGene()</a></code> functions</li>
<li>Minor vignette updates</li>
</ul>
</div>
    <div id="brgenomics-0-7-8" class="section level2">
<h2 class="hasAnchor">
<a href="#brgenomics-0-7-8" class="anchor"></a>BRGenomics 0.7.8</h2>
<ul>
<li>Substantially updated vignette</li>
<li>Fully load <code>rtracklayer</code> (so completely exported to users)</li>
<li>Add <code><a href="../reference/makeGRangesBRG.html">isBRG()</a></code> function</li>
<li>Fixed bug in <code><a href="../reference/getSpikeInNFs.html">spikeInNormGRanges()</a></code> that failed to remove spike-in reads (aside from maintaining those reads, normalization was otherwise correct)</li>
<li>Fixed minor bug in which <code><a href="../reference/bootstrap-signal-by-position.html">metaSubsample()</a></code> automatically added rownames with list input</li>
</ul>
</div>
    <div id="brgenomics-0-7-7" class="section level2">
<h2 class="hasAnchor">
<a href="#brgenomics-0-7-7" class="anchor"></a>BRGenomics 0.7.7</h2>
<ul>
<li>Bug fix in <code>aggregateByNdimensionalBins()</code> affecting simultaneous aggregation of multiple data</li>
<li>Minor updates to documentation, including an error in <code><a href="../reference/getDESeqResults.html">getDESeqResults()</a></code>
</li>
<li>Slightly expanded vignette</li>
</ul>
</div>
    <div id="brgenomics-0-7-5" class="section level2">
<h2 class="hasAnchor">
<a href="#brgenomics-0-7-5" class="anchor"></a>BRGenomics 0.7.5</h2>
<ul>
<li>Update bootstrapping functions
<ul>
<li>Add blacklisting support for <code><a href="../reference/bootstrap-signal-by-position.html">metaSubsample()</a></code>
</li>
<li>Related to blacklisting, NA values now ignored in bootstrapping</li>
</ul>
</li>
<li>Add additional <code>melt</code> options for signal counting functions</li>
<li>Further expanded support for list inputs (lists of GRanges datasets), including in <code><a href="../reference/getStrandedCoverage.html">getStrandedCoverage()</a></code>
</li>
<li>Add explicit support for blacklisting in <code><a href="../reference/getDESeqDataSet.html">getDESeqDataSet()</a></code>
</li>
<li>Rewrite n-dimensional binning functions, and add function for aggregating data within n-dimensional bins
<ul>
<li>Changed arguments in <code><a href="../reference/binNdimensions.html">binNdimensions()</a></code> to only accept dataframe inputs</li>
<li>Add <code>densityInNdimensionalBins()</code> function to count points in each bin</li>
<li>Add <code>aggregateByNdimensionalBins()</code> function to aggregate data within bins using arbitrary functions</li>
</ul>
</li>
<li>Added arguments for setting sample names in spike-in/normalization functions</li>
<li>Various improvements and streamlining for method dispatch and flexibility</li>
</ul>
</div>
    <div id="brgenomics-0-7-0" class="section level2">
<h2 class="hasAnchor">
<a href="#brgenomics-0-7-0" class="anchor"></a>BRGenomics 0.7.0</h2>
<ul>
<li>Added functions for counting and filtering spike-in reads</li>
<li>Added functions for generating spike-in normalization factors</li>
<li>Added support for lists of GRanges datasets throughout, including all signal counting functions</li>
<li>Updating signal counting functions with a <code>blacklist</code> argument, for ignoring reads from user-supplied regions</li>
<li>Added wrappers for <code><a href="../reference/import_bam.html">import_bam()</a></code> for several common use cases</li>
<li>Update <code><a href="../reference/getCountsByPositions.html">getCountsByPositions()</a></code>:
<ul>
<li>Added a <code>melt</code> option for returning melted dataframes</li>
<li>Now returns an error by default if multi-width regions are given (must be explicit)</li>
</ul>
</li>
<li>Changed argument order in <code><a href="../reference/getMaxPositionsBySignal.html">getMaxPositionsBySignal()</a></code>
</li>
</ul>
</div>
    <div id="brgenomics-0-5-6" class="section level2">
<h2 class="hasAnchor">
<a href="#brgenomics-0-5-6" class="anchor"></a>BRGenomics 0.5.6</h2>
<ul>
<li>Update <code><a href="../reference/import_bam.html">import_bam()</a></code> function
<ul>
<li>Added support for paired-end bam files</li>
<li>Added the <code>shift</code> argument</li>
</ul>
</li>
<li>Made <code><a href="../reference/bootstrap-signal-by-position.html">metaSubsample()</a></code> functions robust to unevaluated inputs</li>
<li>Small performance improvement to <code><a href="../reference/genebodies.html">genebodies()</a></code> function</li>
<li>Multicore usage is again the default for <code><a href="../reference/getDESeqResults.html">getDESeqResults()</a></code>
</li>
</ul>
</div>
    <div id="brgenomics-0-5-3" class="section level2">
<h2 class="hasAnchor">
<a href="#brgenomics-0-5-3" class="anchor"></a>BRGenomics 0.5.3</h2>
<ul>
<li>Substantial performance improvement for <code><a href="../reference/mergeGRangesData.html">mergeGRangesData()</a></code>
</li>
<li>Make single-core the default for <code><a href="../reference/getDESeqResults.html">getDESeqResults()</a></code>
</li>
<li>Fixed errant warning message in <code><a href="../reference/binNdimensions.html">binNdimensions()</a></code> for integer inputs</li>
<li>Update namespace to fully import GenomicRanges and S4Vectors</li>
<li>Changed R dependency, evidently required by the updated data objects</li>
<li>Various documentation updates</li>
<li>Added package documentation page</li>
<li>Minor changes to examples</li>
</ul>
</div>
    <div id="brgenomics-0-5-0" class="section level2">
<h2 class="hasAnchor">
<a href="#brgenomics-0-5-0" class="anchor"></a>BRGenomics 0.5.0</h2>
<ul>
<li>Increased support for multiplexed GRanges objects across all functions</li>
<li>Increased performance for bootstrapping over multiple fields/multiplexed GRanges objects</li>
<li>Removed requirement to set ‘field’ argument for gettings counts over multiplexed GRanges</li>
</ul>
</div>
    <div id="brgenomics-0-4-7" class="section level2">
<h2 class="hasAnchor">
<a href="#brgenomics-0-4-7" class="anchor"></a>BRGenomics 0.4.7</h2>
<ul>
<li>Rewrote <code><a href="../reference/mergeGRangesData.html">mergeGRangesData()</a></code> to support the creation of multiplexed GRanges objects</li>
<li>Made <code><a href="../reference/getCountsByRegions.html">getCountsByRegions()</a></code> and <code><a href="../reference/getCountsByPositions.html">getCountsByPositions()</a></code> to return integers if input signal is integer</li>
</ul>
</div>
    <div id="brgenomics-0-4-5" class="section level2">
<h2 class="hasAnchor">
<a href="#brgenomics-0-4-5" class="anchor"></a>BRGenomics 0.4.5</h2>
<ul>
<li>Added and modified numerous arguments</li>
<li>Increased support for normalization factors across signal counting functions</li>
<li>Modified behavior of bootstrapping over multiple fields by removing explicit access to .Random.seed</li>
</ul>
</div>
    <div id="brgenomics-0-4-1" class="section level2">
<h2 class="hasAnchor">
<a href="#brgenomics-0-4-1" class="anchor"></a>BRGenomics 0.4.1</h2>
<ul>
<li>Various documentation updates</li>
</ul>
</div>
    <div id="brgenomics-0-4-0" class="section level2">
<h2 class="hasAnchor">
<a href="#brgenomics-0-4-0" class="anchor"></a>BRGenomics 0.4.0</h2>
<ul>
<li>Added a <code>NEWS.md</code> file to track changes</li>
<li>Added support for bam files</li>
</ul>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <div id="tocnav">
      <h2>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
        <li><a href="#brgenomics-0-99-33">0.99.33</a></li>
        <li><a href="#brgenomics-0-99-30">0.99.30</a></li>
        <li><a href="#brgenomics-0-99-25">0.99.25</a></li>
        <li><a href="#brgenomics-0-99-23">0.99.23</a></li>
        <li><a href="#brgenomics-0-99-15">0.99.15</a></li>
        <li><a href="#brgenomics-0-99-10">0.99.10</a></li>
        <li><a href="#brgenomics-0-99-0">0.99.0</a></li>
        <li><a href="#brgenomics-0-8-1">0.8.1</a></li>
        <li><a href="#brgenomics-0-8-0">0.8.0</a></li>
        <li><a href="#brgenomics-0-7-10">0.7.10</a></li>
        <li><a href="#brgenomics-0-7-8">0.7.8</a></li>
        <li><a href="#brgenomics-0-7-7">0.7.7</a></li>
        <li><a href="#brgenomics-0-7-5">0.7.5</a></li>
        <li><a href="#brgenomics-0-7-0">0.7.0</a></li>
        <li><a href="#brgenomics-0-5-6">0.5.6</a></li>
        <li><a href="#brgenomics-0-5-3">0.5.3</a></li>
        <li><a href="#brgenomics-0-5-0">0.5.0</a></li>
        <li><a href="#brgenomics-0-4-7">0.4.7</a></li>
        <li><a href="#brgenomics-0-4-5">0.4.5</a></li>
        <li><a href="#brgenomics-0-4-1">0.4.1</a></li>
        <li><a href="#brgenomics-0-4-0">0.4.0</a></li>
      </ul>
    </div>
  </div>

</div>


      <footer>
      <div class="copyright">
  <p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


