<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Bootstrapping Mean Signal by Position for Metaplotting — bootstrap-signal-by-position • BRGenomics</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Bootstrapping Mean Signal by Position for Metaplotting — bootstrap-signal-by-position" />
<meta property="og:description" content="These functions perform bootstrap subsampling of mean readcounts at different
positions within regions of interest (metaSubsample), or, in the more
general case of metaSubsampleMatrix, column means of a matrix are
bootstrapped by sampling the rows. Mean signal counts can be calculated at
base-pair resolution, or over larger bins." />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.5.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/BRGenomicsOverview.html">Basepair-Resolution Analysis with BRGenomics</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Bootstrapping Mean Signal by Position for Metaplotting</h1>
    
    <div class="hidden name"><code>bootstrap-signal-by-position.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>These functions perform bootstrap subsampling of mean readcounts at different
positions within regions of interest (<code>metaSubsample</code>), or, in the more
general case of <code>metaSubsampleMatrix</code>, column means of a matrix are
bootstrapped by sampling the rows. Mean signal counts can be calculated at
base-pair resolution, or over larger bins.</p>
    </div>

    <pre class="usage"><span class='fu'>metaSubsample</span>(
  <span class='no'>dataset.gr</span>,
  <span class='no'>regions.gr</span>,
  <span class='kw'>binsize</span> <span class='kw'>=</span> <span class='fl'>1</span>,
  <span class='kw'>first.output.xval</span> <span class='kw'>=</span> <span class='fl'>1</span>,
  <span class='kw'>sample.name</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/deparse.html'>deparse</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/substitute.html'>substitute</a></span>(<span class='no'>dataset.gr</span>)),
  <span class='kw'>n.iter</span> <span class='kw'>=</span> <span class='fl'>1000</span>,
  <span class='kw'>prop.sample</span> <span class='kw'>=</span> <span class='fl'>0.1</span>,
  <span class='kw'>lower</span> <span class='kw'>=</span> <span class='fl'>0.125</span>,
  <span class='kw'>upper</span> <span class='kw'>=</span> <span class='fl'>0.875</span>,
  <span class='kw'>field</span> <span class='kw'>=</span> <span class='st'>"score"</span>,
  <span class='kw'>NF</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>remove.empty</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fu'>detectCores</span>()
)

<span class='fu'>metaSubsampleMatrix</span>(
  <span class='no'>counts.mat</span>,
  <span class='kw'>binsize</span> <span class='kw'>=</span> <span class='fl'>1</span>,
  <span class='kw'>first.output.xval</span> <span class='kw'>=</span> <span class='fl'>1</span>,
  <span class='kw'>sample.name</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>n.iter</span> <span class='kw'>=</span> <span class='fl'>1000</span>,
  <span class='kw'>prop.sample</span> <span class='kw'>=</span> <span class='fl'>0.1</span>,
  <span class='kw'>lower</span> <span class='kw'>=</span> <span class='fl'>0.125</span>,
  <span class='kw'>upper</span> <span class='kw'>=</span> <span class='fl'>0.875</span>,
  <span class='kw'>NF</span> <span class='kw'>=</span> <span class='fl'>1</span>,
  <span class='kw'>remove.empty</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fu'>detectCores</span>()
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>dataset.gr</th>
      <td><p>A GRanges object in which signal is contained in metadata
(typically in the <code>"score"</code> field).</p></td>
    </tr>
    <tr>
      <th>regions.gr</th>
      <td><p>A GRanges object containing intervals over which to
metaplot. All ranges must have the same width.</p></td>
    </tr>
    <tr>
      <th>binsize</th>
      <td><p>The size of bin (in basepairs, or number of columns for
<code>metaSubsampleMatrix</code>) to use for counting signal. Especially
important for counting signal over large or sparse regions.</p></td>
    </tr>
    <tr>
      <th>first.output.xval</th>
      <td><p>The relative start position of the first bin, e.g.
if <code>regions.gr</code> begins at 50 bases upstream of the TSS, set
<code>first.output.xval = -50</code>. This number only affects the x-values that
are returned, which are provided as a convenience.</p></td>
    </tr>
    <tr>
      <th>sample.name</th>
      <td><p>Defaults to the name of the input dataset. This is
included in the output as a convenience, as it allows row-binding outputs
from different samples. If <code>length(field) &gt; 1</code> and the default
<code>sample.name</code> is left, the sample names will be inferred from the
field names.</p></td>
    </tr>
    <tr>
      <th>n.iter</th>
      <td><p>Number of random subsampling iterations to perform. Default is
<code>1000</code>.</p></td>
    </tr>
    <tr>
      <th>prop.sample</th>
      <td><p>The proportion of the ranges in <code>regions.gr</code> (e.g.
the proportion of genes) or the proportion of rows in <code>counts.mat</code> to
sample in each iteration. The default is <code>0.1</code> (10 percent).</p></td>
    </tr>
    <tr>
      <th>lower, upper</th>
      <td><p>The lower and upper quantiles of subsampled signal means
to return. The defaults, <code>0.125</code> and <code>0.875</code> (i.e. the 12.5th and
85.5th percentiles) return a 75 percent confidence interval about the
bootstrapped mean.</p></td>
    </tr>
    <tr>
      <th>field</th>
      <td><p>One or more metadata fields of <code>dataset.gr</code> to be counted.</p></td>
    </tr>
    <tr>
      <th>NF</th>
      <td><p>An optional normalization factor by which to multiply the counts.
If given, <code><a href='https://rdrr.io/r/base/length.html'>length(NF)</a></code> must be equal to <code><a href='https://rdrr.io/r/base/length.html'>length(field)</a></code>.</p></td>
    </tr>
    <tr>
      <th>remove.empty</th>
      <td><p>A logical indicating whether regions
(<code>metaSubsample</code>) or rows (<code>metaSubsampleMatrix</code>) without signal
should be removed from the analysis. Not recommended if using multiple
fields, as the gene lists will no longer be equivalent.</p></td>
    </tr>
    <tr>
      <th>ncores</th>
      <td><p>Number of cores to use for computations.</p></td>
    </tr>
    <tr>
      <th>counts.mat</th>
      <td><p>A matrix over which to bootstrap column means by
subsampling its rows. Typically, a matrix of readcounts with rows for genes
and columns for positions within those genes.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Dataframe containing x-values, means, lower quantiles, upper
  quantiles, and the sample name (as a convenience for row-binding multiple
  of these dataframes). If multiple fields are given, a single dataframe is
  still returned, but will contain data for all fields.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='getCountsByPositions.html'>getCountsByPositions</a></code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='st'>"PROseq"</span>) <span class='co'># import included PROseq data</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='st'>"txs_dm6_chr4"</span>) <span class='co'># import included transcripts</span>

<span class='co'># for each transcript, use promoter-proximal region from TSS to +100</span>
<span class='no'>pr</span> <span class='kw'>&lt;-</span> <span class='fu'>promoters</span>(<span class='no'>txs_dm6_chr4</span>, <span class='fl'>0</span>, <span class='fl'>100</span>)

<span class='co'>#--------------------------------------------------#</span>
<span class='co'># Bootstrap average signal in each 5 bp bin across all transcripts,</span>
<span class='co'># and get confidence bands for middle 30% of bootstrapped means</span>
<span class='co'>#--------------------------------------------------#</span>

<span class='fu'><a href='https://rdrr.io/r/base/Random.html'>set.seed</a></span>(<span class='fl'>11</span>)
<span class='no'>df</span> <span class='kw'>&lt;-</span> <span class='fu'>metaSubsample</span>(<span class='no'>PROseq</span>, <span class='no'>pr</span>, <span class='kw'>binsize</span> <span class='kw'>=</span> <span class='fl'>5</span>,
                    <span class='kw'>lower</span> <span class='kw'>=</span> <span class='fl'>0.35</span>, <span class='kw'>upper</span> <span class='kw'>=</span> <span class='fl'>0.65</span>,
                    <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fl'>2</span>)
<span class='no'>df</span>[<span class='fl'>1</span>:<span class='fl'>10</span>, ]</div><div class='output co'>#&gt;     x       mean      lower      upper sample.name
#&gt; 1   3 0.04117647 0.02941176 0.05882353      PROseq
#&gt; 2   8 0.05294118 0.04117647 0.07058824      PROseq
#&gt; 3  13 0.07058824 0.04705882 0.09411765      PROseq
#&gt; 4  18 0.05882353 0.04117647 0.08235294      PROseq
#&gt; 5  23 0.11764706 0.10000000 0.14705882      PROseq
#&gt; 6  28 0.10000000 0.08823529 0.11176471      PROseq
#&gt; 7  33 0.11764706 0.09411765 0.15294118      PROseq
#&gt; 8  38 0.57058824 0.43529412 0.82941176      PROseq
#&gt; 9  43 0.47058824 0.38235294 0.60205882      PROseq
#&gt; 10 48 0.61470588 0.48823529 0.99617647      PROseq</div><div class='input'>
<span class='co'>#--------------------------------------------------#</span>
<span class='co'># Plot bootstrapped means with confidence intervals</span>
<span class='co'>#--------------------------------------------------#</span>

<span class='fu'><a href='https://rdrr.io/r/graphics/plot.html'>plot</a></span>(<span class='no'>mean</span> ~ <span class='no'>x</span>, <span class='no'>df</span>, <span class='kw'>type</span> <span class='kw'>=</span> <span class='st'>"l"</span>, <span class='kw'>main</span> <span class='kw'>=</span> <span class='st'>"PROseq Signal"</span>,
     <span class='kw'>ylab</span> <span class='kw'>=</span> <span class='st'>"Mean + 30% CI"</span>, <span class='kw'>xlab</span> <span class='kw'>=</span> <span class='st'>"Distance from TSS"</span>)</div><div class='input'><span class='fu'><a href='https://rdrr.io/r/graphics/polygon.html'>polygon</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='no'>df</span>$<span class='no'>x</span>, <span class='fu'><a href='https://rdrr.io/r/base/rev.html'>rev</a></span>(<span class='no'>df</span>$<span class='no'>x</span>)), <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='no'>df</span>$<span class='no'>lower</span>, <span class='fu'><a href='https://rdrr.io/r/base/rev.html'>rev</a></span>(<span class='no'>df</span>$<span class='no'>upper</span>)),
        <span class='kw'>col</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/grDevices/adjustcolor.html'>adjustcolor</a></span>(<span class='st'>"black"</span>, <span class='fl'>0.1</span>), <span class='kw'>border</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)</div><div class='img'><img src='bootstrap-signal-by-position-1.png' alt='' width='700' height='433' /></div><div class='input'>

<span class='co'>#==================================================#</span>
<span class='co'># Using a matrix as input</span>
<span class='co'>#==================================================#</span>

<span class='co'># generate a matrix of counts in each region</span>
<span class='no'>countsmat</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='getCountsByPositions.html'>getCountsByPositions</a></span>(<span class='no'>PROseq</span>, <span class='no'>pr</span>)
<span class='fu'><a href='https://rdrr.io/r/base/dim.html'>dim</a></span>(<span class='no'>countsmat</span>)</div><div class='output co'>#&gt; [1] 339 100</div><div class='input'>
<span class='co'>#--------------------------------------------------#</span>
<span class='co'># bootstrap average signal in 10 bp bins across all transcripts</span>
<span class='co'>#--------------------------------------------------#</span>

<span class='fu'><a href='https://rdrr.io/r/base/Random.html'>set.seed</a></span>(<span class='fl'>11</span>)
 <span class='no'>df</span> <span class='kw'>&lt;-</span> <span class='fu'>metaSubsampleMatrix</span>(<span class='no'>countsmat</span>, <span class='kw'>binsize</span> <span class='kw'>=</span> <span class='fl'>10</span>,
                           <span class='kw'>sample.name</span> <span class='kw'>=</span> <span class='st'>"PROseq"</span>,
                           <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fl'>2</span>)
<span class='no'>df</span>[<span class='fl'>1</span>:<span class='fl'>10</span>, ]</div><div class='output co'>#&gt;       x       mean     lower     upper sample.name
#&gt; 1   5.5  0.4705882 0.2352941  1.147059      PROseq
#&gt; 2  15.5  0.7058824 0.3529412  1.735294      PROseq
#&gt; 3  25.5  1.1470588 0.6764706  2.117647      PROseq
#&gt; 4  35.5  3.6470588 1.7647059 11.419118      PROseq
#&gt; 5  45.5  5.6470588 2.9705882 11.705882      PROseq
#&gt; 6  55.5 10.2794118 5.9705882 15.591912      PROseq
#&gt; 7  65.5  5.1764706 3.2352941  7.352941      PROseq
#&gt; 8  75.5  5.4411765 2.1176471 10.000000      PROseq
#&gt; 9  85.5  3.0588235 2.0588235  4.441176      PROseq
#&gt; 10 95.5  2.8676471 1.6470588  4.558824      PROseq</div><div class='input'>
<span class='co'>#--------------------------------------------------#</span>
<span class='co'># the same, using a normalization factor, and changing the x-values</span>
<span class='co'>#--------------------------------------------------#</span>

<span class='fu'><a href='https://rdrr.io/r/base/Random.html'>set.seed</a></span>(<span class='fl'>11</span>)
<span class='no'>df</span> <span class='kw'>&lt;-</span> <span class='fu'>metaSubsampleMatrix</span>(<span class='no'>countsmat</span>, <span class='kw'>binsize</span> <span class='kw'>=</span> <span class='fl'>10</span>,
                          <span class='kw'>first.output.xval</span> <span class='kw'>=</span> <span class='fl'>0</span>, <span class='kw'>NF</span> <span class='kw'>=</span> <span class='fl'>0.75</span>,
                          <span class='kw'>sample.name</span> <span class='kw'>=</span> <span class='st'>"PROseq"</span>, <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fl'>2</span>)
<span class='no'>df</span>[<span class='fl'>1</span>:<span class='fl'>10</span>, ]</div><div class='output co'>#&gt;       x      mean     lower      upper sample.name
#&gt; 1   4.5 0.3529412 0.1764706  0.8602941      PROseq
#&gt; 2  14.5 0.5294118 0.2647059  1.3014706      PROseq
#&gt; 3  24.5 0.8602941 0.5073529  1.5882353      PROseq
#&gt; 4  34.5 2.7352941 1.3235294  8.5643382      PROseq
#&gt; 5  44.5 4.2352941 2.2279412  8.7794118      PROseq
#&gt; 6  54.5 7.7095588 4.4779412 11.6939338      PROseq
#&gt; 7  64.5 3.8823529 2.4264706  5.5147059      PROseq
#&gt; 8  74.5 4.0808824 1.5882353  7.5000000      PROseq
#&gt; 9  84.5 2.2941176 1.5441176  3.3308824      PROseq
#&gt; 10 94.5 2.1507353 1.2352941  3.4191176      PROseq</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      <li><a href="#value">Value</a></li>
      <li><a href="#see-also">See also</a></li>
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    <p>Mike DeBerardine</p>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


