<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Bootstrapping Mean Signal by Position for Metaplotting — bootstrap-signal-by-position • BRGenomics</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Bootstrapping Mean Signal by Position for Metaplotting — bootstrap-signal-by-position"><meta property="og:description" content="These functions perform bootstrap subsampling of mean readcounts at different
positions within regions of interest (metaSubsample), or, in the more
general case of metaSubsampleMatrix, column means of a matrix are
bootstrapped by sampling the rows. Mean signal counts can be calculated at
base-pair resolution, or over larger bins."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.12.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Overview.html">Overview</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Guides
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/GettingStarted.html">Getting Started</a>
    </li>
    <li>
      <a href="../articles/ImportingModifyingAnnotations.html">Importing &amp; Modifying Annotations</a>
    </li>
    <li>
      <a href="../articles/ImportingProcessingData.html">Importing &amp; Processing Data</a>
    </li>
    <li>
      <a href="../articles/SignalCounting.html">Signal Counting</a>
    </li>
    <li>
      <a href="../articles/AnalyzingMultipleDatasets.html">Analyzing Multiple Datasets</a>
    </li>
    <li>
      <a href="../articles/ProfilePlotsAndBootstrapping.html">Profile Plots &amp; Bootstrapping</a>
    </li>
    <li>
      <a href="../articles/SpikeInNormalization.html">Spike-in Normalization</a>
    </li>
    <li>
      <a href="../articles/DESeq2WithGlobalPerturbations.html">DESeq2 with Global Perturbations</a>
    </li>
    <li>
      <a href="../articles/SequenceExtraction.html">Sequence Extraction</a>
    </li>
  </ul></li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/mdeber/BRGenomics" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Bootstrapping Mean Signal by Position for Metaplotting</h1>
    <small class="dont-index">Source: <a href="https://github.com/mdeber/BRGenomics/blob/HEAD/R/bootstrap_counts.R" class="external-link"><code>R/bootstrap_counts.R</code></a></small>
    <div class="hidden name"><code>bootstrap-signal-by-position.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>These functions perform bootstrap subsampling of mean readcounts at different
positions within regions of interest (<code>metaSubsample</code>), or, in the more
general case of <code>metaSubsampleMatrix</code>, column means of a matrix are
bootstrapped by sampling the rows. Mean signal counts can be calculated at
base-pair resolution, or over larger bins.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">metaSubsample</span><span class="op">(</span></span>
<span>  <span class="va">dataset.gr</span>,</span>
<span>  <span class="va">regions.gr</span>,</span>
<span>  binsize <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  first.output.xval <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  sample.name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/deparse.html" class="external-link">deparse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/substitute.html" class="external-link">substitute</a></span><span class="op">(</span><span class="va">dataset.gr</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  n.iter <span class="op">=</span> <span class="fl">1000L</span>,</span>
<span>  prop.sample <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  lower <span class="op">=</span> <span class="fl">0.125</span>,</span>
<span>  upper <span class="op">=</span> <span class="fl">0.875</span>,</span>
<span>  field <span class="op">=</span> <span class="st">"score"</span>,</span>
<span>  NF <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  remove.empty <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  blacklist <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  zero_blacklisted <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  expand_ranges <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  ncores <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"mc.cores"</span>, <span class="fl">2L</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">metaSubsampleMatrix</span><span class="op">(</span></span>
<span>  <span class="va">counts.mat</span>,</span>
<span>  binsize <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  first.output.xval <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  sample.name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  n.iter <span class="op">=</span> <span class="fl">1000L</span>,</span>
<span>  prop.sample <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  lower <span class="op">=</span> <span class="fl">0.125</span>,</span>
<span>  upper <span class="op">=</span> <span class="fl">0.875</span>,</span>
<span>  NF <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  remove.empty <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  ncores <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"mc.cores"</span>, <span class="fl">2L</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>dataset.gr</dt>
<dd><p>A GRanges object in which signal is contained in metadata
(typically in the <code>"score"</code> field), or a list of such GRanges objects.</p></dd>


<dt>regions.gr</dt>
<dd><p>A GRanges object containing intervals over which to
metaplot. All ranges must have the same width.</p></dd>


<dt>binsize</dt>
<dd><p>The size of bin (in basepairs, or number of columns for
<code>metaSubsampleMatrix</code>) to use for counting signal. Especially
important for counting signal over large or sparse regions.</p></dd>


<dt>first.output.xval</dt>
<dd><p>The relative start position of the first bin, e.g.
if <code>regions.gr</code> begins at 50 bases upstream of the TSS, set
<code>first.output.xval = -50</code>. This number only affects the x-values that
are returned, which are provided as a convenience.</p></dd>


<dt>sample.name</dt>
<dd><p>Defaults to the name of the input dataset. This is
included in the output as a convenience, as it allows row-binding outputs
from different samples. If <code>length(field) &gt; 1</code> and the default
<code>sample.name</code> is left, the sample names will be inferred from the
field names.</p></dd>


<dt>n.iter</dt>
<dd><p>Number of random subsampling iterations to perform. Default is
<code>1000</code>.</p></dd>


<dt>prop.sample</dt>
<dd><p>The proportion of the ranges in <code>regions.gr</code> (e.g.
the proportion of genes) or the proportion of rows in <code>counts.mat</code> to
sample in each iteration. The default is <code>0.1</code> (10 percent).</p></dd>


<dt>lower, upper</dt>
<dd><p>The lower and upper quantiles of subsampled signal means
to return. The defaults, <code>0.125</code> and <code>0.875</code> (i.e. the 12.5th and
85.5th percentiles) return a 75 percent confidence interval about the
bootstrapped mean.</p></dd>


<dt>field</dt>
<dd><p>One or more metadata fields of <code>dataset.gr</code> to be counted.</p></dd>


<dt>NF</dt>
<dd><p>An optional normalization factor by which to multiply the counts.
If given, <code>length(NF)</code> must be equal to <code>length(field)</code>.</p></dd>


<dt>remove.empty</dt>
<dd><p>A logical indicating whether regions
(<code>metaSubsample</code>) or rows (<code>metaSubsampleMatrix</code>) without signal
should be removed from the analysis. Not recommended if using multiple
fields, as the gene lists will no longer be equivalent.</p></dd>


<dt>blacklist</dt>
<dd><p>An optional GRanges object containing regions that should be
excluded from signal counting.</p></dd>


<dt>zero_blacklisted</dt>
<dd><p>When set to <code>FALSE</code> (the default), signal at
blacklisted sites is ignored from computations. If set to <code>TRUE</code>,
signal at blacklisted sites will be treated as equal to zero. For
bootstrapping, the default behavior of ignoring signal at blacklisted sites
should almost always be kept.</p></dd>


<dt>expand_ranges</dt>
<dd><p>Logical indicating if ranges in <code>dataset.gr</code> should
be treated as descriptions of single molecules (<code>FALSE</code>), or if ranges
should be treated as representing multiple adjacent positions with the same
signal (<code>TRUE</code>). See <code><a href="getCountsByRegions.html">
getCountsByRegions</a></code>.</p></dd>


<dt>ncores</dt>
<dd><p>Number of cores to use for computations.</p></dd>


<dt>counts.mat</dt>
<dd><p>A matrix over which to bootstrap column means by
subsampling its rows. Typically, a matrix of readcounts with rows for genes
and columns for positions within those genes.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A dataframe containing x-values, means, lower quantiles, upper
  quantiles, and the sample name (as a convenience for row-binding multiple
  of these dataframes). If a list of GRanges is given as input, or if
  multiple fields are given, a single, combined dataframe is returned
  containing data for all fields/datasets.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="getCountsByPositions.html">getCountsByPositions</a></code></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Mike DeBerardine</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"PROseq"</span><span class="op">)</span> <span class="co"># import included PROseq data</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"txs_dm6_chr4"</span><span class="op">)</span> <span class="co"># import included transcripts</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># for each transcript, use promoter-proximal region from TSS to +100</span></span></span>
<span class="r-in"><span><span class="va">pr</span> <span class="op">&lt;-</span> <span class="fu">promoters</span><span class="op">(</span><span class="va">txs_dm6_chr4</span>, <span class="fl">0</span>, <span class="fl">100</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># Bootstrap average signal in each 5 bp bin across all transcripts,</span></span></span>
<span class="r-in"><span><span class="co"># and get confidence bands for middle 30% of bootstrapped means</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">11</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu">metaSubsample</span><span class="op">(</span><span class="va">PROseq</span>, <span class="va">pr</span>, binsize <span class="op">=</span> <span class="fl">5</span>,</span></span>
<span class="r-in"><span>                    lower <span class="op">=</span> <span class="fl">0.35</span>, upper <span class="op">=</span> <span class="fl">0.65</span>,</span></span>
<span class="r-in"><span>                    ncores <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">df</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     x       mean      lower      upper sample.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1   3 0.03529412 0.02941176 0.04705882      PROseq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2   8 0.05882353 0.04117647 0.07058824      PROseq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3  13 0.07647059 0.05294118 0.10000000      PROseq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4  18 0.07058824 0.05294118 0.09029412      PROseq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5  23 0.11764706 0.09411765 0.15294118      PROseq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6  28 0.09411765 0.08235294 0.10588235      PROseq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7  33 0.17058824 0.14117647 0.21764706      PROseq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8  38 0.63529412 0.46470588 0.92176471      PROseq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9  43 0.48235294 0.37058824 0.61176471      PROseq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10 48 0.83529412 0.66470588 1.11382353      PROseq</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># Plot bootstrapped means with confidence intervals</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">mean</span> <span class="op">~</span> <span class="va">x</span>, <span class="va">df</span>, type <span class="op">=</span> <span class="st">"l"</span>, main <span class="op">=</span> <span class="st">"PROseq Signal"</span>,</span></span>
<span class="r-in"><span>     ylab <span class="op">=</span> <span class="st">"Mean + 30% CI"</span>, xlab <span class="op">=</span> <span class="st">"Distance from TSS"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/graphics/polygon.html" class="external-link">polygon</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">x</span>, <span class="fu"><a href="https://rdrr.io/r/base/rev.html" class="external-link">rev</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">x</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">lower</span>, <span class="fu"><a href="https://rdrr.io/r/base/rev.html" class="external-link">rev</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">upper</span><span class="op">)</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>        col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/adjustcolor.html" class="external-link">adjustcolor</a></span><span class="op">(</span><span class="st">"black"</span>, <span class="fl">0.1</span><span class="op">)</span>, border <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="bootstrap-signal-by-position-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#==================================================#</span></span></span>
<span class="r-in"><span><span class="co"># Using a matrix as input</span></span></span>
<span class="r-in"><span><span class="co">#==================================================#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># generate a matrix of counts in each region</span></span></span>
<span class="r-in"><span><span class="va">countsmat</span> <span class="op">&lt;-</span> <span class="fu"><a href="getCountsByPositions.html">getCountsByPositions</a></span><span class="op">(</span><span class="va">PROseq</span>, <span class="va">pr</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">countsmat</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 339 100</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># bootstrap average signal in 10 bp bins across all transcripts</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">11</span><span class="op">)</span></span></span>
<span class="r-in"><span> <span class="va">df</span> <span class="op">&lt;-</span> <span class="fu">metaSubsampleMatrix</span><span class="op">(</span><span class="va">countsmat</span>, binsize <span class="op">=</span> <span class="fl">10</span>,</span></span>
<span class="r-in"><span>                           sample.name <span class="op">=</span> <span class="st">"PROseq"</span>,</span></span>
<span class="r-in"><span>                           ncores <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">df</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       x      mean     lower     upper sample.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1   5.5 0.4411765 0.2352941  1.117647      PROseq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2  15.5 0.7941176 0.3823529  1.882353      PROseq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3  25.5 1.1176471 0.6764706  2.088235      PROseq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4  35.5 4.2205882 1.9411765 12.091912      PROseq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5  45.5 6.8529412 3.2352941 12.882353      PROseq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6  55.5 9.5588235 5.7022059 14.617647      PROseq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7  65.5 4.9411765 3.2058824  7.000000      PROseq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8  75.5 5.4558824 2.0294118  9.827206      PROseq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9  85.5 3.0882353 2.0882353  4.827206      PROseq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10 95.5 2.5882353 1.5294118  4.003676      PROseq</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># the same, using a normalization factor, and changing the x-values</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">11</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu">metaSubsampleMatrix</span><span class="op">(</span><span class="va">countsmat</span>, binsize <span class="op">=</span> <span class="fl">10</span>,</span></span>
<span class="r-in"><span>                          first.output.xval <span class="op">=</span> <span class="fl">0</span>, NF <span class="op">=</span> <span class="fl">0.75</span>,</span></span>
<span class="r-in"><span>                          sample.name <span class="op">=</span> <span class="st">"PROseq"</span>, ncores <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">df</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       x      mean     lower      upper sample.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1   4.5 0.3308824 0.1764706  0.8382353      PROseq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2  14.5 0.5955882 0.2867647  1.4117647      PROseq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3  24.5 0.8382353 0.5073529  1.5661765      PROseq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4  34.5 3.1654412 1.4558824  9.0689338      PROseq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5  44.5 5.1397059 2.4264706  9.6617647      PROseq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6  54.5 7.1691176 4.2766544 10.9632353      PROseq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7  64.5 3.7058824 2.4044118  5.2500000      PROseq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8  74.5 4.0919118 1.5220588  7.3704044      PROseq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9  84.5 2.3161765 1.5661765  3.6204044      PROseq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10 94.5 1.9411765 1.1470588  3.0027574      PROseq</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

