<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Extract Genebodies — genebodies • BRGenomics</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Extract Genebodies — genebodies" />
<meta property="og:description" content="This function returns ranges that are defined relative to the strand-specific
start and end sites of regions of interest (usually genes). Unlike
GenomicRanges::promoters, distances
can be upstream or downstream based on the sign, and both the start and end
of the returned regions can be defined in terms of either the start of end
site of the input ranges. For example, genebodies(txs, -50, 150,
fix.end = &quot;start&quot;) is equivalent to promoters(txs, 50, 150). The
default arguments return ranges that begin 300 bases downstream of the
original start positions, and end 300 bases upstream of the original end
positions." />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Extract Genebodies</h1>
    
    <div class="hidden name"><code>genebodies.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function returns ranges that are defined relative to the strand-specific
start and end sites of regions of interest (usually genes). Unlike
<code><a href='https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html'>GenomicRanges::promoters</a></code>, distances
can be upstream or downstream based on the sign, and both the start and end
of the returned regions can be defined in terms of either the start of end
site of the input ranges. For example, <code>genebodies(txs, -50, 150,
fix.end = "start")</code> is equivalent to <code>promoters(txs, 50, 150)</code>. The
default arguments return ranges that begin 300 bases downstream of the
original start positions, and end 300 bases upstream of the original end
positions.</p>
    </div>

    <pre class="usage"><span class='fu'>genebodies</span>(
  <span class='no'>genelist</span>,
  <span class='kw'>start</span> <span class='kw'>=</span> <span class='fl'>300</span>,
  <span class='kw'>end</span> <span class='kw'>=</span> -<span class='fl'>300</span>,
  <span class='kw'>fix.start</span> <span class='kw'>=</span> <span class='st'>"start"</span>,
  <span class='kw'>fix.end</span> <span class='kw'>=</span> <span class='st'>"end"</span>,
  <span class='kw'>min.window</span> <span class='kw'>=</span> <span class='fl'>0</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>genelist</th>
      <td><p>A GRanges object containing genes of interest.</p></td>
    </tr>
    <tr>
      <th>start</th>
      <td><p>Depending on <code>fix.start</code>, the distance from either the
strand-specific start or end site to begin the returned ranges. If
positive, the returned range will begin downstream of the reference
position; negative numbers are used to return sites upstream of the
reference. Set <code>start = 0</code> to return the reference position.</p></td>
    </tr>
    <tr>
      <th>end</th>
      <td><p>Identical to the <code>start</code> argument, but defines the
strand-specific end position of returned ranges. <code>end</code> must be
downstream of <code>start</code>.</p></td>
    </tr>
    <tr>
      <th>fix.start</th>
      <td><p>The reference point to use for defining the strand-specific
start positions of returned ranges, either <code>"start"</code> or <code>"end"</code>.</p></td>
    </tr>
    <tr>
      <th>fix.end</th>
      <td><p>The reference point to use for defining the strand-specific
end positions of returned ranges, either <code>"start"</code> or <code>"end"</code>.
Cannot be set to <code>"start"</code> if <code>fix.start = "end"</code>.</p></td>
    </tr>
    <tr>
      <th>min.window</th>
      <td><p>When <code>fix.start = "start"</code> and <code>fix.end = "end"</code>,
<code>min.window</code> defines the minimum size (width) of a returned range.
However, when <code>fix.end = fix.start</code>, all returned ranges have the same
width, and <code>min.window</code> simply size-filters the input ranges.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A GRanges object that may be shorter than <code>genelist</code> due to loss
  of short ranges.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html'>intra-range-methods</a></code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='st'>"txs_dm6_chr4"</span>) <span class='co'># load included transcript data</span>
<span class='no'>len</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span>(<span class='no'>txs_dm6_chr4</span>)
<span class='no'>txs</span> <span class='kw'>&lt;-</span> <span class='no'>txs_dm6_chr4</span>[<span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fl'>1</span>:<span class='fl'>2</span>, <span class='no'>len</span> - <span class='fl'>1</span>, <span class='no'>len</span>)]
<span class='no'>txs</span></div><div class='output co'>#&gt; GRanges object with 4 ranges and 2 metadata columns:
#&gt;       seqnames          ranges strand |     tx_name     gene_id
#&gt;          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
#&gt;   [1]     chr4        879-5039      + | FBtr0346692 FBgn0267363
#&gt;   [2]     chr4     42774-43374      + | FBtr0344900 FBgn0266617
#&gt;   [3]     chr4 1225737-1230713      - | FBtr0100402 FBgn0027101
#&gt;   [4]     chr4 1225737-1230713      - | FBtr0100404 FBgn0027101
#&gt;   -------
#&gt;   seqinfo: 7 sequences from dm6 genome</div><div class='input'>
<span class='co'>#--------------------------------------------------#</span>
<span class='co'># genebodies from +300 (300 bp after TSS) to 300 bp before the poly-A site</span>
<span class='co'>#--------------------------------------------------#</span>

<span class='fu'>genebodies</span>(<span class='no'>txs</span>, <span class='fl'>300</span>, -<span class='fl'>300</span>)</div><div class='output co'>#&gt; GRanges object with 4 ranges and 2 metadata columns:
#&gt;       seqnames          ranges strand |     tx_name     gene_id
#&gt;          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
#&gt;   [1]     chr4       1179-4739      + | FBtr0346692 FBgn0267363
#&gt;   [2]     chr4           43074      + | FBtr0344900 FBgn0266617
#&gt;   [3]     chr4 1226037-1230413      - | FBtr0100402 FBgn0027101
#&gt;   [4]     chr4 1226037-1230413      - | FBtr0100404 FBgn0027101
#&gt;   -------
#&gt;   seqinfo: 7 sequences from dm6 genome</div><div class='input'>
<span class='co'>#--------------------------------------------------#</span>
<span class='co'># promoters from -50 to +100</span>
<span class='co'>#--------------------------------------------------#</span>

<span class='fu'>promoters</span>(<span class='no'>txs</span>, <span class='fl'>50</span>, <span class='fl'>100</span>)</div><div class='output co'>#&gt; GRanges object with 4 ranges and 2 metadata columns:
#&gt;       seqnames          ranges strand |     tx_name     gene_id
#&gt;          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
#&gt;   [1]     chr4         829-978      + | FBtr0346692 FBgn0267363
#&gt;   [2]     chr4     42724-42873      + | FBtr0344900 FBgn0266617
#&gt;   [3]     chr4 1230614-1230763      - | FBtr0100402 FBgn0027101
#&gt;   [4]     chr4 1230614-1230763      - | FBtr0100404 FBgn0027101
#&gt;   -------
#&gt;   seqinfo: 7 sequences from dm6 genome</div><div class='input'>
<span class='fu'>genebodies</span>(<span class='no'>txs</span>, -<span class='fl'>50</span>, <span class='fl'>100</span>, <span class='kw'>fix.end</span> <span class='kw'>=</span> <span class='st'>"start"</span>)</div><div class='output co'>#&gt; GRanges object with 4 ranges and 2 metadata columns:
#&gt;       seqnames          ranges strand |     tx_name     gene_id
#&gt;          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
#&gt;   [1]     chr4         829-979      + | FBtr0346692 FBgn0267363
#&gt;   [2]     chr4     42724-42874      + | FBtr0344900 FBgn0266617
#&gt;   [3]     chr4 1230613-1230763      - | FBtr0100402 FBgn0027101
#&gt;   [4]     chr4 1230613-1230763      - | FBtr0100404 FBgn0027101
#&gt;   -------
#&gt;   seqinfo: 7 sequences from dm6 genome</div><div class='input'>
<span class='co'>#--------------------------------------------------#</span>
<span class='co'># region from 500 to 1000 bases after the poly-A site</span>
<span class='co'>#--------------------------------------------------#</span>

<span class='fu'>genebodies</span>(<span class='no'>txs</span>, <span class='fl'>500</span>, <span class='fl'>1000</span>, <span class='kw'>fix.start</span> <span class='kw'>=</span> <span class='st'>"end"</span>)</div><div class='output co'>#&gt; GRanges object with 4 ranges and 2 metadata columns:
#&gt;       seqnames          ranges strand |     tx_name     gene_id
#&gt;          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
#&gt;   [1]     chr4       5539-6039      + | FBtr0346692 FBgn0267363
#&gt;   [2]     chr4     43874-44374      + | FBtr0344900 FBgn0266617
#&gt;   [3]     chr4 1224737-1225237      - | FBtr0100402 FBgn0027101
#&gt;   [4]     chr4 1224737-1225237      - | FBtr0100404 FBgn0027101
#&gt;   -------
#&gt;   seqinfo: 7 sequences from dm6 genome</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      <li><a href="#value">Value</a></li>
      <li><a href="#see-also">See also</a></li>
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    <p>Mike DeBerardine</p>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


