<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Extract Genebodies — genebodies • BRGenomics</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Extract Genebodies — genebodies"><meta property="og:description" content="This function returns ranges that are defined relative to the strand-specific
start and end sites of regions of interest (usually genes)."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.11.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Overview.html">Overview</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Guides
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/GettingStarted.html">Getting Started</a>
    </li>
    <li>
      <a href="../articles/ImportingModifyingAnnotations.html">Importing &amp; Modifying Annotations</a>
    </li>
    <li>
      <a href="../articles/ImportingProcessingData.html">Importing &amp; Processing Data</a>
    </li>
    <li>
      <a href="../articles/SignalCounting.html">Signal Counting</a>
    </li>
    <li>
      <a href="../articles/AnalyzingMultipleDatasets.html">Analyzing Multiple Datasets</a>
    </li>
    <li>
      <a href="../articles/ProfilePlotsAndBootstrapping.html">Profile Plots &amp; Bootstrapping</a>
    </li>
    <li>
      <a href="../articles/SpikeInNormalization.html">Spike-in Normalization</a>
    </li>
    <li>
      <a href="../articles/DESeq2WithGlobalPerturbations.html">DESeq2 with Global Perturbations</a>
    </li>
    <li>
      <a href="../articles/SequenceExtraction.html">Sequence Extraction</a>
    </li>
  </ul></li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/mdeber/BRGenomics" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Extract Genebodies</h1>
    <small class="dont-index">Source: <a href="https://github.com/mdeber/BRGenomics/blob/HEAD/R/roi_functions.R" class="external-link"><code>R/roi_functions.R</code></a></small>
    <div class="hidden name"><code>genebodies.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function returns ranges that are defined relative to the strand-specific
start and end sites of regions of interest (usually genes).</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">genebodies</span><span class="op">(</span></span>
<span>  <span class="va">genelist</span>,</span>
<span>  start <span class="op">=</span> <span class="fl">300L</span>,</span>
<span>  end <span class="op">=</span> <span class="op">-</span><span class="fl">300L</span>,</span>
<span>  fix.start <span class="op">=</span> <span class="st">"start"</span>,</span>
<span>  fix.end <span class="op">=</span> <span class="st">"end"</span>,</span>
<span>  min.window <span class="op">=</span> <span class="fl">0L</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>genelist</dt>
<dd><p>A GRanges object containing genes of interest.</p></dd>


<dt>start</dt>
<dd><p>Depending on <code>fix.start</code>, the distance from either the
strand-specific start or end site to begin the returned ranges. If
positive, the returned range will begin downstream of the reference
position; negative numbers are used to return sites upstream of the
reference. Set <code>start = 0</code> to return the reference position.</p></dd>


<dt>end</dt>
<dd><p>Identical to the <code>start</code> argument, but defines the
strand-specific end position of returned ranges. <code>end</code> must be
downstream of <code>start</code>.</p></dd>


<dt>fix.start</dt>
<dd><p>The reference point to use for defining the strand-specific
start positions of returned ranges, either <code>"start"</code> or <code>"end"</code>.</p></dd>


<dt>fix.end</dt>
<dd><p>The reference point to use for defining the strand-specific
end positions of returned ranges, either <code>"start"</code> or <code>"end"</code>.
Cannot be set to <code>"start"</code> if <code>fix.start = "end"</code>.</p></dd>


<dt>min.window</dt>
<dd><p>When <code>fix.start = "start"</code> and <code>fix.end = "end"</code>,
<code>min.window</code> defines the minimum size (width) of a returned range.
However, when <code>fix.end = fix.start</code>, all returned ranges have the same
width, and <code>min.window</code> simply size-filters the input ranges.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A GRanges object that may be shorter than <code>genelist</code> due to
  filtering of short ranges. For example, using the default arguments, genes
  shorter than 600 bp would be removed.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Unlike
  <code><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">GenomicRanges::promoters</a></code>,
  distances can be defined to be upstream or downstream by changing the sign
  of the argument, and both the start and end of the returned regions can be
  defined in terms of the strand-specific start or end site of the input
  ranges. For example, <code>genebodies(txs, -50, 150, fix.end = "start")</code> is
  equivalent to <code>promoters(txs, 50, 151)</code> (the downstream edge is off by
  1 because <code>promoters</code> keeps the downstream interval closed). The
  default arguments return ranges that begin 300 bases downstream of the
  original start positions, and end 300 bases upstream of the original end
  positions.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">intra-range-methods</a></code></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Mike DeBerardine</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"txs_dm6_chr4"</span><span class="op">)</span> <span class="co"># load included transcript data</span></span></span>
<span class="r-in"><span><span class="va">txs</span> <span class="op">&lt;-</span> <span class="va">txs_dm6_chr4</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">167</span>, <span class="fl">168</span><span class="op">)</span><span class="op">]</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">txs</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 4 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames      ranges strand |     tx_name     gene_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr4    879-5039      + | FBtr0346692 FBgn0267363</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr4 42774-43374      + | FBtr0344900 FBgn0266617</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chr4  5829-11765      - | FBtr0333684 FBgn0052011</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chr4  6374-11765      - | FBtr0089182 FBgn0052011</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 7 sequences from dm6 genome</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># genebody regions from 300 bp after the TSS to</span></span></span>
<span class="r-in"><span><span class="co"># 300 bp before the polyA site</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">genebodies</span><span class="op">(</span><span class="va">txs</span>, <span class="fl">300</span>, <span class="op">-</span><span class="fl">300</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 4 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames     ranges strand |     tx_name     gene_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt;  &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr4  1179-4739      + | FBtr0346692 FBgn0267363</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr4      43074      + | FBtr0344900 FBgn0266617</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chr4 6129-11465      - | FBtr0333684 FBgn0052011</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chr4 6674-11465      - | FBtr0089182 FBgn0052011</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 7 sequences from dm6 genome</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># promoter-proximal region from 50 bp upstream of</span></span></span>
<span class="r-in"><span><span class="co"># the TSS to 100 bp downstream of the TSS</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">promoters</span><span class="op">(</span><span class="va">txs</span>, <span class="fl">50</span>, <span class="fl">101</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 4 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames      ranges strand |     tx_name     gene_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr4     829-979      + | FBtr0346692 FBgn0267363</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr4 42724-42874      + | FBtr0344900 FBgn0266617</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chr4 11665-11815      - | FBtr0333684 FBgn0052011</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chr4 11665-11815      - | FBtr0089182 FBgn0052011</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 7 sequences from dm6 genome</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">genebodies</span><span class="op">(</span><span class="va">txs</span>, <span class="op">-</span><span class="fl">50</span>, <span class="fl">100</span>, fix.end <span class="op">=</span> <span class="st">"start"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 4 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames      ranges strand |     tx_name     gene_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr4     829-979      + | FBtr0346692 FBgn0267363</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr4 42724-42874      + | FBtr0344900 FBgn0266617</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chr4 11665-11815      - | FBtr0333684 FBgn0052011</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chr4 11665-11815      - | FBtr0089182 FBgn0052011</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 7 sequences from dm6 genome</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># region from 500 to 1000 bp after the polyA site</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">genebodies</span><span class="op">(</span><span class="va">txs</span>, <span class="fl">500</span>, <span class="fl">1000</span>, fix.start <span class="op">=</span> <span class="st">"end"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 4 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames      ranges strand |     tx_name     gene_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr4   5539-6039      + | FBtr0346692 FBgn0267363</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr4 43874-44374      + | FBtr0344900 FBgn0266617</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chr4   4829-5329      - | FBtr0333684 FBgn0052011</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chr4   5374-5874      - | FBtr0089182 FBgn0052011</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 7 sequences from dm6 genome</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

