<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Get signal counts at each position within regions of interest — getCountsByPositions • BRGenomics</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Get signal counts at each position within regions of interest — getCountsByPositions"><meta property="og:description" content="Get the sum of the signal in dataset.gr that overlaps each position
within each range in regions.gr. If binning is used (i.e. positions
are wider than 1 bp), any function can be used to summarize the signal
overlapping each bin. For a description of the critical difference between
expand_ranges = FALSE and expand_ranges = TRUE, see
getCountsByRegions."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.11.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Overview.html">Overview</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Guides
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/GettingStarted.html">Getting Started</a>
    </li>
    <li>
      <a href="../articles/ImportingModifyingAnnotations.html">Importing &amp; Modifying Annotations</a>
    </li>
    <li>
      <a href="../articles/ImportingProcessingData.html">Importing &amp; Processing Data</a>
    </li>
    <li>
      <a href="../articles/SignalCounting.html">Signal Counting</a>
    </li>
    <li>
      <a href="../articles/AnalyzingMultipleDatasets.html">Analyzing Multiple Datasets</a>
    </li>
    <li>
      <a href="../articles/ProfilePlotsAndBootstrapping.html">Profile Plots &amp; Bootstrapping</a>
    </li>
    <li>
      <a href="../articles/SpikeInNormalization.html">Spike-in Normalization</a>
    </li>
    <li>
      <a href="../articles/DESeq2WithGlobalPerturbations.html">DESeq2 with Global Perturbations</a>
    </li>
    <li>
      <a href="../articles/SequenceExtraction.html">Sequence Extraction</a>
    </li>
  </ul></li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/mdeber/BRGenomics" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Get signal counts at each position within regions of interest</h1>
    <small class="dont-index">Source: <a href="https://github.com/mdeber/BRGenomics/blob/HEAD/R/signal_counting.R" class="external-link"><code>R/signal_counting.R</code></a></small>
    <div class="hidden name"><code>getCountsByPositions.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Get the sum of the signal in <code>dataset.gr</code> that overlaps each position
within each range in <code>regions.gr</code>. If binning is used (i.e. positions
are wider than 1 bp), any function can be used to summarize the signal
overlapping each bin. For a description of the critical difference between
<code>expand_ranges = FALSE</code> and <code>expand_ranges = TRUE</code>, see
<code><a href="getCountsByRegions.html">getCountsByRegions</a></code>.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">getCountsByPositions</span><span class="op">(</span></span>
<span>  <span class="va">dataset.gr</span>,</span>
<span>  <span class="va">regions.gr</span>,</span>
<span>  binsize <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  FUN <span class="op">=</span> <span class="va">sum</span>,</span>
<span>  simplify.multi.widths <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"error"</span>, <span class="st">"list"</span>, <span class="st">"pad 0"</span>, <span class="st">"pad NA"</span><span class="op">)</span>,</span>
<span>  field <span class="op">=</span> <span class="st">"score"</span>,</span>
<span>  NF <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  blacklist <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  NA_blacklisted <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  melt <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  expand_ranges <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  ncores <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"mc.cores"</span>, <span class="fl">2L</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>dataset.gr</dt>
<dd><p>A GRanges object in which signal is contained in metadata
(typically in the "score" field), or a named list of such GRanges objects.</p></dd>


<dt>regions.gr</dt>
<dd><p>A GRanges object containing regions of interest.</p></dd>


<dt>binsize</dt>
<dd><p>Size of bins (in bp) to use for counting within each range of
<code>regions.gr</code>. Note that counts will <em>not</em> be length-normalized.</p></dd>


<dt>FUN</dt>
<dd><p>If <code>binsize &gt; 1</code>, the function used to aggregate the signal
within each bin. By default, the signal is summed, but any function
operating on a numeric vector can be used.</p></dd>


<dt>simplify.multi.widths</dt>
<dd><p>A string indicating the output format if the
ranges in <code>regions.gr</code> have variable widths. By default, an error is
returned. See details below.</p></dd>


<dt>field</dt>
<dd><p>The metadata field of <code>dataset.gr</code> to be counted. If
<code>length(field) &gt; 1</code>, the output is a list whose elements contain the
output for generated each field. If <code>field</code> not found in
<code>names(mcols(dataset.gr))</code>, will default to using all fields found in
<code>dataset.gr</code>.</p></dd>


<dt>NF</dt>
<dd><p>An optional normalization factor by which to multiply the counts.
If given, <code>length(NF)</code> must be equal to <code>length(field)</code>.</p></dd>


<dt>blacklist</dt>
<dd><p>An optional GRanges object containing regions that should be
excluded from signal counting.</p></dd>


<dt>NA_blacklisted</dt>
<dd><p>A logical indicating if NA values should be returned
for blacklisted regions. By default, signal in the blacklisted sites is
ignored, i.e. the reads are excluded. If <code>NA_blacklisted = TRUE</code>,
those positions are set to <code>NA</code> in the final output.</p></dd>


<dt>melt</dt>
<dd><p>A logical indicating if the count matrices should be melted. If
set to <code>TRUE</code>, a dataframe is returned in containing columns for
"region", "position", and "signal". If <code>dataset.gr</code> is a list of
multiple GRanges, or if <code>length(field) &gt; 1</code>, a single dataframe is
returned, which contains an additional column "sample", which contains
individual sample names. If used with multi-width <code>regions.gr</code>, the
resulting dataframe will only contain positions that are found within each
respective region.</p></dd>


<dt>expand_ranges</dt>
<dd><p>Logical indicating if ranges in <code>dataset.gr</code> should
be treated as descriptions of single molecules (<code>FALSE</code>), or if ranges
should be treated as representing multiple adjacent positions with the same
signal (<code>TRUE</code>). See <code><a href="getCountsByRegions.html">
getCountsByRegions</a></code>.</p></dd>


<dt>ncores</dt>
<dd><p>Multiple cores will only be used if <code>dataset.gr</code> is a list
of multiple datasets, or if <code>length(field) &gt; 1</code>.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>If the widths of all ranges in <code>regions.gr</code> are equal, a matrix
  is returned that contains a row for each region of interest, and a column
  for each position (each base if <code>binsize = 1</code>) within each region. If</p>
<p></p>
<p><code>dataset.gr</code> is a list, a parallel list is returned containing a
  matrix for each input dataset.</p>
    </div>
    <div id="use-of-multi-width-regions-of-interest">
    <h2>Use of multi-width regions of interest</h2>
    <p>If the input
  <code>regions.gr</code> contains ranges of varying widths, setting
  <code>simplify.multi.widths = "list"</code> will output a list of variable-length
  vectors, with each vector corresponding to an individual input region. If
  <code>simplify.multi.widths = "pad 0"</code> or <code>"pad NA"</code>, the output is a
  matrix containing a row for each range in <code>regions.gr</code>, but the number
  of columns is determined by the largest range in <code>regions.gr</code>. For
  each region of interest, columns that correspond to positions outside of
  the input range are set, depending on the argument, to <code>0</code> or
  <code>NA</code>.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="getCountsByRegions.html">getCountsByRegions</a></code>,
  <code><a href="bootstrap-signal-by-position.html">metaSubsample</a></code></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Mike DeBerardine</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"PROseq"</span><span class="op">)</span> <span class="co"># load included PROseq data</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"txs_dm6_chr4"</span><span class="op">)</span> <span class="co"># load included transcripts</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># counts from 0 to 50 bp after the TSS</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">txs_pr</span> <span class="op">&lt;-</span> <span class="fu">promoters</span><span class="op">(</span><span class="va">txs_dm6_chr4</span>, <span class="fl">0</span>, <span class="fl">50</span><span class="op">)</span> <span class="co"># first 50 bases</span></span></span>
<span class="r-in"><span><span class="va">countsmat</span> <span class="op">&lt;-</span> <span class="fu">getCountsByPositions</span><span class="op">(</span><span class="va">PROseq</span>, <span class="va">txs_pr</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">countsmat</span><span class="op">[</span><span class="fl">10</span><span class="op">:</span><span class="fl">15</span>, <span class="fl">41</span><span class="op">:</span><span class="fl">50</span><span class="op">]</span> <span class="co"># show only 41-50 bp after TSS</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      [,1] [,2] [,3] [,4] [,5] [,6] [,7] [,8] [,9] [,10]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1,]    0    1    0    0    0    0    0    0    0     0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2,]    1    0    0    4    0    2    2    0    1     3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [3,]    1    0    0    4    0    2    2    0    1     3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [4,]    0    0    0    0    0    0    0    0    0     0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [5,]    0    0    0    0    0    0    0    0    0     0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [6,]    0    0    0    5    2    1    6    2    0     5</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># redo with 10 bp bins from 0 to 100</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># column 5 is sums of rows shown above</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">txs_pr</span> <span class="op">&lt;-</span> <span class="fu">promoters</span><span class="op">(</span><span class="va">txs_dm6_chr4</span>, <span class="fl">0</span>, <span class="fl">100</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">countsmat</span> <span class="op">&lt;-</span> <span class="fu">getCountsByPositions</span><span class="op">(</span><span class="va">PROseq</span>, <span class="va">txs_pr</span>, binsize <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">countsmat</span><span class="op">[</span><span class="fl">10</span><span class="op">:</span><span class="fl">15</span>, <span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      [,1] [,2] [,3] [,4] [,5] [,6] [,7] [,8] [,9] [,10]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1,]    0    0    0    0    1   12   19  160    6    12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2,]    2    0    0   21   13    9    6    8    7     0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [3,]    2    0    0   21   13    9    6    8    7     0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [4,]    4    3    3    0    0    0    2    0    1     0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [5,]    2    1    5    1    0    1    0    1    3     0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [6,]    0    0    0    3   21   23    5    0    1     0</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># same as the above, but with the average signal in each bin</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">countsmat</span> <span class="op">&lt;-</span> <span class="fu">getCountsByPositions</span><span class="op">(</span><span class="va">PROseq</span>, <span class="va">txs_pr</span>, binsize <span class="op">=</span> <span class="fl">10</span>, FUN <span class="op">=</span> <span class="va">mean</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">countsmat</span><span class="op">[</span><span class="fl">10</span><span class="op">:</span><span class="fl">15</span>, <span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      [,1] [,2] [,3] [,4] [,5] [,6] [,7] [,8] [,9] [,10]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1,]  0.0  0.0  0.0  0.0  0.1  1.2  1.9 16.0  0.6   1.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2,]  0.2  0.0  0.0  2.1  1.3  0.9  0.6  0.8  0.7   0.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [3,]  0.2  0.0  0.0  2.1  1.3  0.9  0.6  0.8  0.7   0.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [4,]  0.4  0.3  0.3  0.0  0.0  0.0  0.2  0.0  0.1   0.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [5,]  0.2  0.1  0.5  0.1  0.0  0.1  0.0  0.1  0.3   0.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [6,]  0.0  0.0  0.0  0.3  2.1  2.3  0.5  0.0  0.1   0.0</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># standard deviation of signal in each bin</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">countsmat</span> <span class="op">&lt;-</span> <span class="fu">getCountsByPositions</span><span class="op">(</span><span class="va">PROseq</span>, <span class="va">txs_pr</span>, binsize <span class="op">=</span> <span class="fl">10</span>, FUN <span class="op">=</span> <span class="va">sd</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">countsmat</span><span class="op">[</span><span class="fl">10</span><span class="op">:</span><span class="fl">15</span>, <span class="op">]</span>, <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      [,1] [,2] [,3] [,4] [,5] [,6] [,7] [,8] [,9] [,10]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1,]  0.0  0.0  0.0  0.0  0.3  3.8  3.8 38.0  0.8   0.8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2,]  0.4  0.0  0.0  5.3  1.4  0.9  1.0  1.6  1.1   0.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [3,]  0.4  0.0  0.0  5.3  1.4  0.9  1.0  1.6  1.1   0.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [4,]  0.5  0.7  0.7  0.0  0.0  0.0  0.6  0.0  0.3   0.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [5,]  0.4  0.3  0.7  0.3  0.0  0.3  0.0  0.3  0.5   0.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [6,]  0.0  0.0  0.0  0.5  2.4  2.4  1.1  0.0  0.3   0.0</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

