<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Get signal counts in regions of interest — getCountsByRegions • BRGenomics</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Get signal counts in regions of interest — getCountsByRegions"><meta property="og:description" content='Get the sum of the signal in dataset.gr that overlaps each range in
regions.gr. If expand_regions = FALSE,
getCountsByRegions is written to calculate readcounts
overlapping each region, while expand_regions = TRUE will calculate
"coverage signal" (see details below).'><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Overview.html">Overview</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Guides
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/GettingStarted.html">Getting Started</a>
    </li>
    <li>
      <a href="../articles/ImportingModifyingAnnotations.html">Importing &amp; Modifying Annotations</a>
    </li>
    <li>
      <a href="../articles/ImportingProcessingData.html">Importing &amp; Processing Data</a>
    </li>
    <li>
      <a href="../articles/SignalCounting.html">Signal Counting</a>
    </li>
    <li>
      <a href="../articles/AnalyzingMultipleDatasets.html">Analyzing Multiple Datasets</a>
    </li>
    <li>
      <a href="../articles/ProfilePlotsAndBootstrapping.html">Profile Plots &amp; Bootstrapping</a>
    </li>
    <li>
      <a href="../articles/SpikeInNormalization.html">Spike-in Normalization</a>
    </li>
    <li>
      <a href="../articles/DESeq2WithGlobalPerturbations.html">DESeq2 with Global Perturbations</a>
    </li>
    <li>
      <a href="../articles/SequenceExtraction.html">Sequence Extraction</a>
    </li>
  </ul></li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/mdeber/BRGenomics" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Get signal counts in regions of interest</h1>
    <small class="dont-index">Source: <a href="https://github.com/mdeber/BRGenomics/blob/HEAD/R/signal_counting.R" class="external-link"><code>R/signal_counting.R</code></a></small>
    <div class="hidden name"><code>getCountsByRegions.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Get the sum of the signal in <code>dataset.gr</code> that overlaps each range in
<code>regions.gr</code>. If <code>expand_regions = FALSE</code>,
<code>getCountsByRegions</code> is written to calculate <em>readcounts</em>
overlapping each region, while <code>expand_regions = TRUE</code> will calculate
"coverage signal" (see details below).</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">getCountsByRegions</span><span class="op">(</span></span>
<span>  <span class="va">dataset.gr</span>,</span>
<span>  <span class="va">regions.gr</span>,</span>
<span>  field <span class="op">=</span> <span class="st">"score"</span>,</span>
<span>  NF <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  blacklist <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  melt <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  region_names <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  expand_ranges <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  ncores <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"mc.cores"</span>, <span class="fl">2L</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>dataset.gr</dt>
<dd><p>A GRanges object in which signal is contained in metadata
(typically in the "score" field), or a named list of such GRanges objects.
If a list is given, a dataframe is returned containing the counts in each
region for each dataset.</p></dd>


<dt>regions.gr</dt>
<dd><p>A GRanges object containing regions of interest.</p></dd>


<dt>field</dt>
<dd><p>The metadata field of <code>dataset.gr</code> to be counted. If
<code>length(field) &gt; 1</code>, a dataframe is returned containing the counts for
each region in each field. If <code>field</code> not found in
<code>names(mcols(dataset.gr))</code>, will default to using all fields found in
<code>dataset.gr</code>.</p></dd>


<dt>NF</dt>
<dd><p>An optional normalization factor by which to multiply the counts.
If given, <code>length(NF)</code> must be equal to <code>length(field)</code>.</p></dd>


<dt>blacklist</dt>
<dd><p>An optional GRanges object containing regions that should be
excluded from signal counting.</p></dd>


<dt>melt</dt>
<dd><p>If <code>melt = TRUE</code>, a dataframe is returned containing a
column for regions and another column for signal. If multiple datasets are
given (if <code>dataset.gr</code> is a list or if <code>length(field) &gt; 1</code>), the
output dataframe is melted to contain a third column indicating the sample
names. (See section on return values below).</p></dd>


<dt>region_names</dt>
<dd><p>If <code>melt = TRUE</code>, an optional vector of names for
the regions in <code>regions.gr</code>. If left as <code>NULL</code>, indices of
<code>regions.gr</code> are used instead.</p></dd>


<dt>expand_ranges</dt>
<dd><p>Logical indicating if ranges in <code>dataset.gr</code> should
be treated as descriptions of single molecules (<code>FALSE</code>), or if ranges
should be treated as representing multiple adjacent positions with the same
signal (<code>TRUE</code>). If the ranges in <code>dataset.gr</code> do not all have a
 width of 1, this option has a substantial effect on the results
returned. (See details).</p></dd>


<dt>ncores</dt>
<dd><p>Multiple cores will only be used if <code>dataset.gr</code> is a list
of multiple datasets, or if <code>length(field) &gt; 1</code>.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>An atomic vector the same length as <code>regions.gr</code> containing the
  sum of the signal overlapping each range of <code>regions.gr</code>. If</p>
<p></p>
<p><code>dataset.gr</code> is a list of multiple GRanges, or if <code>length(field)
  &gt; 1</code>, a dataframe is returned. If <code>melt = FALSE</code> (the default),
  dataframes have a column for each dataset and a row for each region. If</p>
<p></p>
<p><code>melt = TRUE</code>, dataframes contain one column to indicate regions
  (either by their indices, or by <code>region_names</code>, if given), another
  column to indicate signal, and a third column containing the sample name
  (unless <code>dataset.gr</code> is a single GRanges object).</p>
    </div>
    <div id="expand-ranges-false">
    <h2><code>expand_ranges = FALSE</code></h2>
    <p>In this configuration,
  <code>getCountsByRegions</code> is designed to work with data in which each range
  represents one type of molecule, whether it's a single base (e.g. the 5'
  ends, 3' ends, or centers of reads) or entire reads (i.e. paired 5' and 3'
  ends of reads).</p>
<p>This is in contrast to standard run-length compressed GRanges object, as
  imported using <code><a href="https://rdrr.io/pkg/rtracklayer/man/BigWigFile.html" class="external-link">rtracklayer::import.bw</a></code>,
  in which a single range can represent multiple contiguous positions that
  share the same signal information.</p>
<p>As an example, a range of covering 10 bp with a score of 2 is treated as 2
  reads (each spanning the same 10 bases), not 20 reads.</p>
    </div>
    <div id="expand-ranges-true">
    <h2><code>expand_ranges = TRUE</code></h2>
    <p>In this configuration, this function
  assumes that ranges in <code>dataset.gr</code> that cover multiple bases are
  compressed representations of multiple adjacent positions that contain the
  same signal. This type of representation is typical of "coverage" objects,
  including bedGraphs and bigWigs generated by many command line utilities,
  but <em>not</em> bigWigs as they are imported by
  <code><a href="import-functions.html">BRGenomics::import_bigWig</a></code>.</p>
<p>As an example, a range covering 10 bp with a score of 2 is treated as
  representing 20 signal counts, i.e. there are 10 adjacent positions that
  each contain a signal of 2.</p>
<p>If the data truly represents basepair-resolution coverage, the "coverage
  signal" is equivalent to readcounts. However, users should consider how
  they interpret results from whole-read coverage, as the "coverage signal"
  is determined by both the read counts as well as read lengths.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="getCountsByPositions.html">getCountsByPositions</a></code></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Mike DeBerardine</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"PROseq"</span><span class="op">)</span> <span class="co"># load included PROseq data</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"txs_dm6_chr4"</span><span class="op">)</span> <span class="co"># load included transcripts</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu">getCountsByRegions</span><span class="op">(</span><span class="va">PROseq</span>, <span class="va">txs_dm6_chr4</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">txs_dm6_chr4</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 339</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 339</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1]    1   59   13  126  263 2509</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Assign as metadata to the transcript GRanges</span></span></span>
<span class="r-in"><span><span class="va">txs_dm6_chr4</span><span class="op">$</span><span class="va">PROseq</span> <span class="op">&lt;-</span> <span class="va">counts</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">txs_dm6_chr4</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 6 ranges and 3 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames       ranges strand |     tx_name     gene_id    PROseq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt;    &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr4     879-5039      + | FBtr0346692 FBgn0267363         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr4  42774-43374      + | FBtr0344900 FBgn0266617        59</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chr4  44774-46074      + | FBtr0340499 FBgn0265633        13</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chr4  56497-60974      + | FBtr0333704 FBgn0264617       126</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [5]     chr4  56497-63124      + | FBtr0333705 FBgn0264617       263</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [6]     chr4 69326-101419      + | FBtr0100246 FBgn0085432      2509</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 7 sequences from dm6 genome</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

