<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Get DESeqDataSet objects for downstream analysis — getDESeqDataSet • BRGenomics</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Get DESeqDataSet objects for downstream analysis — getDESeqDataSet" />
<meta property="og:description" content="This is a convenience function for generating DESeqDataSet objects,
but this function also adds support for counting reads across non-contiguous
regions." />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Get DESeqDataSet objects for downstream analysis</h1>
    
    <div class="hidden name"><code>getDESeqDataSet.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This is a convenience function for generating <code>DESeqDataSet</code> objects,
but this function also adds support for counting reads across non-contiguous
regions.</p>
    </div>

    <pre class="usage"><span class='fu'>getDESeqDataSet</span>(
  <span class='no'>dataset.list</span>,
  <span class='no'>regions.gr</span>,
  <span class='kw'>sample_names</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span>(<span class='no'>dataset.list</span>),
  <span class='kw'>gene_names</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>sizeFactors</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>field</span> <span class='kw'>=</span> <span class='st'>"score"</span>,
  <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fu'>detectCores</span>(),
  <span class='kw'>quiet</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>dataset.list</th>
      <td><p>A list of GRanges datasets that can be individually
passed to <code><a href='getCountsByRegions.html'>getCountsByRegions</a></code>.</p></td>
    </tr>
    <tr>
      <th>regions.gr</th>
      <td><p>A GRanges object containing regions of interest.</p></td>
    </tr>
    <tr>
      <th>sample_names</th>
      <td><p>Names for each dataset in <code>dataset.list</code> are
required, and by default the names of the list elements are used. The names
must each contain the string "_rep#", where "#" is a single character
(usually a number) indicating the replicate. Sample names across different
replicates must be otherwise identical.</p></td>
    </tr>
    <tr>
      <th>gene_names</th>
      <td><p>An optional character vector giving gene names, or any
other identifier over which reads should be counted. Gene names are
required if counting is to be performed over non-contiguous ranges, i.e. if
any genes have multiple ranges. If supplied, gene names are added to the
resulting <code>DESeqDataSet</code> object.</p></td>
    </tr>
    <tr>
      <th>sizeFactors</th>
      <td><p>DESeq2 <code>sizeFactors</code> can be optionally applied in to
the <code>DESeqDataSet</code> object in this function, or they can be applied
later on, either by the user or in a call to <code>getDESeqResults</code>.
Applying the <code>sizeFactors</code> later is useful if multiple sets of factors
will be explored, although <code>sizeFactors</code> can be overwritten at any
time.</p></td>
    </tr>
    <tr>
      <th>field</th>
      <td><p>Argument passed to <code>getCountsByRegions</code>.</p></td>
    </tr>
    <tr>
      <th>ncores</th>
      <td><p>Number of cores to use for read counting across all samples.
Default is the total number of cores available.</p></td>
    </tr>
    <tr>
      <th>quiet</th>
      <td><p>If <code>TRUE</code>, all output messages from call to
<code><a href='https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html'>DESeqDataSet</a></code> will be suppressed.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A <code>DESeqData</code> object in which <code>rowData</code> are given as
  <code>rowRanges</code>, which are equivalent to <code>regions.gr</code>, unless there
  are non-contiguous gene regions (see note below). Samples (as seen in
  <code>colData</code>) are factored so that samples are grouped by
  <code>replicate</code> and <code>condition</code>, i.e. all non-replicate samples are
  treated as distinct, and the DESeq2 design = <code>~condition</code>.</p>
    <h2 class="hasAnchor" id="use-of-non-contiguous-gene-regions"><a class="anchor" href="#use-of-non-contiguous-gene-regions"></a>Use of non-contiguous gene regions</h2>

    <p>In DESeq2, genes must be defined
  by single, contiguous chromosomal locations. This function allows
  individual genes to be encompassed by multiple distinct ranges in
  <code>regions.gr</code>. To use non-contiguous gene regions, provide
  <code>gene_names</code> in which some names are duplicated. For each unique gene
  in <code>gene_names</code>, this function will generate counts across all ranges
  for that gene, but be aware that it will only keep the largest range for
  each gene in the resulting <code>DESeqDataSet</code> object's <code>rowRanges</code>.</p>
    <h2 class="hasAnchor" id="a-note-on-deseq-sizefactors"><a class="anchor" href="#a-note-on-deseq-sizefactors"></a>A note on DESeq2 sizeFactors</h2>

    <p>DESeq2 <code>sizeFactors</code> are
  sample-specific normalization factors that are applied by division, i.e.
  \(counts_{norm,i}=counts_i / sizeFactor_i\). This is in contrast to normalization factors as defined in
  this package (and commonly elsewhere), which are applied by multiplication.
  Also note that DESeq2's "<code>normalizationFactors</code>" are not sample
  specific, but rather gene specific factors used to correct for
  ascertainment bias across different genes (e.g. as might be relevant for
  GSEA or Go analysis).</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html'>DESeq2::DESeqDataSet</a></code>,
  <code><a href='getDESeqResults.html'>getDESeqResults</a></code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/message.html'>suppressPackageStartupMessages</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/library.html'>require</a></span>(<span class='no'>DESeq2</span>))
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='st'>"PROseq"</span>) <span class='co'># import included PROseq data</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='st'>"txs_dm6_chr4"</span>) <span class='co'># import included transcripts</span>

<span class='co'># divide PROseq data into 6 toy datasets</span>
<span class='no'>ps_a_rep1</span> <span class='kw'>&lt;-</span> <span class='no'>PROseq</span>[<span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span>(<span class='fl'>1</span>, <span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span>(<span class='no'>PROseq</span>), <span class='fl'>6</span>)]
<span class='no'>ps_b_rep1</span> <span class='kw'>&lt;-</span> <span class='no'>PROseq</span>[<span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span>(<span class='fl'>2</span>, <span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span>(<span class='no'>PROseq</span>), <span class='fl'>6</span>)]
<span class='no'>ps_c_rep1</span> <span class='kw'>&lt;-</span> <span class='no'>PROseq</span>[<span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span>(<span class='fl'>3</span>, <span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span>(<span class='no'>PROseq</span>), <span class='fl'>6</span>)]

<span class='no'>ps_a_rep2</span> <span class='kw'>&lt;-</span> <span class='no'>PROseq</span>[<span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span>(<span class='fl'>4</span>, <span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span>(<span class='no'>PROseq</span>), <span class='fl'>6</span>)]
<span class='no'>ps_b_rep2</span> <span class='kw'>&lt;-</span> <span class='no'>PROseq</span>[<span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span>(<span class='fl'>5</span>, <span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span>(<span class='no'>PROseq</span>), <span class='fl'>6</span>)]
<span class='no'>ps_c_rep2</span> <span class='kw'>&lt;-</span> <span class='no'>PROseq</span>[<span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span>(<span class='fl'>6</span>, <span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span>(<span class='no'>PROseq</span>), <span class='fl'>6</span>)]

<span class='no'>ps_list</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span>(<span class='kw'>A_rep1</span> <span class='kw'>=</span> <span class='no'>ps_a_rep1</span>,
                <span class='kw'>A_rep2</span> <span class='kw'>=</span> <span class='no'>ps_a_rep2</span>,
                <span class='kw'>B_rep1</span> <span class='kw'>=</span> <span class='no'>ps_b_rep1</span>,
                <span class='kw'>B_rep2</span> <span class='kw'>=</span> <span class='no'>ps_b_rep2</span>,
                <span class='kw'>C_rep1</span> <span class='kw'>=</span> <span class='no'>ps_c_rep1</span>,
                <span class='kw'>C_rep2</span> <span class='kw'>=</span> <span class='no'>ps_c_rep2</span>)

<span class='co'># make flawed dataset (ranges in txs_dm6_chr4 not disjoint)</span>
<span class='co'>#    this means there is double-counting</span>
<span class='co'># also using discontinuous gene regions, as gene_ids are repeated</span>
<span class='no'>dds</span> <span class='kw'>&lt;-</span> <span class='fu'>getDESeqDataSet</span>(<span class='no'>ps_list</span>,
                       <span class='no'>txs_dm6_chr4</span>,
                       <span class='kw'>gene_names</span> <span class='kw'>=</span> <span class='no'>txs_dm6_chr4</span>$<span class='no'>gene_id</span>,
                       <span class='kw'>quiet</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
                       <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fl'>2</span>)
<span class='no'>dds</span></div><div class='output co'>#&gt; class: DESeqDataSet 
#&gt; dim: 111 6 
#&gt; metadata(1): version
#&gt; assays(1): counts
#&gt; rownames(111): FBgn0267363 FBgn0266617 ... FBgn0039924 FBgn0027101
#&gt; rowData names(2): tx_name gene_id
#&gt; colnames(6): A_rep1 A_rep2 ... C_rep1 C_rep2
#&gt; colData names(2): condition replicate</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      <li><a href="#value">Value</a></li>
      <li><a href="#use-of-non-contiguous-gene-regions">Use of non-contiguous gene regions</a></li>
      <li><a href="#a-note-on-deseq-sizefactors">A note on DESeq2 sizeFactors</a></li>
      <li><a href="#see-also">See also</a></li>
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    <p>Mike DeBerardine</p>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


