<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Get DESeq2 results using reduced dispersion matrices — getDESeqResults • BRGenomics</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Get DESeq2 results using reduced dispersion matrices — getDESeqResults"><meta property="og:description" content="This function calls DESeq2::DESeq and
DESeq2::results on a pre-existing
DESeqDataSet object and returns a DESeqResults table for one or
more pairwise comparisons. However, unlike a standard call to
DESeq2::results using the contrast argument, this function
subsets the dataset so that DESeq2 only estimates dispersion for the samples
being compared, and not for all samples present."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Overview.html">Overview</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Guides
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/GettingStarted.html">Getting Started</a>
    </li>
    <li>
      <a href="../articles/ImportingModifyingAnnotations.html">Importing &amp; Modifying Annotations</a>
    </li>
    <li>
      <a href="../articles/ImportingProcessingData.html">Importing &amp; Processing Data</a>
    </li>
    <li>
      <a href="../articles/SignalCounting.html">Signal Counting</a>
    </li>
    <li>
      <a href="../articles/AnalyzingMultipleDatasets.html">Analyzing Multiple Datasets</a>
    </li>
    <li>
      <a href="../articles/ProfilePlotsAndBootstrapping.html">Profile Plots &amp; Bootstrapping</a>
    </li>
    <li>
      <a href="../articles/SpikeInNormalization.html">Spike-in Normalization</a>
    </li>
    <li>
      <a href="../articles/DESeq2WithGlobalPerturbations.html">DESeq2 with Global Perturbations</a>
    </li>
    <li>
      <a href="../articles/SequenceExtraction.html">Sequence Extraction</a>
    </li>
  </ul></li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/mdeber/BRGenomics" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Get DESeq2 results using reduced dispersion matrices</h1>
    <small class="dont-index">Source: <a href="https://github.com/mdeber/BRGenomics/blob/HEAD/R/deseq_functions.R" class="external-link"><code>R/deseq_functions.R</code></a></small>
    <div class="hidden name"><code>getDESeqResults.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function calls <code><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html" class="external-link">DESeq2::DESeq</a></code> and
<code><a href="https://rdrr.io/pkg/DESeq2/man/results.html" class="external-link">DESeq2::results</a></code> on a pre-existing
<code>DESeqDataSet</code> object and returns a <code>DESeqResults</code> table for one or
more pairwise comparisons. However, unlike a standard call to
<code><a href="https://rdrr.io/pkg/DESeq2/man/results.html" class="external-link">DESeq2::results</a></code> using the <code>contrast</code> argument, this function
subsets the dataset so that DESeq2 only estimates dispersion for the samples
being compared, and not for all samples present.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">getDESeqResults</span><span class="op">(</span></span>
<span>  <span class="va">dds</span>,</span>
<span>  <span class="va">contrast.numer</span>,</span>
<span>  <span class="va">contrast.denom</span>,</span>
<span>  comparisons <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  sizeFactors <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  alpha <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  lfcShrink <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  args.DESeq <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  args.results <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  args.lfcShrink <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  ncores <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"mc.cores"</span>, <span class="fl">2L</span><span class="op">)</span>,</span>
<span>  quiet <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>dds</dt>
<dd><p>A DESeqDataSet object, produced using either
<code><a href="getDESeqDataSet.html">getDESeqDataSet</a></code> from this package
or <code><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html" class="external-link">DESeqDataSet</a></code> from <code>DESeq2</code>. If
<code>dds</code> was not created using <code>getDESeqDataSet</code>, <code>dds</code> must be
made with <code>design = ~condition</code> such that a unique <code>condition</code>
level exists for each sample/treatment condition.</p></dd>


<dt>contrast.numer</dt>
<dd><p>A string naming the <code>condition</code> to use as the
numerator in the DESeq2 comparison, typically the perturbative condition.</p></dd>


<dt>contrast.denom</dt>
<dd><p>A string naming the <code>condition</code> to use as the
denominator in the DESeq2 comparison, typically the control condition.</p></dd>


<dt>comparisons</dt>
<dd><p>As an optional alternative to supplying a single
<code>contrast.numer</code> and <code>contrast.denom</code>, users can supply a list of
character vectors containing numerator-denominator pairs, e.g.
<code>list(c("B", "A"), c("C", "A"), c("C", "B"))</code>. <code>comparisons</code> can
also be a dataframe in which each row is a comparison, the first column
contains the numerators, and  the second column contains the denominators.</p></dd>


<dt>sizeFactors</dt>
<dd><p>A vector containing DESeq2 <code>sizeFactors</code> to apply to
each sample. Each sample's readcounts are <em>divided</em> by its respective
DESeq2 <code>sizeFactor</code>. A warning will be generated if the
<code>DESeqDataSet</code> already contains <code>sizeFactors</code>, and the previous
<code>sizeFactors</code> will be over-written.</p></dd>


<dt>alpha</dt>
<dd><p>The significance threshold passed to <code>DESeqResults</code>, which
is used for independent filtering of results (see DESeq2 documentation).</p></dd>


<dt>lfcShrink</dt>
<dd><p>Logical indicating if log2FoldChanges and their standard
errors should be shrunk using <code><a href="https://rdrr.io/pkg/DESeq2/man/lfcShrink.html" class="external-link">lfcShrink</a></code>.
LFC shrinkage is very useful for making fold-change values meaningful, as
low-expression/high variance genes are given low fold-changes.
Set to <code>FALSE</code> by default.</p></dd>


<dt>args.DESeq</dt>
<dd><p>Additional arguments passed to
<code><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html" class="external-link">DESeq</a></code>, given as a list of argument-value pairs,
e.g. <code>list(fitType = "local", useT = TRUE)</code>. All arguments given here
will be passed to <code>DESeq</code> except for <code>object</code> and
<code>parallel</code>. If no arguments are given, all defaults will be used.</p></dd>


<dt>args.results</dt>
<dd><p>Additional arguments passed to
<a href="https://rdrr.io/pkg/DESeq2/man/results.html" class="external-link">DESeq2::results</a>, given as a list of argument-value
pairs, e.g. <code>list(altHypothesis = "greater", lfcThreshold = 1.5)</code>. All
arguments given here will be passed to <code>results</code> except for
<code>object</code>, <code>contrast</code>, <code>alpha</code>, and <code>parallel</code>. If no
arguments are given, all defaults will be used.</p></dd>


<dt>args.lfcShrink</dt>
<dd><p>Additional arguments passed to
<code><a href="https://rdrr.io/pkg/DESeq2/man/lfcShrink.html" class="external-link">lfcShrink</a></code>, given as a list of
argument-value pairs. All arguments given here will be passed to
<code>lfcShrink</code> except for <code>dds</code>, <code>coef</code>, <code>contrast</code>, and
<code>parallel</code>. If no arguments are given, all defaults will be used.</p></dd>


<dt>ncores</dt>
<dd><p>The number of cores to use for parallel processing. Multicore
processing is only used if more than one comparison is being made (i.e.
argument <code>comparisons</code> is used), and the number of cores utilized will
not be greater than the number of comparisons being performed.</p></dd>


<dt>quiet</dt>
<dd><p>If <code>TRUE</code>, all output messages from calls to <code>DESeq</code>
and <code>results</code> will be suppressed, although passing option <code>quiet</code>
in <code>args.DESeq</code> will supersede this option for the call to
<code>DESeq</code>.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>For a single comparison, the output is the <code>DESeqResults</code> result
  table. If a <code>comparisons</code> is used to make multiple comparisons, the
  output is a named list of <code>DESeqResults</code> objects, with elements named
  following the pattern <code>"X_vs_Y"</code>, where <code>X</code> is the name of the
  numerator condition, and <code>Y</code> is the name of the denominator condition.</p>
    </div>
    <div id="errors-when-ncores-gt-">
    <h2>Errors when <code>ncores &gt; 1</code></h2>
    <p>If this function returns an error,
  set <code>ncores = 1</code>. Whether or not this occurs can depend on whether
  users are using alternative BLAS libraries (e.g. OpenBLAS or Apple's
  Accelerate framework) and/or how DESeq2 was installed. This is because some
  DESeq2 functions (e.g. <code><a href="https://rdrr.io/pkg/DESeq2/man/nbinomWaldTest.html" class="external-link">
  nbinomWaldTest</a></code>) use C code that can be compiled to use parallelization,
  and this conflicts with our use of process forking (via the
  <code><a href="https://rdrr.io/r/parallel/parallel-package.html" class="external-link">parallel package</a></code>) when
  <code>ncores &gt; 1</code>.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="getDESeqDataSet.html">getDESeqDataSet</a></code>,
  <code><a href="https://rdrr.io/pkg/DESeq2/man/results.html" class="external-link">DESeq2::results</a></code></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Mike DeBerardine</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># getDESeqDataSet</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://github.com/mikelove/DESeq2" class="external-link">DESeq2</a></span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"PROseq"</span><span class="op">)</span> <span class="co"># import included PROseq data</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"txs_dm6_chr4"</span><span class="op">)</span> <span class="co"># import included transcripts</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># divide PROseq data into 6 toy datasets</span></span></span>
<span class="r-in"><span><span class="va">ps_a_rep1</span> <span class="op">&lt;-</span> <span class="va">PROseq</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">PROseq</span><span class="op">)</span>, <span class="fl">6</span><span class="op">)</span><span class="op">]</span></span></span>
<span class="r-in"><span><span class="va">ps_b_rep1</span> <span class="op">&lt;-</span> <span class="va">PROseq</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">PROseq</span><span class="op">)</span>, <span class="fl">6</span><span class="op">)</span><span class="op">]</span></span></span>
<span class="r-in"><span><span class="va">ps_c_rep1</span> <span class="op">&lt;-</span> <span class="va">PROseq</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">PROseq</span><span class="op">)</span>, <span class="fl">6</span><span class="op">)</span><span class="op">]</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">ps_a_rep2</span> <span class="op">&lt;-</span> <span class="va">PROseq</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">4</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">PROseq</span><span class="op">)</span>, <span class="fl">6</span><span class="op">)</span><span class="op">]</span></span></span>
<span class="r-in"><span><span class="va">ps_b_rep2</span> <span class="op">&lt;-</span> <span class="va">PROseq</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">PROseq</span><span class="op">)</span>, <span class="fl">6</span><span class="op">)</span><span class="op">]</span></span></span>
<span class="r-in"><span><span class="va">ps_c_rep2</span> <span class="op">&lt;-</span> <span class="va">PROseq</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">6</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">PROseq</span><span class="op">)</span>, <span class="fl">6</span><span class="op">)</span><span class="op">]</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">ps_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>A_rep1 <span class="op">=</span> <span class="va">ps_a_rep1</span>, A_rep2 <span class="op">=</span> <span class="va">ps_a_rep2</span>,</span></span>
<span class="r-in"><span>                B_rep1 <span class="op">=</span> <span class="va">ps_b_rep1</span>, B_rep2 <span class="op">=</span> <span class="va">ps_b_rep2</span>,</span></span>
<span class="r-in"><span>                C_rep1 <span class="op">=</span> <span class="va">ps_c_rep1</span>, C_rep2 <span class="op">=</span> <span class="va">ps_c_rep2</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># make flawed dataset (ranges in txs_dm6_chr4 not disjoint)</span></span></span>
<span class="r-in"><span><span class="co">#    this means there is double-counting</span></span></span>
<span class="r-in"><span><span class="co"># also using discontinuous gene regions, as gene_ids are repeated</span></span></span>
<span class="r-in"><span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="getDESeqDataSet.html">getDESeqDataSet</a></span><span class="op">(</span><span class="va">ps_list</span>,</span></span>
<span class="r-in"><span>                       <span class="va">txs_dm6_chr4</span>,</span></span>
<span class="r-in"><span>                       gene_names <span class="op">=</span> <span class="va">txs_dm6_chr4</span><span class="op">$</span><span class="va">gene_id</span>,</span></span>
<span class="r-in"><span>                       ncores <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">dds</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> class: DESeqDataSet </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> dim: 111 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> metadata(1): version</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> assays(1): counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rownames(111): FBgn0267363 FBgn0266617 ... FBgn0039924 FBgn0027101</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowData names(2): tx_name gene_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colnames(6): A_rep1 A_rep2 ... C_rep1 C_rep2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colData names(2): condition replicate</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># getDESeqResults</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">getDESeqResults</span><span class="op">(</span><span class="va">dds</span>, <span class="st">"B"</span>, <span class="st">"A"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> estimating size factors</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> estimating dispersions</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> gene-wise dispersion estimates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> mean-dispersion relationship</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> final dispersion estimates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> fitting model and testing</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">res</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> log2 fold change (MLE): condition B vs A </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Wald test p-value: condition B vs A </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DataFrame with 111 rows and 6 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                baseMean log2FoldChange     lfcSE        stat    pvalue</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0267363    0.254249     -1.4228145  4.997084 -0.28472898  0.775852</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0266617   10.800541      0.1563201  1.033936  0.15118944  0.879826</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0265633    2.057706      0.0175256  2.125748  0.00824443  0.993422</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0264617   66.127353      0.2087670  0.467572  0.44649198  0.655242</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0085432 4678.846158     -0.2333084  0.236723 -0.98557490  0.324342</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ...                 ...            ...       ...         ...       ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0039928     685.650       0.126857  0.258212    0.491292  0.623220</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0052017     130.983       0.239103  0.418935    0.570741  0.568175</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0002521     174.375      -0.110680  0.342843   -0.322829  0.746825</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0039924     161.131       0.322486  0.456555    0.706346  0.479973</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0027101     361.983       0.233889  0.294518    0.794142  0.427113</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  padj</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0267363  0.997413</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0266617  0.997413</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0265633  0.997413</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0264617  0.997413</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0085432  0.997413</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ...               ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0039928  0.997413</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0052017  0.997413</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0002521  0.997413</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0039924  0.997413</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0027101  0.997413</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">reslist</span> <span class="op">&lt;-</span> <span class="fu">getDESeqResults</span><span class="op">(</span><span class="va">dds</span>, comparisons <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"B"</span>, <span class="st">"A"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"C"</span>, <span class="st">"A"</span><span class="op">)</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                           ncores <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">reslist</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "B_vs_A" "C_vs_A"</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">reslist</span><span class="op">$</span><span class="va">B_vs_A</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> log2 fold change (MLE): condition B vs A </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Wald test p-value: condition B vs A </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DataFrame with 111 rows and 6 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                baseMean log2FoldChange     lfcSE        stat    pvalue</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0267363    0.254249     -1.4228145  4.997084 -0.28472898  0.775852</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0266617   10.800541      0.1563201  1.033936  0.15118944  0.879826</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0265633    2.057706      0.0175256  2.125748  0.00824443  0.993422</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0264617   66.127353      0.2087670  0.467572  0.44649198  0.655242</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0085432 4678.846158     -0.2333084  0.236723 -0.98557490  0.324342</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ...                 ...            ...       ...         ...       ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0039928     685.650       0.126857  0.258212    0.491292  0.623220</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0052017     130.983       0.239103  0.418935    0.570741  0.568175</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0002521     174.375      -0.110680  0.342843   -0.322829  0.746825</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0039924     161.131       0.322486  0.456555    0.706346  0.479973</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0027101     361.983       0.233889  0.294518    0.794142  0.427113</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  padj</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0267363  0.997413</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0266617  0.997413</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0265633  0.997413</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0264617  0.997413</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0085432  0.997413</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ...               ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0039928  0.997413</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0052017  0.997413</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0002521  0.997413</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0039924  0.997413</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0027101  0.997413</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># or using a dataframe</span></span></span>
<span class="r-in"><span><span class="va">reslist</span> <span class="op">&lt;-</span> <span class="fu">getDESeqResults</span><span class="op">(</span><span class="va">dds</span>, comparisons <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>num <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"B"</span>, <span class="st">"C"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                                                         den <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"A"</span><span class="op">)</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                           ncores <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">reslist</span><span class="op">$</span><span class="va">B_vs_A</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> log2 fold change (MLE): condition B vs A </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Wald test p-value: condition B vs A </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DataFrame with 111 rows and 6 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                baseMean log2FoldChange     lfcSE        stat    pvalue</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0267363    0.254249     -1.4228145  4.997084 -0.28472898  0.775852</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0266617   10.800541      0.1563201  1.033936  0.15118944  0.879826</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0265633    2.057706      0.0175256  2.125748  0.00824443  0.993422</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0264617   66.127353      0.2087670  0.467572  0.44649198  0.655242</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0085432 4678.846158     -0.2333084  0.236723 -0.98557490  0.324342</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ...                 ...            ...       ...         ...       ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0039928     685.650       0.126857  0.258212    0.491292  0.623220</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0052017     130.983       0.239103  0.418935    0.570741  0.568175</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0002521     174.375      -0.110680  0.342843   -0.322829  0.746825</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0039924     161.131       0.322486  0.456555    0.706346  0.479973</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0027101     361.983       0.233889  0.294518    0.794142  0.427113</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  padj</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0267363  0.997413</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0266617  0.997413</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0265633  0.997413</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0264617  0.997413</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0085432  0.997413</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ...               ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0039928  0.997413</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0052017  0.997413</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0002521  0.997413</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0039924  0.997413</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FBgn0027101  0.997413</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

