<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Get DESeqDataSet objects for downstream analysis — getDESeqDataSet • BRGenomics</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Get DESeqDataSet objects for downstream analysis — getDESeqDataSet" />
<meta property="og:description" content="This is a convenience function for generating DESeqDataSet objects,
but this function also adds support for counting reads across non-contiguous
regions." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Overview.html">Overview</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Guides
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/GettingStarted.html">Getting Started</a>
    </li>
    <li>
      <a href="../articles/ImportingModifyingAnnotations.html">Importing &amp; Modifying Annotations</a>
    </li>
    <li>
      <a href="../articles/ImportingProcessingData.html">Importing &amp; Processing Data</a>
    </li>
    <li>
      <a href="../articles/SignalCounting.html">Signal Counting</a>
    </li>
    <li>
      <a href="../articles/AnalyzingMultipleDatasets.html">Analyzing Multiple Datasets</a>
    </li>
    <li>
      <a href="../articles/ProfilePlotsAndBootstrapping.html">Profile Plots &amp; Bootstrapping</a>
    </li>
    <li>
      <a href="../articles/SpikeInNormalization.html">Spike-in Normalization</a>
    </li>
    <li>
      <a href="../articles/DESeq2WithGlobalPerturbations.html">DESeq2 with Global Perturbations</a>
    </li>
    <li>
      <a href="../articles/SequenceExtraction.html">Sequence Extraction</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mdeber/BRGenomics">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Get DESeqDataSet objects for downstream analysis</h1>
    <small class="dont-index">Source: <a href='https://github.com/mdeber/BRGenomics/blob/master/R/deseq_functions.R'><code>R/deseq_functions.R</code></a></small>
    <div class="hidden name"><code>getDESeqDataSet.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This is a convenience function for generating <code>DESeqDataSet</code> objects,
but this function also adds support for counting reads across non-contiguous
regions.</p>
    </div>

    <pre class="usage"><span class='fu'>getDESeqDataSet</span>(
  <span class='no'>dataset.list</span>,
  <span class='no'>regions.gr</span>,
  <span class='kw'>sample_names</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>gene_names</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>sizeFactors</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>field</span> <span class='kw'>=</span> <span class='st'>"score"</span>,
  <span class='kw'>blacklist</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>expand_ranges</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/options.html'>getOption</a></span>(<span class='st'>"mc.cores"</span>, <span class='fl'>2L</span>),
  <span class='kw'>quiet</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>dataset.list</th>
      <td><p>An object containing GRanges datasets that can be passed
to <code><a href='getCountsByRegions.html'>getCountsByRegions</a></code>,
typically a list of GRanges objects, or a
<code><a href='mergeGRangesData.html'> multiplexed GRanges</a></code> object (see
details below).</p></td>
    </tr>
    <tr>
      <th>regions.gr</th>
      <td><p>A GRanges object containing regions of interest.</p></td>
    </tr>
    <tr>
      <th>sample_names</th>
      <td><p>Names for each dataset in <code>dataset.list</code> are
required. By default (<code>sample_names = NULL</code>), if <code>dataset.list</code>
is a list, the names of the list elements are used; for a multiplexed
GRanges object, the field names are used. The names must each contain the
string "_rep#", where "#" is a single character (usually a number)
indicating the replicate. Sample names across different replicates must be
otherwise identical.</p></td>
    </tr>
    <tr>
      <th>gene_names</th>
      <td><p>An optional character vector giving gene names, or any
other identifier over which reads should be counted. Gene names are
required if counting is to be performed over non-contiguous ranges, i.e. if
any genes have multiple ranges. If supplied, gene names are added to the
resulting <code>DESeqDataSet</code> object.</p></td>
    </tr>
    <tr>
      <th>sizeFactors</th>
      <td><p>DESeq2 <code>sizeFactors</code> can be optionally applied in to
the <code>DESeqDataSet</code> object in this function, or they can be applied
later on, either by the user or in a call to <code>getDESeqResults</code>.
Applying the <code>sizeFactors</code> later is useful if multiple sets of factors
will be explored, although <code>sizeFactors</code> can be overwritten at any
time. Note that DESeq2 <code>sizeFactors</code> are <em>not</em> the same as
normalization factors defined elsewhere in this package. See details below.</p></td>
    </tr>
    <tr>
      <th>field</th>
      <td><p>Argument passed to <code>getCountsByRegions</code>. Can be used to
specify fields in a single multiplexed GRanges object, or individual fields
for each GRanges object in <code>dataset.list</code>.</p></td>
    </tr>
    <tr>
      <th>blacklist</th>
      <td><p>An optional GRanges object containing regions that should be
excluded from signal counting. Use of this argument is distinct from the
use of non-contiguous gene regions (see details below), and the two can be
used simultaneously. Blacklisting doesn't affect the ranges returned as
rowRanges in the output DESeqDataSet object (unlike the use of
non-contiguous regions).</p></td>
    </tr>
    <tr>
      <th>expand_ranges</th>
      <td><p>Logical indicating if ranges in <code>dataset.gr</code> should
be treated as descriptions of single molecules (<code>FALSE</code>), or if ranges
should be treated as representing multiple adjacent positions with the same
signal (<code>TRUE</code>). See <code><a href='getCountsByRegions.html'>
getCountsByRegions</a></code>.</p></td>
    </tr>
    <tr>
      <th>ncores</th>
      <td><p>Number of cores to use for read counting across all samples. By
default, all available cores are used.</p></td>
    </tr>
    <tr>
      <th>quiet</th>
      <td><p>If <code>TRUE</code>, all output messages from call to
<code><a href='https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html'>DESeqDataSet</a></code> will be suppressed.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A <code>DESeqData</code> object in which <code>rowData</code> are given as
  <code>rowRanges</code>, which are equivalent to <code>regions.gr</code>, unless there
  are non-contiguous gene regions (see note below). Samples (as seen in
  <code>colData</code>) are factored so that samples are grouped by
  <code>replicate</code> and <code>condition</code>, i.e. all non-replicate samples are
  treated as distinct, and the DESeq2 design = <code>~condition</code>.</p>
    <h2 class="hasAnchor" id="use-of-non-contiguous-gene-regions"><a class="anchor" href="#use-of-non-contiguous-gene-regions"></a>Use of non-contiguous gene regions</h2>

    <p>In DESeq2, genes must be defined
  by single, contiguous chromosomal locations. In contrast, this function
  allows individual genes to be encompassed by multiple distinct ranges in
  <code>regions.gr</code>. To use non-contiguous gene regions, provide
  <code>gene_names</code> in which some names are duplicated. For each unique gene
  in <code>gene_names</code>, this function will generate counts across all ranges
  for that gene, but be aware that it will only keep the largest range for
  each gene in the resulting <code>DESeqDataSet</code> object's <code>rowRanges</code>.
  If the desired use is to blacklist certain sites in a genelist, note that
  the <code>blacklist</code> argument can be used.</p>
    <h2 class="hasAnchor" id="a-note-on-deseq-sizefactors"><a class="anchor" href="#a-note-on-deseq-sizefactors"></a>A note on DESeq2 sizeFactors</h2>

    <p>DESeq2 <code>sizeFactors</code> are
  sample-specific normalization factors that are applied by division, i.e.
  \(counts_{norm,i}=counts_i / sizeFactor_i\). This is in contrast to normalization factors as defined in
  this package (and commonly elsewhere), which are applied by multiplication.
  Also note that DESeq2's "<code>normalizationFactors</code>" are not sample
  specific, but rather gene specific factors used to correct for
  ascertainment bias across different genes (e.g. as might be relevant for
  GSEA or Go analysis).</p>
    <h2 class="hasAnchor" id="on-gene-names-and-unexpected-errors"><a class="anchor" href="#on-gene-names-and-unexpected-errors"></a>On gene names and unexpected errors</h2>

    <p>Certain gene names can cause
this function to return an error. We've never encountered errors using
conventional, systematic naming schemes (e.g. ensembl IDs), but we have
seen errors when using Drosophila (Flybase) "symbols". We expect this is due
to the unconventional use of non-alphanumeric characters in some Drosophila
gene names.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html'>DESeq2::DESeqDataSet</a></code>,
  <code><a href='getDESeqResults.html'>getDESeqResults</a></code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/message.html'>suppressPackageStartupMessages</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/library.html'>require</a></span>(<span class='no'>DESeq2</span>))
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='st'>"PROseq"</span>) <span class='co'># import included PROseq data</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='st'>"txs_dm6_chr4"</span>) <span class='co'># import included transcripts</span>

<span class='co'># divide PROseq data into 6 toy datasets</span>
<span class='no'>ps_a_rep1</span> <span class='kw'>&lt;-</span> <span class='no'>PROseq</span>[<span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span>(<span class='fl'>1</span>, <span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span>(<span class='no'>PROseq</span>), <span class='fl'>6</span>)]
<span class='no'>ps_b_rep1</span> <span class='kw'>&lt;-</span> <span class='no'>PROseq</span>[<span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span>(<span class='fl'>2</span>, <span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span>(<span class='no'>PROseq</span>), <span class='fl'>6</span>)]
<span class='no'>ps_c_rep1</span> <span class='kw'>&lt;-</span> <span class='no'>PROseq</span>[<span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span>(<span class='fl'>3</span>, <span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span>(<span class='no'>PROseq</span>), <span class='fl'>6</span>)]

<span class='no'>ps_a_rep2</span> <span class='kw'>&lt;-</span> <span class='no'>PROseq</span>[<span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span>(<span class='fl'>4</span>, <span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span>(<span class='no'>PROseq</span>), <span class='fl'>6</span>)]
<span class='no'>ps_b_rep2</span> <span class='kw'>&lt;-</span> <span class='no'>PROseq</span>[<span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span>(<span class='fl'>5</span>, <span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span>(<span class='no'>PROseq</span>), <span class='fl'>6</span>)]
<span class='no'>ps_c_rep2</span> <span class='kw'>&lt;-</span> <span class='no'>PROseq</span>[<span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span>(<span class='fl'>6</span>, <span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span>(<span class='no'>PROseq</span>), <span class='fl'>6</span>)]

<span class='no'>ps_list</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span>(<span class='kw'>A_rep1</span> <span class='kw'>=</span> <span class='no'>ps_a_rep1</span>, <span class='kw'>A_rep2</span> <span class='kw'>=</span> <span class='no'>ps_a_rep2</span>,
                <span class='kw'>B_rep1</span> <span class='kw'>=</span> <span class='no'>ps_b_rep1</span>, <span class='kw'>B_rep2</span> <span class='kw'>=</span> <span class='no'>ps_b_rep2</span>,
                <span class='kw'>C_rep1</span> <span class='kw'>=</span> <span class='no'>ps_c_rep1</span>, <span class='kw'>C_rep2</span> <span class='kw'>=</span> <span class='no'>ps_c_rep2</span>)

<span class='co'># make flawed dataset (ranges in txs_dm6_chr4 not disjoint)</span>
<span class='co'>#    this means there is double-counting</span>
<span class='co'># also using discontinuous gene regions, as gene_ids are repeated</span>
<span class='no'>dds</span> <span class='kw'>&lt;-</span> <span class='fu'>getDESeqDataSet</span>(<span class='no'>ps_list</span>,
                       <span class='no'>txs_dm6_chr4</span>,
                       <span class='kw'>gene_names</span> <span class='kw'>=</span> <span class='no'>txs_dm6_chr4</span>$<span class='no'>gene_id</span>,
                       <span class='kw'>quiet</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
                       <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fl'>1</span>)
<span class='no'>dds</span></div><div class='output co'>#&gt; class: DESeqDataSet 
#&gt; dim: 111 6 
#&gt; metadata(1): version
#&gt; assays(1): counts
#&gt; rownames(111): FBgn0267363 FBgn0266617 ... FBgn0039924 FBgn0027101
#&gt; rowData names(2): tx_name gene_id
#&gt; colnames(6): A_rep1 A_rep2 ... C_rep1 C_rep2
#&gt; colData names(2): condition replicate</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


