<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Get DESeq2 results using reduced dispersion matrices — getDESeqResults • BRGenomics</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Get DESeq2 results using reduced dispersion matrices — getDESeqResults" />
<meta property="og:description" content="This function calls DESeq2::DESeq and
DESeq2::results on a pre-existing
DESeqDataSet object and returns a DESeqResults table for one or
more pairwise comparisons. However, unlike a standard call to
DESeq2::results using the contrast argument, this function
subsets the dataset so that DESeq2 only estimates dispersion for the samples
being compared, and not for all samples present." />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.26</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/BRGenomicsOverview.html">Basepair-Resolution Analysis with BRGenomics</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Get DESeq2 results using reduced dispersion matrices</h1>
    
    <div class="hidden name"><code>getDESeqResults.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function calls <code><a href='https://rdrr.io/pkg/DESeq2/man/DESeq.html'>DESeq2::DESeq</a></code> and
<code><a href='https://rdrr.io/pkg/DESeq2/man/results.html'>DESeq2::results</a></code> on a pre-existing
<code>DESeqDataSet</code> object and returns a <code>DESeqResults</code> table for one or
more pairwise comparisons. However, unlike a standard call to
<code><a href='https://rdrr.io/pkg/DESeq2/man/results.html'>DESeq2::results</a></code> using the <code>contrast</code> argument, this function
subsets the dataset so that DESeq2 only estimates dispersion for the samples
being compared, and not for all samples present.</p>
    </div>

    <pre class="usage"><span class='fu'>getDESeqResults</span>(
  <span class='no'>dds</span>,
  <span class='no'>contrast.numer</span>,
  <span class='no'>contrast.denom</span>,
  <span class='kw'>comparisons</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>sizeFactors</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>alpha</span> <span class='kw'>=</span> <span class='fl'>0.1</span>,
  <span class='kw'>args.DESeq</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>args.results</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fu'>detectCores</span>(),
  <span class='kw'>quiet</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>dds</th>
      <td><p>A DESeqDataSet object, produced using either
<code><a href='getDESeqDataSet.html'>getDESeqDataSet</a></code> from this package
or <code><a href='https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html'>DESeqDataSet</a></code> from <code>DESeq2</code>. If
<code>dds</code> was not created using <code>getDESeqDataSet</code>, <code>dds</code> must be
made with <code>design = ~condition</code> such that a unique <code>condition</code>
level exists for each sample/treatment condition.</p></td>
    </tr>
    <tr>
      <th>contrast.numer</th>
      <td><p>A string naming the <code>condition</code> to use as the
numerator in the DESeq2 comparison, typically the perturbative condition.</p></td>
    </tr>
    <tr>
      <th>contrast.denom</th>
      <td><p>A string naming the <code>condition</code> to use as the
denominator in the DESeq2 comparison, typically the control condition.</p></td>
    </tr>
    <tr>
      <th>comparisons</th>
      <td><p>As an optional alternative to supplying a single
<code>contrast.numer</code> and <code>contrast.denom</code>, users can supply a list of
character vectors containing numerator-denominator pairs, e.g.
<code><a href='https://rdrr.io/r/base/list.html'>list(c("B", "A"), c("C", "A"), c("C", "B"))</a></code>. <code>comparisons</code> can
also be a dataframe in which each row is a comparison, the first column
contains the numerators, and  the second column contains the denominators.</p></td>
    </tr>
    <tr>
      <th>sizeFactors</th>
      <td><p>A vector containing DESeq2 <code>sizeFactors</code> to apply to
each sample. Each sample's readcounts are <em>divided</em> by its respective
DESeq2 <code>sizeFactor</code>. A warning will be generated if the
<code>DESeqDataSet</code> already contains <code>sizeFactors</code>, and the previous
<code>sizeFactors</code> will be over-written.</p></td>
    </tr>
    <tr>
      <th>alpha</th>
      <td><p>The significance threshold passed to <code>DESeqResults</code>, which
is used for independent filtering of results (see DESeq2 documentation).</p></td>
    </tr>
    <tr>
      <th>args.DESeq</th>
      <td><p>Additional arguments passed to
<code><a href='https://rdrr.io/pkg/DESeq2/man/DESeq.html'>DESeq</a></code>, given as a list of argument-value pairs,
e.g. <code><a href='https://rdrr.io/r/base/list.html'>list(fitType = "local", useT = TRUE)</a></code>. All arguments given here
will be passed to <code>DESeq</code> except for <code>object</code> and
<code>parallel</code>. If no arguments are given, all defaults will be used.</p></td>
    </tr>
    <tr>
      <th>args.results</th>
      <td><p>Additional arguments passed to
<a href='https://rdrr.io/pkg/DESeq2/man/results.html'>DESeq2::results</a>, given as a list of argument-value
pairs, e.g. <code><a href='https://rdrr.io/r/base/list.html'>list(altHypothesis = "greater", lfcThreshold = 1.5)</a></code>. All
arguments given here will be passed to <code>results</code> except for
<code>object</code>, <code>contrast</code>, <code>alpha</code>, and <code>parallel</code>. If no
arguments are given, all defaults will be used.</p></td>
    </tr>
    <tr>
      <th>ncores</th>
      <td><p>The number of cores to use for parallel processing. Multicore
processing is only used if more than one comparison is being made (i.e.
argument <code>comparisons</code> is used), and the number of cores utilized will
not be greater than the number of comparisons being performed.</p></td>
    </tr>
    <tr>
      <th>quiet</th>
      <td><p>If <code>TRUE</code>, all output messages from calls to <code>DESeq</code>
and <code>results</code> will be suppressed, although passing option <code>quiet</code>
in <code>args.DESeq</code> will supersede this option for the call to
<code>DESeq</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>For a single comparison, the output is the <code>DESeqResults</code> result
  table. If a <code>comparisons</code> is used to make multiple comparisons, the
  output is a named list of <code>DESeqResults</code> objects, with elements named
  following the pattern <code>"X_vs_Y"</code>, where <code>X</code> is the name of the
  numerator condition, and <code>Y</code> is the name of the denominator condition.</p>
    <h2 class="hasAnchor" id="errors-when-ncores-gt-"><a class="anchor" href="#errors-when-ncores-gt-"></a>Errors when <code>ncores &gt; 1</code></h2>

    <p>If this function returns an error,
  set <code>ncores = 1</code>. Whether or not this occurs can depend on whether
  users are using alternative BLAS libraries (e.g. OpenBLAS or Apple's
  Accelerate framework) and/or how DESeq2 was installed. This is because some
  DESeq2 functions (e.g. <code><a href='https://rdrr.io/pkg/DESeq2/man/nbinomWaldTest.html'>
  nbinomWaldTest</a></code>) use C code that can be compiled to use parallelization,
  and this conflicts with our use of process forking (via the
  <code><a href='https://rdrr.io/r/parallel/parallel-package.html'>parallel package</a></code>) when
  <code>ncores &gt; 1</code>.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='getDESeqDataSet.html'>getDESeqDataSet</a></code>,
  <code><a href='https://rdrr.io/pkg/DESeq2/man/results.html'>DESeq2::results</a></code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>#--------------------------------------------------#</span>
<span class='co'># getDESeqDataSet</span>
<span class='co'>#--------------------------------------------------#</span>
<span class='fu'><a href='https://rdrr.io/r/base/message.html'>suppressPackageStartupMessages</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/library.html'>require</a></span>(<span class='no'>DESeq2</span>))
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='st'>"PROseq"</span>) <span class='co'># import included PROseq data</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='st'>"txs_dm6_chr4"</span>) <span class='co'># import included transcripts</span>

<span class='co'># divide PROseq data into 6 toy datasets</span>
<span class='no'>ps_a_rep1</span> <span class='kw'>&lt;-</span> <span class='no'>PROseq</span>[<span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span>(<span class='fl'>1</span>, <span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span>(<span class='no'>PROseq</span>), <span class='fl'>6</span>)]
<span class='no'>ps_b_rep1</span> <span class='kw'>&lt;-</span> <span class='no'>PROseq</span>[<span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span>(<span class='fl'>2</span>, <span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span>(<span class='no'>PROseq</span>), <span class='fl'>6</span>)]
<span class='no'>ps_c_rep1</span> <span class='kw'>&lt;-</span> <span class='no'>PROseq</span>[<span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span>(<span class='fl'>3</span>, <span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span>(<span class='no'>PROseq</span>), <span class='fl'>6</span>)]

<span class='no'>ps_a_rep2</span> <span class='kw'>&lt;-</span> <span class='no'>PROseq</span>[<span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span>(<span class='fl'>4</span>, <span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span>(<span class='no'>PROseq</span>), <span class='fl'>6</span>)]
<span class='no'>ps_b_rep2</span> <span class='kw'>&lt;-</span> <span class='no'>PROseq</span>[<span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span>(<span class='fl'>5</span>, <span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span>(<span class='no'>PROseq</span>), <span class='fl'>6</span>)]
<span class='no'>ps_c_rep2</span> <span class='kw'>&lt;-</span> <span class='no'>PROseq</span>[<span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span>(<span class='fl'>6</span>, <span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span>(<span class='no'>PROseq</span>), <span class='fl'>6</span>)]

<span class='no'>ps_list</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span>(<span class='kw'>A_rep1</span> <span class='kw'>=</span> <span class='no'>ps_a_rep1</span>, <span class='kw'>A_rep2</span> <span class='kw'>=</span> <span class='no'>ps_a_rep2</span>,
                <span class='kw'>B_rep1</span> <span class='kw'>=</span> <span class='no'>ps_b_rep1</span>, <span class='kw'>B_rep2</span> <span class='kw'>=</span> <span class='no'>ps_b_rep2</span>,
                <span class='kw'>C_rep1</span> <span class='kw'>=</span> <span class='no'>ps_c_rep1</span>, <span class='kw'>C_rep2</span> <span class='kw'>=</span> <span class='no'>ps_c_rep2</span>)

<span class='co'># make flawed dataset (ranges in txs_dm6_chr4 not disjoint)</span>
<span class='co'>#    this means there is double-counting</span>
<span class='co'># also using discontinuous gene regions, as gene_ids are repeated</span>
<span class='no'>dds</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='getDESeqDataSet.html'>getDESeqDataSet</a></span>(<span class='no'>ps_list</span>,
                       <span class='no'>txs_dm6_chr4</span>,
                       <span class='kw'>gene_names</span> <span class='kw'>=</span> <span class='no'>txs_dm6_chr4</span>$<span class='no'>gene_id</span>,
                       <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fl'>1</span>)

<span class='no'>dds</span></div><div class='output co'>#&gt; class: DESeqDataSet 
#&gt; dim: 111 6 
#&gt; metadata(1): version
#&gt; assays(1): counts
#&gt; rownames(111): FBgn0267363 FBgn0266617 ... FBgn0039924 FBgn0027101
#&gt; rowData names(2): tx_name gene_id
#&gt; colnames(6): A_rep1 A_rep2 ... C_rep1 C_rep2
#&gt; colData names(2): condition replicate</div><div class='input'>
<span class='co'>#--------------------------------------------------#</span>
<span class='co'># getDESeqResults</span>
<span class='co'>#--------------------------------------------------#</span>

<span class='no'>res</span> <span class='kw'>&lt;-</span> <span class='fu'>getDESeqResults</span>(<span class='no'>dds</span>, <span class='st'>"B"</span>, <span class='st'>"A"</span>)</div><div class='output co'>#&gt; <span class='message'>estimating size factors</span></div><div class='output co'>#&gt; <span class='message'>estimating dispersions</span></div><div class='output co'>#&gt; <span class='message'>gene-wise dispersion estimates</span></div><div class='output co'>#&gt; <span class='message'>mean-dispersion relationship</span></div><div class='output co'>#&gt; <span class='message'>final dispersion estimates</span></div><div class='output co'>#&gt; <span class='message'>fitting model and testing</span></div><div class='input'>
<span class='no'>res</span></div><div class='output co'>#&gt; log2 fold change (MLE): condition B vs A 
#&gt; Wald test p-value: condition B vs A 
#&gt; DataFrame with 111 rows and 6 columns
#&gt;                baseMean log2FoldChange     lfcSE        stat    pvalue
#&gt;               &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt; &lt;numeric&gt;
#&gt; FBgn0267363    0.254249     -1.4228145  4.997084 -0.28472898  0.775852
#&gt; FBgn0266617   10.800541      0.1563201  1.033936  0.15118944  0.879826
#&gt; FBgn0265633    2.057706      0.0175256  2.125748  0.00824443  0.993422
#&gt; FBgn0264617   66.127353      0.2087670  0.467572  0.44649198  0.655242
#&gt; FBgn0085432 4678.846158     -0.2333084  0.236723 -0.98557490  0.324342
#&gt; ...                 ...            ...       ...         ...       ...
#&gt; FBgn0039928     685.650       0.126857  0.258212    0.491292  0.623220
#&gt; FBgn0052017     130.983       0.239103  0.418935    0.570741  0.568175
#&gt; FBgn0002521     174.375      -0.110680  0.342843   -0.322829  0.746825
#&gt; FBgn0039924     161.131       0.322486  0.456555    0.706346  0.479973
#&gt; FBgn0027101     361.983       0.233889  0.294518    0.794142  0.427113
#&gt;                  padj
#&gt;             &lt;numeric&gt;
#&gt; FBgn0267363  0.997413
#&gt; FBgn0266617  0.997413
#&gt; FBgn0265633  0.997413
#&gt; FBgn0264617  0.997413
#&gt; FBgn0085432  0.997413
#&gt; ...               ...
#&gt; FBgn0039928  0.997413
#&gt; FBgn0052017  0.997413
#&gt; FBgn0002521  0.997413
#&gt; FBgn0039924  0.997413
#&gt; FBgn0027101  0.997413</div><div class='input'>
<span class='no'>reslist</span> <span class='kw'>&lt;-</span> <span class='fu'>getDESeqResults</span>(<span class='no'>dds</span>, <span class='kw'>comparisons</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"B"</span>, <span class='st'>"A"</span>), <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"C"</span>, <span class='st'>"A"</span>)),
                           <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fl'>1</span>)
<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span>(<span class='no'>reslist</span>)</div><div class='output co'>#&gt; [1] "B_vs_A" "C_vs_A"</div><div class='input'>
<span class='no'>reslist</span>$<span class='no'>B_vs_A</span></div><div class='output co'>#&gt; log2 fold change (MLE): condition B vs A 
#&gt; Wald test p-value: condition B vs A 
#&gt; DataFrame with 111 rows and 6 columns
#&gt;                baseMean log2FoldChange     lfcSE        stat    pvalue
#&gt;               &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt; &lt;numeric&gt;
#&gt; FBgn0267363    0.254249     -1.4228145  4.997084 -0.28472898  0.775852
#&gt; FBgn0266617   10.800541      0.1563201  1.033936  0.15118944  0.879826
#&gt; FBgn0265633    2.057706      0.0175256  2.125748  0.00824443  0.993422
#&gt; FBgn0264617   66.127353      0.2087670  0.467572  0.44649198  0.655242
#&gt; FBgn0085432 4678.846158     -0.2333084  0.236723 -0.98557490  0.324342
#&gt; ...                 ...            ...       ...         ...       ...
#&gt; FBgn0039928     685.650       0.126857  0.258212    0.491292  0.623220
#&gt; FBgn0052017     130.983       0.239103  0.418935    0.570741  0.568175
#&gt; FBgn0002521     174.375      -0.110680  0.342843   -0.322829  0.746825
#&gt; FBgn0039924     161.131       0.322486  0.456555    0.706346  0.479973
#&gt; FBgn0027101     361.983       0.233889  0.294518    0.794142  0.427113
#&gt;                  padj
#&gt;             &lt;numeric&gt;
#&gt; FBgn0267363  0.997413
#&gt; FBgn0266617  0.997413
#&gt; FBgn0265633  0.997413
#&gt; FBgn0264617  0.997413
#&gt; FBgn0085432  0.997413
#&gt; ...               ...
#&gt; FBgn0039928  0.997413
#&gt; FBgn0052017  0.997413
#&gt; FBgn0002521  0.997413
#&gt; FBgn0039924  0.997413
#&gt; FBgn0027101  0.997413</div><div class='input'>
<span class='co'># or using a dataframe</span>
<span class='no'>reslist</span> <span class='kw'>&lt;-</span> <span class='fu'>getDESeqResults</span>(<span class='no'>dds</span>, <span class='kw'>comparisons</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span>(<span class='kw'>num</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"B"</span>, <span class='st'>"C"</span>),
                                                         <span class='kw'>den</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"A"</span>, <span class='st'>"A"</span>)),
                           <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fl'>1</span>)
<span class='no'>reslist</span>$<span class='no'>B_vs_A</span></div><div class='output co'>#&gt; log2 fold change (MLE): condition B vs A 
#&gt; Wald test p-value: condition B vs A 
#&gt; DataFrame with 111 rows and 6 columns
#&gt;                baseMean log2FoldChange     lfcSE        stat    pvalue
#&gt;               &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt; &lt;numeric&gt;
#&gt; FBgn0267363    0.254249     -1.4228145  4.997084 -0.28472898  0.775852
#&gt; FBgn0266617   10.800541      0.1563201  1.033936  0.15118944  0.879826
#&gt; FBgn0265633    2.057706      0.0175256  2.125748  0.00824443  0.993422
#&gt; FBgn0264617   66.127353      0.2087670  0.467572  0.44649198  0.655242
#&gt; FBgn0085432 4678.846158     -0.2333084  0.236723 -0.98557490  0.324342
#&gt; ...                 ...            ...       ...         ...       ...
#&gt; FBgn0039928     685.650       0.126857  0.258212    0.491292  0.623220
#&gt; FBgn0052017     130.983       0.239103  0.418935    0.570741  0.568175
#&gt; FBgn0002521     174.375      -0.110680  0.342843   -0.322829  0.746825
#&gt; FBgn0039924     161.131       0.322486  0.456555    0.706346  0.479973
#&gt; FBgn0027101     361.983       0.233889  0.294518    0.794142  0.427113
#&gt;                  padj
#&gt;             &lt;numeric&gt;
#&gt; FBgn0267363  0.997413
#&gt; FBgn0266617  0.997413
#&gt; FBgn0265633  0.997413
#&gt; FBgn0264617  0.997413
#&gt; FBgn0085432  0.997413
#&gt; ...               ...
#&gt; FBgn0039928  0.997413
#&gt; FBgn0052017  0.997413
#&gt; FBgn0002521  0.997413
#&gt; FBgn0039924  0.997413
#&gt; FBgn0027101  0.997413</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      <li><a href="#value">Value</a></li>
      <li><a href="#errors-when-ncores-gt-">Errors when <code>ncores &gt; 1</code></a></li>
      <li><a href="#see-also">See also</a></li>
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    <p>Mike DeBerardine</p>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


