<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Calculate pausing indices from user-supplied promoters &amp; genebodies — getPausingIndices • BRGenomics</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Calculate pausing indices from user-supplied promoters &amp; genebodies — getPausingIndices"><meta property="og:description" content="Pausing index (PI) is calculated for each gene (within matched
promoters.gr and genebodies.gr) as promoter-proximal (or pause
region) signal counts divided by genebody signal counts. If
length.normalize = TRUE (recommended), the signal counts within each
range in promoters.gr and genebodies.gr are divided by their
respective range widths (region lengths) before pausing indices are
calculated."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Overview.html">Overview</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Guides
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/GettingStarted.html">Getting Started</a>
    </li>
    <li>
      <a href="../articles/ImportingModifyingAnnotations.html">Importing &amp; Modifying Annotations</a>
    </li>
    <li>
      <a href="../articles/ImportingProcessingData.html">Importing &amp; Processing Data</a>
    </li>
    <li>
      <a href="../articles/SignalCounting.html">Signal Counting</a>
    </li>
    <li>
      <a href="../articles/AnalyzingMultipleDatasets.html">Analyzing Multiple Datasets</a>
    </li>
    <li>
      <a href="../articles/ProfilePlotsAndBootstrapping.html">Profile Plots &amp; Bootstrapping</a>
    </li>
    <li>
      <a href="../articles/SpikeInNormalization.html">Spike-in Normalization</a>
    </li>
    <li>
      <a href="../articles/DESeq2WithGlobalPerturbations.html">DESeq2 with Global Perturbations</a>
    </li>
    <li>
      <a href="../articles/SequenceExtraction.html">Sequence Extraction</a>
    </li>
  </ul></li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/mdeber/BRGenomics" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Calculate pausing indices from user-supplied promoters &amp; genebodies</h1>
    <small class="dont-index">Source: <a href="https://github.com/mdeber/BRGenomics/blob/HEAD/R/signal_counting.R" class="external-link"><code>R/signal_counting.R</code></a></small>
    <div class="hidden name"><code>getPausingIndices.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Pausing index (PI) is calculated for each gene (within matched
<code>promoters.gr</code> and <code>genebodies.gr</code>) as promoter-proximal (or pause
region) signal counts divided by genebody signal counts. If
<code>length.normalize = TRUE</code> (recommended), the signal counts within each
range in <code>promoters.gr</code> and <code>genebodies.gr</code> are divided by their
respective range widths (region lengths) before pausing indices are
calculated.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">getPausingIndices</span><span class="op">(</span></span>
<span>  <span class="va">dataset.gr</span>,</span>
<span>  <span class="va">promoters.gr</span>,</span>
<span>  <span class="va">genebodies.gr</span>,</span>
<span>  field <span class="op">=</span> <span class="st">"score"</span>,</span>
<span>  length.normalize <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  remove.empty <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  blacklist <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  melt <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  region_names <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  expand_ranges <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  ncores <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"mc.cores"</span>, <span class="fl">2L</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>dataset.gr</dt>
<dd><p>A GRanges object in which signal is contained in metadata
(typically in the "score" field), or a named list of such GRanges objects.</p></dd>


<dt>promoters.gr</dt>
<dd><p>A GRanges object containing promoter-proximal regions of
interest.</p></dd>


<dt>genebodies.gr</dt>
<dd><p>A GRanges object containing genebody regions of
interest.</p></dd>


<dt>field</dt>
<dd><p>The metadata field of <code>dataset.gr</code> to be counted. If
<code>length(field) &gt; 1</code>, a dataframe is returned containing the pausing
indices for each region in each field. If <code>field</code> not found in
<code>names(mcols(dataset.gr))</code>, will default to using all fields found in
<code>dataset.gr</code>. If <code>dataset.gr</code> is a list, a single <code>field</code>
should be given, or <code>length(field)</code> should be the equal to the number
of datasets in <code>dataset.gr</code>.</p></dd>


<dt>length.normalize</dt>
<dd><p>A logical indicating if signal counts within regions
of interest should be length normalized. The default is <code>TRUE</code>, which
is recommended, especially if input regions don't all have the same width.</p></dd>


<dt>remove.empty</dt>
<dd><p>A logical indicating if genes without any signal in
<code>promoters.gr</code> should be removed. No genes are filtered by default. If
<code>dataset.gr</code> is a list of datasets, or if <code>length(field) &gt; 1</code>,
regions are filtered unless they have promoter signal in all datasets.</p></dd>


<dt>blacklist</dt>
<dd><p>An optional GRanges object containing regions that should be
excluded from signal counting. If <code>length.normalize = TRUE</code>,
blacklisted positions will be excluded from length calculations. Users
should take care to note if regions of interest substantially overlap
blacklisted positions.</p></dd>


<dt>melt</dt>
<dd><p>If <code>melt = TRUE</code>, a dataframe is returned containing a
column for regions and another column for pausing indices. If multiple
datasets are given (if <code>dataset.gr</code> is a list or if
<code>length(field) &gt; 1</code>), the output dataframe is melted to contain a
third column indicating the sample names. (See section on return values
below).</p></dd>


<dt>region_names</dt>
<dd><p>If <code>melt = TRUE</code>, an optional vector of names for
the regions in <code>regions.gr</code>. If left as <code>NULL</code>, indices of
<code>regions.gr</code> are used instead.</p></dd>


<dt>expand_ranges</dt>
<dd><p>Logical indicating if ranges in <code>dataset.gr</code> should
be treated as descriptions of single molecules (<code>FALSE</code>), or if ranges
should be treated as representing multiple adjacent positions with the same
signal (<code>TRUE</code>). See <code><a href="getCountsByRegions.html">
getCountsByRegions</a></code>.</p></dd>


<dt>ncores</dt>
<dd><p>Multiple cores will only be used if <code>dataset.gr</code> is a list
of multiple datasets, or if <code>length(field) &gt; 1</code>.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A vector parallel to the input genelist, unless <code>remove.empty =
  TRUE</code>, in which case the vector may be shorter. If <code>dataset.gr</code> is a
  list, or if <code>length(field) &gt; 1</code>, a dataframe is returned, containing a
  column for each field. However, if <code>melt = TRUE</code>, dataframes contain
  one column to indicate regions (either by their indices, or by</p>
<p></p>
<p><code>region_names</code>, if given), another column to indicate signal, and a
  third column containing the sample name (unless <code>dataset.gr</code> is a
  single GRanges object).</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="getCountsByRegions.html">getCountsByRegions</a></code></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Mike DeBerardine</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"PROseq"</span><span class="op">)</span> <span class="co"># load included PROseq data</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"txs_dm6_chr4"</span><span class="op">)</span> <span class="co"># load included transcripts</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># Get promoter-proximal and genebody regions</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># genebodies from +300 to 300 bp before the poly-A site</span></span></span>
<span class="r-in"><span><span class="va">gb</span> <span class="op">&lt;-</span> <span class="fu"><a href="genebodies.html">genebodies</a></span><span class="op">(</span><span class="va">txs_dm6_chr4</span>, <span class="fl">300</span>, <span class="op">-</span><span class="fl">300</span>, min.window <span class="op">=</span> <span class="fl">400</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># get the transcripts that are large enough (&gt;1kb in size)</span></span></span>
<span class="r-in"><span><span class="va">txs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">txs_dm6_chr4</span>, <span class="va">tx_name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">gb</span><span class="op">$</span><span class="va">tx_name</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># for the same transcripts, promoter-proximal region from 0 to +100</span></span></span>
<span class="r-in"><span><span class="va">pr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/intra-range-methods.html" class="external-link">promoters</a></span><span class="op">(</span><span class="va">txs</span>, <span class="fl">0</span>, <span class="fl">100</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># Calculate pausing indices</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">pidx</span> <span class="op">&lt;-</span> <span class="fu">getPausingIndices</span><span class="op">(</span><span class="va">PROseq</span>, <span class="va">pr</span>, <span class="va">gb</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">txs</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 318</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">pidx</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 318</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pidx</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1]  0.00000  3.50500  0.00000  0.00000 29.34224 28.41707</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># Without length normalization</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span> <span class="fu">getPausingIndices</span><span class="op">(</span><span class="va">PROseq</span>, <span class="va">pr</span>, <span class="va">gb</span>, length.normalize <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 0.00000000 0.50000000 0.00000000 0.00000000 0.09316770 0.07329843</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># Removing empty means the values no longer match the genelist</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">pidx_signal</span> <span class="op">&lt;-</span> <span class="fu">getPausingIndices</span><span class="op">(</span><span class="va">PROseq</span>, <span class="va">pr</span>, <span class="va">gb</span>, remove.empty <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">pidx_signal</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 238</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

