<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Calculate pausing indices from user-supplied promoters &amp; genebodies — getPausingIndices • BRGenomics</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Calculate pausing indices from user-supplied promoters &amp; genebodies — getPausingIndices" />
<meta property="og:description" content="Pausing index (PI) is calculated for each gene (within matched
promoters.gr and genebodies.gr) as promoter-proximal (or pause
region) signal counts divided by genebody signal counts. If
length.normalize = TRUE (recommended), the signal counts within each
range in promoters.gr and genebodies.gr are divided by their
respective range widths (region lengths) before pausing indices are
calculated." />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.26</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/BRGenomicsOverview.html">Basepair-Resolution Analysis with BRGenomics</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Calculate pausing indices from user-supplied promoters &amp; genebodies</h1>
    
    <div class="hidden name"><code>getPausingIndices.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Pausing index (PI) is calculated for each gene (within matched
<code>promoters.gr</code> and <code>genebodies.gr</code>) as promoter-proximal (or pause
region) signal counts divided by genebody signal counts. If
<code>length.normalize = TRUE</code> (recommended), the signal counts within each
range in <code>promoters.gr</code> and <code>genebodies.gr</code> are divided by their
respective range widths (region lengths) before pausing indices are
calculated.</p>
    </div>

    <pre class="usage"><span class='fu'>getPausingIndices</span>(
  <span class='no'>dataset.gr</span>,
  <span class='no'>promoters.gr</span>,
  <span class='no'>genebodies.gr</span>,
  <span class='kw'>field</span> <span class='kw'>=</span> <span class='st'>"score"</span>,
  <span class='kw'>length.normalize</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>remove.empty</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>blacklist</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>melt</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>region_names</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>expand_ranges</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fu'>detectCores</span>()
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>dataset.gr</th>
      <td><p>A GRanges object in which signal is contained in metadata
(typically in the "score" field), or a named list of such GRanges objects.</p></td>
    </tr>
    <tr>
      <th>promoters.gr</th>
      <td><p>A GRanges object containing promoter-proximal regions of
interest.</p></td>
    </tr>
    <tr>
      <th>genebodies.gr</th>
      <td><p>A GRanges object containing genebody regions of
interest.</p></td>
    </tr>
    <tr>
      <th>field</th>
      <td><p>The metadata field of <code>dataset.gr</code> to be counted. If
<code>length(field) &gt; 1</code>, a dataframe is returned containing the pausing
indices for each region in each field. If <code>field</code> not found in
<code><a href='https://rdrr.io/r/base/names.html'>names(mcols(dataset.gr))</a></code>, will default to using all fields found in
<code>dataset.gr</code>. If <code>dataset.gr</code> is a list, a single <code>field</code>
should be given, or <code><a href='https://rdrr.io/r/base/length.html'>length(field)</a></code> should be the equal to the number
of datasets in <code>dataset.gr</code>.</p></td>
    </tr>
    <tr>
      <th>length.normalize</th>
      <td><p>A logical indicating if signal counts within regions
of interest should be length normalized. The default is <code>TRUE</code>, which
is recommended, especially if input regions don't all have the same width.</p></td>
    </tr>
    <tr>
      <th>remove.empty</th>
      <td><p>A logical indicating if genes without any signal in
<code>promoters.gr</code> should be removed. No genes are filtered by default. If
<code>dataset.gr</code> is a list of datasets, or if <code>length(field) &gt; 1</code>,
regions are filtered unless they have promoter signal in all datasets.</p></td>
    </tr>
    <tr>
      <th>blacklist</th>
      <td><p>An optional GRanges object containing regions that should be
excluded from signal counting. If <code>length.normalize = TRUE</code>,
blacklisted positions will be excluded from length calculations. Users
should take care to note if regions of interest substantially overlap
blacklisted positions.</p></td>
    </tr>
    <tr>
      <th>melt</th>
      <td><p>If <code>melt = TRUE</code>, a dataframe is returned containing a
column for regions and another column for pausing indices. If multiple
datasets are given (if <code>dataset.gr</code> is a list or if
<code>length(field) &gt; 1</code>), the output dataframe is melted to contain a
third column indicating the sample names. (See section on return values
below).</p></td>
    </tr>
    <tr>
      <th>region_names</th>
      <td><p>If <code>melt = TRUE</code>, an optional vector of names for
the regions in <code>regions.gr</code>. If left as <code>NULL</code>, indices of
<code>regions.gr</code> are used instead.</p></td>
    </tr>
    <tr>
      <th>expand_ranges</th>
      <td><p>Logical indicating if ranges in <code>dataset.gr</code> should
be treated as descriptions of single molecules (<code>FALSE</code>), or if ranges
should be treated as representing multiple adjacent positions with the same
signal (<code>TRUE</code>). See <code><a href='getCountsByRegions.html'>
getCountsByRegions</a></code>.</p></td>
    </tr>
    <tr>
      <th>ncores</th>
      <td><p>Multiple cores will only be used if <code>dataset.gr</code> is a list
of multiple datasets, or if <code>length(field) &gt; 1</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A vector parallel to the input genelist, unless <code>remove.empty =
  TRUE</code>, in which case the vector may be shorter. If <code>dataset.gr</code> is a
  list, or if <code>length(field) &gt; 1</code>, a dataframe is returned, containing a
  column for each field. However, if <code>melt = TRUE</code>, dataframes contain
  one column to indicate regions (either by their indices, or by
  <code>region_names</code>, if given), another column to indicate signal, and a
  third column containing the sample name (unless <code>dataset.gr</code> is a
  single GRanges object).</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='getCountsByRegions.html'>getCountsByRegions</a></code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='st'>"PROseq"</span>) <span class='co'># load included PROseq data</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='st'>"txs_dm6_chr4"</span>) <span class='co'># load included transcripts</span>

<span class='co'>#--------------------------------------------------#</span>
<span class='co'># Get promoter-proximal and genebody regions</span>
<span class='co'>#--------------------------------------------------#</span>

<span class='co'># genebodies from +300 to 300 bp before the poly-A site</span>
<span class='no'>gb</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='genebodies.html'>genebodies</a></span>(<span class='no'>txs_dm6_chr4</span>, <span class='fl'>300</span>, -<span class='fl'>300</span>, <span class='kw'>min.window</span> <span class='kw'>=</span> <span class='fl'>400</span>)

<span class='co'># get the transcripts that are large enough (&gt;1kb in size)</span>
<span class='no'>txs</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/subset.html'>subset</a></span>(<span class='no'>txs_dm6_chr4</span>, <span class='no'>tx_name</span> <span class='kw'>%in%</span> <span class='no'>gb</span>$<span class='no'>tx_name</span>)

<span class='co'># for the same transcripts, promoter-proximal region from 0 to +100</span>
<span class='no'>pr</span> <span class='kw'>&lt;-</span> <span class='fu'>promoters</span>(<span class='no'>txs</span>, <span class='fl'>0</span>, <span class='fl'>100</span>)

<span class='co'>#--------------------------------------------------#</span>
<span class='co'># Calculate pausing indices</span>
<span class='co'>#--------------------------------------------------#</span>

<span class='no'>pidx</span> <span class='kw'>&lt;-</span> <span class='fu'>getPausingIndices</span>(<span class='no'>PROseq</span>, <span class='no'>pr</span>, <span class='no'>gb</span>)

<span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span>(<span class='no'>txs</span>)</div><div class='output co'>#&gt; [1] 318</div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span>(<span class='no'>pidx</span>)</div><div class='output co'>#&gt; [1] 318</div><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span>(<span class='no'>pidx</span>)</div><div class='output co'>#&gt; [1]  0.00000  3.50500  0.00000  0.00000 29.34224 28.41707</div><div class='input'>
<span class='co'>#--------------------------------------------------#</span>
<span class='co'># Without length normalization</span>
<span class='co'>#--------------------------------------------------#</span>

<span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span>( <span class='fu'>getPausingIndices</span>(<span class='no'>PROseq</span>, <span class='no'>pr</span>, <span class='no'>gb</span>, <span class='kw'>length.normalize</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>) )</div><div class='output co'>#&gt; [1] 0.00000000 0.50000000 0.00000000 0.00000000 0.09316770 0.07329843</div><div class='input'>
<span class='co'>#--------------------------------------------------#</span>
<span class='co'># Removing empty means the values no longer match the genelist</span>
<span class='co'>#--------------------------------------------------#</span>

<span class='no'>pidx_signal</span> <span class='kw'>&lt;-</span> <span class='fu'>getPausingIndices</span>(<span class='no'>PROseq</span>, <span class='no'>pr</span>, <span class='no'>gb</span>, <span class='kw'>remove.empty</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)

<span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span>(<span class='no'>pidx_signal</span>)</div><div class='output co'>#&gt; [1] 238</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      <li><a href="#value">Value</a></li>
      <li><a href="#see-also">See also</a></li>
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    <p>Mike DeBerardine</p>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


