<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Get strand-specific coverage — getStrandedCoverage • BRGenomics</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Get strand-specific coverage — getStrandedCoverage"><meta property="og:description" content="Computes strand-specific coverage signal, and returns a GRanges object.
Function also works for non-strand-specific data."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Overview.html">Overview</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Guides
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/GettingStarted.html">Getting Started</a>
    </li>
    <li>
      <a href="../articles/ImportingModifyingAnnotations.html">Importing &amp; Modifying Annotations</a>
    </li>
    <li>
      <a href="../articles/ImportingProcessingData.html">Importing &amp; Processing Data</a>
    </li>
    <li>
      <a href="../articles/SignalCounting.html">Signal Counting</a>
    </li>
    <li>
      <a href="../articles/AnalyzingMultipleDatasets.html">Analyzing Multiple Datasets</a>
    </li>
    <li>
      <a href="../articles/ProfilePlotsAndBootstrapping.html">Profile Plots &amp; Bootstrapping</a>
    </li>
    <li>
      <a href="../articles/SpikeInNormalization.html">Spike-in Normalization</a>
    </li>
    <li>
      <a href="../articles/DESeq2WithGlobalPerturbations.html">DESeq2 with Global Perturbations</a>
    </li>
    <li>
      <a href="../articles/SequenceExtraction.html">Sequence Extraction</a>
    </li>
  </ul></li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/mdeber/BRGenomics" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Get strand-specific coverage</h1>
    <small class="dont-index">Source: <a href="https://github.com/mdeber/BRGenomics/blob/HEAD/R/dataset_functions.R" class="external-link"><code>R/dataset_functions.R</code></a></small>
    <div class="hidden name"><code>getStrandedCoverage.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Computes strand-specific coverage signal, and returns a GRanges object.
Function also works for non-strand-specific data.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">getStrandedCoverage</span><span class="op">(</span></span>
<span>  <span class="va">dataset.gr</span>,</span>
<span>  field <span class="op">=</span> <span class="st">"score"</span>,</span>
<span>  ncores <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"mc.cores"</span>, <span class="fl">2L</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>dataset.gr</dt>
<dd><p>A GRanges object either containing ranges for each read, or
one in which readcounts for individual ranges are contained in metadata
(typically in the "score" field). <code>dataset.gr</code> can also be a list of
such GRanges objects.</p></dd>


<dt>field</dt>
<dd><p>The name of the metadata field that contains readcounts. If no
metadata field contains readcounts, and each range represents a single
read, set to NULL.</p></dd>


<dt>ncores</dt>
<dd><p>Number of cores to use for calculating coverage. For a single
dataset, the max that will be used is 3, one for each possible strand
(plus, minus, and unstranded). More cores can be used if <code>dataset.gr</code>
is a list.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A GRanges object with signal in the "score" metadata column. Note
  that the output is <em>not</em> automatically converted into a</p>
<p></p>
<p><code><a href="makeGRangesBRG.html">"basepair-resolution"</a></code> GRanges
  object.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="makeGRangesBRG.html">makeGRangesBRG</a></code>,
  <code><a href="https://rdrr.io/pkg/IRanges/man/coverage-methods.html" class="external-link">GenomicRanges::coverage</a></code></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Mike DeBerardine</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># Using included full-read data</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># -&gt; whole-read coverage sacrifices meaningful readcount</span></span></span>
<span class="r-in"><span><span class="co">#    information, but can be useful for visualization,</span></span></span>
<span class="r-in"><span><span class="co">#    e.g. for looking at RNA-seq data in a genome browser</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"PROseq_paired"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">PROseq_paired</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 6 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames      ranges strand |     score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr4   1270-1295      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr4 41411-41428      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chr4 42556-42590      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chr4 42559-42588      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [5]     chr4 42559-42593      + |         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [6]     chr4 42560-42593      + |         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 7 sequences from an unspecified genome</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">getStrandedCoverage</span><span class="op">(</span><span class="va">PROseq_paired</span>, ncores <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 6 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames      ranges strand |     score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr4   1270-1295      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr4 41411-41428      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chr4 42556-42558      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chr4       42559      + |         5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [5]     chr4       42560      + |         9</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [6]     chr4 42561-42562      + |        10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 7 sequences from an unspecified genome</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># Getting coverage from single bases of single reads</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># included PROseq data is already single-base coverage</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"PROseq"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/range.html" class="external-link">range</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">PROseq</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 1 1</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># undo coverage for the first 100 positions</span></span></span>
<span class="r-in"><span><span class="va">ps</span> <span class="op">&lt;-</span> <span class="va">PROseq</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">100</span><span class="op">]</span></span></span>
<span class="r-in"><span><span class="va">ps_reads</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="va">ps</span>, times <span class="op">=</span> <span class="va">ps</span><span class="op">$</span><span class="va">score</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols</a></span><span class="op">(</span><span class="va">ps_reads</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">ps_reads</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 6 ranges and 0 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr4      1295      +</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr4     41428      +</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chr4     42588      +</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chr4     42590      +</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [5]     chr4     42590      +</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [6]     chr4     42593      +</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 7 sequences from an unspecified genome</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># re-create coverage</span></span></span>
<span class="r-in"><span><span class="fu">getStrandedCoverage</span><span class="op">(</span><span class="va">ps_reads</span>, field <span class="op">=</span> <span class="cn">NULL</span>, ncores <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 6 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |     score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr4      1295      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr4     41428      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chr4     42588      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chr4     42590      + |         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [5]     chr4     42593      + |         5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [6]     chr4     42594      + |         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 7 sequences from an unspecified genome</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># Reversing makeGRangesBRG</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># -&gt; getStrandedCoverage doesn't return single-width</span></span></span>
<span class="r-in"><span><span class="co">#    GRanges, which is useful because getting coverage</span></span></span>
<span class="r-in"><span><span class="co">#    will merge adjacent bases with equivalent scores</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># included PROseq data is already single-width</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/range.html" class="external-link">range</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">PROseq</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 1 1</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/inter-range-methods.html" class="external-link">isDisjoint</a></span><span class="op">(</span><span class="va">PROseq</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] TRUE</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">ps_cov</span> <span class="op">&lt;-</span> <span class="fu">getStrandedCoverage</span><span class="op">(</span><span class="va">PROseq</span>, ncores <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/range.html" class="external-link">range</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">ps_cov</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 1 6</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu">score</span><span class="op">(</span><span class="va">PROseq</span><span class="op">)</span><span class="op">)</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu">score</span><span class="op">(</span><span class="va">ps_cov</span><span class="op">)</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">ps_cov</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] TRUE</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># -&gt; Look specifically at ranges that could be combined</span></span></span>
<span class="r-in"><span><span class="va">neighbors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/intra-range-methods.html" class="external-link">shift</a></span><span class="op">(</span><span class="va">PROseq</span>, <span class="fl">1</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/intra-range-methods.html" class="external-link">shift</a></span><span class="op">(</span><span class="va">PROseq</span>, <span class="op">-</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">findOverlaps</a></span><span class="op">(</span><span class="va">PROseq</span>, <span class="va">neighbors</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Hits-class.html" class="external-link">from</a></span><span class="op">(</span><span class="va">hits</span><span class="op">)</span><span class="op">)</span> <span class="co"># indices for PROseq with neighbor</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">PROseq</span><span class="op">[</span><span class="va">idx</span><span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 13699 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           seqnames    ranges strand |     score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       [1]     chr4     42593      + |         5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       [2]     chr4     42594      + |         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       [3]     chr4     42595      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       [4]     chr4     42596      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       [5]     chr4     42618      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       ...      ...       ...    ... .       ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [13695]     chr4   1306889      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [13696]     chr4   1307114      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [13697]     chr4   1307115      - |         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [13698]     chr4   1307300      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [13699]     chr4   1307301      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 7 sequences from an unspecified genome</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">getStrandedCoverage</span><span class="op">(</span><span class="va">PROseq</span><span class="op">[</span><span class="va">idx</span><span class="op">]</span>, ncores <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 9704 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          seqnames          ranges strand |     score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      [1]     chr4           42593      + |         5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      [2]     chr4           42594      + |         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      [3]     chr4     42595-42596      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      [4]     chr4           42618      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      [5]     chr4           42619      + |         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      ...      ...             ...    ... .       ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [9700]     chr4 1305972-1305973      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [9701]     chr4 1306888-1306889      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [9702]     chr4         1307114      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [9703]     chr4         1307115      - |         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [9704]     chr4 1307300-1307301      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 7 sequences from an unspecified genome</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

