<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Get strand-specific coverage — getStrandedCoverage • BRGenomics</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Get strand-specific coverage — getStrandedCoverage" />
<meta property="og:description" content="Computes strand-specific coverage signal, and returns a GRanges object.
Function also works for non-strand-specific data." />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.7.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/BRGenomicsOverview.html">Basepair-Resolution Analysis with BRGenomics</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Get strand-specific coverage</h1>
    
    <div class="hidden name"><code>getStrandedCoverage.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Computes strand-specific coverage signal, and returns a GRanges object.
Function also works for non-strand-specific data.</p>
    </div>

    <pre class="usage"><span class='fu'>getStrandedCoverage</span>(<span class='no'>dataset.gr</span>, <span class='kw'>field</span> <span class='kw'>=</span> <span class='st'>"score"</span>, <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fu'>detectCores</span>())</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>dataset.gr</th>
      <td><p>A GRanges object either containing ranges for each read, or
one in which readcounts for individual ranges are contained in metadata
(typically in the "score" field). <code>dataset.gr</code> can also be a list of
such GRanges objects.</p></td>
    </tr>
    <tr>
      <th>field</th>
      <td><p>The name of the metadata field that contains readcounts. If no
metadata field contains readcounts, and each range represents a single
read, set to NULL.</p></td>
    </tr>
    <tr>
      <th>ncores</th>
      <td><p>Number of cores to use for calculating coverage. For a single
dataset, the max that will be used is 3, one for each possible strand
(plus, minus, and unstranded). More cores can be used if <code>dataset.gr</code>
is a list.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A GRanges object with signal in the "score" metadata column. Note
  that the output is <em>not</em> automatically converted into a
  <code><a href='makeGRangesBRG.html'>"basepair-resolution"</a></code> GRanges
  object.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='makeGRangesBRG.html'>makeGRangesBRG</a></code>,
  <code><a href='https://rdrr.io/pkg/GenomicRanges/man/coverage-methods.html'>GenomicRanges::coverage</a></code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>#--------------------------------------------------#</span>
<span class='co'># Using included full-read data</span>
<span class='co'>#--------------------------------------------------#</span>
<span class='co'># -&gt; whole-read coverage sacrifices meaningful readcount</span>
<span class='co'>#    information, but can be useful for visualization,</span>
<span class='co'>#    e.g. for looking at RNA-seq data in a genome browser</span>

<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='st'>"PROseq_paired"</span>)

<span class='no'>PROseq_paired</span>[<span class='fl'>1</span>:<span class='fl'>6</span>]</div><div class='output co'>#&gt; GRanges object with 6 ranges and 1 metadata column:
#&gt;       seqnames      ranges strand |     score
#&gt;          &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
#&gt;   [1]     chr4   1270-1296      + |         1
#&gt;   [2]     chr4 41411-41429      + |         1
#&gt;   [3]     chr4 42556-42591      + |         1
#&gt;   [4]     chr4 42559-42589      + |         1
#&gt;   [5]     chr4 42559-42594      + |         3
#&gt;   [6]     chr4 42560-42594      + |         2
#&gt;   -------
#&gt;   seqinfo: 1870 sequences from an unspecified genome</div><div class='input'>
<span class='fu'>getStrandedCoverage</span>(<span class='no'>PROseq_paired</span>, <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fl'>2</span>)[<span class='fl'>1</span>:<span class='fl'>6</span>]</div><div class='output co'>#&gt; GRanges object with 6 ranges and 1 metadata column:
#&gt;       seqnames      ranges strand |     score
#&gt;          &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
#&gt;   [1]     chr4   1270-1296      + |         1
#&gt;   [2]     chr4 41411-41429      + |         1
#&gt;   [3]     chr4 42556-42558      + |         1
#&gt;   [4]     chr4       42559      + |         5
#&gt;   [5]     chr4       42560      + |         9
#&gt;   [6]     chr4 42561-42562      + |        10
#&gt;   -------
#&gt;   seqinfo: 1870 sequences from an unspecified genome</div><div class='input'>
<span class='co'>#--------------------------------------------------#</span>
<span class='co'># Getting coverage from single bases of single reads</span>
<span class='co'>#--------------------------------------------------#</span>

<span class='co'># included PROseq data is already single-base coverage</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='st'>"PROseq"</span>)
<span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span>(<span class='fu'>width</span>(<span class='no'>PROseq</span>))</div><div class='output co'>#&gt; [1] 1 1</div><div class='input'>
<span class='co'># undo coverage for the first 100 positions</span>
<span class='no'>ps</span> <span class='kw'>&lt;-</span> <span class='no'>PROseq</span>[<span class='fl'>1</span>:<span class='fl'>100</span>]
<span class='no'>ps_reads</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span>(<span class='no'>ps</span>, <span class='kw'>times</span> <span class='kw'>=</span> <span class='no'>ps</span>$<span class='no'>score</span>)
<span class='fu'>mcols</span>(<span class='no'>ps_reads</span>) <span class='kw'>&lt;-</span> <span class='kw'>NULL</span>

<span class='no'>ps_reads</span>[<span class='fl'>1</span>:<span class='fl'>6</span>]</div><div class='output co'>#&gt; GRanges object with 6 ranges and 0 metadata columns:
#&gt;       seqnames    ranges strand
#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
#&gt;   [1]     chr4      1296      +
#&gt;   [2]     chr4     41429      +
#&gt;   [3]     chr4     42589      +
#&gt;   [4]     chr4     42591      +
#&gt;   [5]     chr4     42591      +
#&gt;   [6]     chr4     42594      +
#&gt;   -------
#&gt;   seqinfo: 1870 sequences from an unspecified genome</div><div class='input'>
<span class='co'># re-create coverage</span>
<span class='fu'>getStrandedCoverage</span>(<span class='no'>ps_reads</span>, <span class='kw'>field</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fl'>2</span>)[<span class='fl'>1</span>:<span class='fl'>6</span>]</div><div class='output co'>#&gt; GRanges object with 6 ranges and 1 metadata column:
#&gt;       seqnames    ranges strand |     score
#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
#&gt;   [1]     chr4      1296      + |         1
#&gt;   [2]     chr4     41429      + |         1
#&gt;   [3]     chr4     42589      + |         1
#&gt;   [4]     chr4     42591      + |         2
#&gt;   [5]     chr4     42594      + |         5
#&gt;   [6]     chr4     42595      + |         2
#&gt;   -------
#&gt;   seqinfo: 1870 sequences from an unspecified genome</div><div class='input'>
<span class='co'>#--------------------------------------------------#</span>
<span class='co'># Reversing makeGRangesBRG</span>
<span class='co'>#--------------------------------------------------#</span>
<span class='co'># -&gt; getStrandedCoverage doesn't return single-width</span>
<span class='co'>#    GRanges, which is useful because getting coverage</span>
<span class='co'>#    will merge adjacent bases with equivalent scores</span>

<span class='co'># included PROseq data is already single-width</span>
<span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span>(<span class='fu'>width</span>(<span class='no'>PROseq</span>))</div><div class='output co'>#&gt; [1] 1 1</div><div class='input'><span class='fu'>isDisjoint</span>(<span class='no'>PROseq</span>)</div><div class='output co'>#&gt; [1] TRUE</div><div class='input'>
<span class='no'>ps_cov</span> <span class='kw'>&lt;-</span> <span class='fu'>getStrandedCoverage</span>(<span class='no'>PROseq</span>, <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fl'>2</span>)

<span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span>(<span class='fu'>width</span>(<span class='no'>ps_cov</span>))</div><div class='output co'>#&gt; [1] 1 6</div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/sum.html'>sum</a></span>(<span class='fu'>score</span>(<span class='no'>PROseq</span>)) <span class='kw'>==</span> <span class='fu'><a href='https://rdrr.io/r/base/sum.html'>sum</a></span>(<span class='fu'>score</span>(<span class='no'>ps_cov</span>) * <span class='fu'>width</span>(<span class='no'>ps_cov</span>))</div><div class='output co'>#&gt; [1] TRUE</div><div class='input'>
<span class='co'># -&gt; Look specifically at ranges that could be combined</span>
<span class='no'>neighbors</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fu'>shift</span>(<span class='no'>PROseq</span>, <span class='fl'>1</span>), <span class='fu'>shift</span>(<span class='no'>PROseq</span>, -<span class='fl'>1</span>))
<span class='no'>hits</span> <span class='kw'>&lt;-</span> <span class='fu'>findOverlaps</span>(<span class='no'>PROseq</span>, <span class='no'>neighbors</span>)
<span class='no'>idx</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/unique.html'>unique</a></span>(<span class='fu'>from</span>(<span class='no'>hits</span>)) <span class='co'># indices for PROseq with neighbor</span>

<span class='no'>PROseq</span>[<span class='no'>idx</span>]</div><div class='output co'>#&gt; GRanges object with 13699 ranges and 1 metadata column:
#&gt;           seqnames    ranges strand |     score
#&gt;              &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
#&gt;       [1]     chr4     42594      + |         5
#&gt;       [2]     chr4     42595      + |         2
#&gt;       [3]     chr4     42596      + |         1
#&gt;       [4]     chr4     42597      + |         1
#&gt;       [5]     chr4     42619      + |         1
#&gt;       ...      ...       ...    ... .       ...
#&gt;   [13695]     chr4   1306888      - |         1
#&gt;   [13696]     chr4   1307113      - |         1
#&gt;   [13697]     chr4   1307114      - |         2
#&gt;   [13698]     chr4   1307299      - |         1
#&gt;   [13699]     chr4   1307300      - |         1
#&gt;   -------
#&gt;   seqinfo: 1870 sequences from an unspecified genome</div><div class='input'>
<span class='fu'>getStrandedCoverage</span>(<span class='no'>PROseq</span>[<span class='no'>idx</span>], <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fl'>2</span>)</div><div class='output co'>#&gt; GRanges object with 9704 ranges and 1 metadata column:
#&gt;          seqnames          ranges strand |     score
#&gt;             &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
#&gt;      [1]     chr4           42594      + |         5
#&gt;      [2]     chr4           42595      + |         2
#&gt;      [3]     chr4     42596-42597      + |         1
#&gt;      [4]     chr4           42619      + |         1
#&gt;      [5]     chr4           42620      + |         2
#&gt;      ...      ...             ...    ... .       ...
#&gt;   [9700]     chr4 1305971-1305972      - |         1
#&gt;   [9701]     chr4 1306887-1306888      - |         1
#&gt;   [9702]     chr4         1307113      - |         1
#&gt;   [9703]     chr4         1307114      - |         2
#&gt;   [9704]     chr4 1307299-1307300      - |         1
#&gt;   -------
#&gt;   seqinfo: 1870 sequences from an unspecified genome</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      <li><a href="#value">Value</a></li>
      <li><a href="#see-also">See also</a></li>
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    <p>Mike DeBerardine</p>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


