<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Calculating spike-in normalization factors — getSpikeInNFs • BRGenomics</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Calculating spike-in normalization factors — getSpikeInNFs" />
<meta property="og:description" content="Use getSpikeInNFs to obtain the spike-in normalization factors, or
spikeInNormGRanges to return the input GRanges objects with their
readcounts spike-in normalized." />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.22</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/BRGenomicsOverview.html">Basepair-Resolution Analysis with BRGenomics</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Calculating spike-in normalization factors</h1>
    
    <div class="hidden name"><code>getSpikeInNFs.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Use <code>getSpikeInNFs</code> to obtain the spike-in normalization factors, or
<code>spikeInNormGRanges</code> to return the input GRanges objects with their
readcounts spike-in normalized.</p>
    </div>

    <pre class="usage"><span class='fu'>getSpikeInNFs</span>(
  <span class='no'>dataset.gr</span>,
  <span class='kw'>si_pattern</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>si_names</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>method</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"SRPMC"</span>, <span class='st'>"SNR"</span>, <span class='st'>"RPM"</span>),
  <span class='kw'>batch_norm</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>ctrl_pattern</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>ctrl_names</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>field</span> <span class='kw'>=</span> <span class='st'>"score"</span>,
  <span class='kw'>sample_names</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>expand_ranges</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fu'>detectCores</span>()
)

<span class='fu'>spikeInNormGRanges</span>(
  <span class='no'>dataset.gr</span>,
  <span class='kw'>si_pattern</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>si_names</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>method</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"SRPMC"</span>, <span class='st'>"SNR"</span>, <span class='st'>"RPM"</span>),
  <span class='kw'>batch_norm</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>ctrl_pattern</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>ctrl_names</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>field</span> <span class='kw'>=</span> <span class='st'>"score"</span>,
  <span class='kw'>sample_names</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>expand_ranges</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fu'>detectCores</span>()
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>dataset.gr</th>
      <td><p>A GRanges object, or (more typically) a list of GRanges
objects.</p></td>
    </tr>
    <tr>
      <th>si_pattern</th>
      <td><p>A regular expression that matches spike-in chromosomes. Can
be used in addition to, or as an alternative to <code>si_names</code>.</p></td>
    </tr>
    <tr>
      <th>si_names</th>
      <td><p>A character vector giving the names of the spike-in
chromosomes. Can be used in addition to, or as an alternative to
<code>si_pattern</code>.</p></td>
    </tr>
    <tr>
      <th>method</th>
      <td><p>One of the shown methods, which generate normalization factors
for converting raw readcounts into "Spike-in normalized Reads Per Million
mapped in Control" (the default), "Spike-in Normalized Read counts", or
"Reads Per Million mapped". See descriptions below.</p></td>
    </tr>
    <tr>
      <th>batch_norm</th>
      <td><p>A logical indicating if batch normalization should be used
(<code>TRUE</code> by default). See descriptions below. If batch normalization is
used, sample names must end with "rep#", wherein "#" is one or more
characters (usually a number) giving the replicate. If this is not the
case, users can use the <code>sample_names</code> argument to make the names
conform.</p></td>
    </tr>
    <tr>
      <th>ctrl_pattern</th>
      <td><p>A regular expression that matches negative control sample
names.</p></td>
    </tr>
    <tr>
      <th>ctrl_names</th>
      <td><p>A character vector giving the names of the negative control
samples. Can be used as an alternative to <code>ctrl_pattern</code>.</p></td>
    </tr>
    <tr>
      <th>field</th>
      <td><p>The metadata field in <code>dataset.gr</code> that contains raw
readcounts. If each range is an individual read, set <code>field = NULL</code>.</p></td>
    </tr>
    <tr>
      <th>sample_names</th>
      <td><p>An optional character vector that can be used to rename
the samples in <code>dataset.gr</code>. Intended use is if <code>dataset.gr</code> is
an unnamed list, or if <code>batch_norm = TRUE</code> but the sample names don't
conform to the required naming scheme.</p></td>
    </tr>
    <tr>
      <th>expand_ranges</th>
      <td><p>Logical indicating if ranges in <code>dataset.gr</code> should
be treated as descriptions of single molecules (<code>FALSE</code>), or if ranges
should be treated as representing multiple adjacent positions with the same
signal (<code>TRUE</code>). See <code><a href='getCountsByRegions.html'>
getCountsByRegions</a></code>.</p></td>
    </tr>
    <tr>
      <th>ncores</th>
      <td><p>The number of cores to use for computations.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A numeric vector of normalization factors for each sample in
  <code>dataset.gr</code>. Normalization factors are to be applied by
  multiplication.</p>
    <h2 class="hasAnchor" id="spike-in-normalized-reads-per-million-mapped-in-control-srpmc-"><a class="anchor" href="#spike-in-normalized-reads-per-million-mapped-in-control-srpmc-"></a>Spike-in normalized Reads Per Million mapped in Control (SRPMC)</h2>

    

<p>This is the default spike-in normalization method, as its meaning is the
  most portable and generalizable. Experimental Reads Per Spike-in read (RPS)
  are calculated for each sample, \(i\):</p>
<p>$$RPS_i=\frac{experimental\_reads_i}{
  spikein\_reads_i}$$</p>
<p>RPS for each sample is divided by RPS for the negative control, which
  measures the change in total material vs. the negative control. This global
  adjustment is applied to standard RPM normalization for each sample:</p>
<p>$$NF_i=\frac{RPS_i}{RPS_{control}} \cdot \frac{1 x
  10^6}{experimental\_reads_i}$$</p>
<p>Thus, the negative control(s) are simply RPM-normalized, while the other
  conditions are in equivalent, directly-comparable units ("Reads Per Million
  mapped reads in a negative control").</p>
<p>If <code>batch_norm = TRUE</code> (the default), all negative controls will be
  RPM-normalized, and the global changes in material for all other samples
  are calculated <em>within each batch</em> (vs. the negative control within
  the same batch).</p>
<p>If <code>batch_norm = FALSE</code>, all samples are compared to the average RPS
  of the negative controls. This method can only be justified if batch has
  less effect on RPS than other sources of variation.</p>
    <h2 class="hasAnchor" id="spike-in-normalized-reads-snr-"><a class="anchor" href="#spike-in-normalized-reads-snr-"></a>Spike-in Normalized Reads (SNR)</h2>

    <p>If <code>batch_norm = FALSE</code>, these
  normalization factors act to scale down the readcounts in each sample to
  make the spike-in read counts match the sample with the lowest number of
  spike-in reads:</p>
<p>$$NF_i=\frac{min(spikein\_reads)}{spikein\_reads_i}$$</p>
<p>If <code>batch_norm = TRUE</code>, such normalization factors are calculated
  within each batch, but a final batch (replicate) adjustment is performed
  that results in the negative controls having the same normalized
  readcounts. In this way, the negative controls are used to adjust the
  normalized readcounts of their entire replicate. Just as when
  <code>batch_norm = FALSE</code>, one of the normalization factors will be
  <code>1</code>, while the rest will be <code>&lt;1</code>.</p>
<p>One use for these normalization factors is for normalizing-by-subsampling;
  see <code><a href='subsampleBySpikeIn.html'>subsampleBySpikeIn</a></code>.</p>
    <h2 class="hasAnchor" id="reads-per-million-mapped-reads-rpm-"><a class="anchor" href="#reads-per-million-mapped-reads-rpm-"></a>Reads Per Million mapped reads (RPM)</h2>

    <p>A simple convenience wrapper
  for calculating normalization factors for RPM normalization:</p>
<p>$$NF_i=\frac{1 x 10^6}{experimental\_reads_i}$$</p>
<p>If spike-in reads are present, they're removed before the normalization
  factors are calculated.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='getSpikeInCounts.html'>getSpikeInCounts</a></code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>#--------------------------------------------------#</span>
<span class='co'># Make list of dummy GRanges</span>
<span class='co'>#--------------------------------------------------#</span>
<span class='no'>gr1_rep1</span> <span class='kw'>&lt;-</span> <span class='fu'>GRanges</span>(<span class='kw'>seqnames</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"chr1"</span>, <span class='st'>"chr2"</span>, <span class='st'>"spikechr1"</span>, <span class='st'>"spikechr2"</span>),
                    <span class='kw'>ranges</span> <span class='kw'>=</span> <span class='fu'>IRanges</span>(<span class='kw'>start</span> <span class='kw'>=</span> <span class='fl'>1</span>:<span class='fl'>4</span>, <span class='kw'>width</span> <span class='kw'>=</span> <span class='fl'>1</span>),
                    <span class='kw'>strand</span> <span class='kw'>=</span> <span class='st'>"+"</span>)
<span class='no'>gr2_rep2</span> <span class='kw'>&lt;-</span> <span class='no'>gr2_rep1</span> <span class='kw'>&lt;-</span> <span class='no'>gr1_rep2</span> <span class='kw'>&lt;-</span> <span class='no'>gr1_rep1</span>

<span class='co'># set readcounts</span>
<span class='fu'>score</span>(<span class='no'>gr1_rep1</span>) <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fl'>1</span>, <span class='fl'>1</span>, <span class='fl'>1</span>, <span class='fl'>1</span>) <span class='co'># 2 exp + 2 spike = 4 total</span>
<span class='fu'>score</span>(<span class='no'>gr2_rep1</span>) <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fl'>2</span>, <span class='fl'>2</span>, <span class='fl'>1</span>, <span class='fl'>1</span>) <span class='co'># 4 exp + 2 spike = 6 total</span>
<span class='fu'>score</span>(<span class='no'>gr1_rep2</span>) <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fl'>1</span>, <span class='fl'>1</span>, <span class='fl'>2</span>, <span class='fl'>1</span>) <span class='co'># 2 exp + 3 spike = 5 total</span>
<span class='fu'>score</span>(<span class='no'>gr2_rep2</span>) <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fl'>4</span>, <span class='fl'>4</span>, <span class='fl'>2</span>, <span class='fl'>2</span>) <span class='co'># 8 exp + 4 spike = 12 total</span>

<span class='no'>grl</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span>(<span class='no'>gr1_rep1</span>, <span class='no'>gr2_rep1</span>,
            <span class='no'>gr1_rep2</span>, <span class='no'>gr2_rep2</span>)
<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span>(<span class='no'>grl</span>) <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"gr1_rep1"</span>, <span class='st'>"gr2_rep1"</span>,
                <span class='st'>"gr1_rep2"</span>, <span class='st'>"gr2_rep2"</span>)

<span class='no'>grl</span></div><div class='output co'>#&gt; $gr1_rep1
#&gt; GRanges object with 4 ranges and 1 metadata column:
#&gt;        seqnames    ranges strand |     score
#&gt;           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;
#&gt;   [1]      chr1         1      + |         1
#&gt;   [2]      chr2         2      + |         1
#&gt;   [3] spikechr1         3      + |         1
#&gt;   [4] spikechr2         4      + |         1
#&gt;   -------
#&gt;   seqinfo: 4 sequences from an unspecified genome; no seqlengths
#&gt; 
#&gt; $gr2_rep1
#&gt; GRanges object with 4 ranges and 1 metadata column:
#&gt;        seqnames    ranges strand |     score
#&gt;           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;
#&gt;   [1]      chr1         1      + |         2
#&gt;   [2]      chr2         2      + |         2
#&gt;   [3] spikechr1         3      + |         1
#&gt;   [4] spikechr2         4      + |         1
#&gt;   -------
#&gt;   seqinfo: 4 sequences from an unspecified genome; no seqlengths
#&gt; 
#&gt; $gr1_rep2
#&gt; GRanges object with 4 ranges and 1 metadata column:
#&gt;        seqnames    ranges strand |     score
#&gt;           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;
#&gt;   [1]      chr1         1      + |         1
#&gt;   [2]      chr2         2      + |         1
#&gt;   [3] spikechr1         3      + |         2
#&gt;   [4] spikechr2         4      + |         1
#&gt;   -------
#&gt;   seqinfo: 4 sequences from an unspecified genome; no seqlengths
#&gt; 
#&gt; $gr2_rep2
#&gt; GRanges object with 4 ranges and 1 metadata column:
#&gt;        seqnames    ranges strand |     score
#&gt;           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;
#&gt;   [1]      chr1         1      + |         4
#&gt;   [2]      chr2         2      + |         4
#&gt;   [3] spikechr1         3      + |         2
#&gt;   [4] spikechr2         4      + |         2
#&gt;   -------
#&gt;   seqinfo: 4 sequences from an unspecified genome; no seqlengths
#&gt; </div><div class='input'>
<span class='co'>#--------------------------------------------------#</span>
<span class='co'># Get RPM NFs</span>
<span class='co'>#--------------------------------------------------#</span>

<span class='co'># can use the names of all spike-in chromosomes</span>
<span class='fu'>getSpikeInNFs</span>(<span class='no'>grl</span>, <span class='kw'>si_names</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"spikechr1"</span>, <span class='st'>"spikechr2"</span>),
              <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"RPM"</span>, <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fl'>1</span>)</div><div class='output co'>#&gt; [1] 500000 250000 500000 125000</div><div class='input'>
<span class='co'># or use a regular expression that matches the spike-in chromosome names</span>
<span class='fu'><a href='https://rdrr.io/r/base/grep.html'>grep</a></span>(<span class='st'>"spike"</span>, <span class='fu'><a href='https://rdrr.io/r/base/vector.html'>as.vector</a></span>(<span class='fu'>seqnames</span>(<span class='no'>gr1_rep1</span>)))</div><div class='output co'>#&gt; [1] 3 4</div><div class='input'>
<span class='fu'>getSpikeInNFs</span>(<span class='no'>grl</span>, <span class='kw'>si_pattern</span> <span class='kw'>=</span> <span class='st'>"spike"</span>, <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"RPM"</span>, <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fl'>1</span>)</div><div class='output co'>#&gt; [1] 500000 250000 500000 125000</div><div class='input'>

<span class='co'>#--------------------------------------------------#</span>
<span class='co'># Get simple spike-in NFs ("SNR")</span>
<span class='co'>#--------------------------------------------------#</span>

<span class='co'># without batch normalization, NFs make all spike-in readcounts match</span>
<span class='fu'>getSpikeInNFs</span>(<span class='no'>grl</span>, <span class='kw'>si_pattern</span> <span class='kw'>=</span> <span class='st'>"spike"</span>, <span class='kw'>ctrl_pattern</span> <span class='kw'>=</span> <span class='st'>"gr1"</span>,
              <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"SNR"</span>, <span class='kw'>batch_norm</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fl'>1</span>)</div><div class='output co'>#&gt; [1] 1.0000000 1.0000000 0.6666667 0.5000000</div><div class='input'>
<span class='co'># with batch normalization, controls will have the same normalized counts;</span>
<span class='co'># other samples are normalized to have same spike-in reads as their matched</span>
<span class='co'># control</span>
<span class='fu'>getSpikeInNFs</span>(<span class='no'>grl</span>, <span class='kw'>si_pattern</span> <span class='kw'>=</span> <span class='st'>"spike"</span>, <span class='kw'>ctrl_pattern</span> <span class='kw'>=</span> <span class='st'>"gr1"</span>,
              <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"SNR"</span>, <span class='kw'>batch_norm</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fl'>1</span>)</div><div class='output co'>#&gt; [1] 1.00 1.00 1.00 0.75</div><div class='input'>
<span class='co'>#--------------------------------------------------#</span>
<span class='co'># Get spike-in NFs with more meaningful units ("RPMC")</span>
<span class='co'>#--------------------------------------------------#</span>

<span class='co'># compare to raw RPM NFs above; takes into account spike-in reads;</span>
<span class='co'># units are directly comparable to the negative controls</span>

<span class='co'># with batch normalization, these negative controls are the same, as they</span>
<span class='co'># have the same number of non-spike-in readcounts (they're simply RPM)</span>
<span class='fu'>getSpikeInNFs</span>(<span class='no'>grl</span>, <span class='kw'>si_pattern</span> <span class='kw'>=</span> <span class='st'>"spike"</span>, <span class='kw'>ctrl_pattern</span> <span class='kw'>=</span> <span class='st'>"gr1"</span>, <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fl'>1</span>)</div><div class='output co'>#&gt; [1] 500000 500000 500000 375000</div><div class='input'>
<span class='co'># batch_norm = FALSE, the average reads-per-spike-in for the negative</span>
<span class='co'># controls are used to calculate all NFs; unless the controls have the exact</span>
<span class='co'># same ratio of non-spike-in to spike-in reads, nothing is precisely RPM</span>
<span class='fu'>getSpikeInNFs</span>(<span class='no'>grl</span>, <span class='kw'>si_pattern</span> <span class='kw'>=</span> <span class='st'>"spike"</span>, <span class='kw'>ctrl_pattern</span> <span class='kw'>=</span> <span class='st'>"gr1"</span>,
              <span class='kw'>batch_norm</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fl'>1</span>)</div><div class='output co'>#&gt; [1] 6e+05 6e+05 4e+05 3e+05</div><div class='input'>
<span class='co'>#--------------------------------------------------#</span>
<span class='co'># Apply NFs to the GRanges</span>
<span class='co'>#--------------------------------------------------#</span>

<span class='fu'>spikeInNormGRanges</span>(<span class='no'>grl</span>, <span class='kw'>si_pattern</span> <span class='kw'>=</span> <span class='st'>"spike"</span>, <span class='kw'>ctrl_pattern</span> <span class='kw'>=</span> <span class='st'>"gr1"</span>,
                   <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fl'>1</span>)</div><div class='output co'>#&gt; $gr1_rep1
#&gt; GRanges object with 2 ranges and 1 metadata column:
#&gt;       seqnames    ranges strand |     score
#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;
#&gt;   [1]     chr1         1      + |     5e+05
#&gt;   [2]     chr2         2      + |     5e+05
#&gt;   -------
#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths
#&gt; 
#&gt; $gr2_rep1
#&gt; GRanges object with 2 ranges and 1 metadata column:
#&gt;       seqnames    ranges strand |     score
#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;
#&gt;   [1]     chr1         1      + |     1e+06
#&gt;   [2]     chr2         2      + |     1e+06
#&gt;   -------
#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths
#&gt; 
#&gt; $gr1_rep2
#&gt; GRanges object with 2 ranges and 1 metadata column:
#&gt;       seqnames    ranges strand |     score
#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;
#&gt;   [1]     chr1         1      + |     5e+05
#&gt;   [2]     chr2         2      + |     5e+05
#&gt;   -------
#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths
#&gt; 
#&gt; $gr2_rep2
#&gt; GRanges object with 2 ranges and 1 metadata column:
#&gt;       seqnames    ranges strand |     score
#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;
#&gt;   [1]     chr1         1      + |   1500000
#&gt;   [2]     chr2         2      + |   1500000
#&gt;   -------
#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths
#&gt; </div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      <li><a href="#value">Value</a></li>
      <li><a href="#spike-in-normalized-reads-per-million-mapped-in-control-srpmc-">Spike-in normalized Reads Per Million mapped in Control (SRPMC)</a></li>
      <li><a href="#spike-in-normalized-reads-snr-">Spike-in Normalized Reads (SNR)</a></li>
      <li><a href="#reads-per-million-mapped-reads-rpm-">Reads Per Million mapped reads (RPM)</a></li>
      <li><a href="#see-also">See also</a></li>
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    <p>Mike DeBerardine</p>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


