<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Import basepair-resolution files — import-functions • BRGenomics</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Import basepair-resolution files — import-functions"><meta property="og:description" content="Import functions for plus/minus pairs of bigWig or bedGraph
files."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Overview.html">Overview</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Guides
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/GettingStarted.html">Getting Started</a>
    </li>
    <li>
      <a href="../articles/ImportingModifyingAnnotations.html">Importing &amp; Modifying Annotations</a>
    </li>
    <li>
      <a href="../articles/ImportingProcessingData.html">Importing &amp; Processing Data</a>
    </li>
    <li>
      <a href="../articles/SignalCounting.html">Signal Counting</a>
    </li>
    <li>
      <a href="../articles/AnalyzingMultipleDatasets.html">Analyzing Multiple Datasets</a>
    </li>
    <li>
      <a href="../articles/ProfilePlotsAndBootstrapping.html">Profile Plots &amp; Bootstrapping</a>
    </li>
    <li>
      <a href="../articles/SpikeInNormalization.html">Spike-in Normalization</a>
    </li>
    <li>
      <a href="../articles/DESeq2WithGlobalPerturbations.html">DESeq2 with Global Perturbations</a>
    </li>
    <li>
      <a href="../articles/SequenceExtraction.html">Sequence Extraction</a>
    </li>
  </ul></li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/mdeber/BRGenomics" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Import basepair-resolution files</h1>
    <small class="dont-index">Source: <a href="https://github.com/mdeber/BRGenomics/blob/HEAD/R/data_import.R" class="external-link"><code>R/data_import.R</code></a></small>
    <div class="hidden name"><code>import-functions.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Import functions for plus/minus pairs of <code>bigWig</code> or <code>bedGraph</code>
files.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">import_bigWig</span><span class="op">(</span></span>
<span>  plus_file <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  minus_file <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  genome <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  keep.X <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  keep.Y <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  keep.M <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  keep.nonstandard <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  makeBRG <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  ncores <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"mc.cores"</span>, <span class="fl">2L</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">import_bedGraph</span><span class="op">(</span></span>
<span>  plus_file <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  minus_file <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  genome <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  keep.X <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  keep.Y <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  keep.M <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  keep.nonstandard <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  ncores <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"mc.cores"</span>, <span class="fl">2L</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>plus_file, minus_file</dt>
<dd><p>Paths for strand-specific input files, or a
vector of such paths. If vectors are given, the user should take care that
the orders match!</p></dd>


<dt>genome</dt>
<dd><p>Optional string for UCSC reference genome, e.g. "hg38". If
given, non-standard chromosomes are trimmed, and options for sex and
mitochondrial chromosomes are applied.</p></dd>


<dt>keep.X, keep.Y, keep.M, keep.nonstandard</dt>
<dd><p>Logicals indicating which
non-autosomes should be kept. By default, sex chromosomes are kept, but
mitochondrial and non-standard chromosomes are removed.</p></dd>


<dt>makeBRG</dt>
<dd><p>If <code>TRUE</code> (the default), the output ranges are made
single-width using <code><a href="makeGRangesBRG.html">makeGRangesBRG</a></code></p></dd>


<dt>ncores</dt>
<dd><p>Number of cores to use, if importing multiple objects
simultaneously.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>Imports a GRanges object containing base-pair resolution data, with
  the <code>score</code> metadata column indicating the number of reads represented
  by each range.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>For <code>import_bigWig</code>, the output GRanges is formatted by
  <code><a href="makeGRangesBRG.html">makeGRangesBRG</a></code>, such that all
  ranges are disjoint and have width = 1, and the <code>score</code> is single-base
  coverage, i.e. the number of reads for each position.</p>
<p><code>import_bedGraph</code> is useful for when both 5'- and 3'-end information
  is to be maintained for each sequenced molecule. For this purpose, one use
  bedGraphs to store entire reads, with the <code>score</code> representing the
  number of reads sharing identical 5' and 3' ends. However,
  <code>import_bedGraph</code> doesn't modify the information in the bedGraph
  files. If the bedGraph file represents basepair-resolution coverage data,
  then users can coerce it to a basepair-resolution GRanges object by using
  <code><a href="getStrandedCoverage.html">getStrandedCoverage</a></code> followed
  by <code><a href="makeGRangesBRG.html">makeGRangesBRG</a></code>.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="tidyChromosomes.html">tidyChromosomes</a></code>,
  <code><a href="https://rdrr.io/pkg/rtracklayer/man/BigWigFile.html" class="external-link">rtracklayer::import.bw</a></code>,
  <code><a href="https://rdrr.io/pkg/rtracklayer/man/BEDFile-class.html" class="external-link">rtracklayer::import.bedGraph</a></code></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Mike DeBerardine</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># Import PRO-seq bigWigs -&gt; coverage of 3' bases</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># get local address for included bigWig files</span></span></span>
<span class="r-in"><span><span class="va">p.bw</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"PROseq_dm6_chr4_plus.bw"</span>,</span></span>
<span class="r-in"><span>                    package <span class="op">=</span> <span class="st">"BRGenomics"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">m.bw</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"PROseq_dm6_chr4_minus.bw"</span>,</span></span>
<span class="r-in"><span>                    package <span class="op">=</span> <span class="st">"BRGenomics"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># import bigWigs (not supported on windows)</span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="va">.Platform</span><span class="op">$</span><span class="va">OS.type</span> <span class="op">==</span> <span class="st">"unix"</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>    <span class="va">PROseq</span> <span class="op">&lt;-</span> <span class="fu">import_bigWig</span><span class="op">(</span><span class="va">p.bw</span>, <span class="va">m.bw</span>, genome <span class="op">=</span> <span class="st">"dm6"</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="va">PROseq</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 150 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seqnames    ranges strand |     score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [1]     chr4      1295      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [2]     chr4     41428      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [3]     chr4     42588      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [4]     chr4     42590      + |         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [5]     chr4     42593      + |         5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ...      ...       ...    ... .       ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [146]     chr4   1307742      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [147]     chr4   1316537      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [148]     chr4   1318960      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [149]     chr4   1319004      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [150]     chr4   1319369      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from dm6 genome</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># Import PRO-seq bedGraphs -&gt; whole reads (matched 5' and 3' ends)</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># get local address for included bedGraph files</span></span></span>
<span class="r-in"><span><span class="va">p.bg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"PROseq_dm6_chr4_plus.bedGraph"</span>,</span></span>
<span class="r-in"><span>                    package <span class="op">=</span> <span class="st">"BRGenomics"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">m.bg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"PROseq_dm6_chr4_minus.bedGraph"</span>,</span></span>
<span class="r-in"><span>                    package <span class="op">=</span> <span class="st">"BRGenomics"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># import bedGraphs</span></span></span>
<span class="r-in"><span><span class="va">PROseq_paired</span> <span class="op">&lt;-</span> <span class="fu">import_bedGraph</span><span class="op">(</span><span class="va">p.bg</span>, <span class="va">m.bg</span>, genome <span class="op">=</span> <span class="st">"dm6"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">PROseq_paired</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 164 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seqnames          ranges strand |     score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [1]     chr4       1270-1295      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [2]     chr4     41411-41428      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [3]     chr4     42556-42590      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [4]     chr4     42559-42588      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [5]     chr4     42559-42593      + |         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ...      ...             ...    ... .       ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [160]     chr4 1307742-1307776      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [161]     chr4 1316537-1316563      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [162]     chr4 1318960-1318994      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [163]     chr4 1319004-1319038      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [164]     chr4 1319369-1319403      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from dm6 genome; no seqlengths</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

