<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Import bam files — import_bam • BRGenomics</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Import bam files — import_bam" />
<meta property="og:description" content="Import bam files" />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.4.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Import bam files</h1>
    
    <div class="hidden name"><code>import_bam.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Import bam files</p>
    </div>

    <pre class="usage"><span class='fu'>import_bam</span>(
  <span class='no'>file</span>,
  <span class='kw'>mapq</span> <span class='kw'>=</span> <span class='fl'>20</span>,
  <span class='kw'>revcomp</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>trim.to</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"whole"</span>, <span class='st'>"5p"</span>, <span class='st'>"3p"</span>, <span class='st'>"center"</span>),
  <span class='kw'>ignore.strand</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>field</span> <span class='kw'>=</span> <span class='st'>"score"</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>file</th>
      <td><p>Path of a bam file.</p></td>
    </tr>
    <tr>
      <th>mapq</th>
      <td><p>Filter reads by a minimum MapQ score. This is the correct way to
filter multi-aligners.</p></td>
    </tr>
    <tr>
      <th>revcomp</th>
      <td><p>Logical indicating if aligned reads should be
reverse-complemented.</p></td>
    </tr>
    <tr>
      <th>trim.to</th>
      <td><p>Option for selecting specific bases from the reads, applied
after the <code>revcomp</code> option. By default, the entire read is maintained.
Other options are to take only the 5' base, only the 3' base, or the only
the center base of the read.</p></td>
    </tr>
    <tr>
      <th>ignore.strand</th>
      <td><p>Logical indicating if the strand information should be
discarded. If <code>TRUE</code>, strand information is discarded <em>after</em>
<code>revcomp</code> and <code>trim.to</code> options are applied.</p></td>
    </tr>
    <tr>
      <th>field</th>
      <td><p>Metadata field name to use for readcounts, usually "score". If
set to <code>NULL</code>, identical reads (or identical positions if
<code>trim.to</code> options applied) are not combined, and the length of the
output GRanges will be equal to the number of input reads.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A GRanges object.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># get local address for included bam file</span>
<span class='no'>ps.bam</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"extdata"</span>, <span class='st'>"PROseq_dm6_chr4.bam"</span>,
                      <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>"BRGenomics"</span>)

<span class='co'>#--------------------------------------------------#</span>
<span class='co'># Import entire reads</span>
<span class='co'>#--------------------------------------------------#</span>

<span class='co'># Note that PRO-seq reads are sequenced as reverse complement</span>
<span class='fu'>import_bam</span>(<span class='no'>ps.bam</span>, <span class='kw'>revcomp</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)</div><div class='output co'>#&gt; GRanges object with 53179 ranges and 1 metadata column:
#&gt;           seqnames          ranges strand |     score
#&gt;              &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
#&gt;       [1]     chr4       1189-1224      + |         1
#&gt;       [2]     chr4     31442-31477      + |         1
#&gt;       [3]     chr4     31707-31728      + |         1
#&gt;       [4]     chr4     33805-33831      + |         1
#&gt;       [5]     chr4     34253-34288      + |         1
#&gt;       ...      ...             ...    ... .       ...
#&gt;   [53175]     chr4 1307518-1307553      - |         1
#&gt;   [53176]     chr4 1307615-1307650      - |         1
#&gt;   [53177]     chr4 1316417-1316452      - |         1
#&gt;   [53178]     chr4 1316506-1316541      - |         1
#&gt;   [53179]     chr4 1316671-1316706      - |         2
#&gt;   -------
#&gt;   seqinfo: 1870 sequences from an unspecified genome</div><div class='input'>
<span class='co'>#--------------------------------------------------#</span>
<span class='co'># Import entire reads, 1 range per read</span>
<span class='co'>#--------------------------------------------------#</span>

<span class='fu'>import_bam</span>(<span class='no'>ps.bam</span>, <span class='kw'>revcomp</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>field</span> <span class='kw'>=</span> <span class='kw'>NULL</span>)</div><div class='output co'>#&gt; GRanges object with 73887 ranges and 0 metadata columns:
#&gt;           seqnames          ranges strand
#&gt;              &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt;
#&gt;       [1]     chr4       1189-1224      +
#&gt;       [2]     chr4     31442-31477      +
#&gt;       [3]     chr4     31707-31728      +
#&gt;       [4]     chr4     33805-33831      +
#&gt;       [5]     chr4     34253-34288      +
#&gt;       ...      ...             ...    ...
#&gt;   [73883]     chr4 1307615-1307650      -
#&gt;   [73884]     chr4 1316417-1316452      -
#&gt;   [73885]     chr4 1316506-1316541      -
#&gt;   [73886]     chr4 1316671-1316706      -
#&gt;   [73887]     chr4 1316671-1316706      -
#&gt;   -------
#&gt;   seqinfo: 1870 sequences from an unspecified genome</div><div class='input'>
<span class='co'>#--------------------------------------------------#</span>
<span class='co'># Import PRO-seq reads at basepair-resolution</span>
<span class='co'>#--------------------------------------------------#</span>

<span class='co'># the typical manner to import PRO-seq data:</span>
<span class='fu'>import_bam</span>(<span class='no'>ps.bam</span>, <span class='kw'>revcomp</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>trim.to</span> <span class='kw'>=</span> <span class='st'>"3p"</span>)</div><div class='output co'>#&gt; GRanges object with 48697 ranges and 1 metadata column:
#&gt;           seqnames    ranges strand |     score
#&gt;              &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
#&gt;       [1]     chr4      1224      + |         1
#&gt;       [2]     chr4     31477      + |         1
#&gt;       [3]     chr4     31728      + |         1
#&gt;       [4]     chr4     33831      + |         1
#&gt;       [5]     chr4     34285      + |         1
#&gt;       ...      ...       ...    ... .       ...
#&gt;   [48693]     chr4   1307518      - |         1
#&gt;   [48694]     chr4   1307615      - |         1
#&gt;   [48695]     chr4   1316417      - |         1
#&gt;   [48696]     chr4   1316506      - |         1
#&gt;   [48697]     chr4   1316671      - |         2
#&gt;   -------
#&gt;   seqinfo: 1870 sequences from an unspecified genome</div><div class='input'>
<span class='co'>#--------------------------------------------------#</span>
<span class='co'># Import 5' ends of PRO-seq reads</span>
<span class='co'>#--------------------------------------------------#</span>

<span class='co'># will include bona fide TSSes as well as hydrolysis products</span>
<span class='fu'>import_bam</span>(<span class='no'>ps.bam</span>, <span class='kw'>revcomp</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>trim.to</span> <span class='kw'>=</span> <span class='st'>"5p"</span>)</div><div class='output co'>#&gt; GRanges object with 47380 ranges and 1 metadata column:
#&gt;           seqnames    ranges strand |     score
#&gt;              &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
#&gt;       [1]     chr4      1189      + |         1
#&gt;       [2]     chr4     31442      + |         1
#&gt;       [3]     chr4     31707      + |         1
#&gt;       [4]     chr4     33805      + |         1
#&gt;       [5]     chr4     34253      + |         1
#&gt;       ...      ...       ...    ... .       ...
#&gt;   [47376]     chr4   1307553      - |         1
#&gt;   [47377]     chr4   1307650      - |         1
#&gt;   [47378]     chr4   1316452      - |         1
#&gt;   [47379]     chr4   1316541      - |         1
#&gt;   [47380]     chr4   1316706      - |         2
#&gt;   -------
#&gt;   seqinfo: 1870 sequences from an unspecified genome</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      <li><a href="#value">Value</a></li>
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


