<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Import basepair-resolution files — import-functions • BRGenomics</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Import basepair-resolution files — import-functions" />
<meta property="og:description" content="Import functions for plus/minus pairs of bigWig or bedGraph
files." />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.20</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/BRGenomicsOverview.html">Basepair-Resolution Analysis with BRGenomics</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Import basepair-resolution files</h1>
    
    <div class="hidden name"><code>import-functions.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Import functions for plus/minus pairs of <code>bigWig</code> or <code>bedGraph</code>
files.</p>
    </div>

    <pre class="usage"><span class='fu'>import_bigWig</span>(
  <span class='kw'>plus_file</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>minus_file</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>genome</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>keep.X</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>keep.Y</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>keep.M</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>keep.nonstandard</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fu'>detectCores</span>()
)

<span class='fu'>import_bedGraph</span>(
  <span class='kw'>plus_file</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>minus_file</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>genome</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>keep.X</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>keep.Y</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>keep.M</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>keep.nonstandard</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fu'>detectCores</span>()
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>plus_file, minus_file</th>
      <td><p>Paths for strand-specific input files, or a
vector of such paths. If vectors are given, the user should take care that
the orders match!</p></td>
    </tr>
    <tr>
      <th>genome</th>
      <td><p>Optional string for UCSC reference genome, e.g. "hg38". If
given, non-standard chromosomes are trimmed, and options for sex and
mitochondrial chromosomes are applied.</p></td>
    </tr>
    <tr>
      <th>keep.X, keep.Y, keep.M, keep.nonstandard</th>
      <td><p>Logicals indicating which
non-autosomes should be kept. By default, sex chromosomes are kept, but
mitochondrial and non-standard chromosomes are removed.</p></td>
    </tr>
    <tr>
      <th>ncores</th>
      <td><p>Number of cores to use, if importing multiple objects
simultaneously.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Imports a GRanges object containing base-pair resolution data, with
  the <code>score</code> metadata column indicating the number of reads represented
  by each range.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>For <code>import_bigWig</code>, the output GRanges is formatted by
  <code><a href='makeGRangesBRG.html'>makeGRangesBRG</a></code>, such that all
  ranges are disjoint and have width = 1, and the <code>score</code> is single-base
  coverage, i.e. the number of reads for each position.</p>
<p><code>import_bedGraph</code> is useful for when both 5'- and 3'-end information
  is to be maintained for each sequenced molecule. For this purpose, one use
  bedGraphs to store entire reads, with the <code>score</code> representing the
  number of reads sharing identical 5' and 3' ends. However,
  <code>import_bedGraph</code> doesn't modify the information in the bedGraph
  files. If the bedGraph file represents basepair-resolution coverage data,
  then users can coerce it to a basepair-resolution GRanges object by using
  <code><a href='getStrandedCoverage.html'>getStrandedCoverage</a></code> followed
  by <code><a href='makeGRangesBRG.html'>makeGRangesBRG</a></code>.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='tidyChromosomes.html'>tidyChromosomes</a></code>,
  <code><a href='https://rdrr.io/pkg/rtracklayer/man/export.html'>rtracklayer::import</a></code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>#--------------------------------------------------#</span>
<span class='co'># Import PRO-seq bigWigs -&gt; coverage of 3' bases</span>
<span class='co'>#--------------------------------------------------#</span>

<span class='co'># get local address for included bigWig files</span>
<span class='no'>p.bw</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"extdata"</span>, <span class='st'>"PROseq_dm6_chr4_plus.bw"</span>,
                    <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>"BRGenomics"</span>)
<span class='no'>m.bw</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"extdata"</span>, <span class='st'>"PROseq_dm6_chr4_minus.bw"</span>,
                    <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>"BRGenomics"</span>)

<span class='co'># import bigWigs (not supported on windows)</span>
<span class='kw'>if</span> (<span class='no'>.Platform</span>$<span class='no'>OS.type</span> <span class='kw'>==</span> <span class='st'>"unix"</span>) {
    <span class='no'>PROseq</span> <span class='kw'>&lt;-</span> <span class='fu'>import_bigWig</span>(<span class='no'>p.bw</span>, <span class='no'>m.bw</span>, <span class='kw'>genome</span> <span class='kw'>=</span> <span class='st'>"dm6"</span>)
    <span class='no'>PROseq</span>
}</div><div class='output co'>#&gt; GRanges object with 150 ranges and 1 metadata column:
#&gt;         seqnames    ranges strand |     score
#&gt;            &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
#&gt;     [1]     chr4      1295      + |         1
#&gt;     [2]     chr4     41428      + |         1
#&gt;     [3]     chr4     42588      + |         1
#&gt;     [4]     chr4     42590      + |         2
#&gt;     [5]     chr4     42593      + |         5
#&gt;     ...      ...       ...    ... .       ...
#&gt;   [146]     chr4   1307742      - |         1
#&gt;   [147]     chr4   1316537      - |         1
#&gt;   [148]     chr4   1318960      - |         1
#&gt;   [149]     chr4   1319004      - |         1
#&gt;   [150]     chr4   1319369      - |         1
#&gt;   -------
#&gt;   seqinfo: 1 sequence from dm6 genome</div><div class='input'>

<span class='co'>#--------------------------------------------------#</span>
<span class='co'># Import PRO-seq bedGraphs -&gt; whole reads (matched 5' and 3' ends)</span>
<span class='co'>#--------------------------------------------------#</span>

<span class='co'># get local address for included bedGraph files</span>
<span class='no'>p.bg</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"extdata"</span>, <span class='st'>"PROseq_dm6_chr4_plus.bedGraph"</span>,
                    <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>"BRGenomics"</span>)
<span class='no'>m.bg</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"extdata"</span>, <span class='st'>"PROseq_dm6_chr4_minus.bedGraph"</span>,
                    <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>"BRGenomics"</span>)

<span class='co'># import bedGraphs</span>
<span class='no'>PROseq_paired</span> <span class='kw'>&lt;-</span> <span class='fu'>import_bedGraph</span>(<span class='no'>p.bg</span>, <span class='no'>m.bg</span>, <span class='kw'>genome</span> <span class='kw'>=</span> <span class='st'>"dm6"</span>)
<span class='no'>PROseq_paired</span></div><div class='output co'>#&gt; GRanges object with 164 ranges and 1 metadata column:
#&gt;         seqnames          ranges strand |     score
#&gt;            &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
#&gt;     [1]     chr4       1270-1295      + |         1
#&gt;     [2]     chr4     41411-41428      + |         1
#&gt;     [3]     chr4     42556-42590      + |         1
#&gt;     [4]     chr4     42559-42588      + |         1
#&gt;     [5]     chr4     42559-42593      + |         3
#&gt;     ...      ...             ...    ... .       ...
#&gt;   [160]     chr4 1307742-1307776      - |         1
#&gt;   [161]     chr4 1316537-1316563      - |         1
#&gt;   [162]     chr4 1318960-1318994      - |         1
#&gt;   [163]     chr4 1319004-1319038      - |         1
#&gt;   [164]     chr4 1319369-1319403      - |         1
#&gt;   -------
#&gt;   seqinfo: 1 sequence from dm6 genome; no seqlengths</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      <li><a href="#value">Value</a></li>
      <li><a href="#details">Details</a></li>
      <li><a href="#see-also">See also</a></li>
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    <p>Mike DeBerardine</p>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


