<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Import bam files — import_bam • BRGenomics</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Import bam files — import_bam"><meta property="og:description" content="Import single-end or paired-end bam files as GRanges objects, with various
processing options. It is highly recommend to index the BAM file first."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.11.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Overview.html">Overview</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Guides
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/GettingStarted.html">Getting Started</a>
    </li>
    <li>
      <a href="../articles/ImportingModifyingAnnotations.html">Importing &amp; Modifying Annotations</a>
    </li>
    <li>
      <a href="../articles/ImportingProcessingData.html">Importing &amp; Processing Data</a>
    </li>
    <li>
      <a href="../articles/SignalCounting.html">Signal Counting</a>
    </li>
    <li>
      <a href="../articles/AnalyzingMultipleDatasets.html">Analyzing Multiple Datasets</a>
    </li>
    <li>
      <a href="../articles/ProfilePlotsAndBootstrapping.html">Profile Plots &amp; Bootstrapping</a>
    </li>
    <li>
      <a href="../articles/SpikeInNormalization.html">Spike-in Normalization</a>
    </li>
    <li>
      <a href="../articles/DESeq2WithGlobalPerturbations.html">DESeq2 with Global Perturbations</a>
    </li>
    <li>
      <a href="../articles/SequenceExtraction.html">Sequence Extraction</a>
    </li>
  </ul></li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/mdeber/BRGenomics" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Import bam files</h1>
    <small class="dont-index">Source: <a href="https://github.com/mdeber/BRGenomics/blob/HEAD/R/data_import.R" class="external-link"><code>R/data_import.R</code></a></small>
    <div class="hidden name"><code>import_bam.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Import single-end or paired-end bam files as GRanges objects, with various
processing options. It is highly recommend to index the BAM file first.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">import_bam</span><span class="op">(</span></span>
<span>  <span class="va">file</span>,</span>
<span>  mapq <span class="op">=</span> <span class="fl">20L</span>,</span>
<span>  revcomp <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  shift <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  trim.to <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"whole"</span>, <span class="st">"5p"</span>, <span class="st">"3p"</span>, <span class="st">"center"</span><span class="op">)</span>,</span>
<span>  ignore.strand <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  field <span class="op">=</span> <span class="st">"score"</span>,</span>
<span>  paired_end <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  yieldSize <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  ncores <span class="op">=</span> <span class="fl">1L</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">import_bam_PROseq</span><span class="op">(</span></span>
<span>  <span class="va">file</span>,</span>
<span>  mapq <span class="op">=</span> <span class="fl">20L</span>,</span>
<span>  revcomp <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  shift <span class="op">=</span> <span class="op">-</span><span class="fl">1L</span>,</span>
<span>  trim.to <span class="op">=</span> <span class="st">"3p"</span>,</span>
<span>  ignore.strand <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  field <span class="op">=</span> <span class="st">"score"</span>,</span>
<span>  paired_end <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  yieldSize <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  ncores <span class="op">=</span> <span class="fl">1L</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">import_bam_PROcap</span><span class="op">(</span></span>
<span>  <span class="va">file</span>,</span>
<span>  mapq <span class="op">=</span> <span class="fl">20L</span>,</span>
<span>  revcomp <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  shift <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  trim.to <span class="op">=</span> <span class="st">"5p"</span>,</span>
<span>  ignore.strand <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  field <span class="op">=</span> <span class="st">"score"</span>,</span>
<span>  paired_end <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  yieldSize <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  ncores <span class="op">=</span> <span class="fl">1L</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">import_bam_ATACseq</span><span class="op">(</span></span>
<span>  <span class="va">file</span>,</span>
<span>  mapq <span class="op">=</span> <span class="fl">20L</span>,</span>
<span>  revcomp <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  shift <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  plus_offset <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  minus_offset <span class="op">=</span> <span class="op">-</span><span class="fl">4</span>,</span>
<span>  trim.to <span class="op">=</span> <span class="st">"5p"</span>,</span>
<span>  ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  field <span class="op">=</span> <span class="st">"score"</span>,</span>
<span>  paired_end <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  yieldSize <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  ncores <span class="op">=</span> <span class="fl">1L</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>file</dt>
<dd><p>Path of a bam file, or a vector of paths.</p></dd>


<dt>mapq</dt>
<dd><p>Filter reads by a minimum MAPQ score. This is the correct way to
filter multi-aligners.</p></dd>


<dt>revcomp</dt>
<dd><p>Logical indicating if aligned reads should be
reverse-complemented.</p></dd>


<dt>shift</dt>
<dd><p>Either an integer giving the number of bases by which to shift
the entire read upstream or downstream, or a pair of integers indicating
shifts to be applied to the 5' and 3' ends of the reads, respectively.
Shifting is strand-specific, with negative numbers shifting the reads
upstream, and positive numbers shiftem them downstream. This option is
applied <em>after</em> the <code>revcomp</code>, but before <code>trim.to</code> and
<code>ignore.strand</code> options are applied.</p></dd>


<dt>trim.to</dt>
<dd><p>Option for selecting specific bases from the reads, applied
after the <code>revcomp</code> and <code>shift</code> options. By default, the entire
read is maintained. Other options are to take only the 5' base, only the 3'
base, or the only the center base of the read.</p></dd>


<dt>ignore.strand</dt>
<dd><p>Logical indicating if the strand information should be
discarded. If <code>TRUE</code>, strand information is discarded <em>after</em>
<code>revcomp</code>, <code>trim.to</code>, and <code>shift</code> options are applied.</p></dd>


<dt>field</dt>
<dd><p>Metadata field name to use for readcounts, usually "score". If
set to <code>NULL</code>, identical reads (or identical positions if
<code>trim.to</code> options applied) are not combined, and the length of the
output GRanges will be equal to the number of input reads.</p></dd>


<dt>paired_end</dt>
<dd><p>Logical indicating if reads should be treated as paired end
reads. When set to <code>NULL</code> (the default), the first 100k reads are
checked.</p></dd>


<dt>yieldSize</dt>
<dd><p>The number of bam file records to process simultaneously,
e.g. the "chunk size". Setting a higher chunk size will use more memory,
which can increase speed if there is enough memory available. If chunking
is not necessary, set to <code>NA</code>.</p></dd>


<dt>ncores</dt>
<dd><p>Number of cores to use for importing bam files. Currently,
multicore is only implemented for simultaneously importing multiple bam
files. For smaller datasets or machines with higher memory, this can
increase performance, but can otherwise lead to substantial performance
penalties.</p></dd>


<dt>plus_offset</dt>
<dd><p>For importing ATAC-seq, the shift to apply to plus strand
alignments. By default, plus strand reads are shifted 4 bp downstream.</p></dd>


<dt>minus_offset</dt>
<dd><p>For importing ATAC-seq, the shift to apply to minus
strand alignments. By default, minus strand reads are shifted 4 bp upstream
(in terms of the genomic coordinates).</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A GRanges object.</p>
    </div>
    <div id="atac-seq-data-import">
    <h2>ATAC-seq data import</h2>
    <p>By default, <code>import_bam_ATACseq</code> will
shift plus-aligned reads downstream 4 bp, minus-aligned reads upstream 4 bp,
and then take the strand-specific start site of the reads before removing
strand information and collapsing identical reads. These steps account for
the 9bp gap between opposing fragments generated from the same Tn5 reaction,
selecting the central base in the 9bp duplication.</p>
<p>While other sources often state that the offset should be +4 on plus strand
and -5 on minus strand alignments (or alternatively +5, -4), this does not
result in the two positions overlapping. I have verified that this is true
based on the expected result of the Tn5 reaction and adapter ligation and
sequencing steps, and also using real sequencing data, which confirms that
only the +4/-4 shift results in a significant increase in the number
positions that overlap. However, these arguments are left to the user if they
insist on doing it differently.</p>
<p>Note that the order of operations performed is the same as the order of the
associated arguments in the function proper, but not in the argument
documentation i.e., the <code>plus_offset</code> and <code>minus_offset</code> arguments
are applied <em>after</em> the <code>shift</code> argument and before the
<code>trim.to</code> argument.</p>
<p>While this single-base precision analysis of ATAC-seq may be useful in some
cases, for most users it is unlikely to be useful. Instead, you might use the
<code>plus_offset</code> and <code>minus_offset</code> arguments correctly, but set
<code>trim.to = "whole"</code> (and keep <code>ignore.strand = TRUE</code>). This will
keep the entire ATAC-seq reads, which is the most common analysis approach.
It is also common to use coverage data with ATAC-seq, but this eliminates
read count information.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>Hojoong Kwak, Nicholas J. Fuda, Leighton J. Core, John T. Lis
  (2013). Precise Maps of RNA Polymerase Reveal How Promoters Direct
  Initiation and Pausing. <em>Science</em> 339(6122): 950–953.
  <a href="https://doi.org/10.1126/science.1229386" class="external-link">https://doi.org/10.1126/science.1229386</a></p>
<p>Jason D. Buenrostro, Paul G. Giresi, Lisa C. Zaba, Howard Y. Chang, William
  J. Greenleaf (2013). Transposition of native chromatin for fast and
  sensitive epigenomic profiling of open chromatin, dna-binding proteins and
  nucleosome position. <em>Nature Methods</em> 10: 1213–1218.
  <a href="https://doi.org/10.1038/nmeth.2688" class="external-link">https://doi.org/10.1038/nmeth.2688</a></p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Mike DeBerardine</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># get local address for included bam file</span></span></span>
<span class="r-in"><span><span class="va">ps.bam</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"PROseq_dm6_chr4.bam"</span>,</span></span>
<span class="r-in"><span>                      package <span class="op">=</span> <span class="st">"BRGenomics"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># Import entire reads</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Note that PRO-seq reads are sequenced as reverse complement</span></span></span>
<span class="r-in"><span><span class="fu">import_bam</span><span class="op">(</span><span class="va">ps.bam</span>, revcomp <span class="op">=</span> <span class="cn">TRUE</span>, paired_end <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 164 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seqnames          ranges strand |     score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [1]     chr4       1270-1296      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [2]     chr4     41411-41429      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [3]     chr4     42556-42591      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [4]     chr4     42559-42589      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [5]     chr4     42559-42594      + |         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ...      ...             ...    ... .       ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [160]     chr4 1307741-1307776      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [161]     chr4 1316536-1316563      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [162]     chr4 1318959-1318994      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [163]     chr4 1319003-1319038      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [164]     chr4 1319368-1319403      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1870 sequences from an unspecified genome</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># Import entire reads, 1 range per read</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">import_bam</span><span class="op">(</span><span class="va">ps.bam</span>, revcomp <span class="op">=</span> <span class="cn">TRUE</span>, field <span class="op">=</span> <span class="cn">NULL</span>,</span></span>
<span class="r-in"><span>           paired_end <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 190 ranges and 0 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seqnames          ranges strand</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [1]     chr4       1270-1296      +</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [2]     chr4     41411-41429      +</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [3]     chr4     42556-42591      +</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [4]     chr4     42559-42589      +</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [5]     chr4     42559-42594      +</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ...      ...             ...    ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [186]     chr4 1307741-1307776      -</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [187]     chr4 1316536-1316563      -</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [188]     chr4 1318959-1318994      -</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [189]     chr4 1319003-1319038      -</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [190]     chr4 1319368-1319403      -</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1870 sequences from an unspecified genome</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># Import PRO-seq reads at basepair-resolution</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># the typical manner to import PRO-seq data:</span></span></span>
<span class="r-in"><span><span class="fu">import_bam</span><span class="op">(</span><span class="va">ps.bam</span>, revcomp <span class="op">=</span> <span class="cn">TRUE</span>, trim.to <span class="op">=</span> <span class="st">"3p"</span>,</span></span>
<span class="r-in"><span>           paired_end <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 150 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seqnames    ranges strand |     score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [1]     chr4      1296      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [2]     chr4     41429      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [3]     chr4     42589      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [4]     chr4     42591      + |         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [5]     chr4     42594      + |         5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ...      ...       ...    ... .       ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [146]     chr4   1307741      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [147]     chr4   1316536      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [148]     chr4   1318959      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [149]     chr4   1319003      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [150]     chr4   1319368      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1870 sequences from an unspecified genome</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># Import PRO-seq reads, removing the run-on base</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># the best way to import PRO-seq data; removes the</span></span></span>
<span class="r-in"><span><span class="co"># most 3' base, which was added in the run-on</span></span></span>
<span class="r-in"><span><span class="fu">import_bam</span><span class="op">(</span><span class="va">ps.bam</span>, revcomp <span class="op">=</span> <span class="cn">TRUE</span>, trim.to <span class="op">=</span> <span class="st">"3p"</span>,</span></span>
<span class="r-in"><span>           shift <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, paired_end <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 150 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seqnames    ranges strand |     score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [1]     chr4      1295      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [2]     chr4     41428      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [3]     chr4     42588      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [4]     chr4     42590      + |         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [5]     chr4     42593      + |         5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ...      ...       ...    ... .       ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [146]     chr4   1307742      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [147]     chr4   1316537      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [148]     chr4   1318960      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [149]     chr4   1319004      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [150]     chr4   1319369      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1870 sequences from an unspecified genome</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># Import 5' ends of PRO-seq reads</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># will include bona fide TSSes as well as hydrolysis products</span></span></span>
<span class="r-in"><span><span class="fu">import_bam</span><span class="op">(</span><span class="va">ps.bam</span>, revcomp <span class="op">=</span> <span class="cn">TRUE</span>, trim.to <span class="op">=</span> <span class="st">"5p"</span>,</span></span>
<span class="r-in"><span>           paired_end <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 145 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seqnames    ranges strand |     score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [1]     chr4      1270      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [2]     chr4     41411      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [3]     chr4     42556      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [4]     chr4     42559      + |         4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [5]     chr4     42560      + |         4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ...      ...       ...    ... .       ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [141]     chr4   1307776      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [142]     chr4   1316563      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [143]     chr4   1318994      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [144]     chr4   1319038      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [145]     chr4   1319403      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1870 sequences from an unspecified genome</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

