<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Intersect or reduce ranges according to gene names — intersectByGene • BRGenomics</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Intersect or reduce ranges according to gene names — intersectByGene" />
<meta property="og:description" content="These functions divide up regions of interest according to associated names,
and perform an inter-range operation on them. intersectByGene returns
the &quot;consensus&quot; segment that is common to all input ranges, and returns no
more than one range per gene. reduceByGene collapses the input ranges
into one or more non-overlapping ranges that encompass all segments from the
input ranges." />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/BRGenomicsOverview.html">Basepair-Resolution Analysis with BRGenomics</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Intersect or reduce ranges according to gene names</h1>
    
    <div class="hidden name"><code>intersectByGene.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>These functions divide up regions of interest according to associated names,
and perform an inter-range operation on them. <code>intersectByGene</code> returns
the "consensus" segment that is common to all input ranges, and returns no
more than one range per gene. <code>reduceByGene</code> collapses the input ranges
into one or more non-overlapping ranges that encompass all segments from the
input ranges.</p>
    </div>

    <pre class="usage"><span class='fu'>intersectByGene</span>(<span class='no'>regions.gr</span>, <span class='no'>gene_names</span>)

<span class='fu'>reduceByGene</span>(<span class='no'>regions.gr</span>, <span class='no'>gene_names</span>, <span class='kw'>disjoin</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>regions.gr</th>
      <td><p>A GRanges object containing regions of interest. If
<code>regions.gr</code> has the class <code>list</code>, <code>GRangesList</code>, or
<code>CompressedGRangesList</code>, it will be treated as if each list element is
a gene, and the GRanges within are the ranges associated with that gene.</p></td>
    </tr>
    <tr>
      <th>gene_names</th>
      <td><p>A character vector with the same length as
<code>regions.gr</code>.</p></td>
    </tr>
    <tr>
      <th>disjoin</th>
      <td><p>Logical. If <code>disjoin = TRUE</code>, the output GRanges is
disjoint, and each output range will match a single gene name. If
<code>FALSE</code>, segments from different genes can overlap.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A GRanges object whose individual ranges are named for the associated
  gene.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>These functions modify regions of interest that have associated
  names, such that several ranges share the same name, e.g. transcripts with
  associated gene names. Both functions "combine" the ranges on a
  gene-by-gene basis.</p>
<p><strong>intersectByGene</strong></p>
<p><em>For each unique gene</em>, the segment that overlaps <em>all</em> input
  ranges is returned. If no single range can be constructed that overlaps all
  input ranges, no range is returned for that gene (i.e. the gene is
  effectively filtered).</p>
<p>In other words, for all the ranges associated with a gene, the
  most-downstream start site is selected, and the most upstream end site is
  selected.</p>
<p><strong>reduceByGene</strong></p>
<p><em>For each unique gene</em>, the associated ranges are
  <code><a href='https://rdrr.io/pkg/IRanges/man/inter-range-methods.html'>reduced</a></code> to produce one or
  more non-overlapping ranges. The output range(s) are effectively a
  <code>union</code> of the input ranges, and cover every input base.</p>
<p>With <code>disjoin = FALSE</code>, no genomic segment is overlapped by more than
  one range <em>of the same gene</em>, but ranges from different genes can
  overlap. With <code>disjoin = TRUE</code>, the output ranges are disjoint, and no
  genomic position is overlapped more than once. Any segment that overlaps
  more than one gene is removed, but any segment (i.e. any section of an
  input range) that overlaps only one gene is still maintained.</p>
    <h2 class="hasAnchor" id="typical-uses"><a class="anchor" href="#typical-uses"></a>Typical Uses</h2>

    


<p>A typical use for <code>intersectByGene</code> is to avoid transcript isoform
  selection, as the returned range is found in every isoform.</p>
<p><code>reduceByGene</code> can be used to count any and all reads that overlap any
  part of a gene's annotation, but without double-counting any of them. With
  <code>disjoin = FALSE</code>, no reads will be double-counted for the same gene,
  but the same read can be counted for multiple genes. With <code>disjoin =
  TRUE</code>, no read can be double-counted.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># Make example data:</span>
<span class='co'>#  Ranges 1 and 2 overlap,</span>
<span class='co'>#  Ranges 3 and 4 are adjacent</span>
<span class='no'>gr</span> <span class='kw'>&lt;-</span> <span class='fu'>GRanges</span>(<span class='kw'>seqnames</span> <span class='kw'>=</span> <span class='st'>"chr1"</span>,
              <span class='kw'>ranges</span> <span class='kw'>=</span> <span class='fu'>IRanges</span>(<span class='kw'>start</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fl'>1</span>, <span class='fl'>3</span>, <span class='fl'>7</span>, <span class='fl'>10</span>),
                               <span class='kw'>end</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fl'>4</span>, <span class='fl'>5</span>, <span class='fl'>9</span>, <span class='fl'>11</span>)))
<span class='no'>gr</span></div><div class='output co'>#&gt; GRanges object with 4 ranges and 0 metadata columns:
#&gt;       seqnames    ranges strand
#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
#&gt;   [1]     chr1       1-4      *
#&gt;   [2]     chr1       3-5      *
#&gt;   [3]     chr1       7-9      *
#&gt;   [4]     chr1     10-11      *
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'>
<span class='co'>#--------------------------------------------------#</span>
<span class='co'># intersectByGene</span>
<span class='co'>#--------------------------------------------------#</span>

<span class='fu'>intersectByGene</span>(<span class='no'>gr</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"A"</span>, <span class='st'>"A"</span>, <span class='st'>"B"</span>, <span class='st'>"B"</span>))</div><div class='output co'>#&gt; GRanges object with 1 range and 0 metadata columns:
#&gt;     seqnames    ranges strand
#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
#&gt;   A     chr1       3-4      *
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'>
<span class='fu'>intersectByGene</span>(<span class='no'>gr</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"A"</span>, <span class='st'>"A"</span>, <span class='st'>"B"</span>, <span class='st'>"C"</span>))</div><div class='output co'>#&gt; GRanges object with 3 ranges and 0 metadata columns:
#&gt;     seqnames    ranges strand
#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
#&gt;   A     chr1       3-4      *
#&gt;   B     chr1       7-9      *
#&gt;   C     chr1     10-11      *
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'>
<span class='no'>gr2</span> <span class='kw'>&lt;-</span> <span class='no'>gr</span>
<span class='fu'><a href='https://rdrr.io/r/stats/start.html'>end</a></span>(<span class='no'>gr2</span>)[<span class='fl'>1</span>] <span class='kw'>&lt;-</span> <span class='fl'>10</span>
<span class='no'>gr2</span></div><div class='output co'>#&gt; GRanges object with 4 ranges and 0 metadata columns:
#&gt;       seqnames    ranges strand
#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
#&gt;   [1]     chr1      1-10      *
#&gt;   [2]     chr1       3-5      *
#&gt;   [3]     chr1       7-9      *
#&gt;   [4]     chr1     10-11      *
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'>
<span class='fu'>intersectByGene</span>(<span class='no'>gr2</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"A"</span>, <span class='st'>"A"</span>, <span class='st'>"B"</span>, <span class='st'>"C"</span>))</div><div class='output co'>#&gt; GRanges object with 3 ranges and 0 metadata columns:
#&gt;     seqnames    ranges strand
#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
#&gt;   A     chr1       3-5      *
#&gt;   B     chr1       7-9      *
#&gt;   C     chr1     10-11      *
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'>
<span class='fu'>intersectByGene</span>(<span class='no'>gr2</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"A"</span>, <span class='st'>"A"</span>, <span class='st'>"A"</span>, <span class='st'>"C"</span>))</div><div class='output co'>#&gt; GRanges object with 1 range and 0 metadata columns:
#&gt;     seqnames    ranges strand
#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
#&gt;   C     chr1     10-11      *
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'>
<span class='co'>#--------------------------------------------------#</span>
<span class='co'># reduceByGene</span>
<span class='co'>#--------------------------------------------------#</span>

<span class='co'># For a given gene, overlapping/adjacent ranges are combined;</span>
<span class='co'>#  gaps result in multiple ranges for that gene</span>
<span class='no'>gr</span></div><div class='output co'>#&gt; GRanges object with 4 ranges and 0 metadata columns:
#&gt;       seqnames    ranges strand
#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
#&gt;   [1]     chr1       1-4      *
#&gt;   [2]     chr1       3-5      *
#&gt;   [3]     chr1       7-9      *
#&gt;   [4]     chr1     10-11      *
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'>
<span class='fu'>reduceByGene</span>(<span class='no'>gr</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"A"</span>, <span class='st'>"A"</span>, <span class='st'>"A"</span>, <span class='st'>"A"</span>))</div><div class='output co'>#&gt; GRanges object with 2 ranges and 0 metadata columns:
#&gt;     seqnames    ranges strand
#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
#&gt;   A     chr1       1-5      *
#&gt;   A     chr1      7-11      *
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'>
<span class='co'># With disjoin = FALSE, ranges from different genes can overlap</span>
<span class='no'>gnames</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"A"</span>, <span class='st'>"B"</span>, <span class='st'>"B"</span>, <span class='st'>"B"</span>)
<span class='fu'>reduceByGene</span>(<span class='no'>gr</span>, <span class='no'>gnames</span>)</div><div class='output co'>#&gt; GRanges object with 3 ranges and 0 metadata columns:
#&gt;     seqnames    ranges strand
#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
#&gt;   A     chr1       1-4      *
#&gt;   B     chr1       3-5      *
#&gt;   B     chr1      7-11      *
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'>
<span class='co'># With disjoin = TRUE, segments overlapping &gt;1 gene are removed as well</span>
<span class='fu'>reduceByGene</span>(<span class='no'>gr</span>, <span class='no'>gnames</span>, <span class='kw'>disjoin</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)</div><div class='output co'>#&gt; GRanges object with 3 ranges and 0 metadata columns:
#&gt;     seqnames    ranges strand
#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
#&gt;   A     chr1       1-2      *
#&gt;   B     chr1         5      *
#&gt;   B     chr1      7-11      *
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'>
<span class='co'># Will use one more example to demonstrate how all</span>
<span class='co'>#  unambiguous segments are identified and returned</span>
<span class='no'>gr2</span></div><div class='output co'>#&gt; GRanges object with 4 ranges and 0 metadata columns:
#&gt;       seqnames    ranges strand
#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
#&gt;   [1]     chr1      1-10      *
#&gt;   [2]     chr1       3-5      *
#&gt;   [3]     chr1       7-9      *
#&gt;   [4]     chr1     10-11      *
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'>
<span class='no'>gnames</span></div><div class='output co'>#&gt; [1] "A" "B" "B" "B"</div><div class='input'><span class='fu'>reduceByGene</span>(<span class='no'>gr2</span>, <span class='no'>gnames</span>, <span class='kw'>disjoin</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)</div><div class='output co'>#&gt; GRanges object with 3 ranges and 0 metadata columns:
#&gt;     seqnames    ranges strand
#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
#&gt;   A     chr1       1-2      *
#&gt;   A     chr1         6      *
#&gt;   B     chr1        11      *
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'>
<span class='co'>#--------------------------------------------------#</span>
<span class='co'># reduceByGene, then aggregate counts by gene</span>
<span class='co'>#--------------------------------------------------#</span>

<span class='co'># Consider if you did getCountsByRegions on the last output,</span>
<span class='co'>#  you can aggregate those counts according to the genes</span>
<span class='no'>gr2_redux</span> <span class='kw'>&lt;-</span> <span class='fu'>reduceByGene</span>(<span class='no'>gr2</span>, <span class='no'>gnames</span>, <span class='kw'>disjoin</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)
<span class='no'>counts</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fl'>5</span>, <span class='fl'>2</span>, <span class='fl'>3</span>) <span class='co'># if these were the counts-by-regions</span>
<span class='fu'><a href='https://rdrr.io/r/stats/aggregate.html'>aggregate</a></span>(<span class='no'>counts</span> ~ <span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span>(<span class='no'>gr2_redux</span>), <span class='kw'>FUN</span> <span class='kw'>=</span> <span class='no'>sum</span>)</div><div class='output co'>#&gt;   names(gr2_redux) counts
#&gt; 1                A      7
#&gt; 2                B      3</div><div class='input'>
<span class='co'># even more convenient if using a melted dataframe</span>
<span class='no'>df</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span>(<span class='kw'>gene</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span>(<span class='no'>gr2_redux</span>),
                 <span class='kw'>reads</span> <span class='kw'>=</span> <span class='no'>counts</span>)
<span class='fu'><a href='https://rdrr.io/r/stats/aggregate.html'>aggregate</a></span>(<span class='no'>reads</span> ~ <span class='no'>gene</span>, <span class='no'>df</span>, <span class='kw'>FUN</span> <span class='kw'>=</span> <span class='no'>sum</span>)</div><div class='output co'>#&gt;   gene reads
#&gt; 1    A     7
#&gt; 2    B     3</div><div class='input'>
<span class='co'># can be extended to multiple samples</span>
<span class='no'>df</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/cbind.html'>rbind</a></span>(<span class='no'>df</span>, <span class='no'>df</span>)
<span class='no'>df</span>$<span class='no'>sample</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"s1"</span>, <span class='st'>"s2"</span>), <span class='kw'>each</span> <span class='kw'>=</span> <span class='fl'>3</span>)
<span class='no'>df</span>$<span class='no'>reads</span>[<span class='fl'>4</span>:<span class='fl'>6</span>] <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fl'>3</span>, <span class='fl'>1</span>, <span class='fl'>2</span>)
<span class='no'>df</span></div><div class='output co'>#&gt;   gene reads sample
#&gt; 1    A     5     s1
#&gt; 2    A     2     s1
#&gt; 3    B     3     s1
#&gt; 4    A     3     s2
#&gt; 5    A     1     s2
#&gt; 6    B     2     s2</div><div class='input'>
<span class='fu'><a href='https://rdrr.io/r/stats/aggregate.html'>aggregate</a></span>(<span class='no'>reads</span> ~ <span class='no'>sample</span>*<span class='no'>gene</span>, <span class='no'>df</span>, <span class='kw'>FUN</span> <span class='kw'>=</span> <span class='no'>sum</span>)</div><div class='output co'>#&gt;   sample gene reads
#&gt; 1     s1    A     7
#&gt; 2     s2    A     4
#&gt; 3     s1    B     3
#&gt; 4     s2    B     2</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      <li><a href="#value">Value</a></li>
      <li><a href="#details">Details</a></li>
      <li><a href="#typical-uses">Typical Uses</a></li>
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    <p>Mike DeBerardine</p>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


