<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Intersect or reduce ranges according to gene names — intersectByGene • BRGenomics</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Intersect or reduce ranges according to gene names — intersectByGene"><meta property="og:description" content='These functions divide up regions of interest according to associated names,
and perform an inter-range operation on them. intersectByGene returns
the "consensus" segment that is common to all input ranges, and returns no
more than one range per gene. reduceByGene collapses the input ranges
into one or more non-overlapping ranges that encompass all segments from the
input ranges.'><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Overview.html">Overview</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Guides
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/GettingStarted.html">Getting Started</a>
    </li>
    <li>
      <a href="../articles/ImportingModifyingAnnotations.html">Importing &amp; Modifying Annotations</a>
    </li>
    <li>
      <a href="../articles/ImportingProcessingData.html">Importing &amp; Processing Data</a>
    </li>
    <li>
      <a href="../articles/SignalCounting.html">Signal Counting</a>
    </li>
    <li>
      <a href="../articles/AnalyzingMultipleDatasets.html">Analyzing Multiple Datasets</a>
    </li>
    <li>
      <a href="../articles/ProfilePlotsAndBootstrapping.html">Profile Plots &amp; Bootstrapping</a>
    </li>
    <li>
      <a href="../articles/SpikeInNormalization.html">Spike-in Normalization</a>
    </li>
    <li>
      <a href="../articles/DESeq2WithGlobalPerturbations.html">DESeq2 with Global Perturbations</a>
    </li>
    <li>
      <a href="../articles/SequenceExtraction.html">Sequence Extraction</a>
    </li>
  </ul></li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/mdeber/BRGenomics" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Intersect or reduce ranges according to gene names</h1>
    <small class="dont-index">Source: <a href="https://github.com/mdeber/BRGenomics/blob/HEAD/R/roi_functions.R" class="external-link"><code>R/roi_functions.R</code></a></small>
    <div class="hidden name"><code>intersectByGene.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>These functions divide up regions of interest according to associated names,
and perform an inter-range operation on them. <code>intersectByGene</code> returns
the "consensus" segment that is common to all input ranges, and returns no
more than one range per gene. <code>reduceByGene</code> collapses the input ranges
into one or more non-overlapping ranges that encompass all segments from the
input ranges.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">intersectByGene</span><span class="op">(</span><span class="va">regions.gr</span>, <span class="va">gene_names</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">reduceByGene</span><span class="op">(</span><span class="va">regions.gr</span>, <span class="va">gene_names</span>, disjoin <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>regions.gr</dt>
<dd><p>A GRanges object containing regions of interest. If
<code>regions.gr</code> has the class <code>list</code>, <code>GRangesList</code>, or
<code>CompressedGRangesList</code>, it will be treated as if each list element is
a gene, and the GRanges within are the ranges associated with that gene.</p></dd>


<dt>gene_names</dt>
<dd><p>A character vector with the same length as
<code>regions.gr</code>.</p></dd>


<dt>disjoin</dt>
<dd><p>Logical. If <code>disjoin = TRUE</code>, the output GRanges is
disjoint, and each output range will match a single gene name. If
<code>FALSE</code>, segments from different genes can overlap.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A GRanges object whose individual ranges are named for the associated
  gene.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>These functions modify regions of interest that have associated
  names, such that several ranges share the same name, e.g. transcripts with
  associated gene names. Both functions "combine" the ranges on a
  gene-by-gene basis.</p>
<p><strong>intersectByGene</strong></p>
<p><em>For each unique gene</em>, the segment that overlaps <em>all</em> input
  ranges is returned. If no single range can be constructed that overlaps all
  input ranges, no range is returned for that gene (i.e. the gene is
  effectively filtered).</p>
<p>In other words, for all the ranges associated with a gene, the
  most-downstream start site is selected, and the most upstream end site is
  selected.</p>
<p><strong>reduceByGene</strong></p>
<p><em>For each unique gene</em>, the associated ranges are
  <code><a href="https://rdrr.io/pkg/IRanges/man/inter-range-methods.html" class="external-link">reduced</a></code> to produce one or
  more non-overlapping ranges. The output range(s) are effectively a
  <code>union</code> of the input ranges, and cover every input base.</p>
<p>With <code>disjoin = FALSE</code>, no genomic segment is overlapped by more than
  one range <em>of the same gene</em>, but ranges from different genes can
  overlap. With <code>disjoin = TRUE</code>, the output ranges are disjoint, and no
  genomic position is overlapped more than once. Any segment that overlaps
  more than one gene is removed, but any segment (i.e. any section of an
  input range) that overlaps only one gene is still maintained.</p>
    </div>
    <div id="typical-uses">
    <h2>Typical Uses</h2>
    


<p>A typical use for <code>intersectByGene</code> is to avoid transcript isoform
  selection, as the returned range is found in every isoform.</p>
<p><code>reduceByGene</code> can be used to count any and all reads that overlap any
  part of a gene's annotation, but without double-counting any of them. With
  <code>disjoin = FALSE</code>, no reads will be double-counted for the same gene,
  but the same read can be counted for multiple genes. With <code>disjoin =
  TRUE</code>, no read can be double-counted.</p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Mike DeBerardine</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Make example data:</span></span></span>
<span class="r-in"><span><span class="co">#  Ranges 1 and 2 overlap,</span></span></span>
<span class="r-in"><span><span class="co">#  Ranges 3 and 4 are adjacent</span></span></span>
<span class="r-in"><span><span class="va">gr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="st">"chr1"</span>,</span></span>
<span class="r-in"><span>              ranges <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">3</span>, <span class="fl">7</span>, <span class="fl">10</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                               end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span>, <span class="fl">5</span>, <span class="fl">9</span>, <span class="fl">11</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">gr</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 4 ranges and 0 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr1       1-4      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr1       3-5      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chr1       7-9      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chr1     10-11      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># intersectByGene</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">intersectByGene</span><span class="op">(</span><span class="va">gr</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"A"</span>, <span class="st">"B"</span>, <span class="st">"B"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 1 range and 0 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     seqnames    ranges strand</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   A     chr1       3-4      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">intersectByGene</span><span class="op">(</span><span class="va">gr</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"A"</span>, <span class="st">"B"</span>, <span class="st">"C"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 3 ranges and 0 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     seqnames    ranges strand</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   A     chr1       3-4      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   B     chr1       7-9      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   C     chr1     10-11      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">gr2</span> <span class="op">&lt;-</span> <span class="va">gr</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">end</a></span><span class="op">(</span><span class="va">gr2</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">10</span></span></span>
<span class="r-in"><span><span class="va">gr2</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 4 ranges and 0 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr1      1-10      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr1       3-5      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chr1       7-9      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chr1     10-11      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">intersectByGene</span><span class="op">(</span><span class="va">gr2</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"A"</span>, <span class="st">"B"</span>, <span class="st">"C"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 3 ranges and 0 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     seqnames    ranges strand</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   A     chr1       3-5      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   B     chr1       7-9      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   C     chr1     10-11      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">intersectByGene</span><span class="op">(</span><span class="va">gr2</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"A"</span>, <span class="st">"A"</span>, <span class="st">"C"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 1 range and 0 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     seqnames    ranges strand</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   C     chr1     10-11      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># reduceByGene</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># For a given gene, overlapping/adjacent ranges are combined;</span></span></span>
<span class="r-in"><span><span class="co">#  gaps result in multiple ranges for that gene</span></span></span>
<span class="r-in"><span><span class="va">gr</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 4 ranges and 0 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr1       1-4      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr1       3-5      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chr1       7-9      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chr1     10-11      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">reduceByGene</span><span class="op">(</span><span class="va">gr</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"A"</span>, <span class="st">"A"</span>, <span class="st">"A"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 2 ranges and 0 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     seqnames    ranges strand</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   A     chr1       1-5      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   A     chr1      7-11      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># With disjoin = FALSE, ranges from different genes can overlap</span></span></span>
<span class="r-in"><span><span class="va">gnames</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"B"</span>, <span class="st">"B"</span>, <span class="st">"B"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">reduceByGene</span><span class="op">(</span><span class="va">gr</span>, <span class="va">gnames</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 3 ranges and 0 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     seqnames    ranges strand</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   A     chr1       1-4      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   B     chr1       3-5      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   B     chr1      7-11      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># With disjoin = TRUE, segments overlapping &gt;1 gene are removed as well</span></span></span>
<span class="r-in"><span><span class="fu">reduceByGene</span><span class="op">(</span><span class="va">gr</span>, <span class="va">gnames</span>, disjoin <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 3 ranges and 0 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     seqnames    ranges strand</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   A     chr1       1-2      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   B     chr1         5      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   B     chr1      7-11      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Will use one more example to demonstrate how all</span></span></span>
<span class="r-in"><span><span class="co">#  unambiguous segments are identified and returned</span></span></span>
<span class="r-in"><span><span class="va">gr2</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 4 ranges and 0 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr1      1-10      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr1       3-5      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chr1       7-9      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chr1     10-11      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">gnames</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "A" "B" "B" "B"</span>
<span class="r-in"><span><span class="fu">reduceByGene</span><span class="op">(</span><span class="va">gr2</span>, <span class="va">gnames</span>, disjoin <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 3 ranges and 0 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     seqnames    ranges strand</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   A     chr1       1-2      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   A     chr1         6      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   B     chr1        11      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># reduceByGene, then aggregate counts by gene</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Consider if you did getCountsByRegions on the last output,</span></span></span>
<span class="r-in"><span><span class="co">#  you can aggregate those counts according to the genes</span></span></span>
<span class="r-in"><span><span class="va">gr2_redux</span> <span class="op">&lt;-</span> <span class="fu">reduceByGene</span><span class="op">(</span><span class="va">gr2</span>, <span class="va">gnames</span>, disjoin <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">2</span>, <span class="fl">3</span><span class="op">)</span> <span class="co"># if these were the counts-by-regions</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/stats/aggregate.html" class="external-link">aggregate</a></span><span class="op">(</span><span class="va">counts</span> <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gr2_redux</span><span class="op">)</span>, FUN <span class="op">=</span> <span class="va">sum</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   names(gr2_redux) counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1                A      7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2                B      3</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># even more convenient if using a melted dataframe</span></span></span>
<span class="r-in"><span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gr2_redux</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                 reads <span class="op">=</span> <span class="va">counts</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/stats/aggregate.html" class="external-link">aggregate</a></span><span class="op">(</span><span class="va">reads</span> <span class="op">~</span> <span class="va">gene</span>, <span class="va">df</span>, FUN <span class="op">=</span> <span class="va">sum</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   gene reads</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1    A     7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2    B     3</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># can be extended to multiple samples</span></span></span>
<span class="r-in"><span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">df</span>, <span class="va">df</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">df</span><span class="op">$</span><span class="va">sample</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"s1"</span>, <span class="st">"s2"</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">df</span><span class="op">$</span><span class="va">reads</span><span class="op">[</span><span class="fl">4</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">df</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   gene reads sample</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1    A     5     s1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2    A     2     s1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3    B     3     s1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4    A     3     s2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5    A     1     s2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6    B     2     s2</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/stats/aggregate.html" class="external-link">aggregate</a></span><span class="op">(</span><span class="va">reads</span> <span class="op">~</span> <span class="va">sample</span><span class="op">*</span><span class="va">gene</span>, <span class="va">df</span>, FUN <span class="op">=</span> <span class="va">sum</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sample gene reads</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1     s1    A     7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2     s2    A     4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3     s1    B     3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4     s2    B     2</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

