<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Iterative Subsampling for Metaplotting (On Count Matrices) — metaSubsampleMatrix • BRGenomics</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Iterative Subsampling for Metaplotting (On Count Matrices) — metaSubsampleMatrix" />
<meta property="og:description" content="In the most general sense, this function performs iterations of randomly
subsampling rows of a matrix, and returns a summary of mean values calculated
for each column. The typical application is for generating metaplots, with
the typical input being a matrix in which each row is a gene or other region
of interest, each column is a position within that gene (either a specific
basepair or a bin), and element values are signal (e.g. read counts) within
those positions." />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Iterative Subsampling for Metaplotting (On Count Matrices)</h1>
    
    <div class="hidden name"><code>metaSubsampleMatrix.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>In the most general sense, this function performs iterations of randomly
subsampling rows of a matrix, and returns a summary of mean values calculated
for each column. The typical application is for generating metaplots, with
the typical input being a matrix in which each row is a gene or other region
of interest, each column is a position within that gene (either a specific
basepair or a bin), and element values are signal (e.g. read counts) within
those positions.</p>
    </div>

    <pre class="usage"><span class='fu'>metaSubsampleMatrix</span>(
  <span class='no'>counts.mat</span>,
  <span class='kw'>binsize</span> <span class='kw'>=</span> <span class='fl'>1</span>,
  <span class='kw'>first.output.xval</span> <span class='kw'>=</span> <span class='fl'>1</span>,
  <span class='kw'>sample.name</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/deparse.html'>deparse</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/substitute.html'>substitute</a></span>(<span class='no'>counts.mat</span>)),
  <span class='kw'>n.iter</span> <span class='kw'>=</span> <span class='fl'>1000</span>,
  <span class='kw'>prop.sample</span> <span class='kw'>=</span> <span class='fl'>0.1</span>,
  <span class='kw'>lower</span> <span class='kw'>=</span> <span class='fl'>0.125</span>,
  <span class='kw'>upper</span> <span class='kw'>=</span> <span class='fl'>0.875</span>,
  <span class='kw'>NF</span> <span class='kw'>=</span> <span class='fl'>1</span>,
  <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fl'>1</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>counts.mat</th>
      <td><p>A matrix of signal counts in which rows are regions of
interest and columns are sites/bins in each region.</p></td>
    </tr>
    <tr>
      <th>binsize</th>
      <td><p>The size of bin (number of columns, e.g. basepairs) to use for
metaplotting. Especially important for metaplots over large/sparse regions.</p></td>
    </tr>
    <tr>
      <th>first.output.xval</th>
      <td><p>The relative start position of the first bin, e.g.
if regions.gr begins at 50 bases upstream of the TSS, set
<code>first.output.xval = -50</code>. This number only affects the x-values that
are returned, which are provided as a convenience.</p></td>
    </tr>
    <tr>
      <th>sample.name</th>
      <td><p>Defaults to the name of <code>dataset.gr</code>.</p></td>
    </tr>
    <tr>
      <th>n.iter</th>
      <td><p>Number of random subsampling iterations to perform. Default is
1000.</p></td>
    </tr>
    <tr>
      <th>prop.sample</th>
      <td><p>The proportion of rows to subsample in each iteration.
The default is 0.1.</p></td>
    </tr>
    <tr>
      <th>lower</th>
      <td><p>The lower quantile of subsampled signal means to return. The
default is 0.125 (12.5th percentile).</p></td>
    </tr>
    <tr>
      <th>upper</th>
      <td><p>The upper quantile of subsampled signal means to return. The
default is 0.875 (85.5th percentile).</p></td>
    </tr>
    <tr>
      <th>NF</th>
      <td><p>Optional normalization factor by which to multiply the counts.</p></td>
    </tr>
    <tr>
      <th>ncores</th>
      <td><p>Number of cores to use for parallel computation. As of writing,
parallel processing doesn't show any benefit for short computation times
(e.g. &lt;1 minute for our typical experience on a laptop).</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Dataframe containing x-values, means, lower quantiles, upper
  quantiles, and the sample name (as a convenience for row-binding multiple
  of these dataframes).</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='metaSubsample.html'>metaSubsample</a></code>,
  <code><a href='getCountsByPositions.html'>getCountsByPositions</a></code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='st'>"PROseq"</span>) <span class='co'># load included PROseq data</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='st'>"txs_dm6_chr4"</span>) <span class='co'># load included transcripts</span>

<span class='co'># for each transcript, use promoter-proximal region from TSS to +100</span>
<span class='no'>pr</span> <span class='kw'>&lt;-</span> <span class='fu'>promoters</span>(<span class='no'>txs_dm6_chr4</span>, <span class='fl'>0</span>, <span class='fl'>100</span>)

<span class='co'># generate a matrix of counts in each region</span>
<span class='no'>countsmat</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='getCountsByPositions.html'>getCountsByPositions</a></span>(<span class='no'>PROseq</span>, <span class='no'>pr</span>)
<span class='fu'><a href='https://rdrr.io/r/base/dim.html'>dim</a></span>(<span class='no'>countsmat</span>)</div><div class='output co'>#&gt; [1] 339 100</div><div class='input'>
<span class='co'>#--------------------------------------------------#</span>
<span class='co'># bootstrap average signal in 10 bp bins across all transcripts</span>
<span class='co'>#--------------------------------------------------#</span>

<span class='fu'><a href='https://rdrr.io/r/base/Random.html'>set.seed</a></span>(<span class='fl'>11</span>)
<span class='no'>df</span> <span class='kw'>&lt;-</span> <span class='fu'>metaSubsampleMatrix</span>(<span class='no'>countsmat</span>, <span class='kw'>binsize</span> <span class='kw'>=</span> <span class='fl'>10</span>, <span class='kw'>sample.name</span> <span class='kw'>=</span> <span class='st'>"PROseq"</span>)
<span class='no'>df</span>[<span class='fl'>1</span>:<span class='fl'>10</span>, ]</div><div class='output co'>#&gt;     x       mean     lower     upper sample.name
#&gt; 1   6  0.4705882 0.2352941  1.147059      PROseq
#&gt; 2  16  0.7058824 0.3529412  1.735294      PROseq
#&gt; 3  26  1.2058824 0.7058824  2.176471      PROseq
#&gt; 4  36  3.6470588 1.7647059 11.687500      PROseq
#&gt; 5  46  5.6617647 2.9963235 11.709559      PROseq
#&gt; 6  56 10.3382353 6.0000000 15.709559      PROseq
#&gt; 7  66  5.2058824 3.2941176  7.411765      PROseq
#&gt; 8  76  5.4558824 2.1433824 10.003676      PROseq
#&gt; 9  86  3.0588235 2.0588235  4.441176      PROseq
#&gt; 10 96  2.8676471 1.6470588  4.558824      PROseq</div><div class='input'>
<span class='co'>#--------------------------------------------------#</span>
<span class='co'># the same, using a normalization factor, and changing the x-values</span>
<span class='co'>#--------------------------------------------------#</span>

<span class='fu'><a href='https://rdrr.io/r/base/Random.html'>set.seed</a></span>(<span class='fl'>11</span>)
<span class='no'>df</span> <span class='kw'>&lt;-</span> <span class='fu'>metaSubsampleMatrix</span>(<span class='no'>countsmat</span>, <span class='kw'>binsize</span> <span class='kw'>=</span> <span class='fl'>10</span>, <span class='kw'>first.output.xval</span> <span class='kw'>=</span> <span class='fl'>0</span>,
                          <span class='kw'>NF</span> <span class='kw'>=</span> <span class='fl'>0.75</span>, <span class='kw'>sample.name</span> <span class='kw'>=</span> <span class='st'>"PROseq"</span>)
<span class='no'>df</span>[<span class='fl'>1</span>:<span class='fl'>10</span>, ]</div><div class='output co'>#&gt;     x      mean     lower      upper sample.name
#&gt; 1   5 0.3529412 0.1764706  0.8602941      PROseq
#&gt; 2  15 0.5294118 0.2647059  1.3014706      PROseq
#&gt; 3  25 0.9044118 0.5294118  1.6323529      PROseq
#&gt; 4  35 2.7352941 1.3235294  8.7656250      PROseq
#&gt; 5  45 4.2463235 2.2472426  8.7821691      PROseq
#&gt; 6  55 7.7536765 4.5000000 11.7821691      PROseq
#&gt; 7  65 3.9044118 2.4705882  5.5588235      PROseq
#&gt; 8  75 4.0919118 1.6075368  7.5027574      PROseq
#&gt; 9  85 2.2941176 1.5441176  3.3308824      PROseq
#&gt; 10 95 2.1507353 1.2352941  3.4191176      PROseq</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      <li><a href="#value">Value</a></li>
      <li><a href="#see-also">See also</a></li>
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    <p>Mike DeBerardine</p>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


