<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Subset regions of interest by quantiles of overlapping signal — subsetRegionsBySignal • BRGenomics</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Subset regions of interest by quantiles of overlapping signal — subsetRegionsBySignal"><meta property="og:description" content="A convenience function to subset regions of interest by the
  amount of signal they contain, according to their quantile (i.e. their
  signal ranks)."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Overview.html">Overview</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Guides
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/GettingStarted.html">Getting Started</a>
    </li>
    <li>
      <a href="../articles/ImportingModifyingAnnotations.html">Importing &amp; Modifying Annotations</a>
    </li>
    <li>
      <a href="../articles/ImportingProcessingData.html">Importing &amp; Processing Data</a>
    </li>
    <li>
      <a href="../articles/SignalCounting.html">Signal Counting</a>
    </li>
    <li>
      <a href="../articles/AnalyzingMultipleDatasets.html">Analyzing Multiple Datasets</a>
    </li>
    <li>
      <a href="../articles/ProfilePlotsAndBootstrapping.html">Profile Plots &amp; Bootstrapping</a>
    </li>
    <li>
      <a href="../articles/SpikeInNormalization.html">Spike-in Normalization</a>
    </li>
    <li>
      <a href="../articles/DESeq2WithGlobalPerturbations.html">DESeq2 with Global Perturbations</a>
    </li>
    <li>
      <a href="../articles/SequenceExtraction.html">Sequence Extraction</a>
    </li>
  </ul></li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/mdeber/BRGenomics" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Subset regions of interest by quantiles of overlapping signal</h1>
    <small class="dont-index">Source: <a href="https://github.com/mdeber/BRGenomics/blob/HEAD/R/roi_functions.R" class="external-link"><code>R/roi_functions.R</code></a></small>
    <div class="hidden name"><code>subsetRegionsBySignal.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>A convenience function to subset regions of interest by the
  amount of signal they contain, according to their quantile (i.e. their
  signal ranks).</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">subsetRegionsBySignal</span><span class="op">(</span></span>
<span>  <span class="va">regions.gr</span>,</span>
<span>  <span class="va">dataset.gr</span>,</span>
<span>  quantiles <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="fl">1</span><span class="op">)</span>,</span>
<span>  field <span class="op">=</span> <span class="st">"score"</span>,</span>
<span>  order.by.rank <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  density <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  keep.signal <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  expand_ranges <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>regions.gr</dt>
<dd><p>A GRanges object containing regions of interest.</p></dd>


<dt>dataset.gr</dt>
<dd><p>A GRanges object in which signal is contained in metadata
(typically in the "score" field).</p></dd>


<dt>quantiles</dt>
<dd><p>A value pair giving the lower quantile and upper quantile of
regions to keep. Regions with signal quantiles below the lower quantile are
removed, and likewise for regions with signal quantiles above the upper
quantile. Quantiles must be in range <code>(0, 1)</code>. An empty GRanges object
is returned if the lower quantile is set to <code>1</code> or if the upper
quantile is set to <code>0</code>.</p></dd>


<dt>field</dt>
<dd><p>The metadata field of <code>dataset.gr</code> to be counted, typically
"score".</p></dd>


<dt>order.by.rank</dt>
<dd><p>If <code>TRUE</code>, the output regions are sorted based on
the amount of overlapping signal (in decreasing order). If <code>FALSE</code>
(the default), genes are sorted by their positions.</p></dd>


<dt>density</dt>
<dd><p>A logical indicating whether signal counts should be
normalized to the width (chromosomal length) of ranges in
<code>regions.gr</code>. By default, no length normalization is performed.</p></dd>


<dt>keep.signal</dt>
<dd><p>Logical indicating if signal counts should be kept. If set
to <code>TRUE</code>, the signal for each range (length-normalized if
<code>density = TRUE</code>) are kept as a new <code>Signal</code> metadata column in
the output GRanges object.</p></dd>


<dt>expand_ranges</dt>
<dd><p>Logical indicating if ranges in <code>dataset.gr</code> should
be treated as descriptions of single molecules (<code>FALSE</code>), or if ranges
should be treated as representing multiple adjacent positions with the same
signal (<code>TRUE</code>). See <code><a href="getCountsByRegions.html">
getCountsByRegions</a></code>.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A GRanges object of length <code>length(regions.gr) * (upper_quantile
  - lower_quantile)</code>.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="getCountsByRegions.html">getCountsByRegions</a></code></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Mike DeBerardine</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"PROseq"</span><span class="op">)</span> <span class="co"># load included PROseq data</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"txs_dm6_chr4"</span><span class="op">)</span> <span class="co"># load included transcripts</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">txs_dm6_chr4</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 339 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seqnames          ranges strand |     tx_name     gene_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [1]     chr4        879-5039      + | FBtr0346692 FBgn0267363</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [2]     chr4     42774-43374      + | FBtr0344900 FBgn0266617</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [3]     chr4     44774-46074      + | FBtr0340499 FBgn0265633</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [4]     chr4     56497-60974      + | FBtr0333704 FBgn0264617</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [5]     chr4     56497-63124      + | FBtr0333705 FBgn0264617</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ...      ...             ...    ... .         ...         ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [335]     chr4 1192419-1196848      - | FBtr0100543 FBgn0039924</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [336]     chr4 1192419-1196848      - | FBtr0100544 FBgn0039924</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [337]     chr4 1225089-1230713      - | FBtr0100406 FBgn0027101</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [338]     chr4 1225737-1230713      - | FBtr0100402 FBgn0027101</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [339]     chr4 1225737-1230713      - | FBtr0100404 FBgn0027101</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 7 sequences from dm6 genome</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># get the top 50% of transcripts by signal</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">subsetRegionsBySignal</span><span class="op">(</span><span class="va">txs_dm6_chr4</span>, <span class="va">PROseq</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 170 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seqnames          ranges strand |     tx_name     gene_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [1]     chr4    69326-101419      + | FBtr0100246 FBgn0085432</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [2]     chr4    69326-108694      + | FBtr0089159 FBgn0085432</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [3]     chr4    69326-108694      + | FBtr0334301 FBgn0085432</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [4]     chr4    69326-109770      + | FBtr0100245 FBgn0085432</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [5]     chr4    69326-110059      + | FBtr0308615 FBgn0085432</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ...      ...             ...    ... .         ...         ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [166]     chr4 1192419-1196848      - | FBtr0100543 FBgn0039924</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [167]     chr4 1192419-1196848      - | FBtr0100544 FBgn0039924</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [168]     chr4 1225089-1230713      - | FBtr0100406 FBgn0027101</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [169]     chr4 1225737-1230713      - | FBtr0100402 FBgn0027101</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [170]     chr4 1225737-1230713      - | FBtr0100404 FBgn0027101</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 7 sequences from dm6 genome</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># get the middle 50% of transcripts by signal</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">subsetRegionsBySignal</span><span class="op">(</span><span class="va">txs_dm6_chr4</span>, <span class="va">PROseq</span>, quantiles <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.25</span>, <span class="fl">0.75</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 169 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seqnames          ranges strand |     tx_name     gene_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [1]     chr4     56497-60974      + | FBtr0333704 FBgn0264617</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [2]     chr4     56497-63124      + | FBtr0333705 FBgn0264617</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [3]     chr4   136272-140083      + | FBtr0089450 FBgn0052000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [4]     chr4   136272-143123      + | FBtr0089448 FBgn0052000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [5]     chr4   136272-143123      + | FBtr0089449 FBgn0052000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ...      ...             ...    ... .         ...         ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [165]     chr4 1192419-1196848      - | FBtr0100543 FBgn0039924</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [166]     chr4 1192419-1196848      - | FBtr0100544 FBgn0039924</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [167]     chr4 1225089-1230713      - | FBtr0100406 FBgn0027101</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [168]     chr4 1225737-1230713      - | FBtr0100402 FBgn0027101</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [169]     chr4 1225737-1230713      - | FBtr0100404 FBgn0027101</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 7 sequences from dm6 genome</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># get the top 10% of transcripts by signal, and sort them by highest signal</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">subsetRegionsBySignal</span><span class="op">(</span><span class="va">txs_dm6_chr4</span>, <span class="va">PROseq</span>, quantiles <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.9</span>, <span class="fl">1</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                      order.by.rank <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 34 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        seqnames        ranges strand |     tx_name     gene_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [1]     chr4 649041-663103      - | FBtr0310542 FBgn0051992</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [2]     chr4  69653-114270      + | FBtr0309803 FBgn0085432</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [3]     chr4  69326-110059      + | FBtr0308615 FBgn0085432</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [4]     chr4  69326-109770      + | FBtr0100245 FBgn0085432</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [5]     chr4  69326-108694      + | FBtr0334301 FBgn0085432</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    ...      ...           ...    ... .         ...         ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [30]     chr4 649971-659983      - | FBtr0089097 FBgn0051992</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [31]     chr4 184225-193489      - | FBtr0089150 FBgn0039890</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [32]     chr4 184225-193489      - | FBtr0089149 FBgn0039890</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [33]     chr4 132220-140083      + | FBtr0089447 FBgn0052000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [34]     chr4 132220-140083      + | FBtr0089446 FBgn0052000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 7 sequences from dm6 genome</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span><span class="co"># remove the most extreme 10% of regions, and keep scores</span></span></span>
<span class="r-in"><span><span class="co">#--------------------------------------------------#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">subsetRegionsBySignal</span><span class="op">(</span><span class="va">txs_dm6_chr4</span>, <span class="va">PROseq</span>, quantiles <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.05</span>, <span class="fl">0.95</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                      keep.signal <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 305 ranges and 3 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seqnames          ranges strand |     tx_name     gene_id    Signal</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [1]     chr4     42774-43374      + | FBtr0344900 FBgn0266617        59</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [2]     chr4     44774-46074      + | FBtr0340499 FBgn0265633        13</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [3]     chr4     56497-60974      + | FBtr0333704 FBgn0264617       126</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [4]     chr4     56497-63124      + | FBtr0333705 FBgn0264617       263</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [5]     chr4    90355-110059      + | FBtr0112657 FBgn0085432      1464</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ...      ...             ...    ... .         ...         ...       ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [301]     chr4 1192419-1196848      - | FBtr0100543 FBgn0039924       454</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [302]     chr4 1192419-1196848      - | FBtr0100544 FBgn0039924       454</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [303]     chr4 1225089-1230713      - | FBtr0100406 FBgn0027101       766</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [304]     chr4 1225737-1230713      - | FBtr0100402 FBgn0027101       665</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [305]     chr4 1225737-1230713      - | FBtr0100404 FBgn0027101       665</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 7 sequences from dm6 genome</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

