<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Subset regions of interest by quantiles of overlapping signal — subsetRegionsBySignal • BRGenomics</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Subset regions of interest by quantiles of overlapping signal — subsetRegionsBySignal" />
<meta property="og:description" content="A convenience function to subset regions of interest by the
  amount of signal they contain, according to their quantile (i.e. their
  signal ranks)." />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Subset regions of interest by quantiles of overlapping signal</h1>
    
    <div class="hidden name"><code>subsetRegionsBySignal.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>A convenience function to subset regions of interest by the
  amount of signal they contain, according to their quantile (i.e. their
  signal ranks).</p>
    </div>

    <pre class="usage"><span class='fu'>subsetRegionsBySignal</span>(
  <span class='no'>regions.gr</span>,
  <span class='no'>dataset.gr</span>,
  <span class='kw'>quantiles</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fl'>0.5</span>, <span class='fl'>1</span>),
  <span class='kw'>field</span> <span class='kw'>=</span> <span class='st'>"score"</span>,
  <span class='kw'>order.by.rank</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>density</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>regions.gr</th>
      <td><p>A GRanges object containing regions of interest.</p></td>
    </tr>
    <tr>
      <th>dataset.gr</th>
      <td><p>A GRanges object in which signal is contained in metadata
(typically in the "score" field).</p></td>
    </tr>
    <tr>
      <th>quantiles</th>
      <td><p>A value pair giving the lower quantile and upper quantile of
regions to keep. Regions with signal quantiles below than the lower
quantile are removed, while regions with signal quantiles above the upper
quantile are removed. Quantiles must be in range <code>(0, 1)</code>. An empty
GRanges object is returned if <code>lower quantile = 1</code> or <code>upper
quantile = 0</code>.</p></td>
    </tr>
    <tr>
      <th>field</th>
      <td><p>The metadata field of <code>dataset.gr</code> to be counted.</p></td>
    </tr>
    <tr>
      <th>order.by.rank</th>
      <td><p>If <code>TRUE</code>, the output regions are sorted based on
the amount of signal contained (in decreasing order). If <code>FALSE</code> (the
default), genes are sorted by their positions.</p></td>
    </tr>
    <tr>
      <th>density</th>
      <td><p>A logical indicating whether signal counts should be
normalized to the width of ranges in <code>regions.gr</code>. By default, the
function only considers the total signal in each range.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A GRanges object of length <code>length(regions.gr) * (upper_quantile
  - lower_quantile)</code>.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Typical uses may include removing the 5<!-- % of genes with the lowest -->
  signal (<code>lower_quantile = 0.05</code>) and the 5<!-- % with the highest signal -->
  (<code>upper_quantile = 0.95</code>), or returning the middle 50<!-- % of genes by -->
  signal (<code>lower_quantile = 0.25</code>, <code>upper_quantile = 0.75</code>). If
  <code>lower_quantile = 0</code> and <code>upper_quantile = 1</code>, all regions are
  returned, but the returned regions will be sorted by position, or by score
  if <code>order.by.rank = TRUE</code>.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='getCountsByRegions.html'>getCountsByRegions</a></code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='st'>"PROseq"</span>) <span class='co'># load included PROseq data</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='st'>"txs_dm6_chr4"</span>) <span class='co'># load included transcripts</span>

<span class='no'>txs_dm6_chr4</span></div><div class='output co'>#&gt; GRanges object with 339 ranges and 2 metadata columns:
#&gt;         seqnames          ranges strand |     tx_name     gene_id
#&gt;            &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
#&gt;     [1]     chr4        879-5039      + | FBtr0346692 FBgn0267363
#&gt;     [2]     chr4     42774-43374      + | FBtr0344900 FBgn0266617
#&gt;     [3]     chr4     44774-46074      + | FBtr0340499 FBgn0265633
#&gt;     [4]     chr4     56497-60974      + | FBtr0333704 FBgn0264617
#&gt;     [5]     chr4     56497-63124      + | FBtr0333705 FBgn0264617
#&gt;     ...      ...             ...    ... .         ...         ...
#&gt;   [335]     chr4 1192419-1196848      - | FBtr0100543 FBgn0039924
#&gt;   [336]     chr4 1192419-1196848      - | FBtr0100544 FBgn0039924
#&gt;   [337]     chr4 1225089-1230713      - | FBtr0100406 FBgn0027101
#&gt;   [338]     chr4 1225737-1230713      - | FBtr0100402 FBgn0027101
#&gt;   [339]     chr4 1225737-1230713      - | FBtr0100404 FBgn0027101
#&gt;   -------
#&gt;   seqinfo: 7 sequences from dm6 genome</div><div class='input'>
<span class='co'>#--------------------------------------------------#</span>
<span class='co'># get the top 50% of transcripts by signal</span>
<span class='co'>#--------------------------------------------------#</span>

<span class='fu'>subsetRegionsBySignal</span>(<span class='no'>txs_dm6_chr4</span>, <span class='no'>PROseq</span>)</div><div class='output co'>#&gt; GRanges object with 170 ranges and 2 metadata columns:
#&gt;         seqnames          ranges strand |     tx_name     gene_id
#&gt;            &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
#&gt;     [1]     chr4    69326-101419      + | FBtr0100246 FBgn0085432
#&gt;     [2]     chr4    69326-108694      + | FBtr0089159 FBgn0085432
#&gt;     [3]     chr4    69326-108694      + | FBtr0334301 FBgn0085432
#&gt;     [4]     chr4    69326-109770      + | FBtr0100245 FBgn0085432
#&gt;     [5]     chr4    69326-110059      + | FBtr0308615 FBgn0085432
#&gt;     ...      ...             ...    ... .         ...         ...
#&gt;   [166]     chr4 1192419-1196848      - | FBtr0100543 FBgn0039924
#&gt;   [167]     chr4 1192419-1196848      - | FBtr0100544 FBgn0039924
#&gt;   [168]     chr4 1225089-1230713      - | FBtr0100406 FBgn0027101
#&gt;   [169]     chr4 1225737-1230713      - | FBtr0100402 FBgn0027101
#&gt;   [170]     chr4 1225737-1230713      - | FBtr0100404 FBgn0027101
#&gt;   -------
#&gt;   seqinfo: 7 sequences from dm6 genome</div><div class='input'>
<span class='co'>#--------------------------------------------------#</span>
<span class='co'># get the middle 50% of transcripts by signal</span>
<span class='co'>#--------------------------------------------------#</span>

<span class='fu'>subsetRegionsBySignal</span>(<span class='no'>txs_dm6_chr4</span>, <span class='no'>PROseq</span>, <span class='kw'>quantiles</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fl'>0.25</span>, <span class='fl'>0.75</span>))</div><div class='output co'>#&gt; GRanges object with 169 ranges and 2 metadata columns:
#&gt;         seqnames          ranges strand |     tx_name     gene_id
#&gt;            &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
#&gt;     [1]     chr4     56497-60974      + | FBtr0333704 FBgn0264617
#&gt;     [2]     chr4     56497-63124      + | FBtr0333705 FBgn0264617
#&gt;     [3]     chr4   136272-140083      + | FBtr0089450 FBgn0052000
#&gt;     [4]     chr4   136272-143123      + | FBtr0089448 FBgn0052000
#&gt;     [5]     chr4   136272-143123      + | FBtr0089449 FBgn0052000
#&gt;     ...      ...             ...    ... .         ...         ...
#&gt;   [165]     chr4 1192419-1196848      - | FBtr0100543 FBgn0039924
#&gt;   [166]     chr4 1192419-1196848      - | FBtr0100544 FBgn0039924
#&gt;   [167]     chr4 1225089-1230713      - | FBtr0100406 FBgn0027101
#&gt;   [168]     chr4 1225737-1230713      - | FBtr0100402 FBgn0027101
#&gt;   [169]     chr4 1225737-1230713      - | FBtr0100404 FBgn0027101
#&gt;   -------
#&gt;   seqinfo: 7 sequences from dm6 genome</div><div class='input'>
<span class='co'>#--------------------------------------------------#</span>
<span class='co'># get the top 10% of transcripts by signal, and sort them by highest signal</span>
<span class='co'>#--------------------------------------------------#</span>

<span class='fu'>subsetRegionsBySignal</span>(<span class='no'>txs_dm6_chr4</span>, <span class='no'>PROseq</span>,
                      <span class='kw'>quantiles</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fl'>0.9</span>, <span class='fl'>1</span>),
                      <span class='kw'>order.by.rank</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)</div><div class='output co'>#&gt; GRanges object with 34 ranges and 2 metadata columns:
#&gt;        seqnames        ranges strand |     tx_name     gene_id
#&gt;           &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
#&gt;    [1]     chr4 649041-663103      - | FBtr0310542 FBgn0051992
#&gt;    [2]     chr4  69653-114270      + | FBtr0309803 FBgn0085432
#&gt;    [3]     chr4  69326-110059      + | FBtr0308615 FBgn0085432
#&gt;    [4]     chr4  69326-109770      + | FBtr0100245 FBgn0085432
#&gt;    [5]     chr4  69326-108694      + | FBtr0334301 FBgn0085432
#&gt;    ...      ...           ...    ... .         ...         ...
#&gt;   [30]     chr4 649971-659983      - | FBtr0089097 FBgn0051992
#&gt;   [31]     chr4 184225-193489      - | FBtr0089150 FBgn0039890
#&gt;   [32]     chr4 184225-193489      - | FBtr0089149 FBgn0039890
#&gt;   [33]     chr4 132220-140083      + | FBtr0089447 FBgn0052000
#&gt;   [34]     chr4 132220-140083      + | FBtr0089446 FBgn0052000
#&gt;   -------
#&gt;   seqinfo: 7 sequences from dm6 genome</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      <li><a href="#value">Value</a></li>
      <li><a href="#details">Details</a></li>
      <li><a href="#see-also">See also</a></li>
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    <p>Mike DeBerardine</p>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


