<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Basepair-Resolution Analysis with BRGenomics • BRGenomics</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Basepair-Resolution Analysis with BRGenomics">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.7.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BRGenomicsOverview.html">Basepair-Resolution Analysis with BRGenomics</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Basepair-Resolution Analysis with BRGenomics</h1>
            <h3 class="subtitle">
<em>Straightforward and efficient tools for analyzing genomics data</em> [EARLY DRAFT]</h3>
                        <h4 class="author">Mike DeBerardine</h4>
                  <a class="author_email" href="mailto:#"></a><a href="mailto:mike.deberardine@gmail.com" class="email">mike.deberardine@gmail.com</a>
      
                  
      
      
      <div class="hidden name"><code>BRGenomicsOverview.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      BRGenomics is designed to help users avoid code repetition by providing efficient and tested functions to accomplish common, discrete tasks in the analysis of high-throughput sequencing data. The included functions are geared toward analyzing basepair-resolution sequencing data, the properties of which are exploited to increase performance and user-friendliness. We leverage standard Bioconductor methods and classes to maximize compatability with its rich ecoystem of bioinformatics tools, and we aim to make BRGenomics sufficient for most post-alignment data processing. Common data processing and analytical steps are turned into fast-running one-liners that can be simultaneously applied across numerous datasets. BRGenomics is fully-documented, and we aim it to be beginner-friendly.
    </div>
    
<div id="motivation" class="section level1">
<h1 class="hasAnchor">
<a href="#motivation" class="anchor"></a>Motivation</h1>
<p>This package is designed to:</p>
<ul>
<li>Replace the use of command-line utilities for most post-alignment processing, e.g. <code>bedtools</code> and <code>deeptools</code>
</li>
<li>Be easy-to-use and easy-to-install, without requiring external dependencies, e.g. <code>hitslib</code> or the kent source utilities from the UCSC genome browser</li>
<li>Allow users to string together common analysis pipelines with simple, fast-running one-liners</li>
<li>Avoid code repetition by providing tested and validated code</li>
<li>Exploit the properties of basepair-resolution data to optimize performance and increase user-friendliness</li>
<li>Use process forking to make use of multicore processors</li>
<li>Maximize compatability with Bioconductor’s rich ecosystem of analysis software, in addition to leveraging the traditional strengths of R in statistics and data visualization</li>
<li>Fully replace the <code>bigWig</code> R package</li>
</ul>
</div>
<div id="features" class="section level1">
<h1 class="hasAnchor">
<a href="#features" class="anchor"></a>Features</h1>
<ul>
<li>Process and import bedGraoh, bigWig, and bam files quickly and easily, with several pre-configured defaults for typical uses</li>
<li>Count and filter spike-in reads</li>
<li>Calculate spike-in normalization factors using several methods and options, including options for batch normalization</li>
<li>Count reads by regions of interest</li>
<li>Count reads at positions within regions of interest, at single-base resolution or in larger bins, and generate count matrices for heatmapping</li>
<li>Calculate bootstrapped signal (e.g. readcount) profiles with confidence intervals (i.e. meta-profiles)</li>
<li>Modify gene regions (e.g. extract promoters or genebody regions) using a single simple and straightforward function</li>
<li>Conveniently, efficiently, and <em>correctly</em> call <code>DESeq2</code> to calculate differential expression between numerous <em>pairs</em> of samples<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>
<ul>
<li>Use non-contiguous genes in <code>DESeq2</code> analysis, e.g. to exclude of specific sites/peaks from the analysis (not usually supported by DESeq2)</li>
<li>Efficiently generate results across a list of comparisons</li>
</ul>
</li>
<li>Support for blacklisting throughout, and proper accounting of blacklisted sites in relevant calculations</li>
<li>Users interact with an intuitive and computationally efficient data structure (the “basepair resolution <code>GRanges</code>” object), which is already supported by a rich, user-friendly suite of tools that greatly simplify working with datasets and annotations</li>
</ul>
<div id="coming-soon" class="section level3">
<h3 class="hasAnchor">
<a href="#coming-soon" class="anchor"></a>Coming Soon</h3>
<p>Data processing:</p>
<ul>
<li>Summarizing and plotting replicate correlations</li>
<li>Function to use random read sampling to assess if sequencing depth sufficient to stabilize arbitrary calculations (so a user can supply anonymous function to calculate things like RMP-normalized gene expression, or DESeq2 differential expression, pausing indices, etc.)</li>
</ul>
<p>Signal counting and analysis:</p>
<ul>
<li>Two-stranded meta-profile calculations</li>
<li>Automated generation of a list of DESeq2 comparisons using all possible combinations; all possible permutations; or by defining a simple hierarchy of each-vs-one comparisons</li>
</ul>
</div>
</div>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<p>Install development version from <a href="https://github.com/mdeber/BRGenomics">GitHub</a>:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="co"># install.packages("devtools")</span></a>
<a class="sourceLine" id="cb1-2" data-line-number="2">devtools<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/devtools/man/remote-reexports.html">install_github</a></span>(<span class="st">"mdeber/BRGenomics"</span>)</a></code></pre></div>
<p>If you’re using Windows, <a href="https://cran.rstudio.com/bin/windows/Rtools/">Rtools for Windows</a> is required.</p>
</div>
<div id="getting-started" class="section level1">
<h1 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>Getting started</h1>
<p>The package ships with an example dataset of PRO-seq data from Drosophila melanogaster<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>. PRO-seq is a basepair-resolution method that uses 3’-end sequencing of nascent RNA to map the locations of actively engaged RNA polymerases. To keep the dataset small, we’ve only included reads mapping to the fourth chromosome<a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a>. The data is included in several formats, all of which can be derived from the included bam file.</p>
</div>
<div id="working-with-bam-files" class="section level1">
<h1 class="hasAnchor">
<a href="#working-with-bam-files" class="anchor"></a>Working With Bam Files</h1>
<p>The <code>import_bam</code> function provides a number of options for filtering and processing bam files. Because PRO-seq data is sequenced in the 3’-to-5’ direction of the original RNA molecule, we’ll use the <code>revcomp</code> option to reverse-complement all the input reads. We’ll also set a minimum MAPQ score of 20:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(BRGenomics)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2"></a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="co"># path of included bam file</span></a>
<a class="sourceLine" id="cb2-4" data-line-number="4">bfile &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"PROseq_dm6_chr4.bam"</span>, </a>
<a class="sourceLine" id="cb2-5" data-line-number="5">                     <span class="dt">package =</span> <span class="st">"BRGenomics"</span>)</a>
<a class="sourceLine" id="cb2-6" data-line-number="6"></a>
<a class="sourceLine" id="cb2-7" data-line-number="7">ps_reads &lt;-<span class="st"> </span><span class="kw"><a href="../reference/import_bam.html">import_bam</a></span>(bfile, <span class="dt">mapq =</span> <span class="dv">20</span>, <span class="dt">revcomp =</span> <span class="ot">TRUE</span>, <span class="dt">paired_end =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb2-8" data-line-number="8">ps_reads</a></code></pre></div>
<pre><code>## GRanges object with 53179 ranges and 1 metadata column:
##           seqnames          ranges strand |     score
##              &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
##       [1]     chr4       1270-1296      + |         1
##       [2]     chr4     41411-41429      + |         1
##       [3]     chr4     42556-42591      + |         1
##       [4]     chr4     42559-42589      + |         1
##       [5]     chr4     42559-42594      + |         3
##       ...      ...             ...    ... .       ...
##   [53175]     chr4 1307741-1307776      - |         1
##   [53176]     chr4 1316536-1316563      - |         1
##   [53177]     chr4 1318959-1318994      - |         1
##   [53178]     chr4 1319003-1319038      - |         1
##   [53179]     chr4 1319368-1319403      - |         1
##   -------
##   seqinfo: 1870 sequences from an unspecified genome</code></pre>
<p>Files are imported as <code>GRanges</code> objects from the <em><a href="https://bioconductor.org/packages/3.11/GenomicRanges">GenomicRanges</a></em> package. GRanges objects are very easy to work with, and are supported by a plethora of useful functions and packages.</p>
<hr>
<p><em>To learn more about GRanges objects, including a general overview of their components, see the useful vignette <a href="https://bioconductor.org/packages/release/bioc/vignettes/GenomicRanges/inst/doc/GenomicRangesIntroduction.html"><em>An Introduction to the GenomicRanges Package</em></a>. Alternatively, see the archived materials from the 2018 Bioconductor workshop <a href="https://bioconductor.github.io/BiocWorkshops/solving-common-bioinformatic-challenges-using-genomicranges.html"><em>Solving Common Bioinformatic Challenges Using GenomicRanges</em></a>. Note that this package will implement and streamline a number of common operations, but users should still have a basic familiarity with GRanges objects.</em></p>
<hr>
<p>By default, <code>import_bam</code> combines identical reads into the same range, and uses the <code>score</code> metadata column to represent how many times that read occurs. This means that the total number of reads in the dataset is equal to the total score:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="kw">score</span>(ps_reads))</a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="co">## [1] 73887</span></a></code></pre></div>
<p>Alternatively, you can import each read as its own range by setting <code>field = NULL</code>:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">reads_expanded &lt;-<span class="st"> </span><span class="kw"><a href="../reference/import_bam.html">import_bam</a></span>(bfile, <span class="dt">mapq =</span> <span class="dv">20</span>, <span class="dt">revcomp =</span> <span class="ot">TRUE</span>, </a>
<a class="sourceLine" id="cb5-2" data-line-number="2">                             <span class="dt">field =</span> <span class="ot">NULL</span>, <span class="dt">paired_end =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb5-3" data-line-number="3">ps_reads[<span class="dv">1</span><span class="op">:</span><span class="dv">8</span>]</a></code></pre></div>
<pre><code>## GRanges object with 8 ranges and 1 metadata column:
##       seqnames      ranges strand |     score
##          &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
##   [1]     chr4   1270-1296      + |         1
##   [2]     chr4 41411-41429      + |         1
##   [3]     chr4 42556-42591      + |         1
##   [4]     chr4 42559-42589      + |         1
##   [5]     chr4 42559-42594      + |         3
##   [6]     chr4 42560-42594      + |         2
##   [7]     chr4 42560-42595      + |         2
##   [8]     chr4 42561-42596      + |         1
##   -------
##   seqinfo: 1870 sequences from an unspecified genome</code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">reads_expanded[<span class="dv">1</span><span class="op">:</span><span class="dv">8</span>]</a></code></pre></div>
<pre><code>## GRanges object with 8 ranges and 0 metadata columns:
##       seqnames      ranges strand
##          &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt;
##   [1]     chr4   1270-1296      +
##   [2]     chr4 41411-41429      +
##   [3]     chr4 42556-42591      +
##   [4]     chr4 42559-42589      +
##   [5]     chr4 42559-42594      +
##   [6]     chr4 42559-42594      +
##   [7]     chr4 42559-42594      +
##   [8]     chr4 42560-42594      +
##   -------
##   seqinfo: 1870 sequences from an unspecified genome</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(reads_expanded) <span class="op">==</span><span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="kw">score</span>(ps_reads))</a></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
</div>
<div id="basepair-resolution-granges-objects" class="section level1">
<h1 class="hasAnchor">
<a href="#basepair-resolution-granges-objects" class="anchor"></a>Basepair-Resolution GRanges Objects</h1>
<p>The base immediately before the 3’ end of a mapped PRO-seq read gives the position of an RNA polymerase at basepair-resolution. The <code>import_bam</code> function also includes options for selecting specific bases of each read, which are applied after reverse-complementing. Here, we reverse complement, take the 3’ end of the read, and then shift that base upstream by 1 base:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">ps &lt;-<span class="st"> </span><span class="kw"><a href="../reference/import_bam.html">import_bam</a></span>(bfile, </a>
<a class="sourceLine" id="cb11-2" data-line-number="2">                 <span class="dt">mapq =</span> <span class="dv">20</span>, </a>
<a class="sourceLine" id="cb11-3" data-line-number="3">                 <span class="dt">revcomp =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb11-4" data-line-number="4">                 <span class="dt">shift =</span> <span class="dv">-1</span>,</a>
<a class="sourceLine" id="cb11-5" data-line-number="5">                 <span class="dt">trim.to =</span> <span class="st">"3p"</span>,</a>
<a class="sourceLine" id="cb11-6" data-line-number="6">                 <span class="dt">paired_end =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb11-7" data-line-number="7">ps</a></code></pre></div>
<pre><code>## GRanges object with 47380 ranges and 1 metadata column:
##           seqnames    ranges strand |     score
##              &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
##       [1]     chr4      1295      + |         1
##       [2]     chr4     41428      + |         1
##       [3]     chr4     42588      + |         1
##       [4]     chr4     42590      + |         2
##       [5]     chr4     42593      + |         5
##       ...      ...       ...    ... .       ...
##   [47376]     chr4   1307742      - |         1
##   [47377]     chr4   1316537      - |         1
##   [47378]     chr4   1318960      - |         1
##   [47379]     chr4   1319004      - |         1
##   [47380]     chr4   1319369      - |         1
##   -------
##   seqinfo: 1870 sequences from an unspecified genome</code></pre>
<p>You may have noticed that this GRanges object isn’t the same length as when we imported the entire reads. That’s because after trimming, identical positions were collapsed together, and the <code>score</code> column contains the readcounts at each position.</p>
<p>The returned data is a GRanges object in which all ranges represent a single base, and in which a metadata field (usually <code>score</code>) contains readcounts. This data structure underpins much of this package, and we refer to it as a <strong>basepair-resolution GRanges object</strong><a href="#fn4" class="footnote-ref" id="fnref4"><sup>4</sup></a>.</p>
<hr>
<p><em>bigWig files, as well as <code>coverage</code> objects in <code>GenomicRanges</code>, use a form of compression in which adjacent positions sharing the same signal are combined. For whole-read coverage methods, this run-length encoding enhances performance and efficiency. However, basepair resolution data is not nearly as smooth, and its rare that adjacent bases share the same signal. With basepair resolution data, combining adjacent positions increases complexity, while providing a negligible storage benefit.</em></p>
<hr>
<p>Splitting up each base into its own range (its own index) results in a number of downstream computational benefits. Perhaps more importantly, however, is the added simplicity for the user: each index of a basepair-resolution GRanges object addresses a single genomic position and its associated signal.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(ps) <span class="co"># number of unique positions in dataset</span></a>
<a class="sourceLine" id="cb13-2" data-line-number="2"><span class="co">## [1] 47380</span></a>
<a class="sourceLine" id="cb13-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="kw">score</span>(ps)) <span class="co"># total number of reads</span></a>
<a class="sourceLine" id="cb13-4" data-line-number="4"><span class="co">## [1] 73887</span></a>
<a class="sourceLine" id="cb13-5" data-line-number="5"><span class="kw"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="kw">score</span>(ps) <span class="op">==</span><span class="st"> </span><span class="dv">1</span>) <span class="co"># number of sites with only a single read</span></a>
<a class="sourceLine" id="cb13-6" data-line-number="6"><span class="co">## [1] 37559</span></a></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/subset.html">subset</a></span>(ps, score <span class="op">&gt;</span><span class="st"> </span><span class="dv">10</span>) <span class="co"># sites with more than 10 reads</span></a></code></pre></div>
<pre><code>## GRanges object with 309 ranges and 1 metadata column:
##         seqnames    ranges strand |     score
##            &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
##     [1]     chr4     69377      + |        12
##     [2]     chr4     69391      + |        12
##     [3]     chr4     69397      + |        25
##     [4]     chr4     69401      + |       122
##     [5]     chr4     69692      + |        17
##     ...      ...       ...    ... .       ...
##   [305]     chr4   1176185      - |        13
##   [306]     chr4   1182428      - |        52
##   [307]     chr4   1196793      - |        54
##   [308]     chr4   1196794      - |        14
##   [309]     chr4   1226558      - |        13
##   -------
##   seqinfo: 1870 sequences from an unspecified genome</code></pre>
<p>We can write a simple test to determine if a GRanges object is “basepair-resolution”:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/all.html">all</a></span>(<span class="kw">width</span>(ps) <span class="op">==</span><span class="st"> </span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb16-2" data-line-number="2"><span class="co">## [1] TRUE</span></a></code></pre></div>
</div>
<div id="example-pipeline-for-generating-pro-seq-browser-tracks" class="section level1">
<h1 class="hasAnchor">
<a href="#example-pipeline-for-generating-pro-seq-browser-tracks" class="anchor"></a>Example: Pipeline for Generating PRO-seq Browser Tracks</h1>
<p>The <code>import_bam</code> function provides several options for processing aligned reads, and in conjunction with functions from the <em><a href="https://bioconductor.org/packages/3.11/rtracklayer">rtracklayer</a></em> package, we can easily write a post-alignment pipeline for generating bigWig files for viewing in a genome browser.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1">ps &lt;-<span class="st"> </span><span class="kw"><a href="../reference/import_bam.html">import_bam</a></span>(bfile, </a>
<a class="sourceLine" id="cb17-2" data-line-number="2">                 <span class="dt">mapq =</span> <span class="dv">20</span>, </a>
<a class="sourceLine" id="cb17-3" data-line-number="3">                 <span class="dt">revcomp =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb17-4" data-line-number="4">                 <span class="dt">shift =</span> <span class="dv">-1</span>,</a>
<a class="sourceLine" id="cb17-5" data-line-number="5">                 <span class="dt">trim.to =</span> <span class="st">"3p"</span>,</a>
<a class="sourceLine" id="cb17-6" data-line-number="6">                 <span class="dt">paired_end =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb17-7" data-line-number="7"></a>
<a class="sourceLine" id="cb17-8" data-line-number="8">ps_plus &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/subset.html">subset</a></span>(ps, strand <span class="op">==</span><span class="st"> "+"</span>)</a>
<a class="sourceLine" id="cb17-9" data-line-number="9">ps_minus &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/subset.html">subset</a></span>(ps, strand <span class="op">==</span><span class="st"> "-"</span>)</a>
<a class="sourceLine" id="cb17-10" data-line-number="10"><span class="kw">score</span>(ps_minus) &lt;-<span class="st"> </span><span class="op">-</span><span class="kw">score</span>(ps_minus) <span class="co"># make minus-strand scores negative</span></a>
<a class="sourceLine" id="cb17-11" data-line-number="11"></a>
<a class="sourceLine" id="cb17-12" data-line-number="12">rtracklayer<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/rtracklayer/man/BigWigFile.html">export.bw</a></span>(ps_plus, <span class="st">"~/Data/PROseq_plus.bw"</span>)</a>
<a class="sourceLine" id="cb17-13" data-line-number="13">rtracklayer<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/rtracklayer/man/BigWigFile.html">export.bw</a></span>(ps_minus, <span class="st">"~/Data/PROseq_minus.bw"</span>)</a></code></pre></div>
</div>
<div id="working-with-annotations" class="section level1">
<h1 class="hasAnchor">
<a href="#working-with-annotations" class="anchor"></a>Working With Annotations</h1>
</div>
<div id="efficient-analysis-of-multiple-datasets" class="section level1">
<h1 class="hasAnchor">
<a href="#efficient-analysis-of-multiple-datasets" class="anchor"></a>Efficient Analysis of Multiple Datasets</h1>
</div>
<div id="saving-binary-r-files" class="section level1">
<h1 class="hasAnchor">
<a href="#saving-binary-r-files" class="anchor"></a>Saving Binary R Files</h1>
</div>
<div id="deseq" class="section level1">
<h1 class="hasAnchor">
<a href="#deseq" class="anchor"></a>DESeq</h1>
</div>
<div id="combining-replicates" class="section level1">
<h1 class="hasAnchor">
<a href="#combining-replicates" class="anchor"></a>Combining Replicates</h1>
</div>
<div id="heatmaps" class="section level1">
<h1 class="hasAnchor">
<a href="#heatmaps" class="anchor"></a>Heatmaps</h1>
<!-- (Use image, and use ComplexHeatmap; and show reshape2::melt to use ggplot or lattice) -->
</div>
<div id="metaplots" class="section level1">
<h1 class="hasAnchor">
<a href="#metaplots" class="anchor"></a>Metaplots</h1>
<!-- (Use R plotting, and ggplot) -->
</div>
<div id="browser-shots" class="section level1">
<h1 class="hasAnchor">
<a href="#browser-shots" class="anchor"></a>“Browser Shots”</h1>
</div>
<div id="sequence-extraction" class="section level1">
<h1 class="hasAnchor">
<a href="#sequence-extraction" class="anchor"></a>Sequence Extraction</h1>
</div>
<div id="session-info" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<pre><code>## ─ Session info ───────────────────────────────────────────────────────────────
##  setting  value                                             
##  version  R Under development (unstable) (2020-01-10 r77651)
##  os       macOS Catalina 10.15.2                            
##  system   x86_64, darwin15.6.0                              
##  ui       X11                                               
##  language (EN)                                              
##  collate  en_US.UTF-8                                       
##  ctype    en_US.UTF-8                                       
##  tz       America/New_York                                  
##  date     2020-02-15                                        
## 
## ─ Packages ───────────────────────────────────────────────────────────────────
##  package              * version   date       lib source        
##  annotate               1.65.1    2020-01-27 [3] Bioconductor  
##  AnnotationDbi          1.49.1    2020-01-25 [3] Bioconductor  
##  askpass                1.1       2019-01-13 [3] CRAN (R 4.0.0)
##  assertthat             0.2.1     2019-03-21 [3] CRAN (R 4.0.0)
##  backports              1.1.5     2019-10-02 [3] CRAN (R 4.0.0)
##  Biobase                2.47.2    2019-12-19 [3] Bioconductor  
##  BiocFileCache          1.11.4    2019-12-02 [3] Bioconductor  
##  BiocGenerics         * 0.33.0    2019-11-06 [3] Bioconductor  
##  BiocManager            1.30.10   2019-11-16 [3] CRAN (R 4.0.0)
##  BiocParallel           1.21.2    2019-12-21 [3] Bioconductor  
##  BiocStyle            * 2.15.6    2020-02-01 [3] Bioconductor  
##  biomaRt                2.43.3    2020-02-08 [3] Bioconductor  
##  Biostrings             2.55.4    2019-12-14 [3] Bioconductor  
##  bit                    1.1-14    2018-05-29 [3] CRAN (R 4.0.0)
##  bit64                  0.9-7     2017-05-08 [3] CRAN (R 4.0.0)
##  bitops                 1.0-6     2013-08-17 [3] CRAN (R 4.0.0)
##  blob                   1.2.0     2019-07-09 [3] CRAN (R 4.0.0)
##  bookdown               0.17      2020-01-11 [3] CRAN (R 4.0.0)
##  BRGenomics           * 0.7.5     2020-02-15 [1] Bioconductor  
##  BSgenome               1.55.2    2019-12-13 [3] Bioconductor  
##  callr                  3.4.0     2019-12-09 [3] CRAN (R 4.0.0)
##  cli                    2.0.1     2020-01-08 [3] CRAN (R 4.0.0)
##  colorspace             1.4-1     2019-03-18 [3] CRAN (R 4.0.0)
##  crayon                 1.3.4     2017-09-16 [3] CRAN (R 4.0.0)
##  curl                   4.3       2019-12-02 [3] CRAN (R 4.0.0)
##  DBI                    1.1.0     2019-12-15 [3] CRAN (R 4.0.0)
##  dbplyr                 1.4.2     2019-06-17 [3] CRAN (R 4.0.0)
##  DelayedArray           0.13.2    2020-01-06 [3] Bioconductor  
##  desc                   1.2.0     2018-05-01 [3] CRAN (R 4.0.0)
##  DESeq2                 1.27.19   2020-01-07 [3] Bioconductor  
##  devtools               2.2.1     2019-09-24 [3] CRAN (R 4.0.0)
##  digest                 0.6.23    2019-11-23 [3] CRAN (R 4.0.0)
##  dplyr                  0.8.3     2019-07-04 [3] CRAN (R 4.0.0)
##  ellipsis               0.3.0     2019-09-20 [3] CRAN (R 4.0.0)
##  evaluate               0.14      2019-05-28 [3] CRAN (R 4.0.0)
##  fansi                  0.4.1     2020-01-08 [3] CRAN (R 4.0.0)
##  fs                     1.3.1     2019-05-06 [3] CRAN (R 4.0.0)
##  genefilter             1.69.0    2019-11-06 [3] Bioconductor  
##  geneplotter            1.65.0    2019-11-06 [3] Bioconductor  
##  GenomeInfoDb         * 1.23.3    2020-01-11 [3] Bioconductor  
##  GenomeInfoDbData       1.2.2     2020-01-12 [3] Bioconductor  
##  GenomicAlignments      1.23.1    2019-11-12 [3] Bioconductor  
##  GenomicFeatures        1.39.2    2019-12-20 [3] Bioconductor  
##  GenomicFiles           1.23.0    2019-11-06 [3] Bioconductor  
##  GenomicRanges        * 1.39.1    2019-11-07 [3] Bioconductor  
##  ggplot2                3.2.1     2019-08-10 [3] CRAN (R 4.0.0)
##  glue                   1.3.1     2019-03-12 [3] CRAN (R 4.0.0)
##  gtable                 0.3.0     2019-03-25 [3] CRAN (R 4.0.0)
##  hms                    0.5.3     2020-01-08 [3] CRAN (R 4.0.0)
##  htmltools              0.4.0     2019-10-04 [3] CRAN (R 4.0.0)
##  httr                   1.4.1     2019-08-05 [3] CRAN (R 4.0.0)
##  IRanges              * 2.21.3    2020-01-13 [3] Bioconductor  
##  knitr                  1.26      2019-11-12 [3] CRAN (R 4.0.0)
##  lattice                0.20-38   2018-11-04 [3] CRAN (R 4.0.0)
##  lazyeval               0.2.2     2019-03-15 [3] CRAN (R 4.0.0)
##  lifecycle              0.1.0     2019-08-01 [3] CRAN (R 4.0.0)
##  locfit                 1.5-9.1   2013-04-20 [3] CRAN (R 4.0.0)
##  magrittr               1.5       2014-11-22 [3] CRAN (R 4.0.0)
##  MASS                   7.3-51.5  2019-12-20 [3] CRAN (R 4.0.0)
##  Matrix                 1.2-18    2019-11-27 [3] CRAN (R 4.0.0)
##  matrixStats            0.55.0    2019-09-07 [3] CRAN (R 4.0.0)
##  memoise                1.1.0     2017-04-21 [3] CRAN (R 4.0.0)
##  munsell                0.5.0     2018-06-12 [3] CRAN (R 4.0.0)
##  openssl                1.4.1     2019-07-18 [3] CRAN (R 4.0.0)
##  pillar                 1.4.3     2019-12-20 [3] CRAN (R 4.0.0)
##  pkgbuild               1.0.6     2019-10-09 [3] CRAN (R 4.0.0)
##  pkgconfig              2.0.3     2019-09-22 [3] CRAN (R 4.0.0)
##  pkgdown                1.4.1     2019-09-15 [3] CRAN (R 4.0.0)
##  pkgload                1.0.2     2018-10-29 [3] CRAN (R 4.0.0)
##  prettyunits            1.1.0     2020-01-09 [3] CRAN (R 4.0.0)
##  processx               3.4.1     2019-07-18 [3] CRAN (R 4.0.0)
##  progress               1.2.2     2019-05-16 [3] CRAN (R 4.0.0)
##  ps                     1.3.0     2018-12-21 [3] CRAN (R 4.0.0)
##  purrr                  0.3.3     2019-10-18 [3] CRAN (R 4.0.0)
##  R6                     2.4.1     2019-11-12 [3] CRAN (R 4.0.0)
##  rappdirs               0.3.1     2016-03-28 [3] CRAN (R 4.0.0)
##  RColorBrewer           1.1-2     2014-12-07 [3] CRAN (R 4.0.0)
##  Rcpp                   1.0.3     2019-11-08 [3] CRAN (R 4.0.0)
##  RCurl                  1.95-4.12 2019-03-04 [3] CRAN (R 4.0.0)
##  remotes                2.1.0     2019-06-24 [3] CRAN (R 4.0.0)
##  rlang                  0.4.2     2019-11-23 [3] CRAN (R 4.0.0)
##  rmarkdown              2.0       2019-12-12 [3] CRAN (R 4.0.0)
##  rprojroot              1.3-2     2018-01-03 [3] CRAN (R 4.0.0)
##  Rsamtools              2.3.2     2019-11-07 [3] Bioconductor  
##  RSQLite                2.2.0     2020-01-07 [3] CRAN (R 4.0.0)
##  rstudioapi             0.10      2019-03-19 [3] CRAN (R 4.0.0)
##  rtracklayer            1.47.0    2019-11-06 [3] Bioconductor  
##  S4Vectors            * 0.25.10   2020-01-13 [3] Bioconductor  
##  scales                 1.1.0     2019-11-18 [3] CRAN (R 4.0.0)
##  sessioninfo            1.1.1     2018-11-05 [3] CRAN (R 4.0.0)
##  stringi                1.4.5     2020-01-11 [3] CRAN (R 4.0.0)
##  stringr                1.4.0     2019-02-10 [3] CRAN (R 4.0.0)
##  SummarizedExperiment   1.17.1    2019-12-19 [3] Bioconductor  
##  survival               3.1-8     2019-12-03 [3] CRAN (R 4.0.0)
##  testthat               2.3.1     2019-12-01 [3] CRAN (R 4.0.0)
##  tibble                 2.1.3     2019-06-06 [3] CRAN (R 4.0.0)
##  tidyselect             0.2.5     2018-10-11 [3] CRAN (R 4.0.0)
##  usethis                1.5.1     2019-07-04 [3] CRAN (R 4.0.0)
##  VariantAnnotation      1.33.0    2019-11-06 [3] Bioconductor  
##  vctrs                  0.2.1     2019-12-17 [3] CRAN (R 4.0.0)
##  withr                  2.1.2     2018-03-15 [3] CRAN (R 4.0.0)
##  xfun                   0.11      2019-11-12 [3] CRAN (R 4.0.0)
##  XML                    3.98-1.20 2019-06-06 [3] CRAN (R 4.0.0)
##  xtable                 1.8-4     2019-04-21 [3] CRAN (R 4.0.0)
##  XVector                0.27.0    2019-11-06 [3] Bioconductor  
##  yaml                   2.2.0     2018-07-25 [3] CRAN (R 4.0.0)
##  zeallot                0.1.0     2018-01-28 [3] CRAN (R 4.0.0)
##  zlibbioc               1.33.0    2019-11-06 [3] Bioconductor  
## 
## [1] /private/var/folders/zt/m504_3g569g6gqzjbqg34qn40000gn/T/Rtmp123Vub/temp_libpathfb2f617e006f
## [2] /private/var/folders/zt/m504_3g569g6gqzjbqg34qn40000gn/T/Rtmp123Vub/temp_libpathfb2f152237b5
## [3] /Library/Frameworks/R.framework/Versions/4.0/Resources/library</code></pre>
<!-- SCRAP -->
<!-- The `import_bam` function provides several options for processing aligned reads, and in conjunction with functions from the *[rtracklayer](https://bioconductor.org/packages/3.11/rtracklayer)* package, we can easily write a post-alignment pipeline for generating bigWig files for viewing in a genome browser. -->
<!-- As an exmaple, PRO-seq data is sequenced in the antisense orientation to the original RNA molecule, the 3' end of which provides the location of the active site of an engaged RNA polymerase. Therefore, PRO-seq reads need to be reverse-complemented, and the 3' ends need to be extracted. -->
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>By avoiding the default behavior of calculating genewise dispersion across all samples present, which is invalid if any experimental condition causes broad changes<a href="#fnref1" class="footnote-back">↩</a></p></li>
<li id="fn2"><p>Hojoong Kwak, Nicholas J. Fuda, Leighton J. Core, John T. Lis (2013). Precise Maps of RNA Polymerase Reveal How Promoters Direct Initiation and Pausing. <em>Science</em> <strong>339</strong>(6122): 950–953. <a href="https://doi.org/10.1126/science.1229386" class="uri">https://doi.org/10.1126/science.1229386</a><a href="#fnref2" class="footnote-back">↩</a></p></li>
<li id="fn3"><p>Chromosome 4 in Drosophila, often referred to as the “dot” chromosome, is very small and contains very few genes<a href="#fnref3" class="footnote-back">↩</a></p></li>
<li id="fn4"><p>The <code>GPos</code> class is designed for a similar purpose, but we currently do not use it. There is a note on this in the documentation for <code>makeGRangesBRG</code><a href="#fnref4" class="footnote-back">↩</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#motivation">Motivation</a></li>
      <li><a href="#features">Features</a></li>
      <li><a href="#installation">Installation</a></li>
      <li><a href="#getting-started">Getting started</a></li>
      <li><a href="#working-with-bam-files">Working With Bam Files</a></li>
      <li><a href="#basepair-resolution-granges-objects">Basepair-Resolution GRanges Objects</a></li>
      <li><a href="#example-pipeline-for-generating-pro-seq-browser-tracks">Example: Pipeline for Generating PRO-seq Browser Tracks</a></li>
      <li><a href="#working-with-annotations">Working With Annotations</a></li>
      <li><a href="#efficient-analysis-of-multiple-datasets">Efficient Analysis of Multiple Datasets</a></li>
      <li><a href="#saving-binary-r-files">Saving Binary R Files</a></li>
      <li><a href="#deseq">DESeq</a></li>
      <li><a href="#combining-replicates">Combining Replicates</a></li>
      <li><a href="#heatmaps">Heatmaps</a></li>
      <li><a href="#metaplots">Metaplots</a></li>
      <li><a href="#browser-shots">“Browser Shots”</a></li>
      <li><a href="#sequence-extraction">Sequence Extraction</a></li>
      <li><a href="#session-info">Session info</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
