<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>DESeq2 with Global Perturbations • BRGenomics</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="DESeq2 with Global Perturbations">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Overview.html">Overview</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Guides
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/GettingStarted.html">Getting Started</a>
    </li>
    <li>
      <a href="../articles/ImportingModifyingAnnotations.html">Importing &amp; Modifying Annotations</a>
    </li>
    <li>
      <a href="../articles/ImportingProcessingData.html">Importing &amp; Processing Data</a>
    </li>
    <li>
      <a href="../articles/SignalCounting.html">Signal Counting</a>
    </li>
    <li>
      <a href="../articles/AnalyzingMultipleDatasets.html">Analyzing Multiple Datasets</a>
    </li>
    <li>
      <a href="../articles/ProfilePlotsAndBootstrapping.html">Profile Plots &amp; Bootstrapping</a>
    </li>
    <li>
      <a href="../articles/SpikeInNormalization.html">Spike-in Normalization</a>
    </li>
    <li>
      <a href="../articles/DESeq2WithGlobalPerturbations.html">DESeq2 with Global Perturbations</a>
    </li>
    <li>
      <a href="../articles/SequenceExtraction.html">Sequence Extraction</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mdeber/BRGenomics">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>DESeq2 with Global Perturbations</h1>
            
      
      
      <div class="hidden name"><code>DESeq2WithGlobalPerturbations.Rmd</code></div>

    </div>

    
    
<div id="using-deseq2-for-pairwise-differential-expression" class="section level1">
<h1 class="hasAnchor">
<a href="#using-deseq2-for-pairwise-differential-expression" class="anchor"></a>Using DESeq2 for Pairwise Differential Expression</h1>
<div id="rationale" class="section level2">
<h2 class="hasAnchor">
<a href="#rationale" class="anchor"></a>Rationale</h2>
<p>DESeq2’s default treatment of data relies on the assumption that genewise estimates of dispersion are largely unchanged across samples. While this assumption holds for a typical RNA-seq data, it can be violated if there are samples within the <code>DESeqDataSet</code> object for which there are meaningful signal changes across a majority of regions of interest.</p>
<p>The BRGenomics functions <code><a href="../reference/getDESeqDataSet.html">getDESeqDataSet()</a></code> and <code><a href="../reference/getDESeqResults.html">getDESeqResults()</a></code> are simple and flexible wrappers for making pairwise comparisons between individual samples, without relying on the assumption of globally-similar dispersion estimates. In particular, <code><a href="../reference/getDESeqResults.html">getDESeqResults()</a></code> follows the logic that the presence of a dataset <span class="math inline">\(X\)</span> within the <code>DESeqDataSet</code> object will not affect the comparison of datasets <span class="math inline">\(Y\)</span> and <span class="math inline">\(Z\)</span>.</p>
<p>While the intuition above is appealing, users should note that if the globally-similar dispersions assumption <em>does</em> hold, then DESeq2’s default behavior should be more sensitive in its estimates of genewise dispersion. In this case, users can still take advantage of the convenience of the BRGenomics function <code><a href="../reference/getDESeqDataSet.html">getDESeqDataSet()</a></code>, but they should subsequently call <code><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html">DESeq2::DESeq()</a></code> and <code><a href="https://rdrr.io/pkg/DESeq2/man/results.html">DESeq2::results()</a></code> directly.</p>
<p>If the globally-similar dispersions assumption is violated, but something beyond simple pairwise comparisons is desired (e.g. group comparisons or additional model terms), we note that, with some prying, DESeq2 can be used without “blind dispersion estimation” (see the DESeq2 manual).</p>
</div>
<div id="formatting-data-for-deseq2" class="section level2">
<h2 class="hasAnchor">
<a href="#formatting-data-for-deseq2" class="anchor"></a>Formatting Data for DESeq2</h2>
<p>Just like the functions that generate batch-normalized spike-in normalization factors, the DESeq-oriented functions require that the names of the input datasets end in <code>"rep1"</code>, <code>"rep2"</code>, etc.</p>
<p>Let’s first make a toy list of multiple datasets to compare:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(BRGenomics)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"PROseq"</span>)</a></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">ps_list &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="dv">1</span><span class="op">:</span><span class="dv">6</span>, <span class="cf">function</span>(i) PROseq[<span class="kw"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(i, <span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(PROseq), <span class="dv">6</span>)])</a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(ps_list) &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"A_rep1"</span>, <span class="st">"A_rep2"</span>, </a>
<a class="sourceLine" id="cb2-3" data-line-number="3">                    <span class="st">"B_rep1"</span>, <span class="st">"B_rep2"</span>,</a>
<a class="sourceLine" id="cb2-4" data-line-number="4">                    <span class="st">"C_rep1"</span>, <span class="st">"C_rep2"</span>)</a></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">ps_list[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>]</a></code></pre></div>
<pre><code>## $A_rep1
## GRanges object with 7897 ranges and 1 metadata column:
##          seqnames    ranges strand |     score
##             &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
##      [1]     chr4      1295      + |         1
##      [2]     chr4     42595      + |         1
##      [3]     chr4     42622      + |         2
##      [4]     chr4     42718      + |         1
##      [5]     chr4     42789      + |         1
##      ...      ...       ...    ... .       ...
##   [7893]     chr4   1295277      - |         1
##   [7894]     chr4   1306500      - |         1
##   [7895]     chr4   1306889      - |         1
##   [7896]     chr4   1307122      - |         1
##   [7897]     chr4   1316537      - |         1
##   -------
##   seqinfo: 7 sequences from an unspecified genome
## 
## $A_rep2
## GRanges object with 7897 ranges and 1 metadata column:
##          seqnames    ranges strand |     score
##             &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
##      [1]     chr4     41428      + |         1
##      [2]     chr4     42596      + |         1
##      [3]     chr4     42652      + |         3
##      [4]     chr4     42733      + |         1
##      [5]     chr4     42794      + |         2
##      ...      ...       ...    ... .       ...
##   [7893]     chr4   1305972      - |         1
##   [7894]     chr4   1306514      - |         1
##   [7895]     chr4   1307032      - |         1
##   [7896]     chr4   1307126      - |         1
##   [7897]     chr4   1318960      - |         1
##   -------
##   seqinfo: 7 sequences from an unspecified genome</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(ps_list)</a></code></pre></div>
<pre><code>## [1] "A_rep1" "A_rep2" "B_rep1" "B_rep2" "C_rep1" "C_rep2"</code></pre>
<p>As you can see, the names all end in “repX”, where X indicates the replicate. Replicates will be grouped by anything that follows “rep”. If the sample names do not conform to this standard, the <code>sample_names</code> argument can be used to rename the samples within the call to <code><a href="../reference/getDESeqDataSet.html">getDESeqDataSet()</a></code>.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"txs_dm6_chr4"</span>)</a></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">dds &lt;-<span class="st"> </span><span class="kw"><a href="../reference/getDESeqDataSet.html">getDESeqDataSet</a></span>(ps_list, txs_dm6_chr4,</a>
<a class="sourceLine" id="cb8-2" data-line-number="2">                       <span class="dt">gene_names =</span> txs_dm6_chr4<span class="op">$</span>gene_id,</a>
<a class="sourceLine" id="cb8-3" data-line-number="3">                       <span class="dt">ncores =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb8-4" data-line-number="4">dds</a></code></pre></div>
<pre><code>## class: DESeqDataSet 
## dim: 111 6 
## metadata(1): version
## assays(1): counts
## rownames(111): FBgn0267363 FBgn0266617 ... FBgn0039924 FBgn0027101
## rowData names(2): tx_name gene_id
## colnames(6): A_rep1 A_rep2 ... C_rep1 C_rep2
## colData names(2): condition replicate</code></pre>
<p>Notice that the <code>dim</code> attribute of the <code>DESeqDataSet</code> object is <code><a href="https://rdrr.io/r/base/c.html">c(111, 6)</a></code>. There are 6 samples, but <code><a href="https://rdrr.io/r/base/length.html">length(txs_dm6_chr4)</a></code> is not 111. This is because we provided gene names to <code><a href="../reference/getDESeqDataSet.html">getDESeqDataSet()</a></code>, which were non-unique. The feature being exploited here is for use with <strong>discontinuous gene regions</strong>, <em>not for multiple overlapping transcript isoforms</em>.</p>
<hr>
<p><strong>By default, <code><a href="../reference/getDESeqDataSet.html">getDESeqDataSet()</a></code> will combine counts across all ranges belonging to a gene, but if they overlap, they will be counted twice. For addressing issues related to overlaps, see the <code><a href="../reference/intersectByGene.html">reduceByGene()</a></code> and <code><a href="../reference/intersectByGene.html">intersectByGene()</a></code> functions.</strong></p>
<hr>
<p>We could have added normalization factors, which DESeq2 calls “size factors”, in the call to <code><a href="../reference/getDESeqDataSet.html">getDESeqDataSet()</a></code>, or we can supply them to <code><a href="../reference/getDESeqResults.html">getDESeqResults()</a></code> below. (Supplying them twice will overwrite the previous size factors).</p>
<p><em>Important note on normalization factors:</em> DESeq2 “size factors” are the <em>inverse</em> of BRGenomics normalization factors. So if you calculate normalization factors with <code>NF &lt;- getSpikeInNFs(...)</code>, set <code>sizeFactors &lt;- 1/NF</code>.</p>
</div>
<div id="getting-deseq2-results" class="section level2">
<h2 class="hasAnchor">
<a href="#getting-deseq2-results" class="anchor"></a>Getting DESeq2 Results</h2>
<p>Generating DESeq2 results is simple:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="kw"><a href="../reference/getDESeqResults.html">getDESeqResults</a></span>(dds, <span class="dt">contrast.numer =</span> <span class="st">"B"</span>, <span class="dt">contrast.denom =</span> <span class="st">"A"</span>,</a>
<a class="sourceLine" id="cb10-2" data-line-number="2">                <span class="dt">quiet =</span> <span class="ot">TRUE</span>, <span class="dt">ncores =</span> <span class="dv">1</span>)</a></code></pre></div>
<pre><code>## log2 fold change (MLE): condition B vs A 
## Wald test p-value: condition B vs A 
## DataFrame with 111 rows and 6 columns
##                baseMean log2FoldChange     lfcSE       stat    pvalue      padj
##               &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
## FBgn0267363    0.252443     -1.4201507  4.997269 -0.2841853  0.776268  0.990374
## FBgn0266617    9.648855     -0.4357749  0.983164 -0.4432374  0.657594  0.990374
## FBgn0265633    2.291440      0.3467829  1.882149  0.1842484  0.853819  0.990374
## FBgn0264617   67.131764      0.0245325  0.451334  0.0543556  0.956652  0.990374
## FBgn0085432 4688.232636     -0.0469812  0.270431 -0.1737268  0.862080  0.990374
## ...                 ...            ...       ...        ...       ...       ...
## FBgn0039928     653.783    -0.00994936  0.262637 -0.0378825  0.969781  0.990374
## FBgn0052017     114.906     0.02350299  0.369082  0.0636796  0.949225  0.990374
## FBgn0002521     168.778     0.05777885  0.347249  0.1663903  0.867850  0.990374
## FBgn0039924     170.213    -0.20850730  0.409865 -0.5087219  0.610947  0.990374
## FBgn0027101     343.009    -0.21151750  0.290061 -0.7292177  0.465868  0.990374</code></pre>
<p>We can also make multiple pairwise-comparisons by ignoring the <code>contrast.numer</code> and <code>contrast.denom</code> arguments, and instead using the <code>comparisons</code> argument. The resulting list of results is named according to the comparisons:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">comparison_list &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"B"</span>, <span class="st">"A"</span>), </a>
<a class="sourceLine" id="cb12-2" data-line-number="2">                        <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"C"</span>, <span class="st">"A"</span>),</a>
<a class="sourceLine" id="cb12-3" data-line-number="3">                        <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"C"</span>, <span class="st">"B"</span>))</a>
<a class="sourceLine" id="cb12-4" data-line-number="4">dsres &lt;-<span class="st"> </span><span class="kw"><a href="../reference/getDESeqResults.html">getDESeqResults</a></span>(dds, <span class="dt">comparisons =</span> comparison_list,</a>
<a class="sourceLine" id="cb12-5" data-line-number="5">                         <span class="dt">quiet =</span> <span class="ot">TRUE</span>, <span class="dt">ncores =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb12-6" data-line-number="6"><span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(dsres)</a></code></pre></div>
<pre><code>## [1] "B_vs_A" "C_vs_A" "C_vs_B"</code></pre>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1">dsres<span class="op">$</span>C_vs_B</a></code></pre></div>
<pre><code>## log2 fold change (MLE): condition C vs B 
## Wald test p-value: condition C vs B 
## DataFrame with 111 rows and 6 columns
##               baseMean log2FoldChange     lfcSE       stat    pvalue      padj
##              &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
## FBgn0267363    0.00000             NA        NA         NA        NA        NA
## FBgn0266617    9.38822      0.3938871  0.901764   0.436796  0.662259  0.998692
## FBgn0265633    2.28968     -0.3206785  1.718554  -0.186598  0.851976  0.998692
## FBgn0264617   65.30695     -0.0773192  0.415908  -0.185905  0.852520  0.998692
## FBgn0085432 4281.29059     -0.1902162  0.229704  -0.828094  0.407617  0.998692
## ...                ...            ...       ...        ...       ...       ...
## FBgn0039928    696.482      0.2150507  0.248408  0.8657159  0.386646  0.998692
## FBgn0052017    133.743      0.4160583  0.333948  1.2458762  0.212810  0.998692
## FBgn0002521    171.141      0.0159143  0.323995  0.0491188  0.960825  0.998692
## FBgn0039924    138.749     -0.3685073  0.327200 -1.1262461  0.260061  0.998692
## FBgn0027101    347.104      0.2699312  0.271542  0.9940675  0.320190  0.998692</code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#using-deseq2-for-pairwise-differential-expression">Using DESeq2 for Pairwise Differential Expression</a><ul class="nav nav-pills nav-stacked">
<li><a href="#rationale">Rationale</a></li>
      <li><a href="#formatting-data-for-deseq2">Formatting Data for DESeq2</a></li>
      <li><a href="#getting-deseq2-results">Getting DESeq2 Results</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
