<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Getting Started • BRGenomics</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Getting Started">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Overview.html">Overview</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Guides
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/GettingStarted.html">Getting Started</a>
    </li>
    <li>
      <a href="../articles/ImportingModifyingAnnotations.html">Importing &amp; Modifying Annotations</a>
    </li>
    <li>
      <a href="../articles/ImportingProcessingData.html">Importing &amp; Processing Data</a>
    </li>
    <li>
      <a href="../articles/SignalCounting.html">Signal Counting</a>
    </li>
    <li>
      <a href="../articles/AnalyzingMultipleDatasets.html">Analyzing Multiple Datasets</a>
    </li>
    <li>
      <a href="../articles/ProfilePlotsAndBootstrapping.html">Profile Plots &amp; Bootstrapping</a>
    </li>
    <li>
      <a href="../articles/SpikeInNormalization.html">Spike-in Normalization</a>
    </li>
    <li>
      <a href="../articles/DESeq2WithGlobalPerturbations.html">DESeq2 with Global Perturbations</a>
    </li>
    <li>
      <a href="../articles/SequenceExtraction.html">Sequence Extraction</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mdeber/BRGenomics">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Getting Started</h1>
            
      
      
      <div class="hidden name"><code>GettingStarted.Rmd</code></div>

    </div>

    
    
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<p>As of Bioconductor 3.11 (release date April 28, 2020), BRGenomics can be installed directly from Bioconductor:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="co"># install.packages("BiocManager")</span></a>
<a class="sourceLine" id="cb1-2" data-line-number="2">BiocManager<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="st">"BRGenomics"</span>)</a></code></pre></div>
<p>Alternatively, the latest development version can be installed from<br><a href="https://github.com/mdeber/BRGenomics">GitHub</a>:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># install.packages("remotes")</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2">remotes<span class="op">::</span><span class="kw"><a href="https://remotes.r-lib.org/reference/install_github.html">install_github</a></span>(<span class="st">"mdeber/BRGenomics@R3"</span>)</a></code></pre></div>
<p><em>BRGenomics (and Bioconductor 3.11) require R version 4.0 (release April 24, 2020). Installing the <code>R3</code> branch, as shown above, is required to install under R 3.x.</em></p>
<p>If you install the development version from Github and you’re using Windows, <a href="https://cran.rstudio.com/bin/windows/Rtools/">Rtools for Windows</a> is required.</p>
</div>
<div id="parallel-processing" class="section level1">
<h1 class="hasAnchor">
<a href="#parallel-processing" class="anchor"></a>Parallel Processing</h1>
<p>By default, many BRGenomics functions use multicore processing as implemented in the <code>parallel</code> package. BRGenomics functions that can be parallelized always contain the argument <code>ncores</code>. If not specified, the default is to use the global option “mc.cores” (the same option used by the <code>parallel</code> package), or 2 if that option is not set.</p>
<p>If you wanted to change the global default to 4 cores, for example, you would run <code><a href="https://rdrr.io/r/base/options.html">options(mc.cores = 4)</a></code> at the start of your R session. If you’re unsure how many cores your processor has, run <code><a href="https://rdrr.io/r/parallel/detectCores.html">parallel::detectCores()</a></code>.</p>
<p>While performance can be memory constrained in some cases (and thus actually hampered by excessive parallelization), substantial performance benefits can be acheived by maximizing parallelization.</p>
<p>However, parallel processing is not available on Windows. To maintain compatability, all code in this vignette as well as the example code in the documentation is to use a single core, i.e. <code>ncores = 1</code>.</p>
</div>
<div id="included-datasets" class="section level1">
<h1 class="hasAnchor">
<a href="#included-datasets" class="anchor"></a>Included Datasets</h1>
<p>BRGenomics ships with an example dataset of PRO-seq data from Drosophila melanogaster<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>. PRO-seq is a basepair-resolution method that uses 3’-end sequencing of nascent RNA to map the locations of actively engaged RNA polymerases.</p>
<p>To keep the dataset small, we’ve only included reads mapping to the fourth chromosome<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>.</p>
<p>The included datasets can be accessed using the <code><a href="https://rdrr.io/r/utils/data.html">data()</a></code> function:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(BRGenomics)</a></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"PROseq"</span>)</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">PROseq</a></code></pre></div>
<pre><code>## GRanges object with 47380 ranges and 1 metadata column:
##           seqnames    ranges strand |     score
##              &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
##       [1]     chr4      1295      + |         1
##       [2]     chr4     41428      + |         1
##       [3]     chr4     42588      + |         1
##       [4]     chr4     42590      + |         2
##       [5]     chr4     42593      + |         5
##       ...      ...       ...    ... .       ...
##   [47376]     chr4   1307742      - |         1
##   [47377]     chr4   1316537      - |         1
##   [47378]     chr4   1318960      - |         1
##   [47379]     chr4   1319004      - |         1
##   [47380]     chr4   1319369      - |         1
##   -------
##   seqinfo: 7 sequences from an unspecified genome</code></pre>
<p>Notice that the data is contained within a <code>GRanges</code> object. GRanges objects, from the <em><a href="https://bioconductor.org/packages/3.11/GenomicRanges">GenomicRanges</a></em> package, are very easy to work with, and are supported by a plethora of useful functions and packages.</p>
<p>The structure of the data will be described later on (in section <em>“Basepair-Resolution GRanges Objects”</em>). For now, we’ll just note that both annotations (e.g. genelists) and data are contained using the same GRanges class.</p>
<p>We’ve included an example genelist to accompany the PRO-seq data:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"txs_dm6_chr4"</span>)</a>
<a class="sourceLine" id="cb6-2" data-line-number="2">txs_dm6_chr4</a></code></pre></div>
<pre><code>## GRanges object with 339 ranges and 2 metadata columns:
##         seqnames          ranges strand |     tx_name     gene_id
##            &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
##     [1]     chr4        879-5039      + | FBtr0346692 FBgn0267363
##     [2]     chr4     42774-43374      + | FBtr0344900 FBgn0266617
##     [3]     chr4     44774-46074      + | FBtr0340499 FBgn0265633
##     [4]     chr4     56497-60974      + | FBtr0333704 FBgn0264617
##     [5]     chr4     56497-63124      + | FBtr0333705 FBgn0264617
##     ...      ...             ...    ... .         ...         ...
##   [335]     chr4 1192419-1196848      - | FBtr0100543 FBgn0039924
##   [336]     chr4 1192419-1196848      - | FBtr0100544 FBgn0039924
##   [337]     chr4 1225089-1230713      - | FBtr0100406 FBgn0027101
##   [338]     chr4 1225737-1230713      - | FBtr0100402 FBgn0027101
##   [339]     chr4 1225737-1230713      - | FBtr0100404 FBgn0027101
##   -------
##   seqinfo: 7 sequences from dm6 genome</code></pre>
<p>The GRanges above contains all Flybase annotated transcripts from chromosome 4, with no filtering of any kind.</p>
</div>
<div id="basic-operations-on-granges" class="section level1">
<h1 class="hasAnchor">
<a href="#basic-operations-on-granges" class="anchor"></a>Basic Operations on GRanges</h1>
<p>For users who are unfamiliar with GRanges objects, this section demonstrates a number of basic operations.</p>
<p>A quick summary of the general strcture: Each element of a GRanges object is called a “range”. As you can see above, each range consists of several components: <code>seqnames</code>, <code>ranges</code>, and <code>strand</code>. These essential attributes are all found to the left of the vertical divider above; everything to the right of that divider is an optional, metadata attribute.</p>
<p>The core attributes can be accessed using the functions <code>seqnames()</code>, <code>ranges()</code>, and <code>strand()</code>. All metadata can be accessed using <code>mcols()</code>, and individual columns are accessable with the <code><a href="https://rdrr.io/r/base/Extract.html">$</a></code> operator. The only reserved metadata column is the <code>score</code> column, which is just like any other metadata column, except that users can use the <code>score()</code> function to assess it.</p>
<p>All of the above functions are both “getters” and “setters”, e.g. <code>strand(x)</code> returns the strand information, and <code>strand(x) &lt;- "+"</code> assigns it.</p>
<p>These and other operations are demonstrated below.</p>
<hr>
<p><em>To learn more about GRanges objects, including a general overview of their components, see the useful vignette <a href="https://bioconductor.org/packages/release/bioc/vignettes/GenomicRanges/inst/doc/GenomicRangesIntroduction.html"><em>An Introduction to the GenomicRanges Package</em></a>. Alternatively, see the archived materials from the 2018 Bioconductor workshop <a href="https://bioconductor.github.io/BiocWorkshops/solving-common-bioinformatic-challenges-using-genomicranges.html"><em>Solving Common Bioinformatic Challenges Using GenomicRanges</em></a>. Note that this package will implement and streamline a number of common operations, but users should still have a basic familiarity with GRanges objects.</em></p>
<hr>
<p> </p>
<p>Get the length of the genelist:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(txs_dm6_chr4) </a></code></pre></div>
<pre><code>## [1] 339</code></pre>
<p> </p>
<p>Select the 2nd transcript:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">txs_dm6_chr4[<span class="dv">2</span>]</a></code></pre></div>
<pre><code>## GRanges object with 1 range and 2 metadata columns:
##       seqnames      ranges strand |     tx_name     gene_id
##          &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
##   [1]     chr4 42774-43374      + | FBtr0344900 FBgn0266617
##   -------
##   seqinfo: 7 sequences from dm6 genome</code></pre>
<p> </p>
<p>Select 4 transcipts:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">tx4 &lt;-<span class="st"> </span>txs_dm6_chr4[<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">1</span>, <span class="dv">10</span>, <span class="dv">200</span>, <span class="dv">300</span>)]</a>
<a class="sourceLine" id="cb12-2" data-line-number="2">tx4</a></code></pre></div>
<pre><code>## GRanges object with 4 ranges and 2 metadata columns:
##       seqnames          ranges strand |     tx_name     gene_id
##          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
##   [1]     chr4        879-5039      + | FBtr0346692 FBgn0267363
##   [2]     chr4    69326-110059      + | FBtr0308615 FBgn0085432
##   [3]     chr4   184225-193489      - | FBtr0089150 FBgn0039890
##   [4]     chr4 1009895-1027101      - | FBtr0309865 FBgn0025741
##   -------
##   seqinfo: 7 sequences from dm6 genome</code></pre>
<p> </p>
<p>Get the lengths of the first 4 transcripts:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw">width</span>(tx4)</a></code></pre></div>
<pre><code>## [1]  4161 40734  9265 17207</code></pre>
<p> </p>
<p>Get a dataframe of the metadata for the first 4 transcripts:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1"><span class="kw">mcols</span>(tx4)</a></code></pre></div>
<pre><code>## DataFrame with 4 rows and 2 columns
##       tx_name     gene_id
##   &lt;character&gt; &lt;character&gt;
## 1 FBtr0346692 FBgn0267363
## 2 FBtr0308615 FBgn0085432
## 3 FBtr0089150 FBgn0039890
## 4 FBtr0309865 FBgn0025741</code></pre>
<p> </p>
<p>Access a single metadata column for the first 4 transcripts:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1"><span class="kw">mcols</span>(tx4)[, <span class="dv">2</span>]</a></code></pre></div>
<pre><code>## [1] "FBgn0267363" "FBgn0085432" "FBgn0039890" "FBgn0025741"</code></pre>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1"><span class="kw">mcols</span>(tx4)[, <span class="st">"gene_id"</span>]</a></code></pre></div>
<pre><code>## [1] "FBgn0267363" "FBgn0085432" "FBgn0039890" "FBgn0025741"</code></pre>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1">tx4<span class="op">$</span>gene_id</a></code></pre></div>
<pre><code>## [1] "FBgn0267363" "FBgn0085432" "FBgn0039890" "FBgn0025741"</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1">tx4_names &lt;-<span class="st"> </span>tx4<span class="op">$</span>tx_name</a>
<a class="sourceLine" id="cb24-2" data-line-number="2">tx4_names</a></code></pre></div>
<pre><code>## [1] "FBtr0346692" "FBtr0308615" "FBtr0089150" "FBtr0309865"</code></pre>
<p> </p>
<p>Get the first gene_id (a metadata element):</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1">tx4<span class="op">$</span>gene_id[<span class="dv">1</span>]</a></code></pre></div>
<pre><code>## [1] "FBgn0267363"</code></pre>
<p> </p>
<p>Remove a metadata column:</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" data-line-number="1"><span class="kw">mcols</span>(tx4) &lt;-<span class="st"> </span><span class="kw">mcols</span>(tx4)[, <span class="dv">-1</span>]</a>
<a class="sourceLine" id="cb28-2" data-line-number="2">tx4</a></code></pre></div>
<pre><code>## GRanges object with 4 ranges and 1 metadata column:
##       seqnames          ranges strand |           X
##          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;
##   [1]     chr4        879-5039      + | FBgn0267363
##   [2]     chr4    69326-110059      + | FBgn0085432
##   [3]     chr4   184225-193489      - | FBgn0039890
##   [4]     chr4 1009895-1027101      - | FBgn0025741
##   -------
##   seqinfo: 7 sequences from dm6 genome</code></pre>
<p> </p>
<p>Rename metadata:</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb30-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="kw">mcols</span>(tx4)) &lt;-<span class="st"> "gene_id"</span></a>
<a class="sourceLine" id="cb30-2" data-line-number="2">tx4</a></code></pre></div>
<pre><code>## GRanges object with 4 ranges and 1 metadata column:
##       seqnames          ranges strand |     gene_id
##          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;
##   [1]     chr4        879-5039      + | FBgn0267363
##   [2]     chr4    69326-110059      + | FBgn0085432
##   [3]     chr4   184225-193489      - | FBgn0039890
##   [4]     chr4 1009895-1027101      - | FBgn0025741
##   -------
##   seqinfo: 7 sequences from dm6 genome</code></pre>
<p> </p>
<p>Add metadata; same as access methods (<code>mcols()[]</code>, <code>mcols()$</code>, or simply <code><a href="https://rdrr.io/r/base/Extract.html">$</a></code>):</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb32-1" data-line-number="1">tx4<span class="op">$</span>tx_name &lt;-<span class="st"> </span>tx4_names</a>
<a class="sourceLine" id="cb32-2" data-line-number="2">tx4</a></code></pre></div>
<pre><code>## GRanges object with 4 ranges and 2 metadata columns:
##       seqnames          ranges strand |     gene_id     tx_name
##          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
##   [1]     chr4        879-5039      + | FBgn0267363 FBtr0346692
##   [2]     chr4    69326-110059      + | FBgn0085432 FBtr0308615
##   [3]     chr4   184225-193489      - | FBgn0039890 FBtr0089150
##   [4]     chr4 1009895-1027101      - | FBgn0025741 FBtr0309865
##   -------
##   seqinfo: 7 sequences from dm6 genome</code></pre>
<p> </p>
<p>Modify metadata:</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb34-1" data-line-number="1">tx4<span class="op">$</span>gene_id[<span class="dv">1</span>] &lt;-<span class="st"> "gene1"</span></a>
<a class="sourceLine" id="cb34-2" data-line-number="2">tx4<span class="op">$</span>tx_name &lt;-<span class="st"> </span><span class="dv">1</span><span class="op">:</span><span class="dv">4</span></a>
<a class="sourceLine" id="cb34-3" data-line-number="3">tx4</a></code></pre></div>
<pre><code>## GRanges object with 4 ranges and 2 metadata columns:
##       seqnames          ranges strand |     gene_id   tx_name
##          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;integer&gt;
##   [1]     chr4        879-5039      + |       gene1         1
##   [2]     chr4    69326-110059      + | FBgn0085432         2
##   [3]     chr4   184225-193489      - | FBgn0039890         3
##   [4]     chr4 1009895-1027101      - | FBgn0025741         4
##   -------
##   seqinfo: 7 sequences from dm6 genome</code></pre>
<p> </p>
<p>Get beginning of ranges (not strand specific):</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb36-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/stats/start.html">start</a></span>(tx4)</a></code></pre></div>
<pre><code>## [1]     879   69326  184225 1009895</code></pre>
<p> </p>
<p>Get beginning of ranges (strand specific):</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb38-1" data-line-number="1">tx4_tss &lt;-<span class="st"> </span><span class="kw">resize</span>(tx4, <span class="dt">width =</span> <span class="dv">1</span>, <span class="dt">fix =</span> <span class="st">"start"</span>)</a>
<a class="sourceLine" id="cb38-2" data-line-number="2">tx4_tss</a></code></pre></div>
<pre><code>## GRanges object with 4 ranges and 2 metadata columns:
##       seqnames    ranges strand |     gene_id   tx_name
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;integer&gt;
##   [1]     chr4       879      + |       gene1         1
##   [2]     chr4     69326      + | FBgn0085432         2
##   [3]     chr4    193489      - | FBgn0039890         3
##   [4]     chr4   1027101      - | FBgn0025741         4
##   -------
##   seqinfo: 7 sequences from dm6 genome</code></pre>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb40-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/stats/start.html">start</a></span>(tx4_tss)</a></code></pre></div>
<pre><code>## [1]     879   69326  193489 1027101</code></pre>
<p> </p>
<p>Remove all metadata:</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb42-1" data-line-number="1"><span class="kw">mcols</span>(tx4) &lt;-<span class="st"> </span><span class="ot">NULL</span></a>
<a class="sourceLine" id="cb42-2" data-line-number="2">tx4</a></code></pre></div>
<pre><code>## GRanges object with 4 ranges and 0 metadata columns:
##       seqnames          ranges strand
##          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt;
##   [1]     chr4        879-5039      +
##   [2]     chr4    69326-110059      +
##   [3]     chr4   184225-193489      -
##   [4]     chr4 1009895-1027101      -
##   -------
##   seqinfo: 7 sequences from dm6 genome</code></pre>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>Hojoong Kwak, Nicholas J. Fuda, Leighton J. Core, John T. Lis (2013). Precise Maps of RNA Polymerase Reveal How Promoters Direct Initiation and Pausing. <em>Science</em> <strong>339</strong>(6122): 950–953. <a href="https://doi.org/10.1126/science.1229386" class="uri">https://doi.org/10.1126/science.1229386</a><a href="#fnref1" class="footnote-back">↩</a></p></li>
<li id="fn2"><p>Chromosome 4 in Drosophila, often referred to as the “dot” chromosome, is very small and contains very few genes<a href="#fnref2" class="footnote-back">↩</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#installation">Installation</a></li>
      <li><a href="#parallel-processing">Parallel Processing</a></li>
      <li><a href="#included-datasets">Included Datasets</a></li>
      <li><a href="#basic-operations-on-granges">Basic Operations on GRanges</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
