<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Importing &amp; Modifying Annotations • BRGenomics</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Importing &amp; Modifying Annotations">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Overview.html">Overview</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Guides
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/GettingStarted.html">Getting Started</a>
    </li>
    <li>
      <a href="../articles/ImportingModifyingAnnotations.html">Importing &amp; Modifying Annotations</a>
    </li>
    <li>
      <a href="../articles/ImportingProcessingData.html">Importing &amp; Processing Data</a>
    </li>
    <li>
      <a href="../articles/SignalCounting.html">Signal Counting</a>
    </li>
    <li>
      <a href="../articles/AnalyzingMultipleDatasets.html">Analyzing Multiple Datasets</a>
    </li>
    <li>
      <a href="../articles/ProfilePlotsAndBootstrapping.html">Profile Plots &amp; Bootstrapping</a>
    </li>
    <li>
      <a href="../articles/SpikeInNormalization.html">Spike-in Normalization</a>
    </li>
    <li>
      <a href="../articles/DESeq2WithGlobalPerturbations.html">DESeq2 with Global Perturbations</a>
    </li>
    <li>
      <a href="../articles/SequenceExtraction.html">Sequence Extraction</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mdeber/BRGenomics">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Importing &amp; Modifying Annotations</h1>
            
      
      
      <div class="hidden name"><code>ImportingModifyingAnnotations.Rmd</code></div>

    </div>

    
    
<div id="importing-annotations-with-rtracklayer" class="section level1">
<h1 class="hasAnchor">
<a href="#importing-annotations-with-rtracklayer" class="anchor"></a>Importing Annotations with rtracklayer</h1>
<p>Importing genomics files is accomplished using the <code>rtracklayer</code> package, which contains a variety of functions and options for importing and exporting.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="co"># import bed file</span></a>
<a class="sourceLine" id="cb1-2" data-line-number="2">genelist &lt;-<span class="st"> </span><span class="kw">import.bed</span>(<span class="st">"~/data/genelists/genes.bed"</span>)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"></a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="co"># import gff</span></a>
<a class="sourceLine" id="cb1-5" data-line-number="5">genelist &lt;-<span class="st"> </span><span class="kw">import.gff</span>(<span class="st">"~/data/genelists/genes.gff"</span>)</a>
<a class="sourceLine" id="cb1-6" data-line-number="6"></a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="co"># export a bed file after modifying</span></a>
<a class="sourceLine" id="cb1-8" data-line-number="8"><span class="kw">export.bed</span>(genelist, <span class="st">"~/data/genelists/filtered_genes.bed"</span>)</a></code></pre></div>
</div>
<div id="defining-regions-using-the-genebodies-function" class="section level1">
<h1 class="hasAnchor">
<a href="#defining-regions-using-the-genebodies-function" class="anchor"></a>Defining Regions Using the genebodies Function</h1>
<p>One of the more useful <code>GenomicRanges</code> functions is the <code>promoters()</code> function, which returns ranges centered on the strand-specific start of the input ranges:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(BRGenomics)</a></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"txs_dm6_chr4"</span>)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2">tx4 &lt;-<span class="st"> </span>txs_dm6_chr4[<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">1</span>, <span class="dv">10</span>, <span class="dv">200</span>, <span class="dv">300</span>)]</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">tx4</a></code></pre></div>
<pre><code>## GRanges object with 4 ranges and 2 metadata columns:
##       seqnames          ranges strand |     tx_name     gene_id
##          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
##   [1]     chr4        879-5039      + | FBtr0346692 FBgn0267363
##   [2]     chr4    69326-110059      + | FBtr0308615 FBgn0085432
##   [3]     chr4   184225-193489      - | FBtr0089150 FBgn0039890
##   [4]     chr4 1009895-1027101      - | FBtr0309865 FBgn0025741
##   -------
##   seqinfo: 7 sequences from dm6 genome</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">tx4_pr &lt;-<span class="st"> </span><span class="kw">promoters</span>(tx4, <span class="dt">upstream =</span> <span class="dv">50</span>, <span class="dt">downstream =</span> <span class="dv">100</span>)</a>
<a class="sourceLine" id="cb5-2" data-line-number="2">tx4_pr</a></code></pre></div>
<pre><code>## GRanges object with 4 ranges and 2 metadata columns:
##       seqnames          ranges strand |     tx_name     gene_id
##          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
##   [1]     chr4         829-978      + | FBtr0346692 FBgn0267363
##   [2]     chr4     69276-69425      + | FBtr0308615 FBgn0085432
##   [3]     chr4   193390-193539      - | FBtr0089150 FBgn0039890
##   [4]     chr4 1027002-1027151      - | FBtr0309865 FBgn0025741
##   -------
##   seqinfo: 7 sequences from dm6 genome</code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw">width</span>(tx4_pr)</a></code></pre></div>
<pre><code>## [1] 150 150 150 150</code></pre>
<p>BRGenomics ships with a more flexible alternative function called <code><a href="../reference/genebodies.html">genebodies()</a></code>. While <code>promoters()</code> has the arguments <code>upstream</code> and <code>downstream</code>, which take only positive values, the <code><a href="../reference/genebodies.html">genebodies()</a></code> function uses <code>start</code> and <code>end</code> arguments that can be positive or negative, and arguments <code>fix.start</code> and <code>fix.end</code> for determining whether to define the positions in relation to the (strand-specific) beginning or ends of genes.</p>
<p>Below, we demonstrate several uses of the <code><a href="../reference/genebodies.html">genebodies()</a></code> function, using a list of transcripts which start at a transcription start site (TSS) and end at a cleavage and polyadenylation site (CPS).</p>
<hr>
<p>Original regions:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">tx4</a></code></pre></div>
<pre><code>## GRanges object with 4 ranges and 2 metadata columns:
##       seqnames          ranges strand |     tx_name     gene_id
##          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
##   [1]     chr4        879-5039      + | FBtr0346692 FBgn0267363
##   [2]     chr4    69326-110059      + | FBtr0308615 FBgn0085432
##   [3]     chr4   184225-193489      - | FBtr0089150 FBgn0039890
##   [4]     chr4 1009895-1027101      - | FBtr0309865 FBgn0025741
##   -------
##   seqinfo: 7 sequences from dm6 genome</code></pre>
<p> </p>
<p>Genebody regions from 300 bp downstream of the TSS to 300 bp upstream of the CPS:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="kw"><a href="../reference/genebodies.html">genebodies</a></span>(tx4, <span class="dt">start =</span> <span class="dv">300</span>, <span class="dt">end =</span> <span class="dv">-300</span>)</a></code></pre></div>
<pre><code>## GRanges object with 4 ranges and 2 metadata columns:
##       seqnames          ranges strand |     tx_name     gene_id
##          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
##   [1]     chr4       1179-4739      + | FBtr0346692 FBgn0267363
##   [2]     chr4    69626-109759      + | FBtr0308615 FBgn0085432
##   [3]     chr4   184525-193189      - | FBtr0089150 FBgn0039890
##   [4]     chr4 1010195-1026801      - | FBtr0309865 FBgn0025741
##   -------
##   seqinfo: 7 sequences from dm6 genome</code></pre>
<p>By default, <code>fix.start = "start"</code> and <code>fix.end = "end"</code>. But we can change either of them to define ranges based solely on the beginnings or ends of the input regions.</p>
<p>Get promoter regions from 50 bp upstream to 100 bp downstream of the TSS:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw"><a href="../reference/genebodies.html">genebodies</a></span>(tx4, <span class="dv">-50</span>, <span class="dv">100</span>, <span class="dt">fix.end =</span> <span class="st">"start"</span>)</a></code></pre></div>
<pre><code>## GRanges object with 4 ranges and 2 metadata columns:
##       seqnames          ranges strand |     tx_name     gene_id
##          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
##   [1]     chr4         829-979      + | FBtr0346692 FBgn0267363
##   [2]     chr4     69276-69426      + | FBtr0308615 FBgn0085432
##   [3]     chr4   193389-193539      - | FBtr0089150 FBgn0039890
##   [4]     chr4 1027001-1027151      - | FBtr0309865 FBgn0025741
##   -------
##   seqinfo: 7 sequences from dm6 genome</code></pre>
<p> </p>
<p>Regions from 100 bp upstream of to 50 bp upstream of the TSS:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="kw"><a href="../reference/genebodies.html">genebodies</a></span>(tx4, <span class="dv">-100</span>, <span class="dv">-50</span>, <span class="dt">fix.end =</span> <span class="st">"start"</span>)</a></code></pre></div>
<pre><code>## GRanges object with 4 ranges and 2 metadata columns:
##       seqnames          ranges strand |     tx_name     gene_id
##          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
##   [1]     chr4         779-829      + | FBtr0346692 FBgn0267363
##   [2]     chr4     69226-69276      + | FBtr0308615 FBgn0085432
##   [3]     chr4   193539-193589      - | FBtr0089150 FBgn0039890
##   [4]     chr4 1027151-1027201      - | FBtr0309865 FBgn0025741
##   -------
##   seqinfo: 7 sequences from dm6 genome</code></pre>
<p> </p>
<p>Regions from 1kb upstream of the CPS to 1kb downstream of the CPS</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="kw"><a href="../reference/genebodies.html">genebodies</a></span>(tx4, <span class="dv">-1000</span>, <span class="dv">1000</span>, <span class="dt">fix.start =</span> <span class="st">"end"</span>)</a></code></pre></div>
<pre><code>## GRanges object with 4 ranges and 2 metadata columns:
##       seqnames          ranges strand |     tx_name     gene_id
##          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
##   [1]     chr4       4039-6039      + | FBtr0346692 FBgn0267363
##   [2]     chr4   109059-111059      + | FBtr0308615 FBgn0085432
##   [3]     chr4   183225-185225      - | FBtr0089150 FBgn0039890
##   [4]     chr4 1008895-1010895      - | FBtr0309865 FBgn0025741
##   -------
##   seqinfo: 7 sequences from dm6 genome</code></pre>
<p> </p>
<p>Regions within the first 10kb downstream of the CPS:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="kw"><a href="../reference/genebodies.html">genebodies</a></span>(tx4, <span class="dv">0</span>, <span class="dv">10000</span>, <span class="dt">fix.start =</span> <span class="st">"end"</span>)</a></code></pre></div>
<pre><code>## GRanges object with 4 ranges and 2 metadata columns:
##       seqnames         ranges strand |     tx_name     gene_id
##          &lt;Rle&gt;      &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
##   [1]     chr4     5039-15039      + | FBtr0346692 FBgn0267363
##   [2]     chr4  110059-120059      + | FBtr0308615 FBgn0085432
##   [3]     chr4  174225-184225      - | FBtr0089150 FBgn0039890
##   [4]     chr4 999895-1009895      - | FBtr0309865 FBgn0025741
##   -------
##   seqinfo: 7 sequences from dm6 genome</code></pre>
<p> </p>
</div>
<div id="modify-by-gene" class="section level1">
<h1 class="hasAnchor">
<a href="#modify-by-gene" class="anchor"></a>Modify-By-Gene</h1>
<p>The <code><a href="../reference/intersectByGene.html">reduceByGene()</a></code> and <code><a href="../reference/intersectByGene.html">intersectByGene()</a></code> are two other useful functions, which perform two common tasks very efficiently.</p>
<div id="reducebygene" class="section level2">
<h2 class="hasAnchor">
<a href="#reducebygene" class="anchor"></a>reduceByGene</h2>
<p><code><a href="../reference/intersectByGene.html">reduceByGene()</a></code> takes all ranges that share the same gene name (e.g. different transcript isoforms) and combines them such that all positions are represented.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1">txs &lt;-<span class="st"> </span>txs_dm6_chr4[<span class="kw"><a href="https://rdrr.io/r/base/order.html">order</a></span>(txs_dm6_chr4<span class="op">$</span>gene_id)] <span class="co"># sort by gene_id</span></a>
<a class="sourceLine" id="cb21-2" data-line-number="2">txs[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>]</a></code></pre></div>
<pre><code>## GRanges object with 10 ranges and 2 metadata columns:
##        seqnames          ranges strand |     tx_name     gene_id
##           &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
##    [1]     chr4 1172469-1181628      - | FBtr0089204 FBgn0002521
##    [2]     chr4 1172469-1181628      - | FBtr0089205 FBgn0002521
##    [3]     chr4   501810-538373      + | FBtr0332913 FBgn0004607
##    [4]     chr4   501810-539792      + | FBtr0089070 FBgn0004607
##    [5]     chr4   501810-540874      + | FBtr0307167 FBgn0004607
##    [6]     chr4     47710-56331      - | FBtr0089178 FBgn0004859
##    [7]     chr4     47710-56331      - | FBtr0308074 FBgn0004859
##    [8]     chr4     47710-57041      - | FBtr0306168 FBgn0004859
##    [9]     chr4   697689-721173      + | FBtr0089235 FBgn0005558
##   [10]     chr4   704651-721173      + | FBtr0089236 FBgn0005558
##   -------
##   seqinfo: 7 sequences from dm6 genome</code></pre>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1"><span class="kw"><a href="../reference/intersectByGene.html">reduceByGene</a></span>(txs, <span class="dt">gene_names =</span> txs<span class="op">$</span>gene_id)</a></code></pre></div>
<pre><code>## GRanges object with 111 ranges and 0 metadata columns:
##               seqnames          ranges strand
##                  &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt;
##   FBgn0002521     chr4 1172469-1181628      -
##   FBgn0004607     chr4   501810-540874      +
##   FBgn0004859     chr4     47710-57041      -
##   FBgn0005558     chr4   697689-721173      +
##   FBgn0005561     chr4 1088798-1113317      +
##           ...      ...             ...    ...
##   FBgn0266727     chr4   776475-777146      -
##   FBgn0266728     chr4   959224-959434      -
##   FBgn0267363     chr4        879-5039      +
##   FBgn0267734     chr4   228576-229112      -
##   FBgn0283557     chr4   400703-400765      -
##   -------
##   seqinfo: 7 sequences from dm6 genome</code></pre>
<p>By default, the gene names are maintained as the names of the rows (ranges) in the output. To set them into metadata again, we could run:</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1">txs_redux &lt;-<span class="st"> </span><span class="kw"><a href="../reference/intersectByGene.html">reduceByGene</a></span>(txs, <span class="dt">gene_names =</span> txs<span class="op">$</span>gene_id)</a>
<a class="sourceLine" id="cb25-2" data-line-number="2">txs_redux<span class="op">$</span>gene_id &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(txs_redux)</a>
<a class="sourceLine" id="cb25-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(txs_redux) &lt;-<span class="st"> </span><span class="ot">NULL</span></a>
<a class="sourceLine" id="cb25-4" data-line-number="4">txs_redux</a></code></pre></div>
<pre><code>## GRanges object with 111 ranges and 1 metadata column:
##         seqnames          ranges strand |     gene_id
##            &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;
##     [1]     chr4 1172469-1181628      - | FBgn0002521
##     [2]     chr4   501810-540874      + | FBgn0004607
##     [3]     chr4     47710-57041      - | FBgn0004859
##     [4]     chr4   697689-721173      + | FBgn0005558
##     [5]     chr4 1088798-1113317      + | FBgn0005561
##     ...      ...             ...    ... .         ...
##   [107]     chr4   776475-777146      - | FBgn0266727
##   [108]     chr4   959224-959434      - | FBgn0266728
##   [109]     chr4        879-5039      + | FBgn0267363
##   [110]     chr4   228576-229112      - | FBgn0267734
##   [111]     chr4   400703-400765      - | FBgn0283557
##   -------
##   seqinfo: 7 sequences from dm6 genome</code></pre>
<p>Note that <code><a href="../reference/intersectByGene.html">reduceByGene()</a></code> is not guaranteed to produce a single range per gene, but will produce the fewest number of ranges required to represent all input positions.</p>
<p>Also note that while the output ranges for a given gene are disjoint, it is possible for ranges from different genes to overlap one another.</p>
<p>To make all ranges disjoint (no position overlapped more than once), set <code>disjoin = TRUE</code>.</p>
</div>
<div id="intersectbygene" class="section level2">
<h2 class="hasAnchor">
<a href="#intersectbygene" class="anchor"></a>intersectByGene</h2>
<p>While <code><a href="../reference/intersectByGene.html">reduceByGene()</a></code> creates a comprehensive representation of all input ranges (e.g. a “union” of the set of input ranges), <code><a href="../reference/intersectByGene.html">intersectByGene()</a></code> outputs only the consensus region, i.e. the region that is shared across all the ranges of a particular gene.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" data-line-number="1">txs[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>]</a></code></pre></div>
<pre><code>## GRanges object with 10 ranges and 2 metadata columns:
##        seqnames          ranges strand |     tx_name     gene_id
##           &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
##    [1]     chr4 1172469-1181628      - | FBtr0089204 FBgn0002521
##    [2]     chr4 1172469-1181628      - | FBtr0089205 FBgn0002521
##    [3]     chr4   501810-538373      + | FBtr0332913 FBgn0004607
##    [4]     chr4   501810-539792      + | FBtr0089070 FBgn0004607
##    [5]     chr4   501810-540874      + | FBtr0307167 FBgn0004607
##    [6]     chr4     47710-56331      - | FBtr0089178 FBgn0004859
##    [7]     chr4     47710-56331      - | FBtr0308074 FBgn0004859
##    [8]     chr4     47710-57041      - | FBtr0306168 FBgn0004859
##    [9]     chr4   697689-721173      + | FBtr0089235 FBgn0005558
##   [10]     chr4   704651-721173      + | FBtr0089236 FBgn0005558
##   -------
##   seqinfo: 7 sequences from dm6 genome</code></pre>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" data-line-number="1">txs_insxt &lt;-<span class="st"> </span><span class="kw"><a href="../reference/intersectByGene.html">intersectByGene</a></span>(txs, <span class="dt">gene_names =</span> txs<span class="op">$</span>gene_id)</a>
<a class="sourceLine" id="cb29-2" data-line-number="2">txs_insxt[<span class="kw"><a href="https://rdrr.io/r/base/order.html">order</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(txs_insxt))]</a></code></pre></div>
<pre><code>## GRanges object with 110 ranges and 0 metadata columns:
##               seqnames          ranges strand
##                  &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt;
##   FBgn0002521     chr4 1172469-1181628      -
##   FBgn0004607     chr4   501810-538373      +
##   FBgn0004859     chr4     47710-56331      -
##   FBgn0005558     chr4   707429-721173      +
##   FBgn0005561     chr4 1098525-1110974      +
##           ...      ...             ...    ...
##   FBgn0266727     chr4   776475-777146      -
##   FBgn0266728     chr4   959224-959434      -
##   FBgn0267363     chr4        879-5039      +
##   FBgn0267734     chr4   228576-229112      -
##   FBgn0283557     chr4   400703-400765      -
##   -------
##   seqinfo: 7 sequences from dm6 genome</code></pre>
<p>Unlike <code><a href="../reference/intersectByGene.html">reduceByGene()</a></code>, <code><a href="../reference/intersectByGene.html">intersectByGene()</a></code> is guaranteed to return no more than 1 range per gene. However, genes for which no consensus is possible (i.e. no single range can overlap every input range) are dropped from the genelist.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#importing-annotations-with-rtracklayer">Importing Annotations with rtracklayer</a></li>
      <li><a href="#defining-regions-using-the-genebodies-function">Defining Regions Using the genebodies Function</a></li>
      <li>
<a href="#modify-by-gene">Modify-By-Gene</a><ul class="nav nav-pills nav-stacked">
<li><a href="#reducebygene">reduceByGene</a></li>
      <li><a href="#intersectbygene">intersectByGene</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
