<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Importing &amp; Processing Data • BRGenomics</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Importing &amp; Processing Data">
<meta property="og:description" content="BRGenomics">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Overview.html">Overview</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Guides
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/GettingStarted.html">Getting Started</a>
    </li>
    <li>
      <a href="../articles/ImportingModifyingAnnotations.html">Importing &amp; Modifying Annotations</a>
    </li>
    <li>
      <a href="../articles/ImportingProcessingData.html">Importing &amp; Processing Data</a>
    </li>
    <li>
      <a href="../articles/SignalCounting.html">Signal Counting</a>
    </li>
    <li>
      <a href="../articles/AnalyzingMultipleDatasets.html">Analyzing Multiple Datasets</a>
    </li>
    <li>
      <a href="../articles/ProfilePlotsAndBootstrapping.html">Profile Plots &amp; Bootstrapping</a>
    </li>
    <li>
      <a href="../articles/SpikeInNormalization.html">Spike-in Normalization</a>
    </li>
    <li>
      <a href="../articles/DESeq2WithGlobalPerturbations.html">DESeq2 with Global Perturbations</a>
    </li>
    <li>
      <a href="../articles/SequenceExtraction.html">Sequence Extraction</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mdeber/BRGenomics">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Importing &amp; Processing Data</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/mdeber/BRGenomics/blob/master/vignettes/ImportingProcessingData.Rmd"><code>vignettes/ImportingProcessingData.Rmd</code></a></small>
      <div class="hidden name"><code>ImportingProcessingData.Rmd</code></div>

    </div>

    
    
<p>BRGenomics provides several functions for conveniently importing and processing BAM, bigWig, bedGraph files.</p>
<div id="importing-bam-files" class="section level1">
<h1 class="hasAnchor">
<a href="#importing-bam-files" class="anchor"></a>Importing BAM Files</h1>
<p>The <code><a href="../reference/import_bam.html">import_bam()</a></code> function provides a number of options for filtering and processing bam files. BRGenomics includes an example BAM file with a small number of reads from the included PRO-seq data. The file’s local location can be found (on your computer) as follows:</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">BRGenomics</span>)</pre></body></html></div>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="no">bfile</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"PROseq_dm6_chr4.bam"</span>,
                     <span class="kw">package</span> <span class="kw">=</span> <span class="st">"BRGenomics"</span>)</pre></body></html></div>
<p>Because PRO-seq data is sequenced in the 3’-to-5’ direction of the original RNA molecule, we’ll use the <code>revcomp</code> option to reverse-complement all the input reads. We’ll also set a minimum MAPQ score of 20:</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="no">ps_reads</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/import_bam.html">import_bam</a></span>(<span class="no">bfile</span>, <span class="kw">mapq</span> <span class="kw">=</span> <span class="fl">20</span>, <span class="kw">revcomp</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">paired_end</span> <span class="kw">=</span> <span class="fl">FALSE</span>)
<span class="no">ps_reads</span></pre></body></html></div>
<pre><code>## GRanges object with 164 ranges and 1 metadata column:
##         seqnames          ranges strand |     score
##            &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
##     [1]     chr4       1270-1296      + |         1
##     [2]     chr4     41411-41429      + |         1
##     [3]     chr4     42556-42591      + |         1
##     [4]     chr4     42559-42589      + |         1
##     [5]     chr4     42559-42594      + |         3
##     ...      ...             ...    ... .       ...
##   [160]     chr4 1307741-1307776      - |         1
##   [161]     chr4 1316536-1316563      - |         1
##   [162]     chr4 1318959-1318994      - |         1
##   [163]     chr4 1319003-1319038      - |         1
##   [164]     chr4 1319368-1319403      - |         1
##   -------
##   seqinfo: 1870 sequences from an unspecified genome</code></pre>
<p>By default, <code><a href="../reference/import_bam.html">import_bam()</a></code> combines identical reads into the same range, and the <code>score</code> metadata column indicates the number of perfectly-overlapping alignments. This means that the total number of alignments (reads) is equal to the sum of the score:</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="fu">score</span>(<span class="no">ps_reads</span>))</pre></body></html></div>
<pre><code>## [1] 190</code></pre>
<p>Alternatively, you can import each read as its own range by setting <code>field =  NULL</code>:</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="no">reads_expanded</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/import_bam.html">import_bam</a></span>(<span class="no">bfile</span>, <span class="kw">mapq</span> <span class="kw">=</span> <span class="fl">20</span>, <span class="kw">revcomp</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
                             <span class="kw">field</span> <span class="kw">=</span> <span class="kw">NULL</span>, <span class="kw">paired_end</span> <span class="kw">=</span> <span class="fl">FALSE</span>)
<span class="no">ps_reads</span>[<span class="fl">1</span>:<span class="fl">8</span>]</pre></body></html></div>
<pre><code>## GRanges object with 8 ranges and 1 metadata column:
##       seqnames      ranges strand |     score
##          &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
##   [1]     chr4   1270-1296      + |         1
##   [2]     chr4 41411-41429      + |         1
##   [3]     chr4 42556-42591      + |         1
##   [4]     chr4 42559-42589      + |         1
##   [5]     chr4 42559-42594      + |         3
##   [6]     chr4 42560-42594      + |         2
##   [7]     chr4 42560-42595      + |         2
##   [8]     chr4 42561-42596      + |         1
##   -------
##   seqinfo: 1870 sequences from an unspecified genome</code></pre>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="no">reads_expanded</span>[<span class="fl">1</span>:<span class="fl">8</span>]</pre></body></html></div>
<pre><code>## GRanges object with 8 ranges and 0 metadata columns:
##       seqnames      ranges strand
##          &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt;
##   [1]     chr4   1270-1296      +
##   [2]     chr4 41411-41429      +
##   [3]     chr4 42556-42591      +
##   [4]     chr4 42559-42589      +
##   [5]     chr4 42559-42594      +
##   [6]     chr4 42559-42594      +
##   [7]     chr4 42559-42594      +
##   [8]     chr4 42560-42594      +
##   -------
##   seqinfo: 1870 sequences from an unspecified genome</code></pre>
<p>Notice that reads 5-7 are now identical, rather than combined into a single range with a score = 3.</p>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">reads_expanded</span>) <span class="kw">==</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="fu">score</span>(<span class="no">ps_reads</span>))</pre></body></html></div>
<pre><code>## [1] TRUE</code></pre>
<p>Many BRGenomics function have a <code>field</code> argument, and setting <code>field = NULL</code> will treat each range has a single read.</p>
<div id="example-importing-pro-seq-bam-files-at-basepair-resolution" class="section level2">
<h2 class="hasAnchor">
<a href="#example-importing-pro-seq-bam-files-at-basepair-resolution" class="anchor"></a>Example: Importing PRO-seq BAM files at Basepair Resolution</h2>
<p>We can use the <code><a href="../reference/import_bam.html">import_bam()</a></code> function to extract the positions of interest from BAM files. Below, we construct an import function for PRO-seq data that returns a basepair-resolution GRanges object.</p>
<p>In PRO-seq, a “run-on” reaction is performed in which actively engaged RNA polymerases incorporate a biotinylated nucleotide at the 3’ end of a nascent RNA. Our base of interest is therefore the base immediately preceding the RNA 3’ end, as this was the original position of a polymerase active site.</p>
<p>The processing options in <code><a href="../reference/import_bam.html">import_bam()</a></code> are applied in the same order that they’re listed in the documentation page. Following this order, we will apply the options:</p>
<ol style="list-style-type: decimal">
<li>Filter reads by a minimum MAPQ score</li>
<li>Take the reverse complement</li>
<li>Shift reads upstream by 1 base</li>
<li>Extract the 3’ base</li>
</ol>
<div class="sourceCode" id="cb13"><html><body><pre class="r"><span class="no">ps</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/import_bam.html">import_bam</a></span>(<span class="no">bfile</span>,
                 <span class="kw">mapq</span> <span class="kw">=</span> <span class="fl">20</span>,
                 <span class="kw">revcomp</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
                 <span class="kw">shift</span> <span class="kw">=</span> -<span class="fl">1</span>,
                 <span class="kw">trim.to</span> <span class="kw">=</span> <span class="st">"3p"</span>,
                 <span class="kw">paired_end</span> <span class="kw">=</span> <span class="fl">FALSE</span>)
<span class="no">ps</span></pre></body></html></div>
<pre><code>## GRanges object with 150 ranges and 1 metadata column:
##         seqnames    ranges strand |     score
##            &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
##     [1]     chr4      1295      + |         1
##     [2]     chr4     41428      + |         1
##     [3]     chr4     42588      + |         1
##     [4]     chr4     42590      + |         2
##     [5]     chr4     42593      + |         5
##     ...      ...       ...    ... .       ...
##   [146]     chr4   1307742      - |         1
##   [147]     chr4   1316537      - |         1
##   [148]     chr4   1318960      - |         1
##   [149]     chr4   1319004      - |         1
##   [150]     chr4   1319369      - |         1
##   -------
##   seqinfo: 1870 sequences from an unspecified genome</code></pre>
<p><em>Note that for paired-end data, <code><a href="../reference/import_bam.html">import_bam()</a></code> will automatically filter unmatched read pairs.</em></p>
<p>Notice that the number of ranges in <code>ps</code> is not the same as for <code>ps_reads</code>, in which we imported the entire read lengths:</p>
<div class="sourceCode" id="cb15"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">ps_reads</span>)
<span class="co">## [1] 164</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">ps</span>)
<span class="co">## [1] 150</span></pre></body></html></div>
<p>This is because identical positions are collapsed together after applying the processing options. However, we can check that all of the same reads are represented:</p>
<div class="sourceCode" id="cb16"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="fu">score</span>(<span class="no">ps</span>)) <span class="kw">==</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="fu">score</span>(<span class="no">ps_reads</span>))
<span class="co">## [1] TRUE</span></pre></body></html></div>
<p>And we can check that collapsing identical positions has created a basepair-resolution GRanges object:</p>
<div class="sourceCode" id="cb17"><html><body><pre class="r"><span class="fu"><a href="../reference/makeGRangesBRG.html">isBRG</a></span>(<span class="no">ps</span>)
<span class="co">## [1] TRUE</span></pre></body></html></div>
</div>
<div id="pre-formatted-input-functions" class="section level2">
<h2 class="hasAnchor">
<a href="#pre-formatted-input-functions" class="anchor"></a>Pre-formatted Input Functions</h2>
<p>For convenience, we’ve included several functions with default options for several kinds of data, including <code><a href="../reference/import_bam.html">import_bam_PROseq()</a></code>, <code><a href="../reference/import_bam.html">import_bam_PROcap()</a></code>, and <code><a href="../reference/import_bam.html">import_bam_ATACseq()</a></code>, the latter of which corrects for the 9 bp offset between Tn5 insertion sites.<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a></p>
</div>
<div id="example-converting-bams-to-bigwigs" class="section level2">
<h2 class="hasAnchor">
<a href="#example-converting-bams-to-bigwigs" class="anchor"></a>Example: Converting BAMs to bigWigs</h2>
<p>In conjunction with export functions from the <em><a href="https://bioconductor.org/packages/3.12/rtracklayer">rtracklayer</a></em> package, we can use the functions described above to write a post-alignment pipeline for generating bigWig files for PRO-seq data:</p>
<div class="sourceCode" id="cb18"><html><body><pre class="r"><span class="co"># import bam, automatically applying processing steps for PRO-seq</span>
<span class="no">ps</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/import_bam.html">import_bam_PROseq</a></span>(<span class="no">bfile</span>, <span class="kw">mapq</span> <span class="kw">=</span> <span class="fl">30</span>, <span class="kw">paired_end</span> <span class="kw">=</span> <span class="fl">FALSE</span>)

<span class="co"># separate strands, and make minus-strand scores negative</span>
<span class="no">ps_plus</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span>(<span class="no">ps</span>, <span class="no">strand</span> <span class="kw">==</span> <span class="st">"+"</span>)
<span class="no">ps_minus</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span>(<span class="no">ps</span>, <span class="no">strand</span> <span class="kw">==</span> <span class="st">"-"</span>)
<span class="fu">score</span>(<span class="no">ps_minus</span>) <span class="kw">&lt;-</span> -<span class="fu">score</span>(<span class="no">ps_minus</span>)

<span class="co"># use rtracklayer to export bigWig files</span>
<span class="fu">export.bw</span>(<span class="no">ps_plus</span>, <span class="st">"~/Data/PROseq_plus.bw"</span>)
<span class="fu">export.bw</span>(<span class="no">ps_minus</span>, <span class="st">"~/Data/PROseq_minus.bw"</span>)</pre></body></html></div>
</div>
<div id="performance-considerations" class="section level2">
<h2 class="hasAnchor">
<a href="#performance-considerations" class="anchor"></a>Performance Considerations</h2>
<p>For single-ended bam files, import is much faster if the bam files are sorted and indexed (i.e. by <code>samtools index</code>).</p>
<p>For paired-end files, we assume that collating (<code>samtools collate</code>) or sorting by name is faster.</p>
<p>Additionally, while single-ended files can often be imported “all at once” (particularly if sorted and indexed), processing paired-end data is more memory intensive, and requires breaking up the file into chunks for processing. For this, use the <code>yieldSize</code> argument.</p>
<p>For example, to process 500 thousands reads at a time, set the <code>yieldSize = 5e5</code>.</p>
</div>
</div>
<div id="importing-bedgraphs-and-bigwigs" class="section level1">
<h1 class="hasAnchor">
<a href="#importing-bedgraphs-and-bigwigs" class="anchor"></a>Importing bedGraphs and bigWigs</h1>
<p>bedGraph and bigWig files are efficient and portable, but unstranded representations of basepair-resolution genomics data.</p>
<p>As compared to <code><a href="https://rdrr.io/pkg/rtracklayer/man/BEDFile-class.html">rtracklayer::import.bedGraph()</a></code>, the BRGenomics function <code><a href="../reference/import-functions.html">import_bedGraph()</a></code> imports both plus-strand and minus-strand files as a single object, and has options for filtering out odd chromosomes, mitochondrial chromosomes, and sex chromosomes.</p>
<div class="sourceCode" id="cb19"><html><body><pre class="r"><span class="co"># local paths to included bedGraph files</span>
<span class="no">bg.p</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"PROseq_dm6_chr4_plus.bedGraph"</span>,
                    <span class="kw">package</span> <span class="kw">=</span> <span class="st">"BRGenomics"</span>)
<span class="no">bg.m</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"PROseq_dm6_chr4_minus.bedGraph"</span>,
                    <span class="kw">package</span> <span class="kw">=</span> <span class="st">"BRGenomics"</span>)

<span class="fu"><a href="../reference/import-functions.html">import_bedGraph</a></span>(<span class="no">bg.p</span>, <span class="no">bg.m</span>, <span class="kw">genome</span> <span class="kw">=</span> <span class="st">"dm6"</span>)</pre></body></html></div>
<pre><code>## GRanges object with 164 ranges and 1 metadata column:
##         seqnames          ranges strand |     score
##            &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
##     [1]     chr4       1270-1295      + |         1
##     [2]     chr4     41411-41428      + |         1
##     [3]     chr4     42556-42590      + |         1
##     [4]     chr4     42559-42588      + |         1
##     [5]     chr4     42559-42593      + |         3
##     ...      ...             ...    ... .       ...
##   [160]     chr4 1307742-1307776      - |         1
##   [161]     chr4 1316537-1316563      - |         1
##   [162]     chr4 1318960-1318994      - |         1
##   [163]     chr4 1319004-1319038      - |         1
##   [164]     chr4 1319369-1319403      - |         1
##   -------
##   seqinfo: 1 sequence from dm6 genome; no seqlengths</code></pre>
<p>The <code><a href="../reference/import-functions.html">import_bigWig()</a></code> function provides the same added functionality as compared to <code><a href="https://rdrr.io/pkg/rtracklayer/man/BigWigFile.html">rtracklayer::import.bw()</a></code>, but also removes run-length compression and returns a basepair-resolution GRanges object by default.</p>
<div class="sourceCode" id="cb21"><html><body><pre class="r"><span class="co"># local paths to included bigWig files</span>
<span class="no">bw.p</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"PROseq_dm6_chr4_plus.bw"</span>,
                    <span class="kw">package</span> <span class="kw">=</span> <span class="st">"BRGenomics"</span>)
<span class="no">bw.m</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"PROseq_dm6_chr4_minus.bw"</span>,
                    <span class="kw">package</span> <span class="kw">=</span> <span class="st">"BRGenomics"</span>)

<span class="fu"><a href="../reference/import-functions.html">import_bigWig</a></span>(<span class="no">bw.p</span>, <span class="no">bw.m</span>, <span class="kw">genome</span> <span class="kw">=</span> <span class="st">"dm6"</span>)</pre></body></html></div>
<pre><code>## GRanges object with 150 ranges and 1 metadata column:
##         seqnames    ranges strand |     score
##            &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
##     [1]     chr4      1295      + |         1
##     [2]     chr4     41428      + |         1
##     [3]     chr4     42588      + |         1
##     [4]     chr4     42590      + |         2
##     [5]     chr4     42593      + |         5
##     ...      ...       ...    ... .       ...
##   [146]     chr4   1307742      - |         1
##   [147]     chr4   1316537      - |         1
##   [148]     chr4   1318960      - |         1
##   [149]     chr4   1319004      - |         1
##   [150]     chr4   1319369      - |         1
##   -------
##   seqinfo: 1 sequence from dm6 genome</code></pre>
<p>Conversion to a basepair-resolution GRanges object can be turned off by setting <code>makeBRG = FALSE</code>.</p>
</div>
<div id="merging-granges-data" class="section level1">
<h1 class="hasAnchor">
<a href="#merging-granges-data" class="anchor"></a>Merging GRanges Data</h1>
<p>Biological replicates are best used to independently reproduce and measure effects, and therefore we often want to handle them separately. However, there are times when combining replicates can allow for more sensitive measurements, assuming that the replicates are highly concordant.</p>
<p>The <code><a href="../reference/mergeGRangesData.html">mergeGRangesData()</a></code> function can be used to combine basepair-resolution GRanges objects.</p>
<p>We’ll break up the included PRO-seq data into a list of toy datasets:</p>
<div class="sourceCode" id="cb23"><html><body><pre class="r"><span class="no">ps_list</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="fl">1</span>:<span class="fl">6</span>, <span class="kw">function</span>(<span class="no">i</span>) <span class="no">ps</span>[<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="no">i</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">ps</span>), <span class="fl">6</span>)])
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">ps_list</span>) <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"A_rep1"</span>, <span class="st">"A_rep2"</span>,
                    <span class="st">"B_rep1"</span>, <span class="st">"B_rep2"</span>,
                    <span class="st">"C_rep1"</span>, <span class="st">"C_rep2"</span>)</pre></body></html></div>
<div class="sourceCode" id="cb24"><html><body><pre class="r"><span class="no">ps_list</span>[<span class="fl">1</span>:<span class="fl">2</span>]</pre></body></html></div>
<pre><code>## $A_rep1
## GRanges object with 25 ranges and 1 metadata column:
##        seqnames    ranges strand |     score
##           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
##    [1]     chr4      1295      + |         1
##    [2]     chr4     42595      + |         1
##    [3]     chr4     42622      + |         2
##    [4]     chr4     42718      + |         1
##    [5]     chr4     42789      + |         1
##    ...      ...       ...    ... .       ...
##   [21]     chr4   1280795      - |         1
##   [22]     chr4   1306469      - |         1
##   [23]     chr4   1306713      - |         1
##   [24]     chr4   1307115      - |         2
##   [25]     chr4   1307301      - |         1
##   -------
##   seqinfo: 1870 sequences from an unspecified genome
## 
## $A_rep2
## GRanges object with 25 ranges and 1 metadata column:
##        seqnames    ranges strand |     score
##           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
##    [1]     chr4     41428      + |         1
##    [2]     chr4     42596      + |         1
##    [3]     chr4     42652      + |         3
##    [4]     chr4     42733      + |         1
##    [5]     chr4     42794      + |         2
##    ...      ...       ...    ... .       ...
##   [21]     chr4   1280936      - |         1
##   [22]     chr4   1306497      - |         2
##   [23]     chr4   1306888      - |         1
##   [24]     chr4   1307120      - |         1
##   [25]     chr4   1307742      - |         1
##   -------
##   seqinfo: 1870 sequences from an unspecified genome</code></pre>
<div class="sourceCode" id="cb26"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">ps_list</span>)</pre></body></html></div>
<pre><code>## [1] "A_rep1" "A_rep2" "B_rep1" "B_rep2" "C_rep1" "C_rep2"</code></pre>
<p>We can pass a list of GRanges objects directly as an argument:</p>
<div class="sourceCode" id="cb28"><html><body><pre class="r"><span class="fu"><a href="../reference/mergeGRangesData.html">mergeGRangesData</a></span>(<span class="no">ps_list</span>, <span class="kw">ncores</span> <span class="kw">=</span> <span class="fl">1</span>)</pre></body></html></div>
<pre><code>## GRanges object with 150 ranges and 1 metadata column:
##         seqnames    ranges strand |     score
##            &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
##     [1]     chr4      1295      + |         1
##     [2]     chr4     41428      + |         1
##     [3]     chr4     42588      + |         1
##     [4]     chr4     42590      + |         2
##     [5]     chr4     42593      + |         5
##     ...      ...       ...    ... .       ...
##   [146]     chr4   1307742      - |         1
##   [147]     chr4   1316537      - |         1
##   [148]     chr4   1318960      - |         1
##   [149]     chr4   1319004      - |         1
##   [150]     chr4   1319369      - |         1
##   -------
##   seqinfo: 1870 sequences from an unspecified genome</code></pre>
<p>Or we can pass any number of individual GRanges objects as arguments:</p>
<div class="sourceCode" id="cb30"><html><body><pre class="r"><span class="no">merge_ps</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/mergeGRangesData.html">mergeGRangesData</a></span>(<span class="no">ps_list</span><span class="kw">[[</span><span class="fl">1</span>]], <span class="no">ps_list</span><span class="kw">[[</span><span class="fl">2</span>]], <span class="no">ps</span>, <span class="kw">ncores</span> <span class="kw">=</span> <span class="fl">1</span>)
<span class="no">merge_ps</span></pre></body></html></div>
<pre><code>## GRanges object with 150 ranges and 1 metadata column:
##         seqnames    ranges strand |     score
##            &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
##     [1]     chr4      1295      + |         2
##     [2]     chr4     41428      + |         2
##     [3]     chr4     42588      + |         1
##     [4]     chr4     42590      + |         2
##     [5]     chr4     42593      + |         5
##     ...      ...       ...    ... .       ...
##   [146]     chr4   1307742      - |         2
##   [147]     chr4   1316537      - |         1
##   [148]     chr4   1318960      - |         1
##   [149]     chr4   1319004      - |         1
##   [150]     chr4   1319369      - |         1
##   -------
##   seqinfo: 1870 sequences from an unspecified genome</code></pre>
<p><em>Note that the output is also a basepair-resolution GRanges object:</em></p>
<div class="sourceCode" id="cb32"><html><body><pre class="r"><span class="fu"><a href="../reference/makeGRangesBRG.html">isBRG</a></span>(<span class="no">merge_ps</span>)
<span class="co">## [1] TRUE</span></pre></body></html></div>
<p> </p>
<div id="merging-replicates" class="section level2">
<h2 class="hasAnchor">
<a href="#merging-replicates" class="anchor"></a>Merging Replicates</h2>
<p>The <code><a href="../reference/mergeReplicates.html">mergeReplicates()</a></code> function makes combining replicates particularly simple:</p>
<div class="sourceCode" id="cb33"><html><body><pre class="r"><span class="fu"><a href="../reference/mergeReplicates.html">mergeReplicates</a></span>(<span class="no">ps_list</span>, <span class="kw">ncores</span> <span class="kw">=</span> <span class="fl">1</span>)</pre></body></html></div>
<pre><code>## $A
## GRanges object with 50 ranges and 1 metadata column:
##        seqnames    ranges strand |     score
##           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
##    [1]     chr4      1295      + |         1
##    [2]     chr4     41428      + |         1
##    [3]     chr4     42595      + |         1
##    [4]     chr4     42596      + |         1
##    [5]     chr4     42622      + |         2
##    ...      ...       ...    ... .       ...
##   [46]     chr4   1306888      - |         1
##   [47]     chr4   1307115      - |         2
##   [48]     chr4   1307120      - |         1
##   [49]     chr4   1307301      - |         1
##   [50]     chr4   1307742      - |         1
##   -------
##   seqinfo: 1870 sequences from an unspecified genome
## 
## $B
## GRanges object with 50 ranges and 1 metadata column:
##        seqnames    ranges strand |     score
##           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
##    [1]     chr4     42588      + |         1
##    [2]     chr4     42590      + |         2
##    [3]     chr4     42601      + |         1
##    [4]     chr4     42618      + |         1
##    [5]     chr4     42657      + |         1
##    ...      ...       ...    ... .       ...
##   [46]     chr4   1307032      - |         1
##   [47]     chr4   1307122      - |         1
##   [48]     chr4   1307126      - |         1
##   [49]     chr4   1316537      - |         1
##   [50]     chr4   1318960      - |         1
##   -------
##   seqinfo: 1870 sequences from an unspecified genome
## 
## $C
## GRanges object with 50 ranges and 1 metadata column:
##        seqnames    ranges strand |     score
##           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
##    [1]     chr4     42593      + |         5
##    [2]     chr4     42594      + |         2
##    [3]     chr4     42619      + |         2
##    [4]     chr4     42621      + |         1
##    [5]     chr4     42661      + |         1
##    ...      ...       ...    ... .       ...
##   [46]     chr4   1307114      - |         1
##   [47]     chr4   1307283      - |         1
##   [48]     chr4   1307300      - |         1
##   [49]     chr4   1319004      - |         1
##   [50]     chr4   1319369      - |         1
##   -------
##   seqinfo: 1870 sequences from an unspecified genome</code></pre>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>Jason D. Buenrostro, Paul G. Giresi, Lisa C. Zaba, Howard Y. Chang, William J. Greenleaf (2013). Transposition of native chromatin for fast and sensitive epigenomic profiling of open chromatin, dna-binding proteins and nucleosome position.  10: 1213–1218. <a href="#fnref1" class="footnote-back">↩</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
