<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Profile Plots &amp; Bootstrapping • BRGenomics</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Profile Plots &amp; Bootstrapping">
<meta property="og:description" content="BRGenomics">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Overview.html">Overview</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Guides
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/GettingStarted.html">Getting Started</a>
    </li>
    <li>
      <a href="../articles/ImportingModifyingAnnotations.html">Importing &amp; Modifying Annotations</a>
    </li>
    <li>
      <a href="../articles/ImportingProcessingData.html">Importing &amp; Processing Data</a>
    </li>
    <li>
      <a href="../articles/SignalCounting.html">Signal Counting</a>
    </li>
    <li>
      <a href="../articles/AnalyzingMultipleDatasets.html">Analyzing Multiple Datasets</a>
    </li>
    <li>
      <a href="../articles/ProfilePlotsAndBootstrapping.html">Profile Plots &amp; Bootstrapping</a>
    </li>
    <li>
      <a href="../articles/SpikeInNormalization.html">Spike-in Normalization</a>
    </li>
    <li>
      <a href="../articles/DESeq2WithGlobalPerturbations.html">DESeq2 with Global Perturbations</a>
    </li>
    <li>
      <a href="../articles/SequenceExtraction.html">Sequence Extraction</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mdeber/BRGenomics">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Profile Plots &amp; Bootstrapping</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/mdeber/BRGenomics/blob/master/vignettes/ProfilePlotsAndBootstrapping.Rmd"><code>vignettes/ProfilePlotsAndBootstrapping.Rmd</code></a></small>
      <div class="hidden name"><code>ProfilePlotsAndBootstrapping.Rmd</code></div>

    </div>

    
    
<div id="conventional-profiles" class="section level1">
<h1 class="hasAnchor">
<a href="#conventional-profiles" class="anchor"></a>Conventional Profiles</h1>
<p>As an example, we’ll plot PRO-seq signal in promoter-proximal regions by calculating the mean signal intensity at each base in the first 100 bases of a transcript.</p>
<p>First, calculate signal at each base for all promoter-proximal regions:</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">BRGenomics</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"PROseq"</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"txs_dm6_chr4"</span>)
<span class="no">txs_pr</span> <span class="kw">&lt;-</span> <span class="fu">promoters</span>(<span class="no">txs_dm6_chr4</span>, <span class="fl">0</span>, <span class="fl">100</span>)</pre></body></html></div>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="no">countmatrix_pr</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/getCountsByPositions.html">getCountsByPositions</a></span>(<span class="no">PROseq</span>, <span class="no">txs_pr</span>, <span class="kw">ncores</span> <span class="kw">=</span> <span class="fl">1</span>)
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="no">countmatrix_pr</span>)
<span class="co">## [1] 339 100</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="no">countmatrix_pr</span>) <span class="kw">==</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">txs_pr</span>), <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="fu">width</span>(<span class="no">txs_pr</span>)))
<span class="co">## [1] TRUE TRUE</span></pre></body></html></div>
<p>For each position (each column of the matrix), calculate the mean, and plot:</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/plot.html">plot</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="fl">1</span>:<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span>(<span class="no">countmatrix_pr</span>),
     <span class="kw">y</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colMeans</a></span>(<span class="no">countmatrix_pr</span>),
     <span class="kw">type</span> <span class="kw">=</span> <span class="st">"l"</span>,
     <span class="kw">xlab</span> <span class="kw">=</span> <span class="st">"Distance to TSS (bp)"</span>,
     <span class="kw">ylab</span> <span class="kw">=</span> <span class="st">"Mean PRO-seq Reads"</span>)</pre></body></html></div>
<p><img src="ProfilePlotsAndBootstrapping_files/figure-html/unnamed-chunk-3-1.png" width="700"></p>
<p> </p>
<div id="drawbacks" class="section level2">
<h2 class="hasAnchor">
<a href="#drawbacks" class="anchor"></a>Drawbacks</h2>
<p>One drawback of using the arithmetic mean is that means are not robust to outliers. In other words, the mean signal at any position is liable to be determined by a small number of highly influential points. This is especially problematic for high dynamic range data like PRO-seq.</p>
<p>It’s common to see this issue addressed through the use medians/quantiles in place of arithmetic means. However, observe what happens as we plot the median signal across our gene list using several different gene-filtering thresholds:</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="no">plot_meds</span> <span class="kw">&lt;-</span> <span class="kw">function</span>(<span class="no">sig_thresh</span>) {
  <span class="no">idx</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span>(<span class="no">countmatrix_pr</span>) <span class="kw">&gt;</span> <span class="no">sig_thresh</span>)
  <span class="fu"><a href="https://rdrr.io/r/base/plot.html">plot</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="fl">1</span>:<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span>(<span class="no">countmatrix_pr</span>),
       <span class="kw">y</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="no">countmatrix_pr</span>[<span class="no">idx</span>, ], <span class="fl">2</span>, <span class="no">median</span>),
       <span class="kw">type</span> <span class="kw">=</span> <span class="st">"l"</span>,
       <span class="kw">main</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span>(<span class="st">"Regions with &gt;%s reads"</span>, <span class="no">sig_thresh</span>),
       <span class="kw">xlab</span> <span class="kw">=</span> <span class="st">"Distance to TSS (bp)"</span>,
       <span class="kw">ylab</span> <span class="kw">=</span> <span class="st">"Median PRO-seq Reads"</span>)
}

<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(<span class="kw">mfrow</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">3</span>, <span class="fl">2</span>))

<span class="kw">for</span> (<span class="no">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>, <span class="fl">30</span>*<span class="fl">2</span>^(<span class="fl">0</span>:<span class="fl">4</span>))) {
  <span class="fu">plot_meds</span>(<span class="no">i</span>)
}</pre></body></html></div>
<p><img src="ProfilePlotsAndBootstrapping_files/figure-html/unnamed-chunk-4-1.png" width="576"></p>
<p>The paucity of transcription on Drosophila chromosome 4 makes this a somewhat extreme example, and you might find that the above situation is a non-issue for your data. However, it’s important to keep in mind how unexpressed genes can affect the median. A common rule of thumb is that a given cell is only apt to express around half of its genes; if that holds true, the median of an unfiltered genelist will be close to zero, and could be insensitive to changes occurring at expressed genes.</p>
</div>
</div>
<div id="the-rationale-for-bootstrapping" class="section level1">
<h1 class="hasAnchor">
<a href="#the-rationale-for-bootstrapping" class="anchor"></a>The Rationale for Bootstrapping</h1>
<p>A robust alternative to plotting mean or median signal profiles is to plot <strong>bootstrapped</strong> mean signal profiles, known as metaprofile plots or metaplots.</p>
<p>To bootstrap the mean signal by position, a small number of genes are randomly sampled from a genelist, and the mean signal at each position is calculated for that group of genes. This process of randomly sampling genes and calculating mean signals is repeated over many iterations, and the final bootstrapped mean for each position is the median of the sampled means.</p>
<p>One feature of bootstrapping is that it is robust to outliers. But more than that, the a bootstrapped mean provides an <em>expectation</em> for what the mean signal would be for any arbitrary group of genes, and associated with that expectation is a measure of uncertainty. By taking quantiles of the sub-sampled means <em>other than</em> the median (the expected value), we can estimate the extent to which the mean varies across arbitrary groups of genes.</p>
<p>For example, the 75th percentile of the sub-sampled means is a number for which, 25% of the time, we calculated a higher mean. Similarly, a 90% confidence interval about the bootstrapped mean encompasses all values between the 5th and 95th percentiles of the sub-sampled means.</p>
</div>
<div id="generating-and-plotting-metaprofiles" class="section level1">
<h1 class="hasAnchor">
<a href="#generating-and-plotting-metaprofiles" class="anchor"></a>Generating and Plotting Metaprofiles</h1>
<p>We can use the <code><a href="../reference/bootstrap-signal-by-position.html">metaSubsample()</a></code> function to bootstrap mean values by position for a genelist. The function accepts the same arguments as <code><a href="../reference/getCountsByPositions.html">getCountsByPositions()</a></code>, in addition to other arguments related to the bootstrapping.</p>
<p>By default, 10% of the genelist is randomly sampled 1000 times, and confidence bands are returned for the 12.5th and 87.5th percentiles (i.e. a 75% confidence interval).</p>
<p>Given the small size our dataset, we’ll reduce this to a 30% confidence interval, and we’ll additionally use 5 bp bins:</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="no">bootmeans.df</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/bootstrap-signal-by-position.html">metaSubsample</a></span>(<span class="no">PROseq</span>, <span class="no">txs_pr</span>, <span class="kw">binsize</span> <span class="kw">=</span> <span class="fl">5</span>,
                              <span class="kw">lower</span> <span class="kw">=</span> <span class="fl">0.35</span>, <span class="kw">upper</span> <span class="kw">=</span> <span class="fl">0.65</span>, <span class="kw">ncores</span> <span class="kw">=</span> <span class="fl">1</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">bootmeans.df</span>)</pre></body></html></div>
<pre><code>##    x       mean      lower      upper sample.name
## 1  3 0.03529412 0.02941176 0.04705882      PROseq
## 2  8 0.05294118 0.04117647 0.07058824      PROseq
## 3 13 0.07647059 0.05088235 0.10000000      PROseq
## 4 18 0.06470588 0.04705882 0.09411765      PROseq
## 5 23 0.12352941 0.10000000 0.15882353      PROseq
## 6 28 0.09411765 0.08235294 0.11176471      PROseq</code></pre>
<p>A dataframe is returned for plotting, and notice how the x-values have been automatically adjusted to be the center of the bins.</p>
<p>Below, we show how to plot the confidence bands using base R plotting, as well as <code>ggplot2</code>.</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/plot.html">plot</a></span>(<span class="no">mean</span> ~ <span class="no">x</span>, <span class="kw">data</span> <span class="kw">=</span> <span class="no">bootmeans.df</span>, <span class="kw">type</span> <span class="kw">=</span> <span class="st">"l"</span>,
     <span class="kw">main</span> <span class="kw">=</span> <span class="st">"PRO-seq Signal"</span>, <span class="kw">ylim</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>, <span class="fl">1.4</span>),
     <span class="kw">xlab</span> <span class="kw">=</span> <span class="st">"Distance from TSS"</span>,
     <span class="kw">ylab</span> <span class="kw">=</span> <span class="st">"Mean Signal + 30% CI"</span>)

<span class="co"># draw a polygon to add confidence bands,</span>
<span class="co"># and use adjustcolor() to add transparency</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/polygon.html">polygon</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="no">bootmeans.df</span>$<span class="no">x</span>, <span class="fu"><a href="https://rdrr.io/r/base/rev.html">rev</a></span>(<span class="no">bootmeans.df</span>$<span class="no">x</span>)),
        <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="no">bootmeans.df</span>$<span class="no">lower</span>, <span class="fu"><a href="https://rdrr.io/r/base/rev.html">rev</a></span>(<span class="no">bootmeans.df</span>$<span class="no">upper</span>)),
        <span class="kw">col</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/adjustcolor.html">adjustcolor</a></span>(<span class="st">"black"</span>, <span class="fl">0.1</span>), <span class="kw">border</span> <span class="kw">=</span> <span class="fl">FALSE</span>)</pre></body></html></div>
<p><img src="ProfilePlotsAndBootstrapping_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">require</a></span>(<span class="no">ggplot2</span>)</pre></body></html></div>
<pre><code>## Loading required package: ggplot2</code></pre>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="no">bootmeans.df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="no">x</span>, <span class="no">mean</span>)) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_line</a></span>() +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_ribbon.html">geom_ribbon</a></span>(<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="no">x</span>, <span class="kw">ymin</span> <span class="kw">=</span> <span class="no">lower</span>, <span class="kw">ymax</span> <span class="kw">=</span> <span class="no">upper</span>),
              <span class="kw">alpha</span> <span class="kw">=</span> <span class="fl">0.1</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span>(<span class="kw">title</span> <span class="kw">=</span> <span class="st">"PRO-seq Signal"</span>,
       <span class="kw">x</span> <span class="kw">=</span> <span class="st">"Distance from TSS"</span>,
       <span class="kw">y</span> <span class="kw">=</span> <span class="st">"Mean Signal + 30% CI"</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span>()</pre></body></html></div>
<p><img src="ProfilePlotsAndBootstrapping_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
<div id="example-comparative-metaplots" class="section level2">
<h2 class="hasAnchor">
<a href="#example-comparative-metaplots" class="anchor"></a>Example: Comparative Metaplots</h2>
<p>Like other functions in BRGenomics, we can pass a list of GRanges to <code><a href="../reference/bootstrap-signal-by-position.html">metaSubsample()</a></code>, and the output is conveniently combined for plotting.</p>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="co"># make 3 datasets</span>
<span class="no">ps_list</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="kw">ps1</span> <span class="kw">=</span> <span class="no">PROseq</span>[<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="fl">1</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">PROseq</span>), <span class="fl">3</span>)],
                <span class="kw">ps2</span> <span class="kw">=</span> <span class="no">PROseq</span>[<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="fl">2</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">PROseq</span>), <span class="fl">3</span>)],
                <span class="kw">ps3</span> <span class="kw">=</span> <span class="no">PROseq</span>[<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="fl">3</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">PROseq</span>), <span class="fl">3</span>)])</pre></body></html></div>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="no">bm_list.df</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/bootstrap-signal-by-position.html">metaSubsample</a></span>(<span class="no">ps_list</span>, <span class="no">txs_pr</span>, <span class="kw">binsize</span> <span class="kw">=</span> <span class="fl">5</span>,
                            <span class="kw">lower</span> <span class="kw">=</span> <span class="fl">0.35</span>, <span class="kw">upper</span> <span class="kw">=</span> <span class="fl">0.65</span>, <span class="kw">ncores</span> <span class="kw">=</span> <span class="fl">1</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">bm_list.df</span>)</pre></body></html></div>
<pre><code>##    x       mean       lower      upper sample.name
## 1  3 0.01176471 0.005882353 0.02352941         ps1
## 2  8 0.01764706 0.017647059 0.02352941         ps1
## 3 13 0.01764706 0.011764706 0.02352941         ps1
## 4 18 0.04117647 0.023529412 0.05294118         ps1
## 5 23 0.02941176 0.017647059 0.04705882         ps1
## 6 28 0.02941176 0.023529412 0.03529412         ps1</code></pre>
<div class="sourceCode" id="cb14"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">require</a></span>(<span class="no">ggplot2</span>)
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="no">bm_list.df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="no">x</span>, <span class="no">mean</span>, <span class="kw">color</span> <span class="kw">=</span> <span class="no">sample.name</span>)) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_line</a></span>() +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_ribbon.html">geom_ribbon</a></span>(<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="no">x</span>, <span class="kw">ymin</span> <span class="kw">=</span> <span class="no">lower</span>, <span class="kw">ymax</span> <span class="kw">=</span> <span class="no">upper</span>,
                  <span class="kw">color</span> <span class="kw">=</span> <span class="kw">NULL</span>, <span class="kw">fill</span> <span class="kw">=</span> <span class="no">sample.name</span>),
              <span class="kw">alpha</span> <span class="kw">=</span> <span class="fl">0.2</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span>(<span class="kw">title</span> <span class="kw">=</span> <span class="st">"PRO-seq Signal"</span>,
       <span class="kw">x</span> <span class="kw">=</span> <span class="st">"Distance from TSS"</span>,
       <span class="kw">y</span> <span class="kw">=</span> <span class="st">"Mean Signal + 30% CI"</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span>()</pre></body></html></div>
<p><img src="ProfilePlotsAndBootstrapping_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
<p><em>Bear in mind that the above confidence intervals are ignoring 70% of the sub-sampling experiments, which is excessive. More reasonable parameters would reveal how lacking this data is, i.e. how un-confident we are that there is a robust difference in the mean signal at various positions.</em></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
