<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Signal Counting • BRGenomics</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Signal Counting">
<meta property="og:description" content="BRGenomics">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.12.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Overview.html">Overview</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Guides
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/GettingStarted.html">Getting Started</a>
    </li>
    <li>
      <a href="../articles/ImportingModifyingAnnotations.html">Importing &amp; Modifying Annotations</a>
    </li>
    <li>
      <a href="../articles/ImportingProcessingData.html">Importing &amp; Processing Data</a>
    </li>
    <li>
      <a href="../articles/SignalCounting.html">Signal Counting</a>
    </li>
    <li>
      <a href="../articles/AnalyzingMultipleDatasets.html">Analyzing Multiple Datasets</a>
    </li>
    <li>
      <a href="../articles/ProfilePlotsAndBootstrapping.html">Profile Plots &amp; Bootstrapping</a>
    </li>
    <li>
      <a href="../articles/SpikeInNormalization.html">Spike-in Normalization</a>
    </li>
    <li>
      <a href="../articles/DESeq2WithGlobalPerturbations.html">DESeq2 with Global Perturbations</a>
    </li>
    <li>
      <a href="../articles/SequenceExtraction.html">Sequence Extraction</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mdeber/BRGenomics" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Signal Counting</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/mdeber/BRGenomics/blob/HEAD/vignettes/SignalCounting.Rmd" class="external-link"><code>vignettes/SignalCounting.Rmd</code></a></small>
      <div class="hidden name"><code>SignalCounting.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="count-signal-by-region">Count Signal by Region<a class="anchor" aria-label="anchor" href="#count-signal-by-region"></a>
</h2>
<p>The <code>getCountsByRegion()</code> function counts signal within regions of interest, and returns a simple vector.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://mdeber.github.io" class="external-link">BRGenomics</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"PROseq"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"txs_dm6_chr4"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts_txs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getCountsByRegions.html">getCountsByRegions</a></span><span class="op">(</span><span class="va">PROseq</span>, <span class="va">txs_dm6_chr4</span><span class="op">)</span></span>
<span><span class="va">counts_txs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="co">## [1]   1  59  13 126 263</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">txs_dm6_chr4</span><span class="op">)</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">counts_txs</span><span class="op">)</span></span>
<span><span class="co">## [1] TRUE</span></span></code></pre></div>
<p><em>Critically, the default assumes the input is “basepair-resolution”. If the input data is run-length compressed (like a standard bigWig or coverage file), set <code>expand_ranges = TRUE</code>.</em></p>
</div>
<div class="section level2">
<h2 id="count-signal-by-position-and-region">Count Signal by Position and Region<a class="anchor" aria-label="anchor" href="#count-signal-by-position-and-region"></a>
</h2>
<p>The <code><a href="../reference/getCountsByPositions.html">getCountsByPositions()</a></code> function will count signal at each position within each region of interest. By default, a matrix is returned, with a row for each region, and a column for each position.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># get first 100 bases of each transcript</span></span>
<span><span class="va">txs_pr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/intra-range-methods.html" class="external-link">promoters</a></span><span class="op">(</span><span class="va">txs_dm6_chr4</span>, <span class="fl">0</span>, <span class="fl">100</span><span class="op">)</span></span>
<span><span class="co"># get signal at each base within each promoter region</span></span>
<span><span class="va">countmatrix_pr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getCountsByPositions.html">getCountsByPositions</a></span><span class="op">(</span><span class="va">PROseq</span>, <span class="va">txs_pr</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">countmatrix_pr</span><span class="op">)</span></span>
<span><span class="co">## [1] "matrix" "array"</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">countmatrix_pr</span><span class="op">)</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">txs_pr</span><span class="op">)</span></span>
<span><span class="co">## [1] TRUE</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">countmatrix_pr</span><span class="op">)</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">txs_pr</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">## [1] TRUE</span></span></code></pre></div>
<p> </p>
<p><em>Again, be sure to know if your data is “basepair-resolution”, or if the ranges are compressed and should be expanded using the <code>expand_ranges</code> argument.</em></p>
<p>By default, each “position” (each column) is a single base, but counting can be done in bins as well:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># get signal in 10 bp bins within each promoter region</span></span>
<span><span class="va">countmatrix_pr_bin</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getCountsByPositions.html">getCountsByPositions</a></span><span class="op">(</span><span class="va">PROseq</span>, <span class="va">txs_pr</span>, binsize <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">countmatrix_pr_bin</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##      [,1] [,2] [,3] [,4] [,5] [,6] [,7] [,8] [,9] [,10]</span></span>
<span><span class="co">## [1,]    0    0    0    0    0    0    0    0    0     0</span></span>
<span><span class="co">## [2,]    2    1    7    1    0    0    0    3    2     0</span></span>
<span><span class="co">## [3,]    0    0    1    0    0    0    1    0    0     0</span></span>
<span><span class="co">## [4,]    0    0    0    0    0    0    0    0    0     0</span></span>
<span><span class="co">## [5,]    0    0    0    0    0    0    0    0    0     0</span></span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/all.html" class="external-link">all</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">countmatrix_pr_bin</span><span class="op">)</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">countmatrix_pr</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## [1] TRUE</span></span></code></pre></div>
<p>By default, an error is returned if input regions aren’t all the same size:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/getCountsByPositions.html">getCountsByPositions</a></span><span class="op">(</span><span class="va">PROseq</span>, <span class="va">txs_dm6_chr4</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Error in .get_cbp_mw(hits, dataset.gr, regions.gr, binsize, FUN, smw, : regions.gr contains ranges with multiple widths, but simplify.multi.widths is set to 'error'. Did you mean to call getCountsByRegions instead?</span></span></code></pre>
<p>This is intended to avoid accidental use of <code><a href="../reference/getCountsByPositions.html">getCountsByPositions()</a></code> in lieu of <code><a href="../reference/getCountsByRegions.html">getCountsByRegions()</a></code>. However, for the rare case when users do intend to use multi-width regions, there is support for this using the <code>simplify.multi.widths</code> argument, which contains several useful options for how to do this (see the documentation for more details).</p>
<div class="section level3">
<h3 id="example-plot-signal-over-a-gene">Example: Plot Signal Over a Gene<a class="anchor" aria-label="anchor" href="#example-plot-signal-over-a-gene"></a>
</h3>
<p>We can use <code><a href="../reference/getCountsByPositions.html">getCountsByPositions()</a></code> to get the signal profile over a single gene. Let’s have a look at the gene with the highest signal near the TSS:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.min.html" class="external-link">which.max</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">countmatrix_pr</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">idx</span></span>
<span><span class="co">## [1] 135</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">countmatrix_pr</span><span class="op">)</span>, </span>
<span>     y <span class="op">=</span> <span class="va">countmatrix_pr</span><span class="op">[</span><span class="va">idx</span>, <span class="op">]</span>, </span>
<span>     type <span class="op">=</span> <span class="st">"h"</span>, </span>
<span>     main <span class="op">=</span> <span class="va">txs_pr</span><span class="op">$</span><span class="va">tx_name</span><span class="op">[</span><span class="va">idx</span><span class="op">]</span>,</span>
<span>     xlab <span class="op">=</span> <span class="st">"Distance to TSS"</span>,</span>
<span>     ylab <span class="op">=</span> <span class="st">"PRO-seq Signal"</span><span class="op">)</span></span></code></pre></div>
<p><img src="SignalCounting_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="example-plot-signal-heatmap">Example: Plot Signal Heatmap<a class="anchor" aria-label="anchor" href="#example-plot-signal-heatmap"></a>
</h3>
<p>A typical use of <code><a href="../reference/getCountsByPositions.html">getCountsByPositions()</a></code> is to generate a heatmap of signal by position within a list of genes. The <code>ComplexHeatmap</code> package is well documented and offers a high level of functionality, but we can also use <code>ggplot2</code> to generate customizable heatmaps.</p>
<p>To format our matrix for ggplot, we want to “melt” it. The package <code>reshape2</code> can melt matrices into dataframes, but BRGenomics also provides a <code>melt</code> argument for several functions, including <code><a href="../reference/getCountsByPositions.html">getCountsByPositions()</a></code>:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cbp.df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getCountsByPositions.html">getCountsByPositions</a></span><span class="op">(</span><span class="va">PROseq</span>, <span class="va">txs_pr</span>, binsize <span class="op">=</span> <span class="fl">10</span>, </span>
<span>                               melt <span class="op">=</span> <span class="cn">TRUE</span>, ncores <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cbp.df</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   region position signal</span></span>
<span><span class="co">## 1      1        1      0</span></span>
<span><span class="co">## 2      1        2      0</span></span>
<span><span class="co">## 3      1        3      0</span></span>
<span><span class="co">## 4      1        4      0</span></span>
<span><span class="co">## 5      1        5      0</span></span>
<span><span class="co">## 6      1        6      0</span></span></code></pre>
<p>As you can see, the rows and columns of the matrix are now described in columns of this dataframe, and the signal held at each position is in the third column. Now we can plot:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: package 'ggplot2' was built under R version 4.1.2</span></span></code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">cbp.df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fl">10</span><span class="op">*</span><span class="va">position</span> <span class="op">-</span> <span class="fl">5</span>, y <span class="op">=</span> <span class="va">region</span>, fill <span class="op">=</span> <span class="va">signal</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_tile.html" class="external-link">geom_raster</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_cartesian.html" class="external-link">coord_cartesian</a></span><span class="op">(</span>expand <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Distance from TSS"</span>, y <span class="op">=</span> <span class="st">"Transcript"</span>, </span>
<span>       title <span class="op">=</span> <span class="st">"PRO-seq"</span>, fill <span class="op">=</span> <span class="st">"Reads"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="SignalCounting_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<p> </p>
<p>The high dynamic range of PRO-seq data means that regions (rows) with less signal are hard to pick out. To address this, we can row-normalize the data, and let’s sort the rows by the max signal position, as well.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># take only rows decent signal</span></span>
<span><span class="va">row_signal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">countmatrix_pr</span><span class="op">)</span></span>
<span><span class="va">idx_signal</span> <span class="op">&lt;-</span> <span class="va">row_signal</span> <span class="op">&gt;</span> <span class="fu"><a href="https://rdrr.io/r/stats/median.html" class="external-link">median</a></span><span class="op">(</span><span class="va">row_signal</span><span class="op">)</span></span>
<span><span class="va">cbp</span> <span class="op">&lt;-</span> <span class="va">countmatrix_pr</span><span class="op">[</span><span class="va">idx_signal</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co"># row-normalize</span></span>
<span><span class="va">cbp_rn</span> <span class="op">&lt;-</span> <span class="fl">100</span> <span class="op">*</span> <span class="va">cbp</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">cbp</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># get row order (by max position)</span></span>
<span><span class="va">row_order</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/order.html" class="external-link">order</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">cbp_rn</span>, <span class="fl">1</span>, <span class="va">which.max</span><span class="op">)</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># melt into a dataframe</span></span>
<span><span class="va">rn_cbp.df</span> <span class="op">&lt;-</span> <span class="fu">reshape2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/melt.html" class="external-link">melt</a></span><span class="op">(</span><span class="va">cbp_rn</span><span class="op">[</span><span class="va">row_order</span>, <span class="op">]</span>, </span>
<span>                            varnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"region"</span>, <span class="st">"position"</span><span class="op">)</span>,</span>
<span>                            value.name <span class="op">=</span> <span class="st">"signal"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">rn_cbp.df</span>, </span>
<span>       <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">position</span>, y <span class="op">=</span> <span class="va">region</span>, fill <span class="op">=</span> <span class="va">signal</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_tile.html" class="external-link">geom_raster</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_fill_gradient</a></span><span class="op">(</span>low <span class="op">=</span> <span class="st">"white"</span>, high <span class="op">=</span> <span class="st">"blue"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_cartesian.html" class="external-link">coord_cartesian</a></span><span class="op">(</span>expand <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Distance from TSS"</span>, y <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>         title <span class="op">=</span> <span class="st">"Row-Normalized PRO-seq"</span>, fill <span class="op">=</span> <span class="st">"% Signal"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.ticks.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>          axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="SignalCounting_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
<p><em>Note that this is not what a typical PRO-seq heatmap looks like, but this data is very sparse owing to the dot chromosome, low sequencing depth, and no genelist filtering.</em></p>
</div>
</div>
<div class="section level2">
<h2 id="blacklisting">Blacklisting<a class="anchor" aria-label="anchor" href="#blacklisting"></a>
</h2>
<p>Many functions in BRGenomics support blacklisting, or the exclusion of certain sites from analysis. To use this feature, import your blacklist as a GRanges object, and use the blacklist option.</p>
<p>For instance, <code><a href="../reference/getCountsByPositions.html">getCountsByPositions()</a></code> supports blacklisting, and there is an additional option of whether to set all blacklisted sites to 0 counts, or to set those sites equal to NA. Setting them to NA is useful, as many functions have arguments to ignore NA values in calculations, i.e. <code>mean(x, na.rm = TRUE)</code>.</p>
<p>Supplying a blacklist to <code><a href="../reference/bootstrap-signal-by-position.html">metaSubsample()</a></code> will result in blacklisted positions being ignored in the calculations. Regions for which some section overlaps the blacklist are not ignored entirely, but the blacklisted positions themselves won’t contribute to the calculations.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
