<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Spike-in Normalization • BRGenomics</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Spike-in Normalization">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BRGenomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.30</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Overview.html">Overview</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Guides
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/GettingStarted.html">Getting Started</a>
    </li>
    <li>
      <a href="../articles/ImportingModifyingAnnotations.html">Importing &amp; Modifying Annotations</a>
    </li>
    <li>
      <a href="../articles/ImportingProcessingData.html">Importing &amp; Processing Data</a>
    </li>
    <li>
      <a href="../articles/SignalCounting.html">Signal Counting</a>
    </li>
    <li>
      <a href="../articles/AnalyzingMultipleDatasets.html">Analyzing Multiple Datasets</a>
    </li>
    <li>
      <a href="../articles/ProfilePlotsAndBootstrapping.html">Profile Plots &amp; Bootstrapping</a>
    </li>
    <li>
      <a href="../articles/SpikeInNormalization.html">Spike-in Normalization</a>
    </li>
    <li>
      <a href="../articles/DESeq2WithGlobalPerturbations.html">DESeq2 with Global Perturbations</a>
    </li>
    <li>
      <a href="../articles/SequenceExtraction.html">Sequence Extraction</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mdeber/BRGenomics">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Spike-in Normalization</h1>
            
      
      
      <div class="hidden name"><code>SpikeInNormalization.Rmd</code></div>

    </div>

    
    
<p>BRGenomics includes useful utilities for spike-in normalization.</p>
<p>A typical approach is to add the spike-in (either exogenous cells or synthetic oligonucleotides) before library preparation, and to subsequently map to a <strong>combined genome</strong> containing both the target organisms chromosomes (to map the experimental reads) as well as sequences/chromosomes for the spike-in.</p>
<p>This so-called “competitive alignment” results in the creation of BAM files containing a mix of chromosomes, for which it should be straightforward to identify the spike-in chromosomes.</p>
<div id="counting-spike-in-reads" class="section level1">
<h1 class="hasAnchor">
<a href="#counting-spike-in-reads" class="anchor"></a>Counting Spike-in Reads</h1>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(BRGenomics)</a></code></pre></div>
<p>For this section, we’ll use a list of 4 dummy datasets containing normal, as well as spike-in chromosomes. Consider the first 2 datasets:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">grl[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>]</a></code></pre></div>
<pre><code>## $gr1_rep1
## GRanges object with 4 ranges and 1 metadata column:
##        seqnames    ranges strand |     score
##           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;
##   [1]      chr1         1      + |         1
##   [2]      chr2         2      + |         1
##   [3] spikechr1         3      + |         1
##   [4] spikechr2         4      + |         1
##   -------
##   seqinfo: 4 sequences from an unspecified genome; no seqlengths
## 
## $gr2_rep1
## GRanges object with 4 ranges and 1 metadata column:
##        seqnames    ranges strand |     score
##           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;
##   [1]      chr1         1      + |         2
##   [2]      chr2         2      + |         2
##   [3] spikechr1         3      + |         1
##   [4] spikechr2         4      + |         1
##   -------
##   seqinfo: 4 sequences from an unspecified genome; no seqlengths</code></pre>
<p>We can identify the spike-in chromosomes either by full names, or by a regular expression that matches the spike-in chromosomes. In this case, we named our spike-in chromosomes to contain the string “spike” which makes them easy to identify.</p>
<p>To count the reads for each dataset:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw"><a href="../reference/getSpikeInCounts.html">getSpikeInCounts</a></span>(grl, <span class="dt">si_pattern =</span> <span class="st">"spike"</span>, <span class="dt">ncores =</span> <span class="dv">1</span>)</a></code></pre></div>
<pre><code>##     sample total_reads exp_reads spike_reads
## 1 gr1_rep1           4         2           2
## 2 gr2_rep1           6         4           2
## 3 gr1_rep2           5         2           3
## 4 gr2_rep2          12         8           4</code></pre>
</div>
<div id="filtering-spike-in-reads" class="section level1">
<h1 class="hasAnchor">
<a href="#filtering-spike-in-reads" class="anchor"></a>Filtering Spike-in Reads</h1>
<p>We can also remove the spike-in reads from our data:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw"><a href="../reference/getSpikeInCounts.html">removeSpikeInReads</a></span>(grl[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>], <span class="dt">si_pattern =</span> <span class="st">"spike"</span>, <span class="dt">ncores =</span> <span class="dv">1</span>)</a></code></pre></div>
<pre><code>## $gr1_rep1
## GRanges object with 2 ranges and 1 metadata column:
##       seqnames    ranges strand |     score
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;
##   [1]     chr1         1      + |         1
##   [2]     chr2         2      + |         1
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths
## 
## $gr2_rep1
## GRanges object with 2 ranges and 1 metadata column:
##       seqnames    ranges strand |     score
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;
##   [1]     chr1         1      + |         2
##   [2]     chr2         2      + |         2
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</code></pre>
<p>And if we wanted to isolate the spike-in reads, there is an analogous <code><a href="../reference/getSpikeInCounts.html">getSpikeInReads()</a></code> function.</p>
</div>
<div id="the-spike-in-normalization-factor" class="section level1">
<h1 class="hasAnchor">
<a href="#the-spike-in-normalization-factor" class="anchor"></a>The Spike-in Normalization Factor</h1>
<p>There are several methods by which to generate spike-in normalization factors, but we advocate for a particular method, which generates units we call <b>S</b>pike-in normalized <b>R</b>eads <b>P</b>er <b>M</b>illion mapped reads in the negative <b>C</b>ontrol (<b>SRPMC</b>). The SRPMC normalization factor for a given sample <span class="math inline">\(i\)</span> is defined as such:</p>
<p><span class="math display">\[SRPMC:\ NF_i = \frac{\sum reads_{spikein, control}}{\sum reads_{spikein, i}} 
\cdot \frac{10^6}{\sum reads_{experimental, control}}\]</span></p>
<p>This expression effectively calculates Reads Per Million (RPM) normalization for the negative control, and all other samples <span class="math inline">\(i\)</span> are scaled into equivalent units according to the ratio of their spike-in reads. We provide a more explicit derivation below.</p>
<div id="derivation-of-srpmc" class="section level2">
<h2 class="hasAnchor">
<a href="#derivation-of-srpmc" class="anchor"></a>Derivation of SRPMC</h2>
<p>The fundamental concept of spike-in normalization is that the ratio of experimental reads to spike-in reads can be used to correct for global changes in starting material. Let’s call this ratio <u>R</u>eads <u>P</u>er <u>S</u>pike-in read (<b>RPS</b>):</p>
<p><span class="math display">\[RPS = \frac{\sum reads_{experimental}}{\sum reads_{spikein}}\]</span></p>
<p>In isolation, this number only reflects the relative amounts of spike-in material recovered and mapped. The meaningful information about changes in material can only arise from making direct comparisons between samples. For any sample, <span class="math inline">\(i\)</span>, we can calculate the global change in signal as a proportion of the material recovered from a negative control:</p>
<p><span class="math display">\[RelativeSignal_i = \frac{RPS_i}{RPS_{control}}\]</span></p>
<p>The usual purpose of spike-in normalization is to measure a biological difference in total material (e.g. RNA) between samples, and the above ratio is a direct measurement of this.</p>
<p>To generate normalization factors, we use the above ratio to adjust RPM (Reads Per Million mapped reads) normalization factors, which we define below for clarity:</p>
<p><span class="math display">\[RPM:\ NF_i = \frac{1}{\frac{\sum{reads_i}}{10^6}} = \frac{10^6}{\sum{reads_i}}\]</span></p>
<p>(Unless indicated, <span class="math inline">\(reads\)</span> refers to non-spike-in reads).</p>
<p>RPM normalization (i.e. read depth normalization) is the simplest and likely most familiar form of normalization. For a basal (unperturbed) negative control, RPM should produce the most portable metric of signal, given that we intend for the negative control to demonstrate typical physiology, and we hope that this state is reproducible. We therefore want to have our normalized signal in unit terms of RPM in the negative control.</p>
<p>To accomplish this, we multiply the ratio of spike-in normalized reads between the sample <span class="math inline">\(i\)</span> and the negative control to the RPM normalization factor for sample <span class="math inline">\(i\)</span>. This converts readcounts into units we summarize as <u>S</u>pike-in normalized <u>R</u>eads <u>P</u>er <u>M</u>illion mapped reads in the negative <u>C</u>ontrol (SRPMC):</p>
<p><span class="math display">\[SRPMC:\ NF_i = \frac{RPS_i}{RPS_{control}} \cdot \frac{10^6}{\sum{reads_i}}\]</span></p>
<p>Again, SRPMC results in the negative control being RPM (sequencing depth) normalized, while all other samples are in equivalent, directly comparable units. And we’ve effectively determined the relative scaling of those samples based on the ratios of spike-in reads.</p>
<p>This becomes more apparent if we substitute the <span class="math inline">\(RPS\)</span> variables above. <span class="math inline">\(\sum{reads_i}\)</span> cancels, and simplifying the fraction produces the original formula:</p>
<p><span class="math display">\[SRPMC:\ NF_i = \frac{\sum reads_{spikein, control}}{\sum reads_{spikein, i}} 
\cdot \frac{10^6}{\sum reads_{experimental, control}}\]</span></p>
</div>
</div>
<div id="calculating-normalization-factors" class="section level1">
<h1 class="hasAnchor">
<a href="#calculating-normalization-factors" class="anchor"></a>Calculating Normalization Factors</h1>
<p>We can calculate SRPMC normalization factors for each sample using the <code><a href="../reference/getSpikeInNFs.html">getSpikeInNFs()</a></code> function, using the same syntax we used to count the spike-in reads.</p>
<p>However, we have to also identify the negative control, which is the sample that will have the “reference” (RPM) normalization. We do this either using a regular expression (<code>ctrl_pattern</code> argument) or by supplying the name(s) of the negative controls (the <code>ctrl_names</code> argument).</p>
<p>The default method is “SRPMC”, but there are other options, as well.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw"><a href="../reference/getSpikeInNFs.html">getSpikeInNFs</a></span>(grl, <span class="dt">si_pattern =</span> <span class="st">"spike"</span>, <span class="dt">ctrl_pattern =</span> <span class="st">"gr1"</span>, <span class="dt">ncores =</span> <span class="dv">1</span>)</a></code></pre></div>
<pre><code>## [1] 500000 500000 500000 375000</code></pre>
<p>(The NFs are high because our dummy data contain only a few reads).</p>
<p>By default, normalization factors utilize <strong>batch normalization</strong>, such that in any replicate (identified by the characters following “rep” in the sample names), the negative control is RPM normalized, and the other conditions are normalized to the within-replicate negative control (see the documentation for further details).</p>
<p>Currently, batch normalization requires the sample names end with strings matching the format <code>"_rep1"</code>, <code>"_rep2"</code>, etc. If sample names do not conform to this pattern, you can rename them by using the <code>sample_names</code> argument.</p>
</div>
<div id="normalizing-data" class="section level1">
<h1 class="hasAnchor">
<a href="#normalizing-data" class="anchor"></a>Normalizing Data</h1>
<p>We can also use the <code><a href="../reference/getSpikeInNFs.html">spikeInNormGRanges()</a></code> function to simultaneously find the spike-in reads, calculate the spike-in normalization factors, filter out spike-in reads, and normalize the readcounts:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="kw"><a href="../reference/getSpikeInNFs.html">spikeInNormGRanges</a></span>(grl, <span class="dt">si_pattern =</span> <span class="st">"spike"</span>, <span class="dt">ctrl_pattern =</span> <span class="st">"gr1"</span>, <span class="dt">ncores =</span> <span class="dv">1</span>)</a></code></pre></div>
<pre><code>## $gr1_rep1
## GRanges object with 2 ranges and 1 metadata column:
##       seqnames    ranges strand |     score
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;
##   [1]     chr1         1      + |     5e+05
##   [2]     chr2         2      + |     5e+05
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths
## 
## $gr2_rep1
## GRanges object with 2 ranges and 1 metadata column:
##       seqnames    ranges strand |     score
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;
##   [1]     chr1         1      + |     1e+06
##   [2]     chr2         2      + |     1e+06
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths
## 
## $gr1_rep2
## GRanges object with 2 ranges and 1 metadata column:
##       seqnames    ranges strand |     score
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;
##   [1]     chr1         1      + |     5e+05
##   [2]     chr2         2      + |     5e+05
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths
## 
## $gr2_rep2
## GRanges object with 2 ranges and 1 metadata column:
##       seqnames    ranges strand |     score
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;
##   [1]     chr1         1      + |   1500000
##   [2]     chr2         2      + |   1500000
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</code></pre>
</div>
<div id="normalization-by-sub-sampling" class="section level1">
<h1 class="hasAnchor">
<a href="#normalization-by-sub-sampling" class="anchor"></a>Normalization by Sub-sampling</h1>
<div id="rationale" class="section level2">
<h2 class="hasAnchor">
<a href="#rationale" class="anchor"></a>Rationale</h2>
<p>When viewing genomics data in a genome browser (or otherwise plotting signal for a single gene), the sparsity of basepair resolution data can challenge our visual perception.</p>
<p>Consider two datasets from identical samples, but where one is sequenced to a higher depth. The two datasets can be normalized such that the signal counts are equivalent, but the dataset with higher sequencing depth will have also uncovered additional sites. When plotted in a genome browser, the total signal within a region may be the same, but the more highly sequenced dataset will cover more positions but have lower peaks, while the less sequenced dataset will look sparse and spikey in comparison.</p>
<p>Below, we compare PRO-seq data derived from the same dataset over the same gene. In one case, we randomly sample half of the reads over that gene, while in another, we divide all the readcounts by 2.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"PROseq"</span>)</a>
<a class="sourceLine" id="cb12-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"txs_dm6_chr4"</span>)</a></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="co"># choose a single gene</span></a>
<a class="sourceLine" id="cb13-2" data-line-number="2">gene_i &lt;-<span class="st"> </span>txs_dm6_chr4[<span class="dv">185</span>]</a>
<a class="sourceLine" id="cb13-3" data-line-number="3">reads.gene_i &lt;-<span class="st"> </span><span class="kw">subsetByOverlaps</span>(PROseq, gene_i)</a>
<a class="sourceLine" id="cb13-4" data-line-number="4"></a>
<a class="sourceLine" id="cb13-5" data-line-number="5"><span class="co"># sample half the raw reads</span></a>
<a class="sourceLine" id="cb13-6" data-line-number="6"><span class="kw"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="dv">11</span>)</a>
<a class="sourceLine" id="cb13-7" data-line-number="7">sreads.gene_i &lt;-<span class="st"> </span><span class="kw"><a href="../reference/subsampleGRanges.html">subsampleGRanges</a></span>(reads.gene_i, <span class="dt">prop =</span> <span class="fl">0.5</span>, <span class="dt">ncores =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb13-8" data-line-number="8"></a>
<a class="sourceLine" id="cb13-9" data-line-number="9"><span class="co"># downscale raw reads by a factor of 2</span></a>
<a class="sourceLine" id="cb13-10" data-line-number="10"><span class="kw">score</span>(reads.gene_i) &lt;-<span class="st"> </span><span class="fl">0.5</span> <span class="op">*</span><span class="st"> </span><span class="kw">score</span>(reads.gene_i)</a></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="kw">score</span>(reads.gene_i))</a>
<a class="sourceLine" id="cb14-2" data-line-number="2"><span class="co">## [1] 738.5</span></a>
<a class="sourceLine" id="cb14-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="kw">score</span>(sreads.gene_i))</a>
<a class="sourceLine" id="cb14-4" data-line-number="4"><span class="co">## [1] 738</span></a></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="dt">x =</span> <span class="dv">1</span><span class="op">:</span><span class="kw">width</span>(gene_i), </a>
<a class="sourceLine" id="cb15-2" data-line-number="2">     <span class="dt">y =</span> <span class="kw"><a href="../reference/getCountsByPositions.html">getCountsByPositions</a></span>(sreads.gene_i, gene_i),</a>
<a class="sourceLine" id="cb15-3" data-line-number="3">     <span class="dt">type =</span> <span class="st">"h"</span>, <span class="dt">ylim =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">0</span>, <span class="dv">20</span>),</a>
<a class="sourceLine" id="cb15-4" data-line-number="4">     <span class="dt">main =</span> <span class="st">"PRO-seq (down-sampled)"</span>,</a>
<a class="sourceLine" id="cb15-5" data-line-number="5">     <span class="dt">xlab =</span> <span class="st">"Distance from TSS"</span>, <span class="dt">ylab =</span> <span class="st">"Down-sampled PRO-seq Reads"</span>)</a></code></pre></div>
<p><img src="SpikeInNormalization_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="dt">x =</span> <span class="dv">1</span><span class="op">:</span><span class="kw">width</span>(gene_i), </a>
<a class="sourceLine" id="cb16-2" data-line-number="2">     <span class="dt">y =</span> <span class="kw"><a href="../reference/getCountsByPositions.html">getCountsByPositions</a></span>(reads.gene_i, gene_i),</a>
<a class="sourceLine" id="cb16-3" data-line-number="3">     <span class="dt">type =</span> <span class="st">"h"</span>,  <span class="dt">ylim =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">0</span>, <span class="dv">20</span>),</a>
<a class="sourceLine" id="cb16-4" data-line-number="4">     <span class="dt">main =</span> <span class="st">"PRO-seq (down-scaled)"</span>, </a>
<a class="sourceLine" id="cb16-5" data-line-number="5">     <span class="dt">xlab =</span> <span class="st">"Distance from TSS"</span>,  <span class="dt">ylab =</span> <span class="st">"Down-scaled PRO-seq Reads"</span>)</a></code></pre></div>
<p><img src="SpikeInNormalization_files/figure-html/unnamed-chunk-11-2.png" width="700"></p>
<p>The two plots above come from the same data, and contain the same quantity of signal, but their profiles are notably distinct. Particularly when plotting many samples over large regions within a genome browser, differences caused by sequencing depth can be misleading. It can be challenging to estimate differences if some datasets are “tall and spikey” while others are “short and smooth”.</p>
<p>Absent global changes in signal, the above scenario can be resolved beforehand by equal sequencing depths, or by down-sampling to match readcounts.</p>
<p>However, matching raw readcounts is not a solution when significant biological changes in total signal should be accounted for.</p>
<p>For instance, consider an example in which there is a true, two-fold biological difference in transcription between two samples. If we could avoid all technical artifacts and measure the transcription directly in each individual cell, we would expect to uncover half the number of transcribing complexes in the lower condition. Having equivalent sequencing depth across those two conditions is effectively a technical artifact, and down-scaling the signal by multiplication can cause the visual challenges observed above.</p>
</div>
<div id="sub-sampling-for-normalization" class="section level2">
<h2 class="hasAnchor">
<a href="#sub-sampling-for-normalization" class="anchor"></a>Sub-sampling for Normalization</h2>
<p>To address the above concerns, we’ve included a function <code><a href="../reference/subsampleBySpikeIn.html">subsampleBySpikeIn()</a></code> to randomly sample reads to match the normalized signal proportions between datasets.</p>
<p>Internally, the function uses the <code><a href="../reference/getSpikeInNFs.html">getSpikeInNFs()</a></code> function, but instead of SRPMC normalization, using the option <code>method = "SNR"</code>, which calculates normalization factors that downscale each dataset to match the dataset with the least spike-in reads. From this, the number of “desired reads” is established for each dataset, and subsequently that number of reads is randomly sampled.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="kw"><a href="../reference/getSpikeInCounts.html">removeSpikeInReads</a></span>(grl, <span class="dt">si_pattern =</span> <span class="st">"spike"</span>, <span class="dt">ncores =</span> <span class="dv">1</span>)</a></code></pre></div>
<pre><code>## $gr1_rep1
## GRanges object with 2 ranges and 1 metadata column:
##       seqnames    ranges strand |     score
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;
##   [1]     chr1         1      + |         1
##   [2]     chr2         2      + |         1
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths
## 
## $gr2_rep1
## GRanges object with 2 ranges and 1 metadata column:
##       seqnames    ranges strand |     score
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;
##   [1]     chr1         1      + |         2
##   [2]     chr2         2      + |         2
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths
## 
## $gr1_rep2
## GRanges object with 2 ranges and 1 metadata column:
##       seqnames    ranges strand |     score
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;
##   [1]     chr1         1      + |         1
##   [2]     chr2         2      + |         1
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths
## 
## $gr2_rep2
## GRanges object with 2 ranges and 1 metadata column:
##       seqnames    ranges strand |     score
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;
##   [1]     chr1         1      + |         4
##   [2]     chr2         2      + |         4
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="kw"><a href="../reference/getSpikeInNFs.html">getSpikeInNFs</a></span>(grl, <span class="dt">si_pattern =</span> <span class="st">"spike"</span>, <span class="dt">method =</span> <span class="st">"SNR"</span>, <span class="dt">batch_norm =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb19-2" data-line-number="2">              <span class="dt">ncores =</span> <span class="dv">1</span>)</a></code></pre></div>
<pre><code>## [1] 1.0000000 1.0000000 0.6666667 0.5000000</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1"><span class="kw"><a href="../reference/subsampleBySpikeIn.html">subsampleBySpikeIn</a></span>(grl, <span class="dt">si_pattern =</span> <span class="st">"spike"</span>, <span class="dt">batch_norm =</span> <span class="ot">FALSE</span>, <span class="dt">ncores =</span> <span class="dv">1</span>)</a></code></pre></div>
<pre><code>## $gr1_rep1
## GRanges object with 2 ranges and 1 metadata column:
##       seqnames    ranges strand |     score
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
##   [1]     chr1         1      + |         1
##   [2]     chr2         2      + |         1
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths
## 
## $gr2_rep1
## GRanges object with 2 ranges and 1 metadata column:
##       seqnames    ranges strand |     score
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
##   [1]     chr1         1      + |         2
##   [2]     chr2         2      + |         2
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths
## 
## $gr1_rep2
## GRanges object with 1 range and 1 metadata column:
##       seqnames    ranges strand |     score
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
##   [1]     chr1         1      + |         1
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths
## 
## $gr2_rep2
## GRanges object with 2 ranges and 1 metadata column:
##       seqnames    ranges strand |     score
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
##   [1]     chr1         1      + |         1
##   [2]     chr2         2      + |         3
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</code></pre>
<p>Normalization by sub-sampling sacrifices information to reduce biases across datasets.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#counting-spike-in-reads">Counting Spike-in Reads</a></li>
      <li><a href="#filtering-spike-in-reads">Filtering Spike-in Reads</a></li>
      <li>
<a href="#the-spike-in-normalization-factor">The Spike-in Normalization Factor</a><ul class="nav nav-pills nav-stacked">
<li><a href="#derivation-of-srpmc">Derivation of SRPMC</a></li>
      </ul>
</li>
      <li><a href="#calculating-normalization-factors">Calculating Normalization Factors</a></li>
      <li><a href="#normalizing-data">Normalizing Data</a></li>
      <li>
<a href="#normalization-by-sub-sampling">Normalization by Sub-sampling</a><ul class="nav nav-pills nav-stacked">
<li><a href="#rationale">Rationale</a></li>
      <li><a href="#sub-sampling-for-normalization">Sub-sampling for Normalization</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Mike DeBerardine.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
